"
Write plink ped and map files for a callset from Petr

Usage:
callset2plink.R --input_file <DATA_IN> --output_prefix <OUT_PFX> [-hv]

Options:
-i --input_file <DATA_IN>        input file
-o --output_prefix <OUT_PFX>     output file prefix (to produce OUT_PFX.map and OUT_PFX.ped)
-h --help                        show this
-v --version                     print version and stop

The program returns PLINK format OUT_PFX.map and OUT_PFX.ped files from a 
tab-delimited file with variants produced by Petr Danecek.

This program requires the R packages 'tidyverse'.

Davis McCarthy
May 2018
" -> doc

write_map_file <- function(object, filename, chrom_format = "Ensembl") {
    # .map (PLINK text fileset variant information file)
    # Variant information file accompanying a .ped text pedigree + genotype table. Also generated by '--recode rlist'.
    # 
    # A text file with no header file, and one line per variant with the following 3-4 fields:
    #     
    #     Chromosome code. PLINK 1.9 also permits contig names here, but most older programs do not.
    # Variant identifier
    # Position in morgans or centimorgans (optional; also safe to use dummy value of '0')
    # Base-pair coordinate
    # All lines must have the same number of columns (so either no lines contain the morgans/centimorgans column, or all of them do).
    nvars <- nrow(object)
    df <- dplyr::as_data_frame(data.frame(matrix(nrow = nvars, ncol = 4)))
    if (chrom_format == "Ensembl")
        df[, 1] <- object[["chrom"]]
    else if (chrom_format == "UCSC")
        df[, 1] <- paste0("chr", object[["chrom"]])
    else
        stop("Unrecognised chrom_format: either 'Ensembl' or 'UCSC'")
    df[, 2] <- paste0(object[["chrom"]], ":", object[["pos"]], "_", 
                      object[["ref"]], "_", object[["alt"]])
    df[, 3] <- 0
    df[, 4] <- object[["pos"]]
    readr::write_tsv(df, path = filename, col_names = FALSE)
}



write_ped_file <- function(object, filename) {
    # .ped (PLINK/MERLIN/Haploview text pedigree + genotype table)
    # Original standard text format for sample pedigree information and genotype calls. Normally must be accompanied by a .map file; Haploview requires an accompanying .info file instead. Loaded with --file, and produced by --recode.
    # 
    # Contains no header line, and one line per sample with 2V+6 fields where V is the number of variants. The first six fields are the same as those in a .fam file. The seventh and eighth fields are allele calls for the first variant in the .map file ('0' = no call); the 9th and 10th are allele calls for the second variant; and so on.
    # 
    # If all alleles are single-character, PLINK 1.9 will correctly parse the more compact 'compound genotype' variant of this format, where each genotype call is represented as a single two-character string. This does not require the use of an additional loading flag. You can produce such a file with '--recode compound-genotypes'.
    # 
    # It is also possible to load .ped files missing some initial fields.
    # 
    nvars <- nrow(object)
    df <- dplyr::as_data_frame(data.frame(matrix(nrow = 1, ncol = 6)))
    df[, 1] <- 0
    df[, 2] <- "consensus"
    df[, 3] <- 0
    df[, 4] <- 0
    df[, 5] <- 0
    df[, 6] <- 0
    genos <- matrix(nrow = 1, ncol = 2 * nvars)
    refs <- object[["ref"]]
    alts <- object[["alt"]]
    genos[1, seq(1, 2 * nvars - 1, by = 2)] <- refs
    genos[1, seq(2, 2 * nvars, by = 2)] <- alts
    df <- dplyr::bind_cols(df, dplyr::as_data_frame(genos))
    readr::write_tsv(df, path = filename, col_names = FALSE) 
}


main <- function(variants_file, output_prefix) {
    exome_sites <- readr::read_tsv(
        variants_file,
        col_types = "ciccdcciiiiccccccccddcdcll", comment = "#",
        col_names = TRUE)
    write_map_file(exome_sites, paste0(output_prefix, ".map"))
    write_ped_file(exome_sites, paste0(output_prefix, ".ped"))
    return("Done.")
}

## Get command line options
opt <- docopt::docopt(doc, version = "version 0.0.1\n")

message("working directory: ", getwd(), "\n")
message("input file: ", opt$input_file, "\n")
message("output prefix: ", opt$output_prefix, "\n")

## Run main function
main(opt$input_file, opt$output_prefix)

