<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Davis J. McCarthy" />


<title>Analysis for joxm: using Cardelino-relax assignments</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fibroblast-clonality</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/davismcc/fibroblast-clonality">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Analysis for joxm: using Cardelino-relax assignments</h1>
<h4 class="author">Davis J. McCarthy</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-10-30
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>fibroblast-clonality/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20180807code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20180807)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20180807code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20180807)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomdavismccfibroblastclonalitytreef2e01ef23334fbb86d5874dacc97ddb1493358e5targetblankf2e01efa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/davismcc/fibroblast-clonality/tree/f2e01ef23334fbb86d5874dacc97ddb1493358e5" target="_blank">f2e01ef</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomdavismccfibroblastclonalitytreef2e01ef23334fbb86d5874dacc97ddb1493358e5targetblankf2e01efa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .vscode/
    Ignored:    code/.DS_Store
    Ignored:    code/selection/.DS_Store
    Ignored:    code/selection/.Rhistory
    Ignored:    code/selection/figures/
    Ignored:    data/.DS_Store
    Ignored:    logs/
    Ignored:    src/.DS_Store
    Ignored:    src/Rmd/.Rhistory

Untracked files:
    Untracked:  .dockerignore
    Untracked:  .dropbox
    Untracked:  .snakemake/
    Untracked:  Rplots.pdf
    Untracked:  Snakefile_clonality
    Untracked:  Snakefile_somatic_calling
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/assess_mutect2_fibro-ipsc_variant_calls.ipynb
    Untracked:  analysis/cardelino_fig1b.R
    Untracked:  analysis/cardelino_fig2b.R
    Untracked:  code/analysis_for_garx.Rmd
    Untracked:  code/selection/data/
    Untracked:  code/selection/fit-dist.nb
    Untracked:  code/selection/result-figure.R
    Untracked:  code/yuanhua/
    Untracked:  data/Melanoma-RegevGarraway-DFCI-scRNA-Seq/
    Untracked:  data/PRJNA485423/
    Untracked:  data/canopy/
    Untracked:  data/cell_assignment/
    Untracked:  data/cnv/
    Untracked:  data/de_analysis_FTv62/
    Untracked:  data/donor_info_070818.txt
    Untracked:  data/donor_info_core.csv
    Untracked:  data/donor_neutrality.tsv
    Untracked:  data/exome-point-mutations/
    Untracked:  data/fdr10.annot.txt.gz
    Untracked:  data/human_H_v5p2.rdata
    Untracked:  data/human_c2_v5p2.rdata
    Untracked:  data/human_c6_v5p2.rdata
    Untracked:  data/neg-bin-rsquared-petr.csv
    Untracked:  data/neutralitytestr-petr.tsv
    Untracked:  data/raw/
    Untracked:  data/sce_merged_donors_cardelino_donorid_all_qc_filt.rds
    Untracked:  data/sce_merged_donors_cardelino_donorid_all_with_qc_labels.rds
    Untracked:  data/sce_merged_donors_cardelino_donorid_unstim_qc_filt.rds
    Untracked:  data/sces/
    Untracked:  data/selection/
    Untracked:  data/simulations/
    Untracked:  data/variance_components/
    Untracked:  docs/figure/clone_prevalences_cardelino-relax.Rmd/
    Untracked:  figures/
    Untracked:  output/differential_expression/
    Untracked:  output/differential_expression_cardelino-relax/
    Untracked:  output/donor_specific/
    Untracked:  output/line_info.tsv
    Untracked:  output/nvars_by_category_by_donor.tsv
    Untracked:  output/nvars_by_category_by_line.tsv
    Untracked:  output/variance_components/
    Untracked:  qolg_BIC.pdf
    Untracked:  references/
    Untracked:  reports/
    Untracked:  src/Rmd/DE_pathways_FTv62_callset_clones_pairwise_vs_base.unst_cells.carderelax.Rmd
    Untracked:  src/Rmd/Rplots.pdf
    Untracked:  src/Rmd/cell_assignment_cardelino-relax_template.Rmd
    Untracked:  tree.txt

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/f2e01ef23334fbb86d5874dacc97ddb1493358e5/analysis/analysis_for_joxm_cardelino-relax.Rmd" target="_blank">f2e01ef</a>
</td>
<td>
Davis McCarthy
</td>
<td>
2019-10-30
</td>
<td>
Couple of typo fixes
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/550176f57f3f33670ae89659f72e541ad8e8ba41/analysis/analysis_for_joxm_cardelino-relax.Rmd" target="_blank">550176f</a>
</td>
<td>
Davis McCarthy
</td>
<td>
2019-10-30
</td>
<td>
Updating analysis to reflect accepted ms
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries-and-data" class="section level2">
<h2>Load libraries and data</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1">knitr<span class="op">::</span>opts_chunk<span class="op">$</span><span class="kw">set</span>(<span class="dt">echo =</span> <span class="ot">TRUE</span>, <span class="dt">warning =</span> <span class="ot">FALSE</span>, <span class="dt">message =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb1-2" data-line-number="2">                      <span class="dt">fig.height =</span> <span class="dv">10</span>, <span class="dt">fig.width =</span> <span class="dv">14</span>)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw">dir.create</span>(<span class="st">&quot;figures/donor_specific&quot;</span>, <span class="dt">showWarnings =</span> <span class="ot">FALSE</span>, <span class="dt">recursive =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="kw">library</span>(tidyverse)</a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="kw">library</span>(scater)</a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="kw">library</span>(ggridges)</a>
<a class="sourceLine" id="cb1-7" data-line-number="7"><span class="kw">library</span>(GenomicRanges)</a>
<a class="sourceLine" id="cb1-8" data-line-number="8"><span class="kw">library</span>(RColorBrewer)</a>
<a class="sourceLine" id="cb1-9" data-line-number="9"><span class="kw">library</span>(edgeR)</a>
<a class="sourceLine" id="cb1-10" data-line-number="10"><span class="kw">library</span>(ggrepel)</a>
<a class="sourceLine" id="cb1-11" data-line-number="11"><span class="kw">library</span>(rlang)</a>
<a class="sourceLine" id="cb1-12" data-line-number="12"><span class="kw">library</span>(limma)</a>
<a class="sourceLine" id="cb1-13" data-line-number="13"><span class="kw">library</span>(org.Hs.eg.db)</a>
<a class="sourceLine" id="cb1-14" data-line-number="14"><span class="kw">library</span>(ggforce)</a>
<a class="sourceLine" id="cb1-15" data-line-number="15"><span class="kw">library</span>(cardelino)</a>
<a class="sourceLine" id="cb1-16" data-line-number="16"><span class="kw">library</span>(cowplot)</a>
<a class="sourceLine" id="cb1-17" data-line-number="17"><span class="kw">library</span>(IHW)</a>
<a class="sourceLine" id="cb1-18" data-line-number="18"><span class="kw">library</span>(viridis)</a>
<a class="sourceLine" id="cb1-19" data-line-number="19"><span class="kw">library</span>(ggthemes)</a>
<a class="sourceLine" id="cb1-20" data-line-number="20"><span class="kw">library</span>(superheat)</a>
<a class="sourceLine" id="cb1-21" data-line-number="21"><span class="kw">options</span>(<span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)</a></code></pre></div>
<p>Load MSigDB gene sets.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw">load</span>(<span class="st">&quot;data/human_c6_v5p2.rdata&quot;</span>)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw">load</span>(<span class="st">&quot;data/human_H_v5p2.rdata&quot;</span>)</a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="kw">load</span>(<span class="st">&quot;data/human_c2_v5p2.rdata&quot;</span>)</a></code></pre></div>
<p>Load VEP consequence information.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">vep_best &lt;-<span class="st"> </span><span class="kw">read_tsv</span>(<span class="st">&quot;data/exome-point-mutations/high-vs-low-exomes.v62.ft.alldonors-filt_lenient.all_filt_sites.vep_most_severe_csq.txt&quot;</span>)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="kw">colnames</span>(vep_best)[<span class="dv">1</span>] &lt;-<span class="st"> &quot;Uploaded_variation&quot;</span></a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="co">## deduplicate dataframe</span></a>
<a class="sourceLine" id="cb3-4" data-line-number="4">vep_best &lt;-<span class="st"> </span>vep_best[<span class="op">!</span><span class="kw">duplicated</span>(vep_best[[<span class="st">&quot;Uploaded_variation&quot;</span>]]),]</a></code></pre></div>
<p>Load somatic variant sites from whole-exome sequencing data.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">exome_sites &lt;-<span class="st"> </span><span class="kw">read_tsv</span>(<span class="st">&quot;data/exome-point-mutations/high-vs-low-exomes.v62.ft.filt_lenient-alldonors.txt.gz&quot;</span>,</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">    <span class="dt">col_types =</span> <span class="st">&quot;ciccdcciiiiccccccccddcdcll&quot;</span>, <span class="dt">comment =</span> <span class="st">&quot;#&quot;</span>,</a>
<a class="sourceLine" id="cb4-3" data-line-number="3">    <span class="dt">col_names =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb4-4" data-line-number="4">exome_sites &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(</a>
<a class="sourceLine" id="cb4-5" data-line-number="5">    exome_sites, </a>
<a class="sourceLine" id="cb4-6" data-line-number="6">    <span class="dt">chrom =</span> <span class="kw">paste0</span>(<span class="st">&quot;chr&quot;</span>, <span class="kw">gsub</span>(<span class="st">&quot;chr&quot;</span>, <span class="st">&quot;&quot;</span>, chrom)),</a>
<a class="sourceLine" id="cb4-7" data-line-number="7">    <span class="dt">var_id =</span> <span class="kw">paste0</span>(chrom, <span class="st">&quot;:&quot;</span>, pos, <span class="st">&quot;_&quot;</span>, ref, <span class="st">&quot;_&quot;</span>, alt))</a>
<a class="sourceLine" id="cb4-8" data-line-number="8"><span class="co">## deduplicate sites list</span></a>
<a class="sourceLine" id="cb4-9" data-line-number="9">exome_sites &lt;-<span class="st"> </span>exome_sites[<span class="op">!</span><span class="kw">duplicated</span>(exome_sites[[<span class="st">&quot;var_id&quot;</span>]]),]</a></code></pre></div>
<p>Add consequences to exome sites.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">vep_best[[<span class="st">&quot;var_id&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;chr&quot;</span>, vep_best[[<span class="st">&quot;Uploaded_variation&quot;</span>]])</a>
<a class="sourceLine" id="cb5-2" data-line-number="2">exome_sites &lt;-<span class="st"> </span><span class="kw">inner_join</span>(exome_sites, </a>
<a class="sourceLine" id="cb5-3" data-line-number="3">                           vep_best[, <span class="kw">c</span>(<span class="st">&quot;var_id&quot;</span>, <span class="st">&quot;Location&quot;</span>, <span class="st">&quot;Consequence&quot;</span>)], </a>
<a class="sourceLine" id="cb5-4" data-line-number="4">                           <span class="dt">by =</span> <span class="st">&quot;var_id&quot;</span>)</a></code></pre></div>
<p>Load cell-clone assignment results for this donor.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">cell_assign_joxm &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">file.path</span>(<span class="st">&quot;data/cell_assignment&quot;</span>, </a>
<a class="sourceLine" id="cb6-2" data-line-number="2">        <span class="kw">paste0</span>(<span class="st">&quot;cardelino_results_carderelax.joxm.filt_lenient.cell_coverage_sites.rds&quot;</span>)))</a></code></pre></div>
<p>Load SCE objects.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">params &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb7-2" data-line-number="2">params<span class="op">$</span>callset &lt;-<span class="st"> &quot;filt_lenient.cell_coverage_sites&quot;</span></a>
<a class="sourceLine" id="cb7-3" data-line-number="3">fls &lt;-<span class="st"> </span><span class="kw">list.files</span>(<span class="st">&quot;data/sces&quot;</span>)</a>
<a class="sourceLine" id="cb7-4" data-line-number="4">fls &lt;-<span class="st"> </span>fls[<span class="kw">grepl</span>(<span class="kw">paste0</span>(<span class="st">&quot;carderelax.&quot;</span>, params<span class="op">$</span>callset), fls)]</a>
<a class="sourceLine" id="cb7-5" data-line-number="5">donors &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;.*ce_([a-z]+)_with_clone_assignments_carderelax.*&quot;</span>, <span class="st">&quot;</span><span class="ch">\\</span><span class="st">1&quot;</span>, fls)</a>
<a class="sourceLine" id="cb7-6" data-line-number="6"></a>
<a class="sourceLine" id="cb7-7" data-line-number="7">sce_unst_list &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb7-8" data-line-number="8"><span class="cf">for</span> (don <span class="cf">in</span> donors) {</a>
<a class="sourceLine" id="cb7-9" data-line-number="9">    sce_unst_list[[don]] &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">file.path</span>(<span class="st">&quot;data/sces&quot;</span>,</a>
<a class="sourceLine" id="cb7-10" data-line-number="10">        <span class="kw">paste0</span>(<span class="st">&quot;sce_&quot;</span>, don, <span class="st">&quot;_with_clone_assignments_carderelax.&quot;</span>, </a>
<a class="sourceLine" id="cb7-11" data-line-number="11">               params<span class="op">$</span>callset, <span class="st">&quot;.rds&quot;</span>)))</a>
<a class="sourceLine" id="cb7-12" data-line-number="12">    <span class="kw">cat</span>(<span class="kw">paste</span>(<span class="st">&quot;reading&quot;</span>, don, <span class="st">&quot;:   &quot;</span>, <span class="kw">ncol</span>(sce_unst_list[[don]]), </a>
<a class="sourceLine" id="cb7-13" data-line-number="13">              <span class="st">&quot;unstimulated cells.</span><span class="ch">\n</span><span class="st">&quot;</span>))</a>
<a class="sourceLine" id="cb7-14" data-line-number="14">}</a></code></pre></div>
<pre><code>reading euts :    79 unstimulated cells.
reading fawm :    53 unstimulated cells.
reading feec :    75 unstimulated cells.
reading fikt :    39 unstimulated cells.
reading garx :    70 unstimulated cells.
reading gesg :    105 unstimulated cells.
reading heja :    50 unstimulated cells.
reading hipn :    62 unstimulated cells.
reading ieki :    58 unstimulated cells.
reading joxm :    79 unstimulated cells.
reading kuco :    48 unstimulated cells.
reading laey :    55 unstimulated cells.
reading lexy :    63 unstimulated cells.
reading naju :    44 unstimulated cells.
reading nusw :    60 unstimulated cells.
reading oaaz :    38 unstimulated cells.
reading oilg :    90 unstimulated cells.
reading pipw :    107 unstimulated cells.
reading puie :    41 unstimulated cells.
reading qayj :    97 unstimulated cells.
reading qolg :    36 unstimulated cells.
reading qonc :    58 unstimulated cells.
reading rozh :    91 unstimulated cells.
reading sehl :    30 unstimulated cells.
reading ualf :    89 unstimulated cells.
reading vass :    37 unstimulated cells.
reading vils :    37 unstimulated cells.
reading vuna :    71 unstimulated cells.
reading wahn :    82 unstimulated cells.
reading wetu :    77 unstimulated cells.
reading xugn :    35 unstimulated cells.
reading zoxy :    88 unstimulated cells.</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">assignments_lst &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="cf">for</span> (don <span class="cf">in</span> donors) {</a>
<a class="sourceLine" id="cb9-3" data-line-number="3">    assignments_lst[[don]] &lt;-<span class="st"> </span><span class="kw">as_tibble</span>(</a>
<a class="sourceLine" id="cb9-4" data-line-number="4">        <span class="kw">colData</span>(sce_unst_list[[don]])[, </a>
<a class="sourceLine" id="cb9-5" data-line-number="5">                                      <span class="kw">c</span>(<span class="st">&quot;donor_short_id&quot;</span>, <span class="st">&quot;highest_prob&quot;</span>, </a>
<a class="sourceLine" id="cb9-6" data-line-number="6">                                        <span class="st">&quot;assigned&quot;</span>, <span class="st">&quot;total_features&quot;</span>,</a>
<a class="sourceLine" id="cb9-7" data-line-number="7">                                        <span class="st">&quot;total_counts_endogenous&quot;</span>, <span class="st">&quot;num_processed&quot;</span>)])</a>
<a class="sourceLine" id="cb9-8" data-line-number="8">}</a>
<a class="sourceLine" id="cb9-9" data-line-number="9">assignments &lt;-<span class="st"> </span><span class="kw">do.call</span>(<span class="st">&quot;rbind&quot;</span>, assignments_lst)</a></code></pre></div>
<p>Load the SCE object for <code>joxm</code>.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">sce_joxm &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;data/sces/sce_joxm_with_clone_assignments_carderelax.filt_lenient.cell_coverage_sites.rds&quot;</span>)</a>
<a class="sourceLine" id="cb10-2" data-line-number="2">sce_joxm</a></code></pre></div>
<pre><code>class: SingleCellExperiment 
dim: 13225 79 
metadata(1): log.exprs.offset
assays(2): counts logcounts
rownames(13225): ENSG00000000003_TSPAN6 ENSG00000000419_DPM1 ...
  ERCC-00170_NA ERCC-00171_NA
rowData names(11): ensembl_transcript_id ensembl_gene_id ...
  is_feature_control high_var_gene
colnames(79): 22259_2#169 22259_2#173 ... 22666_2#70 22666_2#71
colData names(128): salmon_version samp_type ... nvars_cloneid
  clone_apk2
reducedDimNames(0):
spikeNames(1): ERCC</code></pre>
<p>We can check cell assignments for this donor.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="kw">table</span>(sce_joxm<span class="op">$</span>assigned)</a></code></pre></div>
<pre><code>
    clone1     clone2     clone3 unassigned 
        46         25          7          1 </code></pre>
<p>Load DE results (obtained using the <em>edgeR</em> quasi-likelihood F test and the camera method from the <em>limma</em> package).</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1">de_res &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;data/de_analysis_FTv62/carderelax.filt_lenient.cell_coverage_sites.de_results_unstimulated_cells.rds&quot;</span>)</a></code></pre></div>
</div>
<div id="tree-and-probability-heatmap" class="section level2">
<h2>Tree and probability heatmap</h2>
<p>We can plot the clonal tree inferred with <em>Canopy</em> for this donor along with the cell-clone assignment results from <em>cardelino</em>.</p>
<p>Plot tree (updated tree output by Cardelino) with assignment probability matrix.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1">fig_tree &lt;-<span class="st"> </span><span class="kw">plot_tree</span>(cell_assign_joxm<span class="op">$</span>full_tree, <span class="dt">orient =</span> <span class="st">&quot;v&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb15-2" data-line-number="2"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Clonal tree&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb15-3" data-line-number="3"><span class="st">    </span>cardelino<span class="op">:::</span><span class="kw">heatmap.theme</span>(<span class="dt">size =</span> <span class="dv">16</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb15-4" data-line-number="4"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.title.y =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">20</span>))</a>
<a class="sourceLine" id="cb15-5" data-line-number="5"></a>
<a class="sourceLine" id="cb15-6" data-line-number="6">prob_to_plot &lt;-<span class="st"> </span>cell_assign_joxm<span class="op">$</span>prob_mat[</a>
<a class="sourceLine" id="cb15-7" data-line-number="7">    <span class="kw">colnames</span>(sce_joxm)[sce_joxm<span class="op">$</span>well_condition <span class="op">==</span><span class="st"> &quot;unstimulated&quot;</span>], ]</a>
<a class="sourceLine" id="cb15-8" data-line-number="8">hc &lt;-<span class="st"> </span><span class="kw">hclust</span>(<span class="kw">dist</span>(prob_to_plot))</a>
<a class="sourceLine" id="cb15-9" data-line-number="9"></a>
<a class="sourceLine" id="cb15-10" data-line-number="10">clone_ids &lt;-<span class="st"> </span><span class="kw">colnames</span>(prob_to_plot)</a>
<a class="sourceLine" id="cb15-11" data-line-number="11">clone_frac &lt;-<span class="st"> </span><span class="kw">colMeans</span>(prob_to_plot[matrixStats<span class="op">::</span><span class="kw">rowMaxs</span>(prob_to_plot) <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.5</span>,])</a>
<a class="sourceLine" id="cb15-12" data-line-number="12">clone_perc &lt;-<span class="st"> </span><span class="kw">paste0</span>(clone_ids, <span class="st">&quot;: &quot;</span>, </a>
<a class="sourceLine" id="cb15-13" data-line-number="13">                          <span class="kw">round</span>(clone_frac<span class="op">*</span><span class="dv">100</span>, <span class="dt">digits =</span> <span class="dv">1</span>), <span class="st">&quot;%&quot;</span>)</a>
<a class="sourceLine" id="cb15-14" data-line-number="14"></a>
<a class="sourceLine" id="cb15-15" data-line-number="15"><span class="kw">colnames</span>(prob_to_plot) &lt;-<span class="st"> </span>clone_perc</a>
<a class="sourceLine" id="cb15-16" data-line-number="16">nba.m &lt;-<span class="st"> </span><span class="kw">as_data_frame</span>(prob_to_plot[hc<span class="op">$</span>order,]) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb15-17" data-line-number="17"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">cell =</span> <span class="kw">rownames</span>(prob_to_plot[hc<span class="op">$</span>order,])) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb15-18" data-line-number="18"><span class="st">    </span><span class="kw">gather</span>(<span class="dt">key =</span> <span class="st">&quot;clone&quot;</span>, <span class="dt">value =</span> <span class="st">&quot;prob&quot;</span>, <span class="op">-</span>cell)</a>
<a class="sourceLine" id="cb15-19" data-line-number="19">nba.m &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(nba.m, <span class="dt">cell =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb15-20" data-line-number="20">    cell, <span class="dt">levels =</span> <span class="kw">rownames</span>(prob_to_plot[hc<span class="op">$</span>order,])))</a>
<a class="sourceLine" id="cb15-21" data-line-number="21">fig_assign &lt;-<span class="st"> </span><span class="kw">ggplot</span>(nba.m, <span class="kw">aes</span>(clone, cell, <span class="dt">fill =</span> prob)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb15-22" data-line-number="22"><span class="st">    </span><span class="kw">geom_tile</span>(<span class="dt">show.legend =</span> <span class="ot">TRUE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb15-23" data-line-number="23"><span class="st">    </span><span class="co"># scale_fill_gradient(low = &quot;white&quot;, high = &quot;firebrick4&quot;,</span></a>
<a class="sourceLine" id="cb15-24" data-line-number="24"><span class="st">    </span><span class="co">#                     name = &quot;posterior probability of assignment&quot;) +</span></a>
<a class="sourceLine" id="cb15-25" data-line-number="25"><span class="st">    </span>scico<span class="op">::</span><span class="kw">scale_fill_scico</span>(<span class="dt">palette =</span> <span class="st">&quot;oleron&quot;</span>, <span class="dt">direction =</span> <span class="dv">1</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb15-26" data-line-number="26"><span class="st">    </span><span class="kw">ylab</span>(<span class="kw">paste</span>(<span class="st">&quot;Single cells&quot;</span>)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb15-27" data-line-number="27"><span class="st">    </span>cardelino<span class="op">:::</span><span class="kw">heatmap.theme</span>(<span class="dt">size =</span> <span class="dv">16</span>) <span class="op">+</span><span class="st"> </span><span class="co">#cardelino:::pub.theme() +</span></a>
<a class="sourceLine" id="cb15-28" data-line-number="28"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.title.y =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">20</span>), <span class="dt">legend.position =</span> <span class="st">&quot;bottom&quot;</span>,</a>
<a class="sourceLine" id="cb15-29" data-line-number="29">          <span class="dt">legend.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">12</span>), <span class="dt">legend.key.size =</span> <span class="kw">unit</span>(<span class="fl">0.05</span>, <span class="st">&quot;npc&quot;</span>))</a>
<a class="sourceLine" id="cb15-30" data-line-number="30"></a>
<a class="sourceLine" id="cb15-31" data-line-number="31"><span class="kw">plot_grid</span>(fig_tree, fig_assign, <span class="dt">nrow =</span> <span class="dv">2</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.46</span>, <span class="fl">0.52</span>))</a></code></pre></div>
<p><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/plot-tree-1.png" width="1344" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_tree_probmat.png&quot;</span>, <span class="dt">height =</span> <span class="dv">10</span>, <span class="dt">width =</span> <span class="fl">7.5</span>)</a>
<a class="sourceLine" id="cb16-2" data-line-number="2"><span class="kw">ggsave</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_tree_probmat.pdf&quot;</span>, <span class="dt">height =</span> <span class="dv">10</span>, <span class="dt">width =</span> <span class="fl">7.5</span>)</a>
<a class="sourceLine" id="cb16-3" data-line-number="3"><span class="kw">ggsave</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_tree_probmat.svg&quot;</span>, <span class="dt">height =</span> <span class="dv">10</span>, <span class="dt">width =</span> <span class="fl">7.5</span>)</a>
<a class="sourceLine" id="cb16-4" data-line-number="4"></a>
<a class="sourceLine" id="cb16-5" data-line-number="5"><span class="kw">ggsave</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_tree_probmat_wide.png&quot;</span>, <span class="dt">height =</span> <span class="dv">9</span>, <span class="dt">width =</span> <span class="dv">10</span>)</a>
<a class="sourceLine" id="cb16-6" data-line-number="6"><span class="kw">ggsave</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_tree_probmat_wide.pdf&quot;</span>, <span class="dt">height =</span> <span class="dv">9</span>, <span class="dt">width =</span> <span class="dv">10</span>)</a>
<a class="sourceLine" id="cb16-7" data-line-number="7"><span class="kw">ggsave</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_tree_probmat_wide.svg&quot;</span>, <span class="dt">height =</span> <span class="dv">9</span>, <span class="dt">width =</span> <span class="dv">10</span>)</a></code></pre></div>
</div>
<div id="analysis-of-direct-effects-of-variants-on-gene-expression" class="section level2">
<h2>Analysis of direct effects of variants on gene expression</h2>
<p>Load SCE object and cell assignment results.</p>
<p>First, plot the VEP consequences of somatic variants in this donor used to infer the clonal tree.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1">joxm_config &lt;-<span class="st"> </span><span class="kw">as_data_frame</span>(cell_assign_joxm<span class="op">$</span>full_tree<span class="op">$</span>Z)</a>
<a class="sourceLine" id="cb17-2" data-line-number="2">joxm_config[[<span class="st">&quot;var_id&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">rownames</span>(cell_assign_joxm<span class="op">$</span>full_tree<span class="op">$</span>Z)</a>
<a class="sourceLine" id="cb17-3" data-line-number="3">exome_sites_joxm &lt;-<span class="st"> </span><span class="kw">inner_join</span>(exome_sites, joxm_config)</a>
<a class="sourceLine" id="cb17-4" data-line-number="4">exome_sites_joxm[[<span class="st">&quot;clone_presence&quot;</span>]] &lt;-<span class="st"> &quot;&quot;</span></a>
<a class="sourceLine" id="cb17-5" data-line-number="5"><span class="cf">for</span> (cln <span class="cf">in</span> <span class="kw">colnames</span>(cell_assign_joxm<span class="op">$</span>full_tree<span class="op">$</span>Z)[<span class="op">-</span><span class="dv">1</span>]) {</a>
<a class="sourceLine" id="cb17-6" data-line-number="6">    exome_sites_joxm[[<span class="st">&quot;clone_presence&quot;</span>]][</a>
<a class="sourceLine" id="cb17-7" data-line-number="7">        <span class="kw">as.logical</span>(exome_sites_joxm[[cln]])] &lt;-<span class="st"> </span><span class="kw">paste</span>(</a>
<a class="sourceLine" id="cb17-8" data-line-number="8">            exome_sites_joxm[[<span class="st">&quot;clone_presence&quot;</span>]][</a>
<a class="sourceLine" id="cb17-9" data-line-number="9">                <span class="kw">as.logical</span>(exome_sites_joxm[[cln]])], cln, <span class="dt">sep =</span> <span class="st">&quot;&amp;&quot;</span>)</a>
<a class="sourceLine" id="cb17-10" data-line-number="10">}</a>
<a class="sourceLine" id="cb17-11" data-line-number="11">exome_sites_joxm[[<span class="st">&quot;clone_presence&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;^&amp;&quot;</span>, <span class="st">&quot;&quot;</span>,</a>
<a class="sourceLine" id="cb17-12" data-line-number="12">                                        exome_sites_joxm[[<span class="st">&quot;clone_presence&quot;</span>]])</a>
<a class="sourceLine" id="cb17-13" data-line-number="13"></a>
<a class="sourceLine" id="cb17-14" data-line-number="14">exome_sites_joxm <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(Consequence, clone_presence) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb17-15" data-line-number="15"><span class="st">    </span><span class="kw">summarise</span>(<span class="dt">n_vars =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb17-16" data-line-number="16"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> n_vars, <span class="dt">y =</span> <span class="kw">reorder</span>(Consequence, n_vars, max), </a>
<a class="sourceLine" id="cb17-17" data-line-number="17">       <span class="dt">colour =</span> <span class="kw">reorder</span>(Consequence, n_vars, max))) <span class="op">+</span></a>
<a class="sourceLine" id="cb17-18" data-line-number="18"><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">5</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb17-19" data-line-number="19"><span class="st">    </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">x =</span> <span class="dv">0</span>, <span class="dt">y =</span> Consequence, <span class="dt">xend =</span> n_vars, <span class="dt">yend =</span> Consequence)) <span class="op">+</span></a>
<a class="sourceLine" id="cb17-20" data-line-number="20"><span class="st">    </span><span class="kw">facet_wrap</span>(<span class="op">~</span>clone_presence) <span class="op">+</span></a>
<a class="sourceLine" id="cb17-21" data-line-number="21"><span class="co">#    scale_color_brewer(palette = &quot;Set2&quot;) +</span></a>
<a class="sourceLine" id="cb17-22" data-line-number="22"><span class="st">    </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> <span class="kw">colorRampPalette</span>(<span class="kw">brewer.pal</span>(<span class="dv">8</span>, <span class="st">&quot;Accent&quot;</span>))(<span class="dv">12</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb17-23" data-line-number="23"><span class="st">    </span><span class="kw">guides</span>(<span class="dt">colour =</span> <span class="ot">FALSE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb17-24" data-line-number="24"><span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;joxm clone tagging variants by consequence class&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb17-25" data-line-number="25"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;number of variants&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;consequence&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb17-26" data-line-number="26"><span class="st">    </span><span class="kw">theme_bw</span>(<span class="dv">16</span>)</a></code></pre></div>
<p><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/load-sce-canopy-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>Look at expression of genes with mutations.</p>
<p>Organise data for analysis.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1"><span class="co">## filter out any remaining ERCC genes</span></a>
<a class="sourceLine" id="cb18-2" data-line-number="2">sce_joxm &lt;-<span class="st"> </span>sce_joxm[<span class="op">!</span><span class="kw">rowData</span>(sce_joxm)<span class="op">$</span>is_feature_control,]</a>
<a class="sourceLine" id="cb18-3" data-line-number="3">sce_joxm_gr &lt;-<span class="st"> </span><span class="kw">makeGRangesFromDataFrame</span>(<span class="kw">rowData</span>(sce_joxm),</a>
<a class="sourceLine" id="cb18-4" data-line-number="4">                                   <span class="dt">start.field =</span> <span class="st">&quot;start_position&quot;</span>,</a>
<a class="sourceLine" id="cb18-5" data-line-number="5">                                   <span class="dt">end.field =</span> <span class="st">&quot;end_position&quot;</span>,</a>
<a class="sourceLine" id="cb18-6" data-line-number="6">                                   <span class="dt">keep.extra.columns =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb18-7" data-line-number="7">exome_sites_joxm[[<span class="st">&quot;chrom&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;chr&quot;</span>, <span class="st">&quot;&quot;</span>, exome_sites_joxm[[<span class="st">&quot;chrom&quot;</span>]])</a>
<a class="sourceLine" id="cb18-8" data-line-number="8">exome_sites_joxm_gr &lt;-<span class="st"> </span><span class="kw">makeGRangesFromDataFrame</span>(exome_sites_joxm,</a>
<a class="sourceLine" id="cb18-9" data-line-number="9">                                   <span class="dt">start.field =</span> <span class="st">&quot;pos&quot;</span>,</a>
<a class="sourceLine" id="cb18-10" data-line-number="10">                                   <span class="dt">end.field =</span> <span class="st">&quot;pos&quot;</span>,</a>
<a class="sourceLine" id="cb18-11" data-line-number="11">                                   <span class="dt">keep.extra.columns =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb18-12" data-line-number="12"><span class="co"># find overlaps</span></a>
<a class="sourceLine" id="cb18-13" data-line-number="13">ov_joxm &lt;-<span class="st"> </span><span class="kw">findOverlaps</span>(sce_joxm_gr, exome_sites_joxm_gr)</a>
<a class="sourceLine" id="cb18-14" data-line-number="14">tmp_cols &lt;-<span class="st"> </span><span class="kw">colnames</span>(<span class="kw">mcols</span>(exome_sites_joxm_gr))</a>
<a class="sourceLine" id="cb18-15" data-line-number="15">tmp_cols &lt;-<span class="st"> </span>tmp_cols[<span class="kw">grepl</span>(<span class="st">&quot;clone&quot;</span>, tmp_cols)]</a>
<a class="sourceLine" id="cb18-16" data-line-number="16">tmp_cols &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Consequence&quot;</span>, tmp_cols, <span class="st">&quot;var_id&quot;</span>)</a>
<a class="sourceLine" id="cb18-17" data-line-number="17">mut_genes_exprs_joxm &lt;-<span class="st"> </span><span class="kw">logcounts</span>(sce_joxm)[<span class="kw">queryHits</span>(ov_joxm),]</a>
<a class="sourceLine" id="cb18-18" data-line-number="18">mut_genes_df_joxm &lt;-<span class="st"> </span><span class="kw">as_data_frame</span>(mut_genes_exprs_joxm)</a>
<a class="sourceLine" id="cb18-19" data-line-number="19">mut_genes_df_joxm[[<span class="st">&quot;gene&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">rownames</span>(mut_genes_exprs_joxm)</a>
<a class="sourceLine" id="cb18-20" data-line-number="20">mut_genes_df_joxm &lt;-<span class="st"> </span><span class="kw">bind_cols</span>(mut_genes_df_joxm,</a>
<a class="sourceLine" id="cb18-21" data-line-number="21">                          <span class="kw">as_data_frame</span>(</a>
<a class="sourceLine" id="cb18-22" data-line-number="22">                              exome_sites_joxm_gr[</a>
<a class="sourceLine" id="cb18-23" data-line-number="23">                                  <span class="kw">subjectHits</span>(ov_joxm)])[, tmp_cols]</a>
<a class="sourceLine" id="cb18-24" data-line-number="24">)</a></code></pre></div>
<div id="de-comparing-mutated-clone-to-all-other-clones" class="section level3">
<h3>DE comparing mutated clone to all other clones</h3>
<p>Get DE results comparing mutated clone to all unmutated clones.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1">cell_assign_list &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb19-2" data-line-number="2"><span class="cf">for</span> (don <span class="cf">in</span> donors) {</a>
<a class="sourceLine" id="cb19-3" data-line-number="3">    cell_assign_list[[don]] &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">file.path</span>(<span class="st">&quot;data/cell_assignment&quot;</span>, </a>
<a class="sourceLine" id="cb19-4" data-line-number="4">        <span class="kw">paste0</span>(<span class="st">&quot;cardelino_results.&quot;</span>, don, <span class="st">&quot;.&quot;</span>, params<span class="op">$</span>callset, <span class="st">&quot;.rds&quot;</span>)))</a>
<a class="sourceLine" id="cb19-5" data-line-number="5">    <span class="kw">cat</span>(<span class="kw">paste</span>(<span class="st">&quot;reading&quot;</span>, don, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>))</a>
<a class="sourceLine" id="cb19-6" data-line-number="6">}   </a></code></pre></div>
<pre><code>reading euts 
reading fawm 
reading feec 
reading fikt 
reading garx 
reading gesg 
reading heja 
reading hipn 
reading ieki 
reading joxm 
reading kuco 
reading laey 
reading lexy 
reading naju 
reading nusw 
reading oaaz 
reading oilg 
reading pipw 
reading puie 
reading qayj 
reading qolg 
reading qonc 
reading rozh 
reading sehl 
reading ualf 
reading vass 
reading vils 
reading vuna 
reading wahn 
reading wetu 
reading xugn 
reading zoxy </code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1">get_sites_by_donor &lt;-<span class="st"> </span><span class="cf">function</span>(sites_df, sce_list, assign_list) {</a>
<a class="sourceLine" id="cb21-2" data-line-number="2">    <span class="cf">if</span> (<span class="op">!</span><span class="kw">identical</span>(<span class="kw">sort</span>(<span class="kw">names</span>(sce_list)), <span class="kw">sort</span>(<span class="kw">names</span>(assign_list))))</a>
<a class="sourceLine" id="cb21-3" data-line-number="3">        <span class="kw">stop</span>(<span class="st">&quot;donors do not match between sce_list and assign_list.&quot;</span>)</a>
<a class="sourceLine" id="cb21-4" data-line-number="4">    sites_by_donor &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb21-5" data-line-number="5">    <span class="cf">for</span> (don <span class="cf">in</span> <span class="kw">names</span>(sce_list)) {</a>
<a class="sourceLine" id="cb21-6" data-line-number="6">        config &lt;-<span class="st"> </span><span class="kw">as_data_frame</span>(assign_list[[don]]<span class="op">$</span>tree<span class="op">$</span>Z)</a>
<a class="sourceLine" id="cb21-7" data-line-number="7">        config[[<span class="st">&quot;var_id&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">rownames</span>(assign_list[[don]]<span class="op">$</span>tree<span class="op">$</span>Z)</a>
<a class="sourceLine" id="cb21-8" data-line-number="8">        sites_donor &lt;-<span class="st"> </span><span class="kw">inner_join</span>(sites_df, config)</a>
<a class="sourceLine" id="cb21-9" data-line-number="9">        sites_donor[[<span class="st">&quot;clone_presence&quot;</span>]] &lt;-<span class="st"> &quot;&quot;</span></a>
<a class="sourceLine" id="cb21-10" data-line-number="10">        <span class="cf">for</span> (cln <span class="cf">in</span> <span class="kw">colnames</span>(assign_list[[don]]<span class="op">$</span>tree<span class="op">$</span>Z)[<span class="op">-</span><span class="dv">1</span>]) {</a>
<a class="sourceLine" id="cb21-11" data-line-number="11">            sites_donor[[<span class="st">&quot;clone_presence&quot;</span>]][</a>
<a class="sourceLine" id="cb21-12" data-line-number="12">                <span class="kw">as.logical</span>(sites_donor[[cln]])] &lt;-<span class="st"> </span><span class="kw">paste</span>(</a>
<a class="sourceLine" id="cb21-13" data-line-number="13">                    sites_donor[[<span class="st">&quot;clone_presence&quot;</span>]][</a>
<a class="sourceLine" id="cb21-14" data-line-number="14">                        <span class="kw">as.logical</span>(sites_donor[[cln]])], cln, <span class="dt">sep =</span> <span class="st">&quot;&amp;&quot;</span>)</a>
<a class="sourceLine" id="cb21-15" data-line-number="15">        }</a>
<a class="sourceLine" id="cb21-16" data-line-number="16">        sites_donor[[<span class="st">&quot;clone_presence&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;^&amp;&quot;</span>, <span class="st">&quot;&quot;</span>,</a>
<a class="sourceLine" id="cb21-17" data-line-number="17">                                                sites_donor[[<span class="st">&quot;clone_presence&quot;</span>]])</a>
<a class="sourceLine" id="cb21-18" data-line-number="18">        <span class="co">## drop config columns as these won&#39;t match up between donors</span></a>
<a class="sourceLine" id="cb21-19" data-line-number="19">        keep_cols &lt;-<span class="st"> </span><span class="kw">grep</span>(<span class="st">&quot;^clone[0-9]$&quot;</span>, <span class="kw">colnames</span>(sites_donor), <span class="dt">invert =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb21-20" data-line-number="20">        sites_by_donor[[don]] &lt;-<span class="st"> </span>sites_donor[, keep_cols]</a>
<a class="sourceLine" id="cb21-21" data-line-number="21">    }</a>
<a class="sourceLine" id="cb21-22" data-line-number="22">    <span class="kw">do.call</span>(<span class="st">&quot;bind_rows&quot;</span>, sites_by_donor)</a>
<a class="sourceLine" id="cb21-23" data-line-number="23">}</a>
<a class="sourceLine" id="cb21-24" data-line-number="24"></a>
<a class="sourceLine" id="cb21-25" data-line-number="25">sites_by_donor &lt;-<span class="st"> </span><span class="kw">get_sites_by_donor</span>(exome_sites, sce_unst_list, cell_assign_list)</a>
<a class="sourceLine" id="cb21-26" data-line-number="26"></a>
<a class="sourceLine" id="cb21-27" data-line-number="27">sites_by_donor_gr &lt;-<span class="st"> </span><span class="kw">makeGRangesFromDataFrame</span>(sites_by_donor,</a>
<a class="sourceLine" id="cb21-28" data-line-number="28">                                              <span class="dt">start.field =</span> <span class="st">&quot;pos&quot;</span>,</a>
<a class="sourceLine" id="cb21-29" data-line-number="29">                                              <span class="dt">end.field =</span> <span class="st">&quot;pos&quot;</span>,</a>
<a class="sourceLine" id="cb21-30" data-line-number="30">                                              <span class="dt">keep.extra.columns =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb21-31" data-line-number="31"></a>
<a class="sourceLine" id="cb21-32" data-line-number="32"><span class="co">## run DE for mutated cells vs unmutated cells using existing DE results</span></a>
<a class="sourceLine" id="cb21-33" data-line-number="33"><span class="co">## filter out any remaining ERCC genes</span></a>
<a class="sourceLine" id="cb21-34" data-line-number="34"><span class="cf">for</span> (don <span class="cf">in</span> <span class="kw">names</span>(de_res[[<span class="st">&quot;sce_list_unst&quot;</span>]]))</a>
<a class="sourceLine" id="cb21-35" data-line-number="35">    de_res[[<span class="st">&quot;sce_list_unst&quot;</span>]][[don]] &lt;-<span class="st"> </span>de_res[[<span class="st">&quot;sce_list_unst&quot;</span>]][[don]][</a>
<a class="sourceLine" id="cb21-36" data-line-number="36">        <span class="op">!</span><span class="kw">rowData</span>(de_res[[<span class="st">&quot;sce_list_unst&quot;</span>]][[don]])<span class="op">$</span>is_feature_control,]  </a>
<a class="sourceLine" id="cb21-37" data-line-number="37">sce_de_list_gr &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb21-38" data-line-number="38"><span class="cf">for</span> (don <span class="cf">in</span> <span class="kw">names</span>(de_res[[<span class="st">&quot;sce_list_unst&quot;</span>]])) {</a>
<a class="sourceLine" id="cb21-39" data-line-number="39">    sce_de_list_gr[[don]] &lt;-<span class="st"> </span><span class="kw">makeGRangesFromDataFrame</span>(</a>
<a class="sourceLine" id="cb21-40" data-line-number="40">        <span class="kw">rowData</span>(de_res[[<span class="st">&quot;sce_list_unst&quot;</span>]][[don]]),</a>
<a class="sourceLine" id="cb21-41" data-line-number="41">        <span class="dt">start.field =</span> <span class="st">&quot;start_position&quot;</span>,</a>
<a class="sourceLine" id="cb21-42" data-line-number="42">        <span class="dt">end.field =</span> <span class="st">&quot;end_position&quot;</span>,</a>
<a class="sourceLine" id="cb21-43" data-line-number="43">        <span class="dt">keep.extra.columns =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb21-44" data-line-number="44">    <span class="kw">seqlevelsStyle</span>(sce_de_list_gr[[don]]) &lt;-<span class="st"> &quot;UCSC&quot;</span></a>
<a class="sourceLine" id="cb21-45" data-line-number="45">}</a>
<a class="sourceLine" id="cb21-46" data-line-number="46">mut_genes_df_allcells_list &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb21-47" data-line-number="47"><span class="cf">for</span> (don <span class="cf">in</span> <span class="kw">names</span>(de_res[[<span class="st">&quot;sce_list_unst&quot;</span>]])) {</a>
<a class="sourceLine" id="cb21-48" data-line-number="48">    <span class="kw">cat</span>(<span class="st">&quot;working on &quot;</span>, don, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)</a>
<a class="sourceLine" id="cb21-49" data-line-number="49">    sites_tmp &lt;-<span class="st"> </span>sites_by_donor_gr[sites_by_donor_gr<span class="op">$</span>donor_short_id <span class="op">==</span><span class="st"> </span>don]</a>
<a class="sourceLine" id="cb21-50" data-line-number="50">    ov_tmp &lt;-<span class="st"> </span><span class="kw">findOverlaps</span>(sce_de_list_gr[[don]], sites_tmp)</a>
<a class="sourceLine" id="cb21-51" data-line-number="51">    sce_tmp &lt;-<span class="st"> </span>de_res[[<span class="st">&quot;sce_list_unst&quot;</span>]][[don]][<span class="kw">queryHits</span>(ov_tmp),]</a>
<a class="sourceLine" id="cb21-52" data-line-number="52">    sites_tmp &lt;-<span class="st"> </span>sites_tmp[<span class="kw">subjectHits</span>(ov_tmp)]</a>
<a class="sourceLine" id="cb21-53" data-line-number="53">    sites_tmp<span class="op">$</span>gene &lt;-<span class="st"> </span><span class="kw">rownames</span>(sce_tmp)</a>
<a class="sourceLine" id="cb21-54" data-line-number="54">    dge_tmp &lt;-<span class="st"> </span>de_res[[<span class="st">&quot;dge_list&quot;</span>]][[don]]</a>
<a class="sourceLine" id="cb21-55" data-line-number="55">    dge_tmp &lt;-<span class="st"> </span>dge_tmp[<span class="kw">intersect</span>(<span class="kw">rownames</span>(dge_tmp), sites_tmp<span class="op">$</span>gene),]</a>
<a class="sourceLine" id="cb21-56" data-line-number="56">    base_design &lt;-<span class="st"> </span>dge_tmp<span class="op">$</span>design[, <span class="op">!</span><span class="kw">grepl</span>(<span class="st">&quot;assigned&quot;</span>, <span class="kw">colnames</span>(dge_tmp<span class="op">$</span>design))]</a>
<a class="sourceLine" id="cb21-57" data-line-number="57">    de_tbl_tmp &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">donor =</span> don,</a>
<a class="sourceLine" id="cb21-58" data-line-number="58">                             <span class="dt">gene =</span> sites_tmp<span class="op">$</span>gene, </a>
<a class="sourceLine" id="cb21-59" data-line-number="59">                             <span class="dt">hgnc_symbol =</span> <span class="kw">gsub</span>(<span class="st">&quot;.*_&quot;</span>, <span class="st">&quot;&quot;</span>, sites_tmp<span class="op">$</span>gene),</a>
<a class="sourceLine" id="cb21-60" data-line-number="60">                             <span class="dt">ensembl_gene_id =</span> <span class="kw">gsub</span>(<span class="st">&quot;_.*&quot;</span>, <span class="st">&quot;&quot;</span>, sites_tmp<span class="op">$</span>gene),</a>
<a class="sourceLine" id="cb21-61" data-line-number="61">                             <span class="dt">var_id =</span> sites_tmp<span class="op">$</span>var_id,</a>
<a class="sourceLine" id="cb21-62" data-line-number="62">                             <span class="dt">location =</span> sites_tmp<span class="op">$</span>Location,</a>
<a class="sourceLine" id="cb21-63" data-line-number="63">                             <span class="dt">consequence =</span> sites_tmp<span class="op">$</span>Consequence,</a>
<a class="sourceLine" id="cb21-64" data-line-number="64">                             <span class="dt">clone_presence =</span> sites_tmp<span class="op">$</span>clone_presence,</a>
<a class="sourceLine" id="cb21-65" data-line-number="65">                             <span class="dt">logFC =</span> <span class="ot">NA</span>, <span class="dt">logCPM =</span> <span class="ot">NA</span>, <span class="dt">F =</span> <span class="ot">NA</span>, <span class="dt">PValue =</span> <span class="ot">NA</span>,</a>
<a class="sourceLine" id="cb21-66" data-line-number="66">                             <span class="dt">comment =</span> <span class="st">&quot;&quot;</span>)</a>
<a class="sourceLine" id="cb21-67" data-line-number="67">    <span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">seq_len</span>(<span class="kw">length</span>(sites_tmp))) {</a>
<a class="sourceLine" id="cb21-68" data-line-number="68">        clones_tmp &lt;-<span class="st"> </span><span class="kw">strsplit</span>(sites_tmp<span class="op">$</span>clone_presence[i], <span class="dt">split =</span> <span class="st">&quot;&amp;&quot;</span>)[[<span class="dv">1</span>]]</a>
<a class="sourceLine" id="cb21-69" data-line-number="69">        mutatedclone &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(sce_tmp<span class="op">$</span>assigned <span class="op">%in%</span><span class="st"> </span>clones_tmp)</a>
<a class="sourceLine" id="cb21-70" data-line-number="70">        dsgn_tmp &lt;-<span class="st"> </span><span class="kw">cbind</span>(base_design, <span class="kw">data.frame</span>(mutatedclone))</a>
<a class="sourceLine" id="cb21-71" data-line-number="71">        <span class="cf">if</span> (sites_tmp<span class="op">$</span>gene[i] <span class="op">%in%</span><span class="st"> </span><span class="kw">rownames</span>(dge_tmp) <span class="op">&amp;&amp;</span><span class="st"> </span><span class="kw">is.fullrank</span>(dsgn_tmp)) {</a>
<a class="sourceLine" id="cb21-72" data-line-number="72">            qlfit_tmp &lt;-<span class="st"> </span><span class="kw">glmQLFit</span>(dge_tmp[sites_tmp<span class="op">$</span>gene[i],], dsgn_tmp)</a>
<a class="sourceLine" id="cb21-73" data-line-number="73">            de_tmp &lt;-<span class="st"> </span><span class="kw">glmQLFTest</span>(qlfit_tmp, <span class="dt">coef =</span> <span class="kw">ncol</span>(dsgn_tmp))</a>
<a class="sourceLine" id="cb21-74" data-line-number="74">            de_tbl_tmp<span class="op">$</span>logFC[i] &lt;-<span class="st"> </span>de_tmp<span class="op">$</span>table<span class="op">$</span>logFC</a>
<a class="sourceLine" id="cb21-75" data-line-number="75">            de_tbl_tmp<span class="op">$</span>logCPM[i] &lt;-<span class="st"> </span>de_tmp<span class="op">$</span>table<span class="op">$</span>logCPM</a>
<a class="sourceLine" id="cb21-76" data-line-number="76">            de_tbl_tmp<span class="op">$</span>F[i] &lt;-<span class="st"> </span>de_tmp<span class="op">$</span>table<span class="op">$</span>F</a>
<a class="sourceLine" id="cb21-77" data-line-number="77">            de_tbl_tmp<span class="op">$</span>PValue[i] &lt;-<span class="st"> </span>de_tmp<span class="op">$</span>table<span class="op">$</span>PValue</a>
<a class="sourceLine" id="cb21-78" data-line-number="78">        }</a>
<a class="sourceLine" id="cb21-79" data-line-number="79">        <span class="cf">if</span> (<span class="op">!</span>(sites_tmp<span class="op">$</span>gene[i] <span class="op">%in%</span><span class="st"> </span><span class="kw">rownames</span>(dge_tmp)))</a>
<a class="sourceLine" id="cb21-80" data-line-number="80">            de_tbl_tmp<span class="op">$</span>comment[i] &lt;-<span class="st"> &quot;gene did not pass DE filters&quot;</span></a>
<a class="sourceLine" id="cb21-81" data-line-number="81">        <span class="cf">if</span> (<span class="op">!</span><span class="kw">is.fullrank</span>(dsgn_tmp))</a>
<a class="sourceLine" id="cb21-82" data-line-number="82">            de_tbl_tmp<span class="op">$</span>comment[i] &lt;-<span class="st"> &quot;insufficient cells assigned to clone&quot;</span></a>
<a class="sourceLine" id="cb21-83" data-line-number="83">    }</a>
<a class="sourceLine" id="cb21-84" data-line-number="84">    mut_genes_df_allcells_list[[don]] &lt;-<span class="st"> </span>de_tbl_tmp</a>
<a class="sourceLine" id="cb21-85" data-line-number="85">}</a></code></pre></div>
<pre><code>working on  euts 
working on  fawm 
working on  feec 
working on  fikt 
working on  garx 
working on  gesg 
working on  heja 
working on  hipn 
working on  ieki 
working on  joxm 
working on  kuco 
working on  laey 
working on  lexy 
working on  naju 
working on  nusw 
working on  oaaz 
working on  oilg 
working on  pipw 
working on  puie 
working on  qayj 
working on  qolg 
working on  qonc 
working on  rozh 
working on  sehl 
working on  ualf 
working on  vass 
working on  vuna 
working on  wahn 
working on  wetu 
working on  xugn 
working on  zoxy </code></pre>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1">mut_genes_df_allcells &lt;-<span class="st"> </span><span class="kw">do.call</span>(<span class="st">&quot;bind_rows&quot;</span>, mut_genes_df_allcells_list)</a>
<a class="sourceLine" id="cb23-2" data-line-number="2"><span class="co">## add FDRs for genes tested here for DE</span></a>
<a class="sourceLine" id="cb23-3" data-line-number="3">ihw_res_all &lt;-<span class="st"> </span><span class="kw">ihw</span>(PValue <span class="op">~</span><span class="st"> </span>logCPM, <span class="dt">data =</span> mut_genes_df_allcells, <span class="dt">alpha =</span> <span class="fl">0.2</span>)</a>
<a class="sourceLine" id="cb23-4" data-line-number="4">mut_genes_df_allcells<span class="op">$</span>FDR &lt;-<span class="st"> </span><span class="kw">adj_pvalues</span>(ihw_res_all)</a>
<a class="sourceLine" id="cb23-5" data-line-number="5"><span class="co">## add simplified consequence categories</span></a>
<a class="sourceLine" id="cb23-6" data-line-number="6">mut_genes_df_allcells<span class="op">$</span>consequence_simplified &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb23-7" data-line-number="7"><span class="st">    </span>mut_genes_df_allcells<span class="op">$</span>consequence</a>
<a class="sourceLine" id="cb23-8" data-line-number="8">mut_genes_df_allcells<span class="op">$</span>consequence_simplified[</a>
<a class="sourceLine" id="cb23-9" data-line-number="9">    mut_genes_df_allcells<span class="op">$</span>consequence_simplified <span class="op">%in%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb23-10" data-line-number="10"><span class="st">        </span><span class="kw">c</span>(<span class="st">&quot;stop_retained_variant&quot;</span>, <span class="st">&quot;start_lost&quot;</span>, <span class="st">&quot;stop_lost&quot;</span>, <span class="st">&quot;stop_gained&quot;</span>)] &lt;-<span class="st"> &quot;nonsense&quot;</span></a>
<a class="sourceLine" id="cb23-11" data-line-number="11">mut_genes_df_allcells<span class="op">$</span>consequence_simplified[</a>
<a class="sourceLine" id="cb23-12" data-line-number="12">    mut_genes_df_allcells<span class="op">$</span>consequence_simplified <span class="op">%in%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb23-13" data-line-number="13"><span class="st">        </span><span class="kw">c</span>(<span class="st">&quot;splice_donor_variant&quot;</span>, <span class="st">&quot;splice_acceptor_variant&quot;</span>, <span class="st">&quot;splice_region_variant&quot;</span>)] &lt;-<span class="st"> &quot;splicing&quot;</span></a>
<a class="sourceLine" id="cb23-14" data-line-number="14"><span class="co"># table(mut_genes_df_allcells$consequence_simplified)</span></a>
<a class="sourceLine" id="cb23-15" data-line-number="15"><span class="co"># dplyr::arrange(mut_genes_df_allcells, FDR) %&gt;% dplyr::select(-location) %&gt;% head(n = 20)</span></a></code></pre></div>
<p>For just the donor <code>joxm</code>.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1">tmp4 &lt;-<span class="st"> </span>mut_genes_df_allcells <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb24-2" data-line-number="2"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(logFC), donor <span class="op">==</span><span class="st"> &quot;joxm&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb24-3" data-line-number="3"><span class="st">    </span><span class="kw">group_by</span>(consequence_simplified) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb24-4" data-line-number="4"><span class="st">    </span><span class="kw">summarise</span>(<span class="dt">med =</span> <span class="kw">median</span>(logFC, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>),</a>
<a class="sourceLine" id="cb24-5" data-line-number="5">              <span class="dt">nvars =</span> <span class="kw">n</span>())</a>
<a class="sourceLine" id="cb24-6" data-line-number="6">tmp4</a></code></pre></div>
<pre><code># A tibble: 6 x 3
  consequence_simplified                 med nvars
  &lt;chr&gt;                                &lt;dbl&gt; &lt;int&gt;
1 5_prime_UTR_variant                 1.02       1
2 intron_variant                     -0.627      2
3 missense_variant                    0.266     70
4 non_coding_transcript_exon_variant  0.501      3
5 splicing                           -1.78       3
6 synonymous_variant                  0.0835    36</code></pre>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" data-line-number="1">df_to_plot &lt;-<span class="st"> </span>mut_genes_df_allcells <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb26-2" data-line-number="2"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(logFC), donor <span class="op">==</span><span class="st"> &quot;joxm&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb26-3" data-line-number="3"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(</a>
<a class="sourceLine" id="cb26-4" data-line-number="4">        <span class="dt">FDR =</span> <span class="kw">p.adjust</span>(PValue, <span class="dt">method =</span> <span class="st">&quot;BH&quot;</span>),</a>
<a class="sourceLine" id="cb26-5" data-line-number="5">        <span class="dt">consequence_simplified =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb26-6" data-line-number="6">        consequence_simplified, </a>
<a class="sourceLine" id="cb26-7" data-line-number="7">        <span class="kw">levels</span>(<span class="kw">as.factor</span>(consequence_simplified))[<span class="kw">order</span>(tmp4[[<span class="st">&quot;med&quot;</span>]])]),</a>
<a class="sourceLine" id="cb26-8" data-line-number="8">        <span class="dt">de  =</span> <span class="kw">ifelse</span>(FDR <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.2</span>, <span class="st">&quot;FDR &lt; 0.2&quot;</span>, <span class="st">&quot;FDR &gt; 0.2&quot;</span>))</a>
<a class="sourceLine" id="cb26-9" data-line-number="9"></a>
<a class="sourceLine" id="cb26-10" data-line-number="10"></a>
<a class="sourceLine" id="cb26-11" data-line-number="11">df_to_plot <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb26-12" data-line-number="12"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">select</span>(gene, hgnc_symbol, consequence, clone_presence, logFC, </a>
<a class="sourceLine" id="cb26-13" data-line-number="13">                  F, FDR, PValue, ) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb26-14" data-line-number="14"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(FDR) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">head</span>(<span class="dt">n =</span> <span class="dv">20</span>)</a></code></pre></div>
<pre><code>                     gene hgnc_symbol             consequence
1  ENSG00000084764_MAPRE3      MAPRE3        missense_variant
2  ENSG00000108821_COL1A1      COL1A1   splice_region_variant
3  ENSG00000108821_COL1A1      COL1A1 splice_acceptor_variant
4    ENSG00000101407_TTI1        TTI1      synonymous_variant
5    ENSG00000101407_TTI1        TTI1      synonymous_variant
6   ENSG00000129295_LRRC6       LRRC6        missense_variant
7    ENSG00000070476_ZXDC        ZXDC      synonymous_variant
8    ENSG00000130881_LRP3        LRP3      synonymous_variant
9    ENSG00000130881_LRP3        LRP3        missense_variant
10  ENSG00000149090_PAMR1       PAMR1        missense_variant
11 ENSG00000171988_JMJD1C      JMJD1C        missense_variant
12 ENSG00000120910_PPP3CC      PPP3CC        missense_variant
13 ENSG00000120910_PPP3CC      PPP3CC        missense_variant
14  ENSG00000148634_HERC4       HERC4        missense_variant
15   ENSG00000113739_STC2        STC2        missense_variant
16   ENSG00000113739_STC2        STC2      synonymous_variant
17   ENSG00000164733_CTSB        CTSB        missense_variant
18 ENSG00000152104_PTPN14      PTPN14        missense_variant
19  ENSG00000107104_KANK1       KANK1      synonymous_variant
20   ENSG00000173933_RBM4        RBM4        missense_variant
   clone_presence      logFC         F          FDR       PValue
1          clone3  3.8852622 23.551694 0.0006174704 6.836602e-06
2          clone3 -1.7755752 21.389937 0.0006174704 1.610792e-05
3          clone3 -1.7755752 21.389937 0.0006174704 1.610792e-05
4          clone3 -3.6592117  9.544347 0.0562436492 2.934451e-03
5          clone3 -3.6592117  9.544347 0.0562436492 2.934451e-03
6          clone3  4.1091706 10.394705 0.0562436492 2.146039e-03
7   clone2&amp;clone3  1.7671121  9.083986 0.0664002981 4.041757e-03
8          clone3 -4.3468332  7.001221 0.1299831952 1.017260e-02
9          clone3 -4.3468332  7.001221 0.1299831952 1.017260e-02
10         clone2  1.6050652  6.297424 0.1642661344 1.434150e-02
11         clone3 -4.6591571  6.121515 0.1642661344 1.571241e-02
12         clone3 -3.8189782  5.637699 0.1663081972 2.024622e-02
13         clone3 -3.8189782  5.637699 0.1663081972 2.024622e-02
14         clone2  0.9119639  5.671353 0.1663081972 1.988993e-02
15         clone3 -2.0906592  5.350110 0.1694907959 2.358133e-02
16         clone3 -2.0906592  5.350110 0.1694907959 2.358133e-02
17         clone3  0.5820432  5.023420 0.1900232634 2.809040e-02
18         clone3 -1.4753141  4.674966 0.2167685421 3.392899e-02
19         clone3 -2.4983404  4.264768 0.2328037625 4.251199e-02
20         clone3  2.2954114  4.288833 0.2328037625 4.194894e-02</code></pre>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" data-line-number="1">df_to_plot <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb28-2" data-line-number="2"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(FDR) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">write_tsv</span>(<span class="st">&quot;output/donor_specific/joxm_mut_genes_de_results.tsv&quot;</span>)</a>
<a class="sourceLine" id="cb28-3" data-line-number="3"></a>
<a class="sourceLine" id="cb28-4" data-line-number="4">p_mutated_clone &lt;-<span class="st"> </span><span class="kw">ggplot</span>(df_to_plot, <span class="kw">aes</span>(<span class="dt">y =</span> logFC, <span class="dt">x =</span> consequence_simplified)) <span class="op">+</span></a>
<a class="sourceLine" id="cb28-5" data-line-number="5"><span class="st">    </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="dv">0</span>, <span class="dt">linetype =</span> <span class="dv">1</span>, <span class="dt">colour =</span> <span class="st">&quot;black&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb28-6" data-line-number="6"><span class="st">    </span><span class="kw">geom_boxplot</span>(<span class="dt">outlier.size =</span> <span class="dv">0</span>, <span class="dt">outlier.alpha =</span> <span class="dv">0</span>, <span class="dt">fill =</span> <span class="st">&quot;gray90&quot;</span>,</a>
<a class="sourceLine" id="cb28-7" data-line-number="7">                 <span class="dt">colour =</span> <span class="st">&quot;firebrick4&quot;</span>, <span class="dt">width =</span> <span class="fl">0.2</span>, <span class="dt">size =</span> <span class="dv">1</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb28-8" data-line-number="8"><span class="st">    </span>ggbeeswarm<span class="op">::</span><span class="kw">geom_quasirandom</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> <span class="op">-</span><span class="kw">log10</span>(PValue)), </a>
<a class="sourceLine" id="cb28-9" data-line-number="9">                                 <span class="dt">colour =</span> <span class="st">&quot;gray40&quot;</span>, <span class="dt">pch =</span> <span class="dv">21</span>, <span class="dt">size =</span> <span class="dv">4</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb28-10" data-line-number="10"><span class="st">    </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">y =</span> <span class="fl">-0.25</span>, <span class="dt">x =</span> <span class="dv">0</span>, <span class="dt">yend =</span> <span class="dv">-1</span>, <span class="dt">xend =</span> <span class="dv">0</span>), </a>
<a class="sourceLine" id="cb28-11" data-line-number="11">                 <span class="dt">colour =</span> <span class="st">&quot;black&quot;</span>, <span class="dt">size =</span> <span class="dv">1</span>, <span class="dt">arrow =</span> <span class="kw">arrow</span>(<span class="dt">length =</span> <span class="kw">unit</span>(<span class="fl">0.5</span>, <span class="st">&quot;cm&quot;</span>))) <span class="op">+</span></a>
<a class="sourceLine" id="cb28-12" data-line-number="12"><span class="st">    </span><span class="kw">annotate</span>(<span class="st">&quot;text&quot;</span>, <span class="dt">y =</span> <span class="dv">-3</span>, <span class="dt">x =</span> <span class="dv">0</span>, <span class="dt">size =</span> <span class="dv">6</span>, <span class="dt">label =</span> <span class="st">&quot;lower in mutated clone&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb28-13" data-line-number="13"><span class="st">    </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">y =</span> <span class="fl">0.25</span>, <span class="dt">x =</span> <span class="dv">0</span>, <span class="dt">yend =</span> <span class="dv">1</span>, <span class="dt">xend =</span> <span class="dv">0</span>), </a>
<a class="sourceLine" id="cb28-14" data-line-number="14">                 <span class="dt">colour =</span> <span class="st">&quot;black&quot;</span>, <span class="dt">size =</span> <span class="dv">1</span>, <span class="dt">arrow =</span> <span class="kw">arrow</span>(<span class="dt">length =</span> <span class="kw">unit</span>(<span class="fl">0.5</span>, <span class="st">&quot;cm&quot;</span>))) <span class="op">+</span></a>
<a class="sourceLine" id="cb28-15" data-line-number="15"><span class="st">    </span><span class="kw">annotate</span>(<span class="st">&quot;text&quot;</span>, <span class="dt">y =</span> <span class="dv">3</span>, <span class="dt">x =</span> <span class="dv">0</span>, <span class="dt">size =</span> <span class="dv">6</span>, <span class="dt">label =</span> <span class="st">&quot;higher in mutated clone&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb28-16" data-line-number="16"><span class="st">    </span><span class="kw">scale_x_discrete</span>(<span class="dt">expand =</span> <span class="kw">c</span>(<span class="fl">0.1</span>, <span class="fl">.05</span>), <span class="dt">name =</span> <span class="st">&quot;consequence&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb28-17" data-line-number="17"><span class="st">    </span><span class="kw">scale_y_continuous</span>(<span class="dt">expand =</span> <span class="kw">c</span>(<span class="fl">0.1</span>, <span class="fl">0.1</span>), <span class="dt">name =</span> <span class="st">&quot;logFC&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb28-18" data-line-number="18"><span class="st">    </span><span class="kw">expand_limits</span>(<span class="dt">x =</span> <span class="kw">c</span>(<span class="op">-</span><span class="fl">0.75</span>, <span class="dv">8</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb28-19" data-line-number="19"><span class="st">    </span><span class="kw">theme_ridges</span>(<span class="dv">22</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb28-20" data-line-number="20"><span class="st">    </span><span class="kw">coord_flip</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb28-21" data-line-number="21"><span class="st">    </span><span class="kw">scale_fill_viridis</span>(<span class="dt">option =</span> <span class="st">&quot;B&quot;</span>, <span class="dt">name =</span> <span class="st">&quot;-log10(P)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb28-22" data-line-number="22"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">strip.background =</span> <span class="kw">element_rect</span>(<span class="dt">fill =</span> <span class="st">&quot;gray90&quot;</span>),</a>
<a class="sourceLine" id="cb28-23" data-line-number="23">          <span class="dt">legend.position =</span> <span class="st">&quot;right&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb28-24" data-line-number="24"><span class="st">    </span><span class="kw">guides</span>(<span class="dt">color =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb28-25" data-line-number="25"></a>
<a class="sourceLine" id="cb28-26" data-line-number="26"><span class="kw">ggsave</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_mutgenes_logfc-box_by_simple_vep_anno_allcells.png&quot;</span>, </a>
<a class="sourceLine" id="cb28-27" data-line-number="27">      <span class="dt">plot =</span> p_mutated_clone, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">width =</span> <span class="fl">11.5</span>)</a>
<a class="sourceLine" id="cb28-28" data-line-number="28"><span class="kw">ggsave</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_mutgenes_logfc-box_by_simple_vep_anno_allcells.pdf&quot;</span>, </a>
<a class="sourceLine" id="cb28-29" data-line-number="29">       <span class="dt">plot =</span> p_mutated_clone, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">width =</span> <span class="fl">11.5</span>)</a>
<a class="sourceLine" id="cb28-30" data-line-number="30"><span class="kw">ggsave</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_mutgenes_logfc-box_by_simple_vep_anno_allcells.svg&quot;</span>, </a>
<a class="sourceLine" id="cb28-31" data-line-number="31">       <span class="dt">plot =</span> p_mutated_clone, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">width =</span> <span class="fl">11.5</span>)</a>
<a class="sourceLine" id="cb28-32" data-line-number="32">p_mutated_clone</a></code></pre></div>
<p><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/de-mutated-joxm-1.png" width="1104" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="differential-expression-transcriptome-wide" class="section level2">
<h2>Differential expression transcriptome-wide</h2>
<p>First we can look for genes that have any significant difference in expression between clones. The summary below shows the number of significant and non-significant genes at a Benjamini-Hochberg FDR threshold of 10%.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb29-1" data-line-number="1">knitr<span class="op">::</span>opts_chunk<span class="op">$</span><span class="kw">set</span>(<span class="dt">fig.height =</span> <span class="dv">6</span>, <span class="dt">fig.width =</span> <span class="dv">8</span>)</a>
<a class="sourceLine" id="cb29-2" data-line-number="2"><span class="kw">summary</span>(<span class="kw">decideTests</span>(de_res<span class="op">$</span>qlf_list<span class="op">$</span>joxm, <span class="dt">p.value =</span> <span class="fl">0.1</span>))</a></code></pre></div>
<pre><code>       assignedclone3-assignedclone2
NotSig                         10150
Sig                              901</code></pre>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb31-1" data-line-number="1"><span class="kw">summary</span>(<span class="kw">decideTests</span>(de_res<span class="op">$</span>qlf_list<span class="op">$</span>joxm, <span class="dt">p.value =</span> <span class="fl">0.05</span>))</a></code></pre></div>
<pre><code>       assignedclone3-assignedclone2
NotSig                         10502
Sig                              549</code></pre>
<p>We can view the 10 genes with strongest evidence for differential expression across clones.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb33-1" data-line-number="1"><span class="kw">topTags</span>(de_res<span class="op">$</span>qlf_list<span class="op">$</span>joxm)</a></code></pre></div>
<pre><code>Coefficient:  assignedclone2 assignedclone3 
                        logFC.assignedclone2 logFC.assignedclone3
ENSG00000205542_TMSB4X            -0.2886522           -5.8130458
ENSG00000124508_BTN2A2            -0.2512275            4.1377818
ENSG00000158882_TOMM40L           -0.2553535            6.2263232
ENSG00000146776_ATXN7L1            5.8013534            3.3752647
ENSG00000026652_AGPAT4             2.3215386            5.8405432
ENSG00000215271_HOMEZ              4.2232674            2.6794625
ENSG00000175395_ZNF25             -2.1170549            4.6804151
ENSG00000135776_ABCB10             0.6713155            5.0054153
ENSG00000095739_BAMBI              4.5309789           -0.3778691
ENSG00000136158_SPRY2             -1.2049610            4.7778728
                           logCPM        F       PValue          FDR
ENSG00000205542_TMSB4X  12.468006 91.46976 2.962531e-22 3.273893e-18
ENSG00000124508_BTN2A2   2.498248 47.36794 1.474039e-12 8.144803e-09
ENSG00000158882_TOMM40L  3.337964 38.97201 2.925525e-12 1.077666e-08
ENSG00000146776_ATXN7L1  3.810016 32.87690 2.116125e-11 5.846324e-08
ENSG00000026652_AGPAT4   2.860100 33.06655 5.771600e-11 1.275639e-07
ENSG00000215271_HOMEZ    2.851795 30.02105 1.122804e-10 2.068018e-07
ENSG00000175395_ZNF25    3.174000 29.63095 1.417301e-10 2.237513e-07
ENSG00000135776_ABCB10   2.677297 29.15750 1.989993e-10 2.748926e-07
ENSG00000095739_BAMBI    3.004732 27.60813 6.651894e-10 8.167787e-07
ENSG00000136158_SPRY2    2.677447 37.27297 8.205003e-10 9.067349e-07</code></pre>
<p>We can check that the estimates of the biological coefficient of variation from the negative binomial model look sensible. Here they do, so we can expect sensible DE results.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb35-1" data-line-number="1"><span class="kw">plotBCV</span>(de_res<span class="op">$</span>dge_list<span class="op">$</span>joxm)</a></code></pre></div>
<p><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/plot-bcv-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>Likewise, a plot of the quasi-likelihood parameter against average gene expression looks smooth and sensible.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb36-1" data-line-number="1"><span class="kw">plotQLDisp</span>(de_res<span class="op">$</span>qlf_list<span class="op">$</span>joxm)</a></code></pre></div>
<p><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/plot-qld-1.png" width="768" style="display: block; margin: auto;" /></p>
<div id="pairwise-comparisons-of-clones" class="section level3">
<h3>Pairwise comparisons of clones</h3>
<p>As well as looking for <em>any</em> difference in expression across clones, we can also inspect specific pairwise contrasts of clones for differential expression.</p>
<p>For this donor, we are able to look at 3 pairwise contrasts.</p>
<p>The output below shows the top 10 DE genes for pair of (testable) clones.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb37-1" data-line-number="1">cntrsts &lt;-<span class="st"> </span><span class="kw">names</span>(de_res<span class="op">$</span>qlf_pairwise<span class="op">$</span>joxm)[<span class="op">-</span><span class="dv">1</span>]</a>
<a class="sourceLine" id="cb37-2" data-line-number="2"><span class="cf">for</span> (i <span class="cf">in</span> cntrsts) {</a>
<a class="sourceLine" id="cb37-3" data-line-number="3">  <span class="kw">print</span>(<span class="kw">topTags</span>(de_res<span class="op">$</span>qlf_pairwise<span class="op">$</span>joxm[[i]]))</a>
<a class="sourceLine" id="cb37-4" data-line-number="4">}</a></code></pre></div>
<pre><code>Coefficient:  assignedclone2 
                             logFC   logCPM        F       PValue
ENSG00000146776_ATXN7L1   5.801353 3.810016 64.75358 3.618933e-12
ENSG00000215271_HOMEZ     4.223267 2.851795 56.91066 3.832169e-11
ENSG00000095739_BAMBI     4.530979 3.004732 46.71443 1.314479e-09
ENSG00000165475_CRYL1     4.020138 3.055068 42.15286 4.773076e-09
ENSG00000256977_LIMS3     3.124759 2.708791 39.15415 2.358976e-08
ENSG00000166896_XRCC6BP1  2.972211 2.628904 35.00854 6.100632e-08
ENSG00000113368_LMNB1    -4.995249 3.932812 34.70055 8.566612e-08
ENSG00000165752_STK32C   -5.481150 4.483212 32.82498 1.405996e-07
ENSG00000143845_ETNK2     3.270634 2.816029 33.28608 1.513342e-07
ENSG00000197465_GYPE      3.010858 2.606880 32.67812 1.687145e-07
                         ensembl_gene_id hgnc_symbol entrezid          FDR
ENSG00000146776_ATXN7L1  ENSG00000146776     ATXN7L1   222255 3.999282e-08
ENSG00000215271_HOMEZ    ENSG00000215271       HOMEZ    57594 2.117465e-07
ENSG00000095739_BAMBI    ENSG00000095739       BAMBI    25805 4.842103e-06
ENSG00000165475_CRYL1    ENSG00000165475       CRYL1    51084 1.318681e-05
ENSG00000256977_LIMS3    ENSG00000256977       LIMS3    96626 5.213808e-05
ENSG00000166896_XRCC6BP1 ENSG00000166896    XRCC6BP1    91419 1.123635e-04
ENSG00000113368_LMNB1    ENSG00000113368       LMNB1     4001 1.352423e-04
ENSG00000165752_STK32C   ENSG00000165752      STK32C   282974 1.858215e-04
ENSG00000143845_ETNK2    ENSG00000143845       ETNK2    55224 1.858215e-04
ENSG00000197465_GYPE     ENSG00000197465        GYPE     2996 1.864463e-04
Coefficient:  assignedclone3 
                             logFC    logCPM         F       PValue
ENSG00000205542_TMSB4X   -5.813046 12.468006 182.84733 3.083273e-23
ENSG00000026652_AGPAT4    5.840543  2.860100  65.80971 8.363155e-12
ENSG00000158882_TOMM40L   6.226323  3.337964  60.47867 3.581635e-11
ENSG00000135776_ABCB10    5.005415  2.677297  54.64197 8.190527e-11
ENSG00000124508_BTN2A2    4.137782  2.498248  62.56667 1.483166e-10
ENSG00000129048_ACKR4     8.028290  3.720048  65.44187 3.310506e-10
ENSG00000173295_FAM86B3P  6.964274  3.295641  47.99588 6.618565e-10
ENSG00000180787_ZFP3      5.608020  3.305761  48.06258 1.091002e-09
ENSG00000136158_SPRY2     4.777873  2.677447  45.43698 4.638210e-08
ENSG00000198168_SVIP      5.917955  3.038543  34.42856 7.554858e-08
                         ensembl_gene_id hgnc_symbol entrezid          FDR
ENSG00000205542_TMSB4X   ENSG00000205542      TMSB4X     7114 3.407325e-19
ENSG00000026652_AGPAT4   ENSG00000026652      AGPAT4    56895 4.621061e-08
ENSG00000158882_TOMM40L  ENSG00000158882     TOMM40L    84134 1.319355e-07
ENSG00000135776_ABCB10   ENSG00000135776      ABCB10    23456 2.262838e-07
ENSG00000124508_BTN2A2   ENSG00000124508      BTN2A2    10385 3.278094e-07
ENSG00000129048_ACKR4    ENSG00000129048       ACKR4    51554 6.097401e-07
ENSG00000173295_FAM86B3P ENSG00000173295    FAM86B3P   286042 1.044882e-06
ENSG00000180787_ZFP3     ENSG00000180787        ZFP3   124961 1.507083e-06
ENSG00000136158_SPRY2    ENSG00000136158       SPRY2    10253 5.695206e-05
ENSG00000198168_SVIP     ENSG00000198168        SVIP   258010 7.523350e-05
Coefficient:  -1*assignedclone2 1*assignedclone3 
                            logFC    logCPM         F       PValue
ENSG00000205542_TMSB4X  -5.524394 12.468006 167.09909 4.400831e-22
ENSG00000175395_ZNF25    6.797470  3.174000  53.10092 1.265384e-10
ENSG00000124508_BTN2A2   4.389009  2.498248  54.14334 1.147322e-09
ENSG00000136158_SPRY2    5.982834  2.677447  60.81381 1.644934e-09
ENSG00000158882_TOMM40L  6.481677  3.337964  43.32876 6.013150e-09
ENSG00000039139_DNAH5    5.420749  3.262779  41.04082 9.408878e-09
ENSG00000100084_HIRA     5.722823  3.119595  43.45990 1.977958e-08
ENSG00000164099_PRSS12   5.677707  3.222941  36.68476 3.308745e-08
ENSG00000148840_PPRC1    5.234448  2.909858  34.39080 1.243506e-07
ENSG00000165752_STK32C   7.004519  4.483212  32.86562 1.384803e-07
                        ensembl_gene_id hgnc_symbol entrezid          FDR
ENSG00000205542_TMSB4X  ENSG00000205542      TMSB4X     7114 4.863358e-18
ENSG00000175395_ZNF25   ENSG00000175395       ZNF25   219749 6.991878e-07
ENSG00000124508_BTN2A2  ENSG00000124508      BTN2A2    10385 4.226351e-06
ENSG00000136158_SPRY2   ENSG00000136158       SPRY2    10253 4.544540e-06
ENSG00000158882_TOMM40L ENSG00000158882     TOMM40L    84134 1.329026e-05
ENSG00000039139_DNAH5   ENSG00000039139       DNAH5     1767 1.732958e-05
ENSG00000100084_HIRA    ENSG00000100084        HIRA     7290 3.122630e-05
ENSG00000164099_PRSS12  ENSG00000164099      PRSS12     8492 4.570617e-05
ENSG00000148840_PPRC1   ENSG00000148840       PPRC1    23082 1.526887e-04
ENSG00000165752_STK32C  ENSG00000165752      STK32C   282974 1.530346e-04</code></pre>
<p>Below we see the following plots for each pairwise comparison:</p>
<ul>
<li>A “mean-difference” plot: log-fold-change (between clones) vs average expression;</li>
<li>A “volcano plot”: -log10(PValue) vs log-fold-change between clones.</li>
</ul>
<p>In the MD plots we see logFC distributions centred around zero as we would hope (gold line in plots shows lowess curve through points).</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb39-1" data-line-number="1"><span class="cf">for</span> (i <span class="cf">in</span> cntrsts) {</a>
<a class="sourceLine" id="cb39-2" data-line-number="2">  <span class="kw">plotMD</span>(de_res<span class="op">$</span>qlf_pairwise<span class="op">$</span>joxm[[i]], <span class="dt">p.value =</span> <span class="fl">0.1</span>)</a>
<a class="sourceLine" id="cb39-3" data-line-number="3">  <span class="kw">lines</span>(<span class="kw">lowess</span>(<span class="dt">x =</span> de_res<span class="op">$</span>qlf_pairwise<span class="op">$</span>joxm[[i]]<span class="op">$</span>table<span class="op">$</span>logCPM,</a>
<a class="sourceLine" id="cb39-4" data-line-number="4">               <span class="dt">y =</span> de_res<span class="op">$</span>qlf_pairwise<span class="op">$</span>joxm[[i]]<span class="op">$</span>table<span class="op">$</span>logFC), </a>
<a class="sourceLine" id="cb39-5" data-line-number="5">        <span class="dt">col =</span> <span class="st">&quot;goldenrod3&quot;</span>, <span class="dt">lwd =</span> <span class="dv">4</span>)</a>
<a class="sourceLine" id="cb39-6" data-line-number="6">  </a>
<a class="sourceLine" id="cb39-7" data-line-number="7">  de_tab &lt;-<span class="st"> </span>de_res<span class="op">$</span>qlf_pairwise<span class="op">$</span>joxm[[i]]<span class="op">$</span>table</a>
<a class="sourceLine" id="cb39-8" data-line-number="8">  de_tab[[<span class="st">&quot;gene&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">rownames</span>(de_tab)</a>
<a class="sourceLine" id="cb39-9" data-line-number="9">  de_tab &lt;-<span class="st"> </span>de_tab <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb39-10" data-line-number="10"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">FDR =</span> <span class="kw">adj_pvalues</span>(<span class="kw">ihw</span>(PValue <span class="op">~</span><span class="st"> </span>logCPM, <span class="dt">alpha =</span> <span class="fl">0.1</span>)), </a>
<a class="sourceLine" id="cb39-11" data-line-number="11">                  <span class="dt">sig =</span> FDR <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.1</span>,</a>
<a class="sourceLine" id="cb39-12" data-line-number="12">                  <span class="dt">signed_F =</span> <span class="kw">sign</span>(logFC) <span class="op">*</span><span class="st"> </span>F) </a>
<a class="sourceLine" id="cb39-13" data-line-number="13">  de_tab[[<span class="st">&quot;lab&quot;</span>]] &lt;-<span class="st"> &quot;&quot;</span></a>
<a class="sourceLine" id="cb39-14" data-line-number="14">  int_genes_entrezid &lt;-<span class="st"> </span><span class="kw">c</span>(Hs.H<span class="op">$</span>HALLMARK_G2M_CHECKPOINT, Hs.H<span class="op">$</span>HALLMARK_E2F_TARGETS,</a>
<a class="sourceLine" id="cb39-15" data-line-number="15">                          Hs.c2<span class="op">$</span>ROSTY_CERVICAL_CANCER_PROLIFERATION_CLUSTER)</a>
<a class="sourceLine" id="cb39-16" data-line-number="16">  mm &lt;-<span class="st"> </span><span class="kw">match</span>(int_genes_entrezid, de_tab<span class="op">$</span>entrezid)</a>
<a class="sourceLine" id="cb39-17" data-line-number="17">  mm &lt;-<span class="st"> </span>mm[<span class="op">!</span><span class="kw">is.na</span>(mm)]</a>
<a class="sourceLine" id="cb39-18" data-line-number="18">  int_genes_hgnc &lt;-<span class="st"> </span>de_tab<span class="op">$</span>hgnc_symbol[mm]</a>
<a class="sourceLine" id="cb39-19" data-line-number="19">  int_genes_hgnc &lt;-<span class="st"> </span><span class="kw">c</span>(int_genes_hgnc, <span class="st">&quot;MYBL1&quot;</span>)</a>
<a class="sourceLine" id="cb39-20" data-line-number="20">  genes_to_label &lt;-<span class="st"> </span>(de_tab[[<span class="st">&quot;hgnc_symbol&quot;</span>]] <span class="op">%in%</span><span class="st"> </span>int_genes_hgnc <span class="op">&amp;</span></a>
<a class="sourceLine" id="cb39-21" data-line-number="21"><span class="st">                       </span>de_tab[[<span class="st">&quot;FDR&quot;</span>]] <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.1</span>)</a>
<a class="sourceLine" id="cb39-22" data-line-number="22">  de_tab[[<span class="st">&quot;lab&quot;</span>]][genes_to_label] &lt;-</a>
<a class="sourceLine" id="cb39-23" data-line-number="23"><span class="st">    </span>de_tab[[<span class="st">&quot;hgnc_symbol&quot;</span>]][genes_to_label]</a>
<a class="sourceLine" id="cb39-24" data-line-number="24">  </a>
<a class="sourceLine" id="cb39-25" data-line-number="25">  p_vulcan &lt;-<span class="st"> </span><span class="kw">ggplot</span>(de_tab, <span class="kw">aes</span>(<span class="dt">x =</span> logFC, <span class="dt">y =</span> <span class="op">-</span><span class="kw">log10</span>(PValue), <span class="dt">fill =</span> sig,</a>
<a class="sourceLine" id="cb39-26" data-line-number="26">                     <span class="dt">label =</span> lab)) <span class="op">+</span></a>
<a class="sourceLine" id="cb39-27" data-line-number="27"><span class="st">    </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">size =</span> sig), <span class="dt">pch =</span> <span class="dv">21</span>, <span class="dt">colour =</span> <span class="st">&quot;gray40&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb39-28" data-line-number="28"><span class="st">    </span><span class="kw">geom_label_repel</span>(<span class="dt">show.legend =</span> <span class="ot">FALSE</span>, </a>
<a class="sourceLine" id="cb39-29" data-line-number="29">                     <span class="dt">arrow =</span> <span class="kw">arrow</span>(<span class="dt">type =</span> <span class="st">&quot;closed&quot;</span>, <span class="dt">length =</span> <span class="kw">unit</span>(<span class="fl">0.25</span>, <span class="st">&quot;cm&quot;</span>)), </a>
<a class="sourceLine" id="cb39-30" data-line-number="30">                     <span class="dt">nudge_x =</span> <span class="fl">0.2</span>, <span class="dt">nudge_y =</span> <span class="fl">0.3</span>, <span class="dt">fill =</span> <span class="st">&quot;gray95&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb39-31" data-line-number="31"><span class="st">    </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">x =</span> <span class="dv">-1</span>, <span class="dt">y =</span> <span class="dv">0</span>, <span class="dt">xend =</span> <span class="dv">-4</span>, <span class="dt">yend =</span> <span class="dv">0</span>), </a>
<a class="sourceLine" id="cb39-32" data-line-number="32">                 <span class="dt">colour =</span> <span class="st">&quot;black&quot;</span>, <span class="dt">size =</span> <span class="dv">1</span>, <span class="dt">arrow =</span> <span class="kw">arrow</span>(<span class="dt">length =</span> <span class="kw">unit</span>(<span class="fl">0.5</span>, <span class="st">&quot;cm&quot;</span>))) <span class="op">+</span></a>
<a class="sourceLine" id="cb39-33" data-line-number="33"><span class="st">    </span><span class="kw">annotate</span>(<span class="st">&quot;text&quot;</span>, <span class="dt">x =</span> <span class="dv">-4</span>, <span class="dt">y =</span> <span class="fl">-0.5</span>, <span class="dt">size =</span> <span class="dv">6</span>,</a>
<a class="sourceLine" id="cb39-34" data-line-number="34">             <span class="dt">label =</span> <span class="kw">paste</span>(<span class="st">&quot;higher in&quot;</span>, <span class="kw">strsplit</span>(i, <span class="st">&quot;_&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>])) <span class="op">+</span></a>
<a class="sourceLine" id="cb39-35" data-line-number="35"><span class="st">    </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">x =</span> <span class="dv">1</span>, <span class="dt">y =</span> <span class="dv">0</span>, <span class="dt">xend =</span> <span class="dv">4</span>, <span class="dt">yend =</span> <span class="dv">0</span>), </a>
<a class="sourceLine" id="cb39-36" data-line-number="36">                 <span class="dt">colour =</span> <span class="st">&quot;black&quot;</span>, <span class="dt">size =</span> <span class="dv">1</span>, <span class="dt">arrow =</span> <span class="kw">arrow</span>(<span class="dt">length =</span> <span class="kw">unit</span>(<span class="fl">0.5</span>, <span class="st">&quot;cm&quot;</span>))) <span class="op">+</span></a>
<a class="sourceLine" id="cb39-37" data-line-number="37"><span class="st">    </span><span class="kw">annotate</span>(<span class="st">&quot;text&quot;</span>, <span class="dt">x =</span> <span class="dv">4</span>, <span class="dt">y =</span> <span class="fl">-0.5</span>, <span class="dt">size =</span> <span class="dv">6</span>,</a>
<a class="sourceLine" id="cb39-38" data-line-number="38">             <span class="dt">label =</span> <span class="kw">paste</span>(<span class="st">&quot;higher in&quot;</span>, <span class="kw">strsplit</span>(i, <span class="st">&quot;_&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>])) <span class="op">+</span></a>
<a class="sourceLine" id="cb39-39" data-line-number="39"><span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;gray60&quot;</span>, <span class="st">&quot;firebrick&quot;</span>), </a>
<a class="sourceLine" id="cb39-40" data-line-number="40">                      <span class="dt">label =</span> <span class="kw">c</span>(<span class="st">&quot;N.S.&quot;</span>, <span class="st">&quot;FDR &lt; 10%&quot;</span>), <span class="dt">name =</span> <span class="st">&quot;&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb39-41" data-line-number="41"><span class="st">    </span><span class="kw">scale_size_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span>), <span class="dt">guide =</span> <span class="ot">FALSE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb39-42" data-line-number="42"><span class="st">    </span><span class="kw">guides</span>(<span class="dt">alpha =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb39-43" data-line-number="43">           <span class="dt">fill =</span> <span class="kw">guide_legend</span>(<span class="dt">override.aes =</span> <span class="kw">list</span>(<span class="dt">size =</span> <span class="dv">5</span>))) <span class="op">+</span></a>
<a class="sourceLine" id="cb39-44" data-line-number="44"><span class="st">    </span><span class="kw">theme_classic</span>(<span class="dv">20</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;right&quot;</span>)</a>
<a class="sourceLine" id="cb39-45" data-line-number="45">  <span class="kw">print</span>(p_vulcan)</a>
<a class="sourceLine" id="cb39-46" data-line-number="46">  </a>
<a class="sourceLine" id="cb39-47" data-line-number="47">  <span class="kw">ggsave</span>(<span class="kw">paste0</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_volcano_&quot;</span>, i, <span class="st">&quot;.png&quot;</span>), </a>
<a class="sourceLine" id="cb39-48" data-line-number="48">         <span class="dt">plot =</span> p_vulcan, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">width =</span> <span class="dv">9</span>)</a>
<a class="sourceLine" id="cb39-49" data-line-number="49">  <span class="kw">ggsave</span>(<span class="kw">paste0</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_volcano_&quot;</span>, i, <span class="st">&quot;.pdf&quot;</span>), </a>
<a class="sourceLine" id="cb39-50" data-line-number="50">         <span class="dt">plot =</span> p_vulcan, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">width =</span> <span class="dv">9</span>)</a>
<a class="sourceLine" id="cb39-51" data-line-number="51">  <span class="kw">ggsave</span>(<span class="kw">paste0</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_volcano_&quot;</span>, i, <span class="st">&quot;.svg&quot;</span>),</a>
<a class="sourceLine" id="cb39-52" data-line-number="52">         <span class="dt">plot =</span> p_vulcan, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">width =</span> <span class="dv">9</span>)</a>
<a class="sourceLine" id="cb39-53" data-line-number="53">}</a></code></pre></div>
<p><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/plots-cntrst-1.png" width="864" style="display: block; margin: auto;" /><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/plots-cntrst-2.png" width="864" style="display: block; margin: auto;" /><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/plots-cntrst-3.png" width="864" style="display: block; margin: auto;" /><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/plots-cntrst-4.png" width="864" style="display: block; margin: auto;" /><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/plots-cntrst-5.png" width="864" style="display: block; margin: auto;" /><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/plots-cntrst-6.png" width="864" style="display: block; margin: auto;" /></p>
<hr />
</div>
</div>
<div id="gene-set-enrichment-results" class="section level2">
<h2>Gene set enrichment results</h2>
<p>We extend our analysis from looking at differential expression for single genes to looking for enrichment in gene sets. We use gene sets from the MSigDB collection.</p>
<p>We use the <em>camera</em> method from the <em>limma</em> package to conduct competitive gene set testing. This method uses the full distributions of logFC statistics from pairwise clone contrasts to identify significantly enriched gene sets.</p>
<div id="msigdb-hallmark-gene-sets" class="section level3">
<h3>MSigDB Hallmark gene sets</h3>
<p>We look primarily at the “Hallmark” collection of gene sets from MSigDB.</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb40-1" data-line-number="1"><span class="cf">for</span> (i <span class="cf">in</span> cntrsts) {</a>
<a class="sourceLine" id="cb40-2" data-line-number="2">  <span class="kw">print</span>(i)</a>
<a class="sourceLine" id="cb40-3" data-line-number="3">  cam_H_pw &lt;-<span class="st"> </span>de_res<span class="op">$</span>camera<span class="op">$</span>H<span class="op">$</span>joxm[[i]]<span class="op">$</span>logFC</a>
<a class="sourceLine" id="cb40-4" data-line-number="4">  cam_H_pw[[<span class="st">&quot;geneset&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">rownames</span>(de_res<span class="op">$</span>camera<span class="op">$</span>H<span class="op">$</span>joxm[[i]]<span class="op">$</span>logFC)</a>
<a class="sourceLine" id="cb40-5" data-line-number="5">  cam_H_pw &lt;-<span class="st"> </span>cam_H_pw <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb40-6" data-line-number="6"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">sig =</span> FDR <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>) </a>
<a class="sourceLine" id="cb40-7" data-line-number="7">  <span class="kw">print</span>(<span class="kw">head</span>(cam_H_pw))</a>
<a class="sourceLine" id="cb40-8" data-line-number="8">}</a></code></pre></div>
<pre><code>[1] &quot;clone2_clone1&quot;
  NGenes Direction       PValue          FDR
1    197      Down 9.059034e-08 4.529517e-06
2    190      Down 2.002810e-06 5.007025e-05
3    181      Down 3.216112e-03 5.360186e-02
4     88      Down 4.855505e-02 6.069381e-01
5     23      Down 9.654137e-02 8.064825e-01
6     90      Down 1.052216e-01 8.064825e-01
                              geneset   sig
1                HALLMARK_E2F_TARGETS  TRUE
2             HALLMARK_G2M_CHECKPOINT  TRUE
3            HALLMARK_MITOTIC_SPINDLE FALSE
4        HALLMARK_ALLOGRAFT_REJECTION FALSE
5 HALLMARK_WNT_BETA_CATENIN_SIGNALING FALSE
6      HALLMARK_INFLAMMATORY_RESPONSE FALSE
[1] &quot;clone3_clone1&quot;
  NGenes Direction     PValue       FDR                            geneset
1    190      Down 0.02567449 0.9567097            HALLMARK_G2M_CHECKPOINT
2     56        Up 0.11722963 0.9567097            HALLMARK_MYC_TARGETS_V2
3    198        Up 0.14312855 0.9567097 HALLMARK_OXIDATIVE_PHOSPHORYLATION
4     71        Up 0.15442973 0.9567097               HALLMARK_COAGULATION
5    181      Down 0.15515689 0.9567097           HALLMARK_MITOTIC_SPINDLE
6     88      Down 0.16432866 0.9567097       HALLMARK_ALLOGRAFT_REJECTION
    sig
1 FALSE
2 FALSE
3 FALSE
4 FALSE
5 FALSE
6 FALSE
[1] &quot;clone3_clone2&quot;
  NGenes Direction     PValue       FDR                            geneset
1    197        Up 0.03522298 0.9512593               HALLMARK_E2F_TARGETS
2     56        Up 0.08925877 0.9512593            HALLMARK_MYC_TARGETS_V2
3    156        Up 0.09149777 0.9512593                HALLMARK_GLYCOLYSIS
4    123        Up 0.17229713 0.9512593       HALLMARK_IL2_STAT5_SIGNALING
5    123      Down 0.23334204 0.9512593 HALLMARK_INTERFERON_GAMMA_RESPONSE
6    116        Up 0.25855405 0.9512593                HALLMARK_MYOGENESIS
    sig
1 FALSE
2 FALSE
3 FALSE
4 FALSE
5 FALSE
6 FALSE</code></pre>
<p>Below we see the following plots for each pairwise comparison:</p>
<ul>
<li>A -log10(PValue) vs direction plot for enrichment of each of the 50 Hallmark gene sets;</li>
<li>A logFC “barcode plot”: distribution of the logFC statistics for genes in the E2F targets and G2M checkpoint gene sets relative to all other genes;</li>
<li>A signed-F “barcode plot”: distribution of the signed F statistics for genes in the E2F targets and G2M checkpoint gene sets relative to all other genes.</li>
</ul>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb42-1" data-line-number="1"><span class="cf">for</span> (i <span class="cf">in</span> cntrsts) {</a>
<a class="sourceLine" id="cb42-2" data-line-number="2">  cam_H_pw &lt;-<span class="st"> </span>de_res<span class="op">$</span>camera<span class="op">$</span>H<span class="op">$</span>joxm[[i]]<span class="op">$</span>logFC</a>
<a class="sourceLine" id="cb42-3" data-line-number="3">  cam_H_pw[[<span class="st">&quot;geneset&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">rownames</span>(de_res<span class="op">$</span>camera<span class="op">$</span>H<span class="op">$</span>joxm[[i]]<span class="op">$</span>logFC)</a>
<a class="sourceLine" id="cb42-4" data-line-number="4">  cam_H_pw &lt;-<span class="st"> </span>cam_H_pw <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb42-5" data-line-number="5"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">sig =</span> FDR <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>) </a>
<a class="sourceLine" id="cb42-6" data-line-number="6">  cam_H_pw[[<span class="st">&quot;lab&quot;</span>]] &lt;-<span class="st"> &quot;&quot;</span></a>
<a class="sourceLine" id="cb42-7" data-line-number="7">  cam_H_pw[[<span class="st">&quot;lab&quot;</span>]][<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>] &lt;-</a>
<a class="sourceLine" id="cb42-8" data-line-number="8"><span class="st">    </span>cam_H_pw[[<span class="st">&quot;geneset&quot;</span>]][<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>]</a>
<a class="sourceLine" id="cb42-9" data-line-number="9">  cam_H_pw[[<span class="st">&quot;Direction&quot;</span>]][cam_H_pw[[<span class="st">&quot;Direction&quot;</span>]] <span class="op">==</span><span class="st"> &quot;Up&quot;</span>] &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb42-10" data-line-number="10"><span class="st">    </span><span class="kw">paste</span>(<span class="st">&quot;Up in&quot;</span>, <span class="kw">strsplit</span>(i, <span class="st">&quot;_&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>], <span class="st">&quot;vs&quot;</span>, <span class="kw">strsplit</span>(i, <span class="st">&quot;_&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>])</a>
<a class="sourceLine" id="cb42-11" data-line-number="11">  cam_H_pw[[<span class="st">&quot;Direction&quot;</span>]][cam_H_pw[[<span class="st">&quot;Direction&quot;</span>]] <span class="op">==</span><span class="st"> &quot;Down&quot;</span>] &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb42-12" data-line-number="12"><span class="st">    </span><span class="kw">paste</span>(<span class="st">&quot;Down in&quot;</span>, <span class="kw">strsplit</span>(i, <span class="st">&quot;_&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>], <span class="st">&quot;vs&quot;</span>, <span class="kw">strsplit</span>(i, <span class="st">&quot;_&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>])</a>
<a class="sourceLine" id="cb42-13" data-line-number="13">  de_tab &lt;-<span class="st"> </span>de_res<span class="op">$</span>qlf_pairwise<span class="op">$</span>joxm[[i]]<span class="op">$</span>table</a>
<a class="sourceLine" id="cb42-14" data-line-number="14">  de_tab[[<span class="st">&quot;gene&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">rownames</span>(de_tab)</a>
<a class="sourceLine" id="cb42-15" data-line-number="15">  de_tab &lt;-<span class="st"> </span>de_tab <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb42-16" data-line-number="16"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">FDR =</span> <span class="kw">adj_pvalues</span>(<span class="kw">ihw</span>(PValue <span class="op">~</span><span class="st"> </span>logCPM, <span class="dt">alpha =</span> <span class="fl">0.1</span>)), </a>
<a class="sourceLine" id="cb42-17" data-line-number="17">                  <span class="dt">sig =</span> FDR <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.1</span>,</a>
<a class="sourceLine" id="cb42-18" data-line-number="18">                  <span class="dt">signed_F =</span> <span class="kw">sign</span>(logFC) <span class="op">*</span><span class="st"> </span>F)</a>
<a class="sourceLine" id="cb42-19" data-line-number="19">  </a>
<a class="sourceLine" id="cb42-20" data-line-number="20">  p_hallmark &lt;-<span class="st"> </span>cam_H_pw <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb42-21" data-line-number="21"><span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> Direction, <span class="dt">y =</span> <span class="op">-</span><span class="kw">log10</span>(PValue), <span class="dt">colour =</span> sig, </a>
<a class="sourceLine" id="cb42-22" data-line-number="22">               <span class="dt">label =</span> lab)) <span class="op">+</span></a>
<a class="sourceLine" id="cb42-23" data-line-number="23"><span class="st">    </span>ggbeeswarm<span class="op">::</span><span class="kw">geom_quasirandom</span>(<span class="kw">aes</span>(<span class="dt">size =</span> NGenes)) <span class="op">+</span></a>
<a class="sourceLine" id="cb42-24" data-line-number="24"><span class="st">    </span><span class="kw">geom_label_repel</span>(<span class="dt">show.legend =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb42-25" data-line-number="25">                     <span class="dt">nudge_y =</span> <span class="fl">0.3</span>, <span class="dt">nudge_x =</span> <span class="fl">0.3</span>, <span class="dt">fill =</span> <span class="st">&quot;gray95&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb42-26" data-line-number="26"><span class="st">    </span><span class="kw">scale_colour_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;gray50&quot;</span>, <span class="st">&quot;firebrick&quot;</span>), </a>
<a class="sourceLine" id="cb42-27" data-line-number="27">                        <span class="dt">label =</span> <span class="kw">c</span>(<span class="st">&quot;N.S.&quot;</span>, <span class="st">&quot;FDR &lt; 5%&quot;</span>), <span class="dt">name =</span> <span class="st">&quot;&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb42-28" data-line-number="28"><span class="st">    </span><span class="kw">guides</span>(<span class="dt">alpha =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb42-29" data-line-number="29">           <span class="dt">fill =</span> <span class="kw">guide_legend</span>(<span class="dt">override.aes =</span> <span class="kw">list</span>(<span class="dt">size =</span> <span class="dv">5</span>))) <span class="op">+</span></a>
<a class="sourceLine" id="cb42-30" data-line-number="30"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Gene set enrichment direction&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb42-31" data-line-number="31"><span class="st">    </span><span class="kw">theme_classic</span>(<span class="dv">20</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;right&quot;</span>)</a>
<a class="sourceLine" id="cb42-32" data-line-number="32">  <span class="kw">print</span>(p_hallmark)</a>
<a class="sourceLine" id="cb42-33" data-line-number="33">  </a>
<a class="sourceLine" id="cb42-34" data-line-number="34">  <span class="kw">ggsave</span>(<span class="kw">paste0</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_camera_H_&quot;</span>, i, <span class="st">&quot;.png&quot;</span>), </a>
<a class="sourceLine" id="cb42-35" data-line-number="35">         <span class="dt">plot =</span> p_hallmark, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">width =</span> <span class="dv">9</span>)</a>
<a class="sourceLine" id="cb42-36" data-line-number="36">  <span class="kw">ggsave</span>(<span class="kw">paste0</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_camera_H_&quot;</span>, i, <span class="st">&quot;.png&quot;</span>), </a>
<a class="sourceLine" id="cb42-37" data-line-number="37">         <span class="dt">plot =</span> p_hallmark, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">width =</span> <span class="dv">9</span>)</a>
<a class="sourceLine" id="cb42-38" data-line-number="38">  <span class="kw">ggsave</span>(<span class="kw">paste0</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_camera_H_&quot;</span>, i, <span class="st">&quot;.png&quot;</span>), </a>
<a class="sourceLine" id="cb42-39" data-line-number="39">         <span class="dt">plot =</span> p_hallmark, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">width =</span> <span class="dv">9</span>)</a>
<a class="sourceLine" id="cb42-40" data-line-number="40">  </a>
<a class="sourceLine" id="cb42-41" data-line-number="41">  idx &lt;-<span class="st"> </span><span class="kw">ids2indices</span>(Hs.H, <span class="dt">id =</span> de_tab<span class="op">$</span>entrezid)</a>
<a class="sourceLine" id="cb42-42" data-line-number="42">  <span class="kw">barcodeplot</span>(de_tab<span class="op">$</span>logFC, <span class="dt">index =</span> idx<span class="op">$</span>HALLMARK_E2F_TARGETS, </a>
<a class="sourceLine" id="cb42-43" data-line-number="43">              <span class="dt">index2 =</span> idx<span class="op">$</span>HALLMARK_G2M_CHECKPOINT, <span class="dt">xlab =</span> <span class="st">&quot;logFC&quot;</span>, </a>
<a class="sourceLine" id="cb42-44" data-line-number="44">              <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;joxm: &quot;</span>, i, <span class="st">&quot;</span><span class="ch">\n</span><span class="st"> HALLMARK_E2F_TARGETS and HALLMARK_G2M_CHECKPOINT&quot;</span>))</a>
<a class="sourceLine" id="cb42-45" data-line-number="45">  </a>
<a class="sourceLine" id="cb42-46" data-line-number="46">  <span class="kw">png</span>(<span class="kw">paste0</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_camera_H_&quot;</span>, i, <span class="st">&quot;_barcode_logFC_E2F_G2M.png&quot;</span>),</a>
<a class="sourceLine" id="cb42-47" data-line-number="47">      <span class="dt">height =</span> <span class="dv">400</span>, <span class="dt">width =</span> <span class="dv">600</span>)</a>
<a class="sourceLine" id="cb42-48" data-line-number="48">  <span class="kw">barcodeplot</span>(de_tab<span class="op">$</span>logFC, <span class="dt">index =</span> idx<span class="op">$</span>HALLMARK_E2F_TARGETS, </a>
<a class="sourceLine" id="cb42-49" data-line-number="49">              <span class="dt">index2 =</span> idx<span class="op">$</span>HALLMARK_G2M_CHECKPOINT, <span class="dt">xlab =</span> <span class="st">&quot;logFC&quot;</span>, </a>
<a class="sourceLine" id="cb42-50" data-line-number="50">              <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;joxm: &quot;</span>, i, <span class="st">&quot;</span><span class="ch">\n</span><span class="st"> HALLMARK_E2F_TARGETS and HALLMARK_G2M_CHECKPOINT&quot;</span>))</a>
<a class="sourceLine" id="cb42-51" data-line-number="51">  <span class="kw">dev.off</span>()</a>
<a class="sourceLine" id="cb42-52" data-line-number="52">  </a>
<a class="sourceLine" id="cb42-53" data-line-number="53">  <span class="kw">barcodeplot</span>(de_tab<span class="op">$</span>signed_F, <span class="dt">index =</span> idx<span class="op">$</span>HALLMARK_E2F_TARGETS, </a>
<a class="sourceLine" id="cb42-54" data-line-number="54">              <span class="dt">index2 =</span> idx<span class="op">$</span>HALLMARK_G2M_CHECKPOINT, <span class="dt">xlab =</span> <span class="st">&quot;signed F statistic&quot;</span>, </a>
<a class="sourceLine" id="cb42-55" data-line-number="55">              <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;joxm: &quot;</span>, i, <span class="st">&quot;</span><span class="ch">\n</span><span class="st"> HALLMARK_E2F_TARGETS and HALLMARK_G2M_CHECKPOINT&quot;</span>))</a>
<a class="sourceLine" id="cb42-56" data-line-number="56">  <span class="kw">png</span>(<span class="kw">paste0</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_camera_H_&quot;</span>, i, <span class="st">&quot;_barcode_signedF_E2F_G2M.png&quot;</span>),</a>
<a class="sourceLine" id="cb42-57" data-line-number="57">      <span class="dt">height =</span> <span class="dv">400</span>, <span class="dt">width =</span> <span class="dv">600</span>)</a>
<a class="sourceLine" id="cb42-58" data-line-number="58">  <span class="kw">barcodeplot</span>(de_tab<span class="op">$</span>signed_F, <span class="dt">index =</span> idx<span class="op">$</span>HALLMARK_E2F_TARGETS, </a>
<a class="sourceLine" id="cb42-59" data-line-number="59">              <span class="dt">index2 =</span> idx<span class="op">$</span>HALLMARK_G2M_CHECKPOINT, <span class="dt">xlab =</span> <span class="st">&quot;signed F statistic&quot;</span>, </a>
<a class="sourceLine" id="cb42-60" data-line-number="60">              <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;joxm: &quot;</span>, i, <span class="st">&quot;</span><span class="ch">\n</span><span class="st"> HALLMARK_E2F_TARGETS and HALLMARK_G2M_CHECKPOINT&quot;</span>))</a>
<a class="sourceLine" id="cb42-61" data-line-number="61">  <span class="kw">dev.off</span>()</a>
<a class="sourceLine" id="cb42-62" data-line-number="62">}  </a></code></pre></div>
<p><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/gene-set-plots-1.png" width="864" style="display: block; margin: auto;" /><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/gene-set-plots-2.png" width="864" style="display: block; margin: auto;" /><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/gene-set-plots-3.png" width="864" style="display: block; margin: auto;" /><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/gene-set-plots-4.png" width="864" style="display: block; margin: auto;" /><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/gene-set-plots-5.png" width="864" style="display: block; margin: auto;" /><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/gene-set-plots-6.png" width="864" style="display: block; margin: auto;" /><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/gene-set-plots-7.png" width="864" style="display: block; margin: auto;" /><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/gene-set-plots-8.png" width="864" style="display: block; margin: auto;" /><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/gene-set-plots-9.png" width="864" style="display: block; margin: auto;" /></p>
<p>One could carry out similar analyses and produce similar plots for the c2 and c6 MSigDB gene set collections.</p>
</div>
</div>
<div id="test-for-difference-in-cell-cycle-phases-by-clone" class="section level2">
<h2>Test for difference in cell cycle phases by clone</h2>
<p>We observe differing proportions of cells in different phases of the cell cycle by clone.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb43-1" data-line-number="1"><span class="kw">as.data.frame</span>(<span class="kw">colData</span>(de_res[[<span class="st">&quot;sce_list_unst&quot;</span>]][[<span class="st">&quot;joxm&quot;</span>]])) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb43-2" data-line-number="2"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">Cell_Cycle =</span> <span class="kw">factor</span>(cyclone_phase, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;G2M&quot;</span>, <span class="st">&quot;S&quot;</span>, <span class="st">&quot;G1&quot;</span>)),</a>
<a class="sourceLine" id="cb43-3" data-line-number="3">                  <span class="dt">assigned =</span> <span class="kw">factor</span>(assigned, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;clone3&quot;</span>, <span class="st">&quot;clone2&quot;</span>, <span class="st">&quot;clone1&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb43-4" data-line-number="4"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> assigned, <span class="dt">fill =</span> Cell_Cycle)) <span class="op">+</span></a>
<a class="sourceLine" id="cb43-5" data-line-number="5"><span class="st">  </span><span class="kw">geom_bar</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb43-6" data-line-number="6"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;#ff6a5c&quot;</span>, <span class="st">&quot;#ccdfcb&quot;</span>, <span class="st">&quot;#414141&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb43-7" data-line-number="7"><span class="st">  </span><span class="kw">coord_flip</span>() <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb43-8" data-line-number="8"><span class="st">  </span><span class="kw">guides</span>(<span class="dt">fill =</span> <span class="kw">guide_legend</span>(<span class="dt">reverse =</span> <span class="ot">TRUE</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb43-9" data-line-number="9"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.title.y =</span> <span class="kw">element_blank</span>())</a></code></pre></div>
<p><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/test-cc-1.png" width="768" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb44-1" data-line-number="1"><span class="kw">as.data.frame</span>(<span class="kw">colData</span>(de_res[[<span class="st">&quot;sce_list_unst&quot;</span>]][[<span class="st">&quot;joxm&quot;</span>]])) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb44-2" data-line-number="2"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">Cell_Cycle =</span> <span class="kw">factor</span>(cyclone_phase, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;G2M&quot;</span>, <span class="st">&quot;S&quot;</span>, <span class="st">&quot;G1&quot;</span>)),</a>
<a class="sourceLine" id="cb44-3" data-line-number="3">                  <span class="dt">assigned =</span> <span class="kw">factor</span>(assigned, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;clone3&quot;</span>, <span class="st">&quot;clone2&quot;</span>, <span class="st">&quot;clone1&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb44-4" data-line-number="4"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> assigned, <span class="dt">fill =</span> Cell_Cycle)) <span class="op">+</span></a>
<a class="sourceLine" id="cb44-5" data-line-number="5"><span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">position =</span> <span class="st">&quot;fill&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb44-6" data-line-number="6"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;#ff6a5c&quot;</span>, <span class="st">&quot;#ccdfcb&quot;</span>, <span class="st">&quot;#414141&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb44-7" data-line-number="7"><span class="st">  </span><span class="kw">coord_flip</span>() <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb44-8" data-line-number="8"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;proportion&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb44-9" data-line-number="9"><span class="st">  </span><span class="kw">guides</span>(<span class="dt">fill =</span> <span class="kw">guide_legend</span>(<span class="dt">reverse =</span> <span class="ot">TRUE</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb44-10" data-line-number="10"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.title.y =</span> <span class="kw">element_blank</span>())</a></code></pre></div>
<p><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/test-cc-2.png" width="768" style="display: block; margin: auto;" /></p>
<p>A Fisher Exact Test can provide some guidance about whether or not these differences in cell cycle proportions are expected by chance.</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb45-1" data-line-number="1">freqs &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(<span class="kw">table</span>(</a>
<a class="sourceLine" id="cb45-2" data-line-number="2">    de_res[[<span class="st">&quot;sce_list_unst&quot;</span>]][[<span class="st">&quot;joxm&quot;</span>]]<span class="op">$</span>assigned,  </a>
<a class="sourceLine" id="cb45-3" data-line-number="3">    de_res[[<span class="st">&quot;sce_list_unst&quot;</span>]][[<span class="st">&quot;joxm&quot;</span>]]<span class="op">$</span>cyclone_phase))</a>
<a class="sourceLine" id="cb45-4" data-line-number="4"><span class="kw">fisher.test</span>(freqs)</a></code></pre></div>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  freqs
p-value = 0.1335
alternative hypothesis: two.sided</code></pre>
<p>We can also test just for differences in proportions between clone1 and clone2.</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb47-1" data-line-number="1"><span class="kw">fisher.test</span>(freqs[<span class="op">-</span><span class="dv">3</span>,])</a></code></pre></div>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  freqs[-3, ]
p-value = 0.07296
alternative hypothesis: two.sided</code></pre>
</div>
<div id="pca-plots" class="section level2">
<h2>PCA plots</h2>
<p>Principal component analysis can reveal global structure from single-cell transcriptomic profiles.</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb49-1" data-line-number="1">choose_joxm_cells &lt;-<span class="st"> </span>(sce_joxm<span class="op">$</span>well_condition <span class="op">==</span><span class="st"> &quot;unstimulated&quot;</span> <span class="op">&amp;</span></a>
<a class="sourceLine" id="cb49-2" data-line-number="2"><span class="st">                          </span>sce_joxm<span class="op">$</span>assigned <span class="op">!=</span><span class="st"> &quot;unassigned&quot;</span>)</a>
<a class="sourceLine" id="cb49-3" data-line-number="3">pca_unst &lt;-<span class="st"> </span><span class="kw">reducedDim</span>(<span class="kw">runPCA</span>(sce_joxm[, choose_joxm_cells], </a>
<a class="sourceLine" id="cb49-4" data-line-number="4">                              <span class="dt">ntop =</span> <span class="dv">500</span>, <span class="dt">ncomponents =</span> <span class="dv">10</span>), <span class="st">&quot;PCA&quot;</span>)</a>
<a class="sourceLine" id="cb49-5" data-line-number="5">pca_unst &lt;-<span class="st"> </span><span class="kw">data.frame</span>(</a>
<a class="sourceLine" id="cb49-6" data-line-number="6">    <span class="dt">PC1 =</span> pca_unst[, <span class="dv">1</span>], <span class="dt">PC2 =</span> pca_unst[, <span class="dv">2</span>], </a>
<a class="sourceLine" id="cb49-7" data-line-number="7">    <span class="dt">PC3 =</span> pca_unst[, <span class="dv">3</span>], <span class="dt">PC4 =</span> pca_unst[, <span class="dv">4</span>],</a>
<a class="sourceLine" id="cb49-8" data-line-number="8">    <span class="dt">PC5 =</span> pca_unst[, <span class="dv">5</span>], <span class="dt">PC6 =</span> pca_unst[, <span class="dv">6</span>],</a>
<a class="sourceLine" id="cb49-9" data-line-number="9">    <span class="dt">clone =</span> sce_joxm[, choose_joxm_cells]<span class="op">$</span>assigned,</a>
<a class="sourceLine" id="cb49-10" data-line-number="10">    <span class="dt">nvars_cloneid =</span> sce_joxm[, choose_joxm_cells]<span class="op">$</span>nvars_cloneid,</a>
<a class="sourceLine" id="cb49-11" data-line-number="11">    <span class="dt">cyclone_phase =</span> sce_joxm[, choose_joxm_cells]<span class="op">$</span>cyclone_phase,</a>
<a class="sourceLine" id="cb49-12" data-line-number="12">    <span class="dt">G1 =</span> sce_joxm[, choose_joxm_cells]<span class="op">$</span>G1,</a>
<a class="sourceLine" id="cb49-13" data-line-number="13">    <span class="dt">G2M =</span> sce_joxm[, choose_joxm_cells]<span class="op">$</span>G2M,</a>
<a class="sourceLine" id="cb49-14" data-line-number="14">    <span class="dt">S =</span> sce_joxm[, choose_joxm_cells]<span class="op">$</span>S,</a>
<a class="sourceLine" id="cb49-15" data-line-number="15">    <span class="dt">clone1_prob =</span> sce_joxm[, choose_joxm_cells]<span class="op">$</span>clone1_prob,</a>
<a class="sourceLine" id="cb49-16" data-line-number="16">    <span class="dt">clone2_prob =</span> sce_joxm[, choose_joxm_cells]<span class="op">$</span>clone2_prob,</a>
<a class="sourceLine" id="cb49-17" data-line-number="17">    <span class="dt">clone3_prob =</span> sce_joxm[, choose_joxm_cells]<span class="op">$</span>clone3_prob,</a>
<a class="sourceLine" id="cb49-18" data-line-number="18">    <span class="dt">RPS6KA2 =</span> <span class="kw">as.vector</span>(<span class="kw">logcounts</span>(sce_joxm[<span class="kw">grep</span>(<span class="st">&quot;RPS6KA2&quot;</span>, <span class="kw">rownames</span>(sce_joxm)), choose_joxm_cells]))</a>
<a class="sourceLine" id="cb49-19" data-line-number="19">    )</a>
<a class="sourceLine" id="cb49-20" data-line-number="20"></a>
<a class="sourceLine" id="cb49-21" data-line-number="21"><span class="kw">ggplot</span>(pca_unst, <span class="kw">aes</span>(<span class="dt">x =</span> PC1, <span class="dt">y =</span> PC2, <span class="dt">fill =</span> clone)) <span class="op">+</span></a>
<a class="sourceLine" id="cb49-22" data-line-number="22"><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">pch =</span> <span class="dv">21</span>, <span class="dt">size =</span> <span class="dv">4</span>, <span class="dt">colour =</span> <span class="st">&quot;gray30&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb49-23" data-line-number="23"><span class="st">    </span><span class="kw">scale_fill_brewer</span>(<span class="dt">palette =</span> <span class="st">&quot;Accent&quot;</span>, <span class="dt">name =</span> <span class="st">&quot;assigned</span><span class="ch">\n</span><span class="st">clone&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb49-24" data-line-number="24"><span class="st">    </span><span class="kw">theme_classic</span>(<span class="dv">14</span>)</a></code></pre></div>
<p><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/pca-1.png" width="768" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb50-1" data-line-number="1"><span class="kw">ggplot</span>(pca_unst, <span class="kw">aes</span>(<span class="dt">x =</span> PC2, <span class="dt">y =</span> PC3, <span class="dt">fill =</span> clone)) <span class="op">+</span></a>
<a class="sourceLine" id="cb50-2" data-line-number="2"><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">pch =</span> <span class="dv">21</span>, <span class="dt">size =</span> <span class="dv">4</span>, <span class="dt">colour =</span> <span class="st">&quot;gray30&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb50-3" data-line-number="3"><span class="st">    </span><span class="kw">scale_fill_brewer</span>(<span class="dt">palette =</span> <span class="st">&quot;Accent&quot;</span>, <span class="dt">name =</span> <span class="st">&quot;assigned</span><span class="ch">\n</span><span class="st">clone&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb50-4" data-line-number="4"><span class="st">    </span><span class="kw">theme_classic</span>(<span class="dv">14</span>)</a></code></pre></div>
<p><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/pca-2.png" width="768" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb51-1" data-line-number="1"><span class="kw">ggplot</span>(pca_unst, <span class="kw">aes</span>(<span class="dt">x =</span> PC2, <span class="dt">y =</span> PC4, <span class="dt">fill =</span> clone)) <span class="op">+</span></a>
<a class="sourceLine" id="cb51-2" data-line-number="2"><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">pch =</span> <span class="dv">21</span>, <span class="dt">size =</span> <span class="dv">4</span>, <span class="dt">colour =</span> <span class="st">&quot;gray30&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb51-3" data-line-number="3"><span class="st">    </span><span class="kw">scale_fill_brewer</span>(<span class="dt">palette =</span> <span class="st">&quot;Accent&quot;</span>, <span class="dt">name =</span> <span class="st">&quot;assigned</span><span class="ch">\n</span><span class="st">clone&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb51-4" data-line-number="4"><span class="st">    </span><span class="kw">theme_classic</span>(<span class="dv">14</span>)</a></code></pre></div>
<p><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/pca-3.png" width="768" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb52-1" data-line-number="1"><span class="kw">ggplot</span>(pca_unst, <span class="kw">aes</span>(<span class="dt">x =</span> PC3, <span class="dt">y =</span> PC4, <span class="dt">fill =</span> clone)) <span class="op">+</span></a>
<a class="sourceLine" id="cb52-2" data-line-number="2"><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">pch =</span> <span class="dv">21</span>, <span class="dt">size =</span> <span class="dv">4</span>, <span class="dt">colour =</span> <span class="st">&quot;gray30&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb52-3" data-line-number="3"><span class="st">    </span><span class="kw">scale_fill_brewer</span>(<span class="dt">palette =</span> <span class="st">&quot;Accent&quot;</span>, <span class="dt">name =</span> <span class="st">&quot;assigned</span><span class="ch">\n</span><span class="st">clone&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb52-4" data-line-number="4"><span class="st">    </span><span class="kw">theme_classic</span>(<span class="dv">14</span>)</a></code></pre></div>
<p><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/pca-4.png" width="768" style="display: block; margin: auto;" /></p>
<p>Let us also look at PCA with cells coloured by the posterior probability of assignment to the various clones.</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb53-1" data-line-number="1">ppc1 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(pca_unst, <span class="kw">aes</span>(<span class="dt">x =</span> PC1, <span class="dt">y =</span> PC2, <span class="dt">fill =</span> clone1_prob)) <span class="op">+</span></a>
<a class="sourceLine" id="cb53-2" data-line-number="2"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">pch =</span> <span class="dv">21</span>, <span class="dt">size =</span> <span class="dv">4</span>, <span class="dt">colour =</span> <span class="st">&quot;gray30&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb53-3" data-line-number="3"><span class="st">  </span><span class="kw">scale_fill_viridis</span>(<span class="dt">option =</span> <span class="st">&quot;A&quot;</span>, <span class="dt">name =</span> <span class="st">&quot;clone1</span><span class="ch">\n</span><span class="st">posterior</span><span class="ch">\n</span><span class="st">probability&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb53-4" data-line-number="4"><span class="st">  </span><span class="kw">theme_classic</span>(<span class="dv">14</span>)</a>
<a class="sourceLine" id="cb53-5" data-line-number="5"></a>
<a class="sourceLine" id="cb53-6" data-line-number="6">ppc2 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(pca_unst, <span class="kw">aes</span>(<span class="dt">x =</span> PC1, <span class="dt">y =</span> PC2, <span class="dt">fill =</span> clone2_prob)) <span class="op">+</span></a>
<a class="sourceLine" id="cb53-7" data-line-number="7"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">pch =</span> <span class="dv">21</span>, <span class="dt">size =</span> <span class="dv">4</span>, <span class="dt">colour =</span> <span class="st">&quot;gray30&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb53-8" data-line-number="8"><span class="st">  </span><span class="kw">scale_fill_viridis</span>(<span class="dt">option =</span> <span class="st">&quot;A&quot;</span>, <span class="dt">name =</span> <span class="st">&quot;clone2</span><span class="ch">\n</span><span class="st">posterior</span><span class="ch">\n</span><span class="st">probability&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb53-9" data-line-number="9"><span class="st">  </span><span class="kw">theme_classic</span>(<span class="dv">14</span>)</a>
<a class="sourceLine" id="cb53-10" data-line-number="10"></a>
<a class="sourceLine" id="cb53-11" data-line-number="11">ppc3 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(pca_unst, <span class="kw">aes</span>(<span class="dt">x =</span> PC1, <span class="dt">y =</span> PC2, <span class="dt">fill =</span> clone3_prob)) <span class="op">+</span></a>
<a class="sourceLine" id="cb53-12" data-line-number="12"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">pch =</span> <span class="dv">21</span>, <span class="dt">size =</span> <span class="dv">4</span>, <span class="dt">colour =</span> <span class="st">&quot;gray30&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb53-13" data-line-number="13"><span class="st">  </span><span class="kw">scale_fill_viridis</span>(<span class="dt">option =</span> <span class="st">&quot;A&quot;</span>, <span class="dt">name =</span> <span class="st">&quot;clone3</span><span class="ch">\n</span><span class="st">posterior</span><span class="ch">\n</span><span class="st">probability&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb53-14" data-line-number="14"><span class="st">  </span><span class="kw">theme_classic</span>(<span class="dv">14</span>)</a>
<a class="sourceLine" id="cb53-15" data-line-number="15"></a>
<a class="sourceLine" id="cb53-16" data-line-number="16"><span class="kw">plot_grid</span>(ppc1, ppc2, ppc3, <span class="dt">labels =</span> <span class="st">&quot;auto&quot;</span>, <span class="dt">ncol =</span> <span class="dv">1</span>)</a></code></pre></div>
<p><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/pca-clone-prob-1.png" width="912" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb54-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_pca_pc1_pc2_clone_probs.png&quot;</span>, <span class="dt">height =</span> <span class="dv">11</span>, <span class="dt">width =</span> <span class="fl">9.5</span>)</a>
<a class="sourceLine" id="cb54-2" data-line-number="2"><span class="kw">ggsave</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_pca_pc1_pc2_clone_probs.pdf&quot;</span>, <span class="dt">height =</span> <span class="dv">11</span>, <span class="dt">width =</span> <span class="fl">9.5</span>)</a>
<a class="sourceLine" id="cb54-3" data-line-number="3"><span class="kw">ggsave</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_pca_pc1_pc2_clone_probs.svg&quot;</span>, <span class="dt">height =</span> <span class="dv">11</span>, <span class="dt">width =</span> <span class="fl">9.5</span>)</a>
<a class="sourceLine" id="cb54-4" data-line-number="4"></a>
<a class="sourceLine" id="cb54-5" data-line-number="5">ppc1 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(pca_unst, <span class="kw">aes</span>(<span class="dt">x =</span> PC2, <span class="dt">y =</span> PC3, <span class="dt">fill =</span> clone1_prob)) <span class="op">+</span></a>
<a class="sourceLine" id="cb54-6" data-line-number="6"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">pch =</span> <span class="dv">21</span>, <span class="dt">size =</span> <span class="dv">4</span>, <span class="dt">colour =</span> <span class="st">&quot;gray30&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb54-7" data-line-number="7"><span class="st">  </span><span class="kw">scale_fill_viridis</span>(<span class="dt">option =</span> <span class="st">&quot;A&quot;</span>, <span class="dt">name =</span> <span class="st">&quot;clone1</span><span class="ch">\n</span><span class="st">posterior</span><span class="ch">\n</span><span class="st">probability&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb54-8" data-line-number="8"><span class="st">  </span><span class="kw">theme_classic</span>(<span class="dv">14</span>)</a>
<a class="sourceLine" id="cb54-9" data-line-number="9"></a>
<a class="sourceLine" id="cb54-10" data-line-number="10">ppc2 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(pca_unst, <span class="kw">aes</span>(<span class="dt">x =</span> PC2, <span class="dt">y =</span> PC3, <span class="dt">fill =</span> clone2_prob)) <span class="op">+</span></a>
<a class="sourceLine" id="cb54-11" data-line-number="11"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">pch =</span> <span class="dv">21</span>, <span class="dt">size =</span> <span class="dv">4</span>, <span class="dt">colour =</span> <span class="st">&quot;gray30&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb54-12" data-line-number="12"><span class="st">  </span><span class="kw">scale_fill_viridis</span>(<span class="dt">option =</span> <span class="st">&quot;A&quot;</span>, <span class="dt">name =</span> <span class="st">&quot;clone2</span><span class="ch">\n</span><span class="st">posterior</span><span class="ch">\n</span><span class="st">probability&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb54-13" data-line-number="13"><span class="st">  </span><span class="kw">theme_classic</span>(<span class="dv">14</span>)</a>
<a class="sourceLine" id="cb54-14" data-line-number="14"></a>
<a class="sourceLine" id="cb54-15" data-line-number="15">ppc3 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(pca_unst, <span class="kw">aes</span>(<span class="dt">x =</span> PC2, <span class="dt">y =</span> PC3, <span class="dt">fill =</span> clone3_prob)) <span class="op">+</span></a>
<a class="sourceLine" id="cb54-16" data-line-number="16"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">pch =</span> <span class="dv">21</span>, <span class="dt">size =</span> <span class="dv">4</span>, <span class="dt">colour =</span> <span class="st">&quot;gray30&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb54-17" data-line-number="17"><span class="st">  </span><span class="kw">scale_fill_viridis</span>(<span class="dt">option =</span> <span class="st">&quot;A&quot;</span>, <span class="dt">name =</span> <span class="st">&quot;clone3</span><span class="ch">\n</span><span class="st">posterior</span><span class="ch">\n</span><span class="st">probability&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb54-18" data-line-number="18"><span class="st">  </span><span class="kw">theme_classic</span>(<span class="dv">14</span>)</a>
<a class="sourceLine" id="cb54-19" data-line-number="19"></a>
<a class="sourceLine" id="cb54-20" data-line-number="20"><span class="kw">plot_grid</span>(ppc1, ppc2, ppc3, <span class="dt">labels =</span> <span class="st">&quot;auto&quot;</span>, <span class="dt">ncol =</span> <span class="dv">1</span>)</a></code></pre></div>
<p><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/pca-clone-prob-2.png" width="912" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb55-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_pca_pc2_pc3_clone_probs.png&quot;</span>, <span class="dt">height =</span> <span class="dv">11</span>, <span class="dt">width =</span> <span class="fl">9.5</span>)</a>
<a class="sourceLine" id="cb55-2" data-line-number="2"><span class="kw">ggsave</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_pca_pc2_pc3_clone_probs.pdf&quot;</span>, <span class="dt">height =</span> <span class="dv">11</span>, <span class="dt">width =</span> <span class="fl">9.5</span>)</a>
<a class="sourceLine" id="cb55-3" data-line-number="3"><span class="kw">ggsave</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_pca_pc2_pc3_clone_probs.svg&quot;</span>, <span class="dt">height =</span> <span class="dv">11</span>, <span class="dt">width =</span> <span class="fl">9.5</span>)</a></code></pre></div>
<p>We can also explore how inferred cell cycle phase information relates to the PCA components.</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb56-1" data-line-number="1">pca_unst<span class="op">$</span>cyclone_phase &lt;-<span class="st"> </span><span class="kw">factor</span>(pca_unst<span class="op">$</span>cyclone_phase, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;G1&quot;</span>, <span class="st">&quot;S&quot;</span>, <span class="st">&quot;G2M&quot;</span>))</a>
<a class="sourceLine" id="cb56-2" data-line-number="2"><span class="kw">ggplot</span>(pca_unst, <span class="kw">aes</span>(<span class="dt">x =</span> PC1, <span class="dt">y =</span> PC2, <span class="dt">colour =</span> cyclone_phase,</a>
<a class="sourceLine" id="cb56-3" data-line-number="3">                     <span class="dt">shape =</span> clone)) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-4" data-line-number="4"><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">6</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-5" data-line-number="5"><span class="st">    </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> <span class="kw">magma</span>(<span class="dv">6</span>)[<span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span>, <span class="dv">5</span>)], <span class="dt">name =</span> <span class="st">&quot;cell cycle</span><span class="ch">\n</span><span class="st">phase&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-6" data-line-number="6"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;PC1 (10% variance)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-7" data-line-number="7"><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;PC2 (5% variance)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-8" data-line-number="8"><span class="st">    </span><span class="kw">theme_classic</span>(<span class="dv">18</span>)</a></code></pre></div>
<p><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/pca-cc-1.png" width="768" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb57-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_pca.png&quot;</span>, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">width =</span> <span class="fl">9.5</span>)</a>
<a class="sourceLine" id="cb57-2" data-line-number="2"><span class="kw">ggsave</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_pca.pdf&quot;</span>, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">width =</span> <span class="fl">9.5</span>)</a>
<a class="sourceLine" id="cb57-3" data-line-number="3"><span class="kw">ggsave</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_pca.svg&quot;</span>, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">width =</span> <span class="fl">9.5</span>)</a>
<a class="sourceLine" id="cb57-4" data-line-number="4"></a>
<a class="sourceLine" id="cb57-5" data-line-number="5"></a>
<a class="sourceLine" id="cb57-6" data-line-number="6">pca_unst<span class="op">$</span>cyclone_phase &lt;-<span class="st"> </span><span class="kw">factor</span>(pca_unst<span class="op">$</span>cyclone_phase, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;G1&quot;</span>, <span class="st">&quot;S&quot;</span>, <span class="st">&quot;G2M&quot;</span>))</a>
<a class="sourceLine" id="cb57-7" data-line-number="7"><span class="kw">ggplot</span>(pca_unst, <span class="kw">aes</span>(<span class="dt">x =</span> PC2, <span class="dt">y =</span> PC3, <span class="dt">colour =</span> cyclone_phase,</a>
<a class="sourceLine" id="cb57-8" data-line-number="8">                     <span class="dt">shape =</span> clone)) <span class="op">+</span></a>
<a class="sourceLine" id="cb57-9" data-line-number="9"><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">6</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb57-10" data-line-number="10"><span class="st">    </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> <span class="kw">magma</span>(<span class="dv">6</span>)[<span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span>, <span class="dv">5</span>)], <span class="dt">name =</span> <span class="st">&quot;cell cycle</span><span class="ch">\n</span><span class="st">phase&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb57-11" data-line-number="11"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;PC2 (5% variance)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb57-12" data-line-number="12"><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;PC3 (3% variance)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb57-13" data-line-number="13"><span class="st">    </span><span class="kw">theme_classic</span>(<span class="dv">18</span>)</a></code></pre></div>
<p><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/pca-cc-2.png" width="768" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb58-1" data-line-number="1"><span class="kw">ggplot</span>(pca_unst, <span class="kw">aes</span>(<span class="dt">x =</span> PC1, <span class="dt">y =</span> PC2, <span class="dt">fill =</span> G2M,</a>
<a class="sourceLine" id="cb58-2" data-line-number="2">                     <span class="dt">shape =</span> clone)) <span class="op">+</span></a>
<a class="sourceLine" id="cb58-3" data-line-number="3"><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">colour =</span> <span class="st">&quot;gray50&quot;</span>, <span class="dt">size =</span> <span class="dv">5</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb58-4" data-line-number="4"><span class="st">    </span><span class="kw">scale_shape_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="dv">21</span>, <span class="dv">23</span>, <span class="dv">25</span>), <span class="dt">name =</span> <span class="st">&quot;clone&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb58-5" data-line-number="5"><span class="st">    </span>scico<span class="op">::</span><span class="kw">scale_fill_scico</span>(<span class="dt">palette =</span> <span class="st">&quot;bilbao&quot;</span>, <span class="dt">name  =</span> <span class="st">&quot;G2/M score&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb58-6" data-line-number="6"><span class="st">    </span><span class="kw">scale_size_continuous</span>(<span class="dt">range =</span> <span class="kw">c</span>(<span class="dv">4</span>, <span class="dv">6</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb58-7" data-line-number="7"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;PC1 (10% variance)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb58-8" data-line-number="8"><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;PC2 (5% variance)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb58-9" data-line-number="9"><span class="st">    </span><span class="kw">theme_classic</span>(<span class="dv">18</span>)</a></code></pre></div>
<p><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/pca-cc-3.png" width="768" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb59-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_pca_g2m_score.png&quot;</span>, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">width =</span> <span class="fl">9.5</span>)</a>
<a class="sourceLine" id="cb59-2" data-line-number="2"><span class="kw">ggsave</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_pca_g2m_score.pdf&quot;</span>, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">width =</span> <span class="fl">9.5</span>)</a>
<a class="sourceLine" id="cb59-3" data-line-number="3"><span class="kw">ggsave</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_pca_g2m_score.svg&quot;</span>, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">width =</span> <span class="fl">9.5</span>)</a>
<a class="sourceLine" id="cb59-4" data-line-number="4"></a>
<a class="sourceLine" id="cb59-5" data-line-number="5"><span class="kw">ggplot</span>(pca_unst, <span class="kw">aes</span>(<span class="dt">x =</span> PC1, <span class="dt">y =</span> PC2, <span class="dt">colour =</span> S,</a>
<a class="sourceLine" id="cb59-6" data-line-number="6">                     <span class="dt">shape =</span> clone)) <span class="op">+</span></a>
<a class="sourceLine" id="cb59-7" data-line-number="7"><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">5</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb59-8" data-line-number="8"><span class="st">    </span><span class="kw">scale_color_viridis</span>(<span class="dt">option =</span> <span class="st">&quot;B&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb59-9" data-line-number="9"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;PC1 (10% variance)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb59-10" data-line-number="10"><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;PC2 (5% variance&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb59-11" data-line-number="11"><span class="st">    </span><span class="kw">theme_classic</span>(<span class="dv">18</span>)</a></code></pre></div>
<p><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/pca-cc-4.png" width="768" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb60-1" data-line-number="1"><span class="kw">ggplot</span>(pca_unst, <span class="kw">aes</span>(<span class="dt">x =</span> PC1, <span class="dt">y =</span> PC2, <span class="dt">colour =</span> G1,</a>
<a class="sourceLine" id="cb60-2" data-line-number="2">                     <span class="dt">shape =</span> clone)) <span class="op">+</span></a>
<a class="sourceLine" id="cb60-3" data-line-number="3"><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">5</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb60-4" data-line-number="4"><span class="st">    </span><span class="kw">scale_color_viridis</span>(<span class="dt">option =</span> <span class="st">&quot;B&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb60-5" data-line-number="5"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;PC1 (10% variance)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb60-6" data-line-number="6"><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;PC2 (5% variance&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb60-7" data-line-number="7"><span class="st">    </span><span class="kw">theme_classic</span>(<span class="dv">18</span>)</a></code></pre></div>
<p><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/pca-cc-5.png" width="768" style="display: block; margin: auto;" /></p>
<p>Number of variants used for clone ID looks to have little relationship to global structure in expression PCA space.</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb61-1" data-line-number="1"><span class="kw">ggplot</span>(pca_unst, <span class="kw">aes</span>(<span class="dt">x =</span> PC1, <span class="dt">y =</span> PC2, <span class="dt">fill =</span> clone2_prob, <span class="dt">size =</span> nvars_cloneid)) <span class="op">+</span></a>
<a class="sourceLine" id="cb61-2" data-line-number="2"><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">pch =</span> <span class="dv">21</span>, <span class="dt">colour =</span> <span class="st">&quot;gray30&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb61-3" data-line-number="3"><span class="st">    </span><span class="kw">scale_fill_viridis</span>(<span class="dt">option =</span> <span class="st">&quot;B&quot;</span>, <span class="dt">name =</span> <span class="st">&quot;clone2</span><span class="ch">\n</span><span class="st">probability&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb61-4" data-line-number="4"><span class="st">    </span><span class="kw">scale_size_continuous</span>(<span class="dt">name =</span> <span class="st">&quot;# variants</span><span class="ch">\n</span><span class="st">for clone ID&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb61-5" data-line-number="5"><span class="st">    </span><span class="kw">theme_classic</span>(<span class="dv">14</span>)</a></code></pre></div>
<p><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/pca-nvars-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="combined-figure" class="section level2">
<h2>Combined figure</h2>
<p>For publication, we put together a combined figure summarising the analyses conducted above.</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb62-1" data-line-number="1"><span class="co">## tree and cell assignment</span></a>
<a class="sourceLine" id="cb62-2" data-line-number="2">fig_tree &lt;-<span class="st"> </span><span class="kw">plot_tree</span>(cell_assign_joxm<span class="op">$</span>tree_updated, <span class="dt">orient =</span> <span class="st">&quot;v&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb62-3" data-line-number="3"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Clonal tree&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-4" data-line-number="4"><span class="st">    </span>cardelino<span class="op">:::</span><span class="kw">heatmap.theme</span>(<span class="dt">size =</span> <span class="dv">16</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-5" data-line-number="5"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.title.y =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">20</span>))</a>
<a class="sourceLine" id="cb62-6" data-line-number="6"></a>
<a class="sourceLine" id="cb62-7" data-line-number="7">prob_to_plot &lt;-<span class="st"> </span>cell_assign_joxm<span class="op">$</span>prob_mat[</a>
<a class="sourceLine" id="cb62-8" data-line-number="8">    <span class="kw">colnames</span>(sce_joxm)[sce_joxm<span class="op">$</span>well_condition <span class="op">==</span><span class="st"> &quot;unstimulated&quot;</span>], ]</a>
<a class="sourceLine" id="cb62-9" data-line-number="9">hc &lt;-<span class="st"> </span><span class="kw">hclust</span>(<span class="kw">dist</span>(prob_to_plot))</a>
<a class="sourceLine" id="cb62-10" data-line-number="10"></a>
<a class="sourceLine" id="cb62-11" data-line-number="11">clone_ids &lt;-<span class="st"> </span><span class="kw">colnames</span>(prob_to_plot)</a>
<a class="sourceLine" id="cb62-12" data-line-number="12">clone_frac &lt;-<span class="st"> </span><span class="kw">colMeans</span>(prob_to_plot[matrixStats<span class="op">::</span><span class="kw">rowMaxs</span>(prob_to_plot) <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.5</span>,])</a>
<a class="sourceLine" id="cb62-13" data-line-number="13">clone_perc &lt;-<span class="st"> </span><span class="kw">paste0</span>(clone_ids, <span class="st">&quot;: &quot;</span>, </a>
<a class="sourceLine" id="cb62-14" data-line-number="14">                          <span class="kw">round</span>(clone_frac<span class="op">*</span><span class="dv">100</span>, <span class="dt">digits =</span> <span class="dv">1</span>), <span class="st">&quot;%&quot;</span>)</a>
<a class="sourceLine" id="cb62-15" data-line-number="15"></a>
<a class="sourceLine" id="cb62-16" data-line-number="16"><span class="kw">colnames</span>(prob_to_plot) &lt;-<span class="st"> </span>clone_perc</a>
<a class="sourceLine" id="cb62-17" data-line-number="17">nba.m &lt;-<span class="st"> </span><span class="kw">as_data_frame</span>(prob_to_plot[hc<span class="op">$</span>order,]) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb62-18" data-line-number="18"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">cell =</span> <span class="kw">rownames</span>(prob_to_plot[hc<span class="op">$</span>order,])) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb62-19" data-line-number="19"><span class="st">    </span><span class="kw">gather</span>(<span class="dt">key =</span> <span class="st">&quot;clone&quot;</span>, <span class="dt">value =</span> <span class="st">&quot;prob&quot;</span>, <span class="op">-</span>cell)</a>
<a class="sourceLine" id="cb62-20" data-line-number="20">nba.m &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(nba.m, <span class="dt">cell =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb62-21" data-line-number="21">    cell, <span class="dt">levels =</span> <span class="kw">rownames</span>(prob_to_plot[hc<span class="op">$</span>order,])))</a>
<a class="sourceLine" id="cb62-22" data-line-number="22">fig_assign &lt;-<span class="st"> </span><span class="kw">ggplot</span>(nba.m, <span class="kw">aes</span>(clone, cell, <span class="dt">fill =</span> prob)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb62-23" data-line-number="23"><span class="st">    </span><span class="kw">geom_tile</span>(<span class="dt">show.legend =</span> <span class="ot">TRUE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-24" data-line-number="24"><span class="st">    </span><span class="co"># scale_fill_gradient(low = &quot;white&quot;, high = &quot;firebrick4&quot;,</span></a>
<a class="sourceLine" id="cb62-25" data-line-number="25"><span class="st">    </span><span class="co">#                     name = &quot;posterior probability of assignment&quot;) +</span></a>
<a class="sourceLine" id="cb62-26" data-line-number="26"><span class="st">    </span>scico<span class="op">::</span><span class="kw">scale_fill_scico</span>(<span class="dt">palette =</span> <span class="st">&quot;oleron&quot;</span>, <span class="dt">direction =</span> <span class="dv">1</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-27" data-line-number="27"><span class="st">    </span><span class="kw">ylab</span>(<span class="kw">paste</span>(<span class="st">&quot;Single cells&quot;</span>)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb62-28" data-line-number="28"><span class="st">    </span>cardelino<span class="op">:::</span><span class="kw">heatmap.theme</span>(<span class="dt">size =</span> <span class="dv">16</span>) <span class="op">+</span><span class="st"> </span><span class="co">#cardelino:::pub.theme() +</span></a>
<a class="sourceLine" id="cb62-29" data-line-number="29"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.title.y =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">20</span>), <span class="dt">legend.position =</span> <span class="st">&quot;bottom&quot;</span>,</a>
<a class="sourceLine" id="cb62-30" data-line-number="30">          <span class="dt">legend.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">12</span>), <span class="dt">legend.key.size =</span> <span class="kw">unit</span>(<span class="fl">0.05</span>, <span class="st">&quot;npc&quot;</span>))</a>
<a class="sourceLine" id="cb62-31" data-line-number="31"></a>
<a class="sourceLine" id="cb62-32" data-line-number="32">p_tree &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(fig_tree, fig_assign, <span class="dt">nrow =</span> <span class="dv">2</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.46</span>, <span class="fl">0.52</span>))</a>
<a class="sourceLine" id="cb62-33" data-line-number="33"></a>
<a class="sourceLine" id="cb62-34" data-line-number="34"></a>
<a class="sourceLine" id="cb62-35" data-line-number="35"><span class="co">## cell cycle barplot</span></a>
<a class="sourceLine" id="cb62-36" data-line-number="36">p_bar &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">colData</span>(de_res[[<span class="st">&quot;sce_list_unst&quot;</span>]][[<span class="st">&quot;joxm&quot;</span>]])) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb62-37" data-line-number="37"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">Cell_Cycle =</span> <span class="kw">factor</span>(cyclone_phase, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;G2M&quot;</span>, <span class="st">&quot;S&quot;</span>, <span class="st">&quot;G1&quot;</span>)),</a>
<a class="sourceLine" id="cb62-38" data-line-number="38">                <span class="dt">assigned =</span> <span class="kw">factor</span>(assigned, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;clone3&quot;</span>, <span class="st">&quot;clone2&quot;</span>, <span class="st">&quot;clone1&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb62-39" data-line-number="39"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> assigned, <span class="dt">fill =</span> Cell_Cycle, <span class="dt">group =</span> Cell_Cycle)) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-40" data-line-number="40"><span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">position =</span> <span class="st">&quot;fill&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-41" data-line-number="41"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;#ff6a5c&quot;</span>, <span class="st">&quot;#ccdfcb&quot;</span>, <span class="st">&quot;#414141&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-42" data-line-number="42"><span class="st">  </span><span class="kw">coord_flip</span>() <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb62-43" data-line-number="43"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;proportion of cells&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-44" data-line-number="44"><span class="st">  </span><span class="kw">guides</span>(<span class="dt">fill =</span> <span class="kw">guide_legend</span>(<span class="dt">reverse =</span> <span class="ot">TRUE</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-45" data-line-number="45"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.title.y =</span> <span class="kw">element_blank</span>())</a>
<a class="sourceLine" id="cb62-46" data-line-number="46"></a>
<a class="sourceLine" id="cb62-47" data-line-number="47"><span class="co">## effects on mutated clone</span></a>
<a class="sourceLine" id="cb62-48" data-line-number="48">df_to_plot &lt;-<span class="st"> </span>mut_genes_df_allcells <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb62-49" data-line-number="49"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(logFC), donor <span class="op">==</span><span class="st"> &quot;joxm&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb62-50" data-line-number="50"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">duplicated</span>(gene)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb62-51" data-line-number="51"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(</a>
<a class="sourceLine" id="cb62-52" data-line-number="52">    <span class="dt">FDR =</span> <span class="kw">p.adjust</span>(PValue, <span class="dt">method =</span> <span class="st">&quot;BH&quot;</span>),</a>
<a class="sourceLine" id="cb62-53" data-line-number="53">    <span class="dt">consequence_simplified =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb62-54" data-line-number="54">      consequence_simplified, </a>
<a class="sourceLine" id="cb62-55" data-line-number="55">      <span class="kw">levels</span>(<span class="kw">as.factor</span>(consequence_simplified))[<span class="kw">order</span>(tmp4[[<span class="st">&quot;med&quot;</span>]])]),</a>
<a class="sourceLine" id="cb62-56" data-line-number="56">    <span class="dt">de  =</span> <span class="kw">ifelse</span>(FDR <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.2</span>, <span class="st">&quot;FDR &lt; 0.2&quot;</span>, <span class="st">&quot;FDR &gt; 0.2&quot;</span>))</a>
<a class="sourceLine" id="cb62-57" data-line-number="57"></a>
<a class="sourceLine" id="cb62-58" data-line-number="58"></a>
<a class="sourceLine" id="cb62-59" data-line-number="59">p_mutated_clone &lt;-<span class="st"> </span><span class="kw">ggplot</span>(df_to_plot, <span class="kw">aes</span>(<span class="dt">y =</span> logFC, <span class="dt">x =</span> consequence_simplified)) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-60" data-line-number="60"><span class="st">    </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="dv">0</span>, <span class="dt">linetype =</span> <span class="dv">1</span>, <span class="dt">colour =</span> <span class="st">&quot;black&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-61" data-line-number="61"><span class="st">    </span><span class="kw">geom_boxplot</span>(<span class="dt">outlier.size =</span> <span class="dv">0</span>, <span class="dt">outlier.alpha =</span> <span class="dv">0</span>, <span class="dt">fill =</span> <span class="st">&quot;gray90&quot;</span>,</a>
<a class="sourceLine" id="cb62-62" data-line-number="62">                 <span class="dt">colour =</span> <span class="st">&quot;firebrick4&quot;</span>, <span class="dt">width =</span> <span class="fl">0.2</span>, <span class="dt">size =</span> <span class="dv">1</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-63" data-line-number="63"><span class="st">    </span>ggbeeswarm<span class="op">::</span><span class="kw">geom_quasirandom</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> <span class="op">-</span><span class="kw">log10</span>(PValue)), </a>
<a class="sourceLine" id="cb62-64" data-line-number="64">                                 <span class="dt">colour =</span> <span class="st">&quot;gray40&quot;</span>, <span class="dt">pch =</span> <span class="dv">21</span>, <span class="dt">size =</span> <span class="dv">4</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-65" data-line-number="65"><span class="st">    </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">y =</span> <span class="fl">-0.25</span>, <span class="dt">x =</span> <span class="dv">0</span>, <span class="dt">yend =</span> <span class="dv">-1</span>, <span class="dt">xend =</span> <span class="dv">0</span>), </a>
<a class="sourceLine" id="cb62-66" data-line-number="66">                 <span class="dt">colour =</span> <span class="st">&quot;black&quot;</span>, <span class="dt">size =</span> <span class="dv">1</span>, <span class="dt">arrow =</span> <span class="kw">arrow</span>(<span class="dt">length =</span> <span class="kw">unit</span>(<span class="fl">0.5</span>, <span class="st">&quot;cm&quot;</span>))) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-67" data-line-number="67"><span class="st">    </span><span class="kw">annotate</span>(<span class="st">&quot;text&quot;</span>, <span class="dt">y =</span> <span class="dv">-3</span>, <span class="dt">x =</span> <span class="dv">0</span>, <span class="dt">size =</span> <span class="dv">6</span>, <span class="dt">label =</span> <span class="st">&quot;lower in mutated clone&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-68" data-line-number="68"><span class="st">    </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">y =</span> <span class="fl">0.25</span>, <span class="dt">x =</span> <span class="dv">0</span>, <span class="dt">yend =</span> <span class="dv">1</span>, <span class="dt">xend =</span> <span class="dv">0</span>), </a>
<a class="sourceLine" id="cb62-69" data-line-number="69">                 <span class="dt">colour =</span> <span class="st">&quot;black&quot;</span>, <span class="dt">size =</span> <span class="dv">1</span>, <span class="dt">arrow =</span> <span class="kw">arrow</span>(<span class="dt">length =</span> <span class="kw">unit</span>(<span class="fl">0.5</span>, <span class="st">&quot;cm&quot;</span>))) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-70" data-line-number="70"><span class="st">    </span><span class="kw">annotate</span>(<span class="st">&quot;text&quot;</span>, <span class="dt">y =</span> <span class="dv">3</span>, <span class="dt">x =</span> <span class="dv">0</span>, <span class="dt">size =</span> <span class="dv">6</span>, <span class="dt">label =</span> <span class="st">&quot;higher in mutated clone&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-71" data-line-number="71"><span class="st">    </span><span class="kw">scale_x_discrete</span>(<span class="dt">expand =</span> <span class="kw">c</span>(<span class="fl">0.1</span>, <span class="fl">.05</span>), <span class="dt">name =</span> <span class="st">&quot;consequence&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-72" data-line-number="72"><span class="st">    </span><span class="kw">scale_y_continuous</span>(<span class="dt">expand =</span> <span class="kw">c</span>(<span class="fl">0.1</span>, <span class="fl">0.1</span>), <span class="dt">name =</span> <span class="st">&quot;logFC&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-73" data-line-number="73"><span class="st">    </span><span class="kw">expand_limits</span>(<span class="dt">x =</span> <span class="kw">c</span>(<span class="op">-</span><span class="fl">0.75</span>, <span class="dv">8</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-74" data-line-number="74"><span class="st">    </span><span class="kw">theme_ridges</span>(<span class="dv">22</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-75" data-line-number="75"><span class="st">    </span><span class="kw">coord_flip</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb62-76" data-line-number="76"><span class="st">    </span><span class="kw">scale_fill_viridis</span>(<span class="dt">option =</span> <span class="st">&quot;B&quot;</span>, <span class="dt">name =</span> <span class="st">&quot;-log10(P)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-77" data-line-number="77"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">strip.background =</span> <span class="kw">element_rect</span>(<span class="dt">fill =</span> <span class="st">&quot;gray90&quot;</span>),</a>
<a class="sourceLine" id="cb62-78" data-line-number="78">          <span class="dt">legend.position =</span> <span class="st">&quot;right&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-79" data-line-number="79"><span class="st">    </span><span class="kw">guides</span>(<span class="dt">color =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb62-80" data-line-number="80"></a>
<a class="sourceLine" id="cb62-81" data-line-number="81"><span class="co">## PCA</span></a>
<a class="sourceLine" id="cb62-82" data-line-number="82">p_pca &lt;-<span class="st"> </span><span class="kw">ggplot</span>(pca_unst, <span class="kw">aes</span>(<span class="dt">x =</span> PC2, <span class="dt">y =</span> PC3, <span class="dt">fill =</span> clone,</a>
<a class="sourceLine" id="cb62-83" data-line-number="83">                     <span class="dt">shape =</span> clone)) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-84" data-line-number="84"><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">colour =</span> <span class="st">&quot;gray50&quot;</span>, <span class="dt">size =</span> <span class="dv">5</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-85" data-line-number="85"><span class="st">    </span><span class="kw">scale_shape_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="dv">21</span>, <span class="dv">23</span>, <span class="dv">25</span>, <span class="dv">22</span>, <span class="dv">24</span>, <span class="dv">26</span>), <span class="dt">name =</span> <span class="st">&quot;clone&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-86" data-line-number="86"><span class="st">    </span><span class="co"># scico::scale_fill_scico(palette = &quot;bilbao&quot;, name  = &quot;G2/M score&quot;) +</span></a>
<a class="sourceLine" id="cb62-87" data-line-number="87"><span class="st">    </span>ggthemes<span class="op">::</span><span class="kw">scale_fill_canva</span>(<span class="dt">palette =</span> <span class="st">&quot;Surf and turf&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-88" data-line-number="88"><span class="st">    </span><span class="kw">scale_size_continuous</span>(<span class="dt">range =</span> <span class="kw">c</span>(<span class="dv">4</span>, <span class="dv">6</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-89" data-line-number="89"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;PC2 (5% variance)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-90" data-line-number="90"><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;PC3 (3% variance)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-91" data-line-number="91"><span class="st">    </span><span class="kw">theme_classic</span>(<span class="dv">18</span>)</a>
<a class="sourceLine" id="cb62-92" data-line-number="92"></a>
<a class="sourceLine" id="cb62-93" data-line-number="93"> <span class="co"># ggplot(pca_unst, aes(x = PC2, y = PC3, colour = clone,</span></a>
<a class="sourceLine" id="cb62-94" data-line-number="94"> <span class="co">#                     shape = cyclone_phase)) +</span></a>
<a class="sourceLine" id="cb62-95" data-line-number="95"> <span class="co">#    geom_point(alpha = 0.9, size = 5) +</span></a>
<a class="sourceLine" id="cb62-96" data-line-number="96"> <span class="co">#    scale_shape_manual(values = c(15, 17, 19), name = &quot;clone&quot;) +</span></a>
<a class="sourceLine" id="cb62-97" data-line-number="97"> <span class="co">#    # scico::scale_fill_scico(palette = &quot;bilbao&quot;, name  = &quot;G2/M score&quot;) +</span></a>
<a class="sourceLine" id="cb62-98" data-line-number="98"> <span class="co">#    ggthemes::scale_color_canva(palette = &quot;Surf and turf&quot;) +</span></a>
<a class="sourceLine" id="cb62-99" data-line-number="99"> <span class="co">#    scale_size_continuous(range = c(4, 6)) +</span></a>
<a class="sourceLine" id="cb62-100" data-line-number="100"> <span class="co">#    xlab(&quot;PC2 (5% variance)&quot;) +</span></a>
<a class="sourceLine" id="cb62-101" data-line-number="101"> <span class="co">#    ylab(&quot;PC3 (3% variance)&quot;) +</span></a>
<a class="sourceLine" id="cb62-102" data-line-number="102"> <span class="co">#    theme_classic(18)</span></a>
<a class="sourceLine" id="cb62-103" data-line-number="103"></a>
<a class="sourceLine" id="cb62-104" data-line-number="104"><span class="co">## volcano</span></a>
<a class="sourceLine" id="cb62-105" data-line-number="105">de_joxm_cl2_vs_cl1 &lt;-<span class="st"> </span>de_res<span class="op">$</span>qlf_pairwise<span class="op">$</span>joxm<span class="op">$</span>clone2_clone1<span class="op">$</span>table</a>
<a class="sourceLine" id="cb62-106" data-line-number="106">de_joxm_cl2_vs_cl1[[<span class="st">&quot;gene&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">rownames</span>(de_joxm_cl2_vs_cl1)</a>
<a class="sourceLine" id="cb62-107" data-line-number="107">de_joxm_cl2_vs_cl1 &lt;-<span class="st"> </span>de_joxm_cl2_vs_cl1 <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb62-108" data-line-number="108"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">FDR =</span> <span class="kw">adj_pvalues</span>(<span class="kw">ihw</span>(PValue <span class="op">~</span><span class="st"> </span>logCPM, <span class="dt">alpha =</span> <span class="fl">0.1</span>)), </a>
<a class="sourceLine" id="cb62-109" data-line-number="109">                <span class="dt">sig =</span> FDR <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.1</span>,</a>
<a class="sourceLine" id="cb62-110" data-line-number="110">                <span class="dt">signed_F =</span> <span class="kw">sign</span>(logFC) <span class="op">*</span><span class="st"> </span>F) </a>
<a class="sourceLine" id="cb62-111" data-line-number="111">de_joxm_cl2_vs_cl1[[<span class="st">&quot;lab&quot;</span>]] &lt;-<span class="st"> &quot;&quot;</span></a>
<a class="sourceLine" id="cb62-112" data-line-number="112">int_genes_entrezid &lt;-<span class="st"> </span><span class="kw">c</span>(Hs.H<span class="op">$</span>HALLMARK_G2M_CHECKPOINT, Hs.H<span class="op">$</span>HALLMARK_E2F_TARGETS,</a>
<a class="sourceLine" id="cb62-113" data-line-number="113">                        Hs.H<span class="op">$</span>HALLMARK_MITOTIC_SPINDLE)</a>
<a class="sourceLine" id="cb62-114" data-line-number="114">mm &lt;-<span class="st"> </span><span class="kw">match</span>(int_genes_entrezid, de_joxm_cl2_vs_cl1<span class="op">$</span>entrezid)</a>
<a class="sourceLine" id="cb62-115" data-line-number="115">mm &lt;-<span class="st"> </span>mm[<span class="op">!</span><span class="kw">is.na</span>(mm)]</a>
<a class="sourceLine" id="cb62-116" data-line-number="116">int_genes_hgnc &lt;-<span class="st"> </span>de_joxm_cl2_vs_cl1<span class="op">$</span>hgnc_symbol[mm]</a>
<a class="sourceLine" id="cb62-117" data-line-number="117">genes_to_label &lt;-<span class="st"> </span>(de_joxm_cl2_vs_cl1[[<span class="st">&quot;hgnc_symbol&quot;</span>]] <span class="op">%in%</span><span class="st"> </span>int_genes_hgnc <span class="op">&amp;</span></a>
<a class="sourceLine" id="cb62-118" data-line-number="118"><span class="st">                     </span>de_joxm_cl2_vs_cl1[[<span class="st">&quot;FDR&quot;</span>]] <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.01</span>)</a>
<a class="sourceLine" id="cb62-119" data-line-number="119">de_joxm_cl2_vs_cl1[[<span class="st">&quot;lab&quot;</span>]][genes_to_label] &lt;-</a>
<a class="sourceLine" id="cb62-120" data-line-number="120"><span class="st">  </span>de_joxm_cl2_vs_cl1[[<span class="st">&quot;hgnc_symbol&quot;</span>]][genes_to_label]</a>
<a class="sourceLine" id="cb62-121" data-line-number="121">de_joxm_cl2_vs_cl1[[<span class="st">&quot;cell_cycle_gene&quot;</span>]] &lt;-<span class="st"> </span>(de_joxm_cl2_vs_cl1<span class="op">$</span>entrezid <span class="op">%in%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb62-122" data-line-number="122"><span class="st">                                              </span>int_genes_entrezid)</a>
<a class="sourceLine" id="cb62-123" data-line-number="123"></a>
<a class="sourceLine" id="cb62-124" data-line-number="124">p_volcano &lt;-<span class="st"> </span><span class="kw">ggplot</span>(de_joxm_cl2_vs_cl1, <span class="kw">aes</span>(<span class="dt">x =</span> logFC, <span class="dt">y =</span> <span class="op">-</span><span class="kw">log10</span>(PValue), </a>
<a class="sourceLine" id="cb62-125" data-line-number="125">                                            <span class="dt">fill =</span> sig, <span class="dt">label =</span> lab)) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-126" data-line-number="126"><span class="st">  </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">size =</span> sig), <span class="dt">pch =</span> <span class="dv">21</span>, <span class="dt">colour =</span> <span class="st">&quot;gray40&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-127" data-line-number="127"><span class="st">  </span><span class="kw">geom_label_repel</span>(<span class="dt">show.legend =</span> <span class="ot">FALSE</span>, </a>
<a class="sourceLine" id="cb62-128" data-line-number="128">                   <span class="dt">arrow =</span> <span class="kw">arrow</span>(<span class="dt">type =</span> <span class="st">&quot;closed&quot;</span>, <span class="dt">length =</span> <span class="kw">unit</span>(<span class="fl">0.25</span>, <span class="st">&quot;cm&quot;</span>)), </a>
<a class="sourceLine" id="cb62-129" data-line-number="129">                   <span class="dt">nudge_x =</span> <span class="fl">0.2</span>, <span class="dt">nudge_y =</span> <span class="fl">0.3</span>, <span class="dt">fill =</span> <span class="st">&quot;gray95&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-130" data-line-number="130"><span class="st">  </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">x =</span> <span class="dv">-1</span>, <span class="dt">y =</span> <span class="dv">0</span>, <span class="dt">xend =</span> <span class="dv">-4</span>, <span class="dt">yend =</span> <span class="dv">0</span>), </a>
<a class="sourceLine" id="cb62-131" data-line-number="131">               <span class="dt">colour =</span> <span class="st">&quot;black&quot;</span>, <span class="dt">size =</span> <span class="dv">1</span>, <span class="dt">arrow =</span> <span class="kw">arrow</span>(<span class="dt">length =</span> <span class="kw">unit</span>(<span class="fl">0.5</span>, <span class="st">&quot;cm&quot;</span>))) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-132" data-line-number="132"><span class="st">  </span><span class="kw">annotate</span>(<span class="st">&quot;text&quot;</span>, <span class="dt">x =</span> <span class="dv">-4</span>, <span class="dt">y =</span> <span class="fl">-0.5</span>, <span class="dt">label =</span> <span class="st">&quot;higher in clone1&quot;</span>, <span class="dt">size =</span> <span class="dv">6</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-133" data-line-number="133"><span class="st">  </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">x =</span> <span class="dv">1</span>, <span class="dt">y =</span> <span class="dv">0</span>, <span class="dt">xend =</span> <span class="dv">4</span>, <span class="dt">yend =</span> <span class="dv">0</span>), </a>
<a class="sourceLine" id="cb62-134" data-line-number="134">               <span class="dt">colour =</span> <span class="st">&quot;black&quot;</span>, <span class="dt">size =</span> <span class="dv">1</span>, <span class="dt">arrow =</span> <span class="kw">arrow</span>(<span class="dt">length =</span> <span class="kw">unit</span>(<span class="fl">0.5</span>, <span class="st">&quot;cm&quot;</span>))) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-135" data-line-number="135"><span class="st">  </span><span class="kw">annotate</span>(<span class="st">&quot;text&quot;</span>, <span class="dt">x =</span> <span class="dv">4</span>, <span class="dt">y =</span> <span class="fl">-0.5</span>, <span class="dt">label =</span> <span class="st">&quot;higher in clone2&quot;</span>, <span class="dt">size =</span> <span class="dv">6</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-136" data-line-number="136"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;gray60&quot;</span>, <span class="st">&quot;firebrick&quot;</span>), </a>
<a class="sourceLine" id="cb62-137" data-line-number="137">                    <span class="dt">label =</span> <span class="kw">c</span>(<span class="st">&quot;N.S.&quot;</span>, <span class="st">&quot;FDR &lt; 10%&quot;</span>), <span class="dt">name =</span> <span class="st">&quot;&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-138" data-line-number="138"><span class="st">  </span><span class="kw">scale_size_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span>), <span class="dt">guide =</span> <span class="ot">FALSE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-139" data-line-number="139"><span class="st">  </span><span class="kw">ylab</span>(<span class="kw">expression</span>(<span class="op">-</span><span class="st">&quot;log&quot;</span>[<span class="dv">10</span>](P))) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-140" data-line-number="140"><span class="st">  </span><span class="kw">xlab</span>(<span class="kw">expression</span>(<span class="st">&quot;log&quot;</span>[<span class="dv">2</span>](FC))) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-141" data-line-number="141"><span class="st">  </span><span class="kw">guides</span>(<span class="dt">alpha =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb62-142" data-line-number="142">         <span class="dt">fill =</span> <span class="kw">guide_legend</span>(<span class="dt">override.aes =</span> <span class="kw">list</span>(<span class="dt">size =</span> <span class="dv">5</span>))) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-143" data-line-number="143"><span class="st">  </span><span class="kw">theme_classic</span>(<span class="dv">20</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;right&quot;</span>)</a>
<a class="sourceLine" id="cb62-144" data-line-number="144"></a>
<a class="sourceLine" id="cb62-145" data-line-number="145"><span class="co"># ggplot(de_joxm_cl2_vs_cl1, aes(x = logFC, y = -log10(PValue), </span></a>
<a class="sourceLine" id="cb62-146" data-line-number="146"><span class="co">#                                fill = cell_cycle_gene, label = lab)) +</span></a>
<a class="sourceLine" id="cb62-147" data-line-number="147"><span class="co">#   geom_point(aes(size = sig), pch = 21, colour = &quot;gray40&quot;) +</span></a>
<a class="sourceLine" id="cb62-148" data-line-number="148"><span class="co">#   geom_point(aes(size = sig), pch = 21, colour = &quot;gray40&quot;, </span></a>
<a class="sourceLine" id="cb62-149" data-line-number="149"><span class="co">#              data = dplyr::filter(de_joxm_cl2_vs_cl1, cell_cycle_gene)) +</span></a>
<a class="sourceLine" id="cb62-150" data-line-number="150"><span class="co">#   geom_label_repel(show.legend = FALSE, </span></a>
<a class="sourceLine" id="cb62-151" data-line-number="151"><span class="co">#                    arrow = arrow(type = &quot;closed&quot;, length = unit(0.25, &quot;cm&quot;)), </span></a>
<a class="sourceLine" id="cb62-152" data-line-number="152"><span class="co">#                    nudge_x = 0.2, nudge_y = 0.3, fill = &quot;gray95&quot;) +</span></a>
<a class="sourceLine" id="cb62-153" data-line-number="153"><span class="co">#   geom_segment(aes(x = -1, y = 0, xend = -4, yend = 0), </span></a>
<a class="sourceLine" id="cb62-154" data-line-number="154"><span class="co">#                colour = &quot;black&quot;, size = 1, arrow = arrow(length = unit(0.5, &quot;cm&quot;))) +</span></a>
<a class="sourceLine" id="cb62-155" data-line-number="155"><span class="co">#   annotate(&quot;text&quot;, x = -4, y = -0.5, label = &quot;higher in clone1&quot;, size = 6) +</span></a>
<a class="sourceLine" id="cb62-156" data-line-number="156"><span class="co">#   geom_segment(aes(x = 1, y = 0, xend = 4, yend = 0), </span></a>
<a class="sourceLine" id="cb62-157" data-line-number="157"><span class="co">#                colour = &quot;black&quot;, size = 1, arrow = arrow(length = unit(0.5, &quot;cm&quot;))) +</span></a>
<a class="sourceLine" id="cb62-158" data-line-number="158"><span class="co">#   annotate(&quot;text&quot;, x = 4, y = -0.5, label = &quot;higher in clone2&quot;, size = 6) +</span></a>
<a class="sourceLine" id="cb62-159" data-line-number="159"><span class="co">#   scale_fill_manual(values = c(&quot;gray60&quot;, &quot;firebrick&quot;), </span></a>
<a class="sourceLine" id="cb62-160" data-line-number="160"><span class="co">#                     label = c(&quot;N.S.&quot;, &quot;FDR &lt; 10%&quot;), name = &quot;&quot;) +</span></a>
<a class="sourceLine" id="cb62-161" data-line-number="161"><span class="co">#   scale_size_manual(values = c(1, 3), guide = FALSE) +</span></a>
<a class="sourceLine" id="cb62-162" data-line-number="162"><span class="co">#   guides(alpha = FALSE) +</span></a>
<a class="sourceLine" id="cb62-163" data-line-number="163"><span class="co">#   theme_classic(20) + theme(legend.position = &quot;right&quot;)</span></a>
<a class="sourceLine" id="cb62-164" data-line-number="164"></a>
<a class="sourceLine" id="cb62-165" data-line-number="165"><span class="co">## genesets</span></a>
<a class="sourceLine" id="cb62-166" data-line-number="166">cam_H_pw &lt;-<span class="st"> </span>de_res<span class="op">$</span>camera<span class="op">$</span>H<span class="op">$</span>joxm<span class="op">$</span>clone2_clone1<span class="op">$</span>logFC</a>
<a class="sourceLine" id="cb62-167" data-line-number="167">cam_H_pw[[<span class="st">&quot;geneset&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">rownames</span>(cam_H_pw)</a>
<a class="sourceLine" id="cb62-168" data-line-number="168">tmp_sig &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="st">&quot;FDR &gt; 10%&quot;</span>, <span class="kw">nrow</span>(cam_H_pw))</a>
<a class="sourceLine" id="cb62-169" data-line-number="169">tmp_sig[cam_H_pw[[<span class="st">&quot;FDR&quot;</span>]] <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>] &lt;-<span class="st"> &quot;FDR &lt; 5%&quot;</span></a>
<a class="sourceLine" id="cb62-170" data-line-number="170">tmp_sig[cam_H_pw[[<span class="st">&quot;FDR&quot;</span>]] <span class="op">&gt;=</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>cam_H_pw[[<span class="st">&quot;FDR&quot;</span>]] <span class="op">&lt;=</span><span class="st"> </span><span class="fl">0.1</span>] &lt;-<span class="st"> &quot;5% &lt; FDR &lt; 10%&quot;</span></a>
<a class="sourceLine" id="cb62-171" data-line-number="171">cam_H_pw[[<span class="st">&quot;sig&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">factor</span>(tmp_sig, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;FDR &gt; 10%&quot;</span>, <span class="st">&quot;5% &lt; FDR &lt; 10%&quot;</span>, <span class="st">&quot;FDR &lt; 5%&quot;</span>))</a>
<a class="sourceLine" id="cb62-172" data-line-number="172">cam_H_pw[[<span class="st">&quot;lab&quot;</span>]] &lt;-<span class="st"> &quot;&quot;</span></a>
<a class="sourceLine" id="cb62-173" data-line-number="173">cam_H_pw[[<span class="st">&quot;lab&quot;</span>]][<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>] &lt;-</a>
<a class="sourceLine" id="cb62-174" data-line-number="174"><span class="st">    </span>cam_H_pw[[<span class="st">&quot;geneset&quot;</span>]][<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>]</a>
<a class="sourceLine" id="cb62-175" data-line-number="175"></a>
<a class="sourceLine" id="cb62-176" data-line-number="176">cam_H_pw &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(</a>
<a class="sourceLine" id="cb62-177" data-line-number="177">    cam_H_pw,</a>
<a class="sourceLine" id="cb62-178" data-line-number="178">    <span class="dt">Direction =</span> <span class="kw">gsub</span>(<span class="st">&quot;clone4&quot;</span>, <span class="st">&quot;clone2&quot;</span>, Direction)</a>
<a class="sourceLine" id="cb62-179" data-line-number="179">)</a>
<a class="sourceLine" id="cb62-180" data-line-number="180">p_genesets &lt;-<span class="st"> </span>cam_H_pw <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb62-181" data-line-number="181"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> Direction, <span class="dt">y =</span> <span class="op">-</span><span class="kw">log10</span>(PValue), <span class="dt">colour =</span> sig, </a>
<a class="sourceLine" id="cb62-182" data-line-number="182">             <span class="dt">label =</span> lab)) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-183" data-line-number="183"><span class="st">  </span>ggbeeswarm<span class="op">::</span><span class="kw">geom_quasirandom</span>(<span class="kw">aes</span>(<span class="dt">size =</span> NGenes)) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-184" data-line-number="184"><span class="st">  </span><span class="kw">geom_label_repel</span>(<span class="dt">show.legend =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb62-185" data-line-number="185">                   <span class="dt">nudge_y =</span> <span class="fl">0.3</span>, <span class="dt">nudge_x =</span> <span class="fl">0.3</span>, <span class="dt">fill =</span> <span class="st">&quot;gray95&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-186" data-line-number="186"><span class="st">  </span><span class="kw">scale_colour_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;gray50&quot;</span>, <span class="st">&quot;darkorange2&quot;</span>, <span class="st">&quot;firebrick&quot;</span>), </a>
<a class="sourceLine" id="cb62-187" data-line-number="187">                      <span class="dt">name =</span> <span class="st">&quot;&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-188" data-line-number="188"><span class="st">  </span><span class="kw">guides</span>(<span class="dt">alpha =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb62-189" data-line-number="189">         <span class="dt">colour =</span> <span class="kw">guide_legend</span>(<span class="dt">override.aes =</span> <span class="kw">list</span>(<span class="dt">size =</span> <span class="dv">5</span>))) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-190" data-line-number="190"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Gene set enrichment direction&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-191" data-line-number="191"><span class="st">  </span><span class="kw">ylab</span>(<span class="kw">expression</span>(<span class="op">-</span><span class="st">&quot;log&quot;</span>[<span class="dv">10</span>](P))) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-192" data-line-number="192"><span class="st">  </span><span class="kw">theme_classic</span>(<span class="dv">20</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;right&quot;</span>)</a>
<a class="sourceLine" id="cb62-193" data-line-number="193"></a>
<a class="sourceLine" id="cb62-194" data-line-number="194"></a>
<a class="sourceLine" id="cb62-195" data-line-number="195"><span class="co">## produce combined fig</span></a>
<a class="sourceLine" id="cb62-196" data-line-number="196"><span class="co">## combine pca and barplot</span></a>
<a class="sourceLine" id="cb62-197" data-line-number="197">p_bar_pca &lt;-<span class="st"> </span><span class="kw">ggdraw</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb62-198" data-line-number="198"><span class="st">  </span><span class="kw">draw_plot</span>(p_pca <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.justification =</span> <span class="st">&quot;top&quot;</span>), <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">1</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-199" data-line-number="199"><span class="st">  </span><span class="kw">draw_plot</span>(p_bar, <span class="dt">x =</span> <span class="fl">0.48</span>, <span class="fl">0.1</span>, <span class="dt">height =</span> <span class="fl">0.35</span>, <span class="dt">width =</span> <span class="fl">0.52</span>, <span class="dt">scale =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb62-200" data-line-number="200"></a>
<a class="sourceLine" id="cb62-201" data-line-number="201">pp &lt;-<span class="st"> </span><span class="kw">ggdraw</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb62-202" data-line-number="202"><span class="st">    </span><span class="kw">draw_plot</span>(p_tree, <span class="dt">x =</span> <span class="dv">0</span>,  <span class="dt">y =</span> <span class="fl">0.45</span>, <span class="dt">width =</span> <span class="fl">0.48</span>, <span class="dt">height =</span> <span class="fl">0.55</span>, <span class="dt">scale =</span> <span class="dv">1</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-203" data-line-number="203"><span class="st">    </span><span class="kw">draw_plot</span>(p_bar_pca, <span class="dt">x =</span> <span class="fl">0.52</span>, <span class="dt">y =</span> <span class="fl">0.45</span>, <span class="dt">width =</span> <span class="fl">0.48</span>, <span class="dt">height =</span> <span class="fl">0.55</span>, <span class="dt">scale =</span> <span class="dv">1</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-204" data-line-number="204"><span class="st">    </span><span class="kw">draw_plot</span>(p_volcano, <span class="dt">x =</span> <span class="dv">0</span>,  <span class="dt">y =</span> <span class="dv">0</span>, <span class="dt">width =</span> <span class="fl">0.48</span>, <span class="dt">height =</span> <span class="fl">0.45</span>, <span class="dt">scale =</span> <span class="dv">1</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-205" data-line-number="205"><span class="st">    </span><span class="kw">draw_plot</span>(p_genesets, <span class="dt">x =</span> <span class="fl">0.52</span>,  <span class="dt">y =</span> <span class="dv">0</span>, <span class="dt">width =</span> <span class="fl">0.48</span>, <span class="dt">height =</span> <span class="fl">0.45</span>, <span class="dt">scale =</span> <span class="dv">1</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-206" data-line-number="206"><span class="st">    </span><span class="kw">draw_plot_label</span>(letters[<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>], <span class="dt">x =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.5</span>, <span class="dv">0</span>, <span class="fl">0.5</span>), </a>
<a class="sourceLine" id="cb62-207" data-line-number="207">                    <span class="dt">y =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">0.45</span>, <span class="fl">0.45</span>), <span class="dt">size =</span> <span class="dv">36</span>)</a>
<a class="sourceLine" id="cb62-208" data-line-number="208">    </a>
<a class="sourceLine" id="cb62-209" data-line-number="209"><span class="kw">ggsave</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_combined_fig.png&quot;</span>, </a>
<a class="sourceLine" id="cb62-210" data-line-number="210">       <span class="dt">height =</span> <span class="dv">16</span>, <span class="dt">width =</span> <span class="dv">18</span>, <span class="dt">plot =</span> pp)</a>
<a class="sourceLine" id="cb62-211" data-line-number="211"><span class="kw">ggsave</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_combined_fig.pdf&quot;</span>, </a>
<a class="sourceLine" id="cb62-212" data-line-number="212">       <span class="dt">height =</span> <span class="dv">16</span>, <span class="dt">width =</span> <span class="dv">18</span>, <span class="dt">plot =</span> pp)</a>
<a class="sourceLine" id="cb62-213" data-line-number="213"></a>
<a class="sourceLine" id="cb62-214" data-line-number="214"></a>
<a class="sourceLine" id="cb62-215" data-line-number="215"><span class="co">## plots for talk</span></a>
<a class="sourceLine" id="cb62-216" data-line-number="216"><span class="kw">ggsave</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_bar_pca.png&quot;</span>, <span class="dt">plot =</span> p_bar_pca,</a>
<a class="sourceLine" id="cb62-217" data-line-number="217">       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="dv">10</span>)</a>
<a class="sourceLine" id="cb62-218" data-line-number="218"><span class="kw">ggsave</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_volcano.png&quot;</span>, <span class="dt">plot =</span> p_volcano,</a>
<a class="sourceLine" id="cb62-219" data-line-number="219">       <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">width =</span> <span class="dv">10</span>)</a>
<a class="sourceLine" id="cb62-220" data-line-number="220"><span class="kw">ggsave</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_genesets.png&quot;</span>, <span class="dt">plot =</span> p_genesets,</a>
<a class="sourceLine" id="cb62-221" data-line-number="221">       <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">width =</span> <span class="dv">10</span>)</a>
<a class="sourceLine" id="cb62-222" data-line-number="222"><span class="kw">ggsave</span>(<span class="st">&quot;figures/donor_specific/carderelax_joxm_mutated_clone.png&quot;</span>, <span class="dt">plot =</span> p_mutated_clone,</a>
<a class="sourceLine" id="cb62-223" data-line-number="223">       <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">width =</span> <span class="dv">14</span>)</a>
<a class="sourceLine" id="cb62-224" data-line-number="224"></a>
<a class="sourceLine" id="cb62-225" data-line-number="225">pp</a></code></pre></div>
<p><img src="figure/analysis_for_joxm_cardelino-relax.Rmd/combined-fig-1.png" width="1728" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb63-1" data-line-number="1"><span class="co"># ggdraw() +</span></a>
<a class="sourceLine" id="cb63-2" data-line-number="2"><span class="co">#     draw_plot(p_tree, x = 0,  y = 0.57, width = 0.48, height = 0.43, scale = 1) +</span></a>
<a class="sourceLine" id="cb63-3" data-line-number="3"><span class="co">#     draw_plot(p_bar_pca, x = 0.52, y = 0.57, width = 0.48, height = 0.43, scale = 1) +</span></a>
<a class="sourceLine" id="cb63-4" data-line-number="4"><span class="co">#     draw_plot(p_volcano, x = 0,  y = 0.3, width = 0.48, height = 0.27, scale = 1) +</span></a>
<a class="sourceLine" id="cb63-5" data-line-number="5"><span class="co">#     draw_plot(p_genesets, x = 0.52,  y = 0.3, width = 0.48, height = 0.27, scale = 1) +</span></a>
<a class="sourceLine" id="cb63-6" data-line-number="6"><span class="co">#     #draw_plot(p_table, x = 0,  y = 0.2, width = 1, height = 0.15, scale = 1) +</span></a>
<a class="sourceLine" id="cb63-7" data-line-number="7"><span class="co">#     draw_plot(p_mutated_clone, x = 0.05,  y = 0, width = 0.9, height = 0.3, scale = 1) +</span></a>
<a class="sourceLine" id="cb63-8" data-line-number="8"><span class="co">#     draw_plot_label(letters[1:5], x = c(0, 0.5, 0, 0.5, 0), </span></a>
<a class="sourceLine" id="cb63-9" data-line-number="9"><span class="co">#                     y = c(1, 1, 0.57, 0.57, 0.3), size = 36)</span></a>
<a class="sourceLine" id="cb63-10" data-line-number="10"><span class="co"># ggsave(&quot;figures/donor_specific/joxm_combined_fig.png&quot;, </span></a>
<a class="sourceLine" id="cb63-11" data-line-number="11"><span class="co">#        height = 20, width = 19)</span></a>
<a class="sourceLine" id="cb63-12" data-line-number="12"><span class="co"># ggsave(&quot;figures/donor_specific/joxm_combined_fig.pdf&quot;, </span></a>
<a class="sourceLine" id="cb63-13" data-line-number="13"><span class="co">#        height = 20, width = 19)</span></a></code></pre></div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb64-1" data-line-number="1">devtools<span class="op">::</span><span class="kw">session_info</span>()</a></code></pre></div>
<pre><code>─ Session info ──────────────────────────────────────────────────────────
 setting  value                       
 version  R version 3.6.0 (2019-04-26)
 os       Ubuntu 18.04.3 LTS          
 system   x86_64, linux-gnu           
 ui       X11                         
 language (EN)                        
 collate  en_AU.UTF-8                 
 ctype    en_AU.UTF-8                 
 tz       Australia/Melbourne         
 date     2019-10-30                  

─ Packages ──────────────────────────────────────────────────────────────
 package              * version   date       lib
 AnnotationDbi        * 1.46.1    2019-08-20 [1]
 ape                    5.3       2019-03-17 [1]
 assertthat             0.2.1     2019-03-21 [1]
 backports              1.1.4     2019-04-10 [1]
 beeswarm               0.2.3     2016-04-25 [1]
 Biobase              * 2.44.0    2019-05-02 [1]
 BiocGenerics         * 0.30.0    2019-05-02 [1]
 BiocNeighbors          1.2.0     2019-05-02 [1]
 BiocParallel         * 1.18.1    2019-08-06 [1]
 BiocSingular           1.0.0     2019-05-02 [1]
 biomaRt                2.40.4    2019-08-19 [1]
 Biostrings             2.52.0    2019-05-02 [1]
 bit                    1.1-14    2018-05-29 [1]
 bit64                  0.9-7     2017-05-08 [1]
 bitops                 1.0-6     2013-08-17 [1]
 blob                   1.2.0     2019-07-09 [1]
 broom                  0.5.2     2019-04-07 [1]
 BSgenome               1.52.0    2019-05-02 [1]
 callr                  3.3.2     2019-09-22 [1]
 cardelino            * 0.6.4     2019-08-21 [1]
 cellranger             1.1.0     2016-07-27 [1]
 cli                    1.1.0     2019-03-19 [1]
 colorspace             1.4-1     2019-03-18 [1]
 cowplot              * 1.0.0     2019-07-11 [1]
 crayon                 1.3.4     2017-09-16 [1]
 DBI                    1.0.0     2018-05-02 [1]
 DelayedArray         * 0.10.0    2019-05-02 [1]
 DelayedMatrixStats     1.6.1     2019-09-08 [1]
 desc                   1.2.0     2018-05-01 [1]
 devtools               2.2.1     2019-09-24 [1]
 digest                 0.6.21    2019-09-20 [1]
 dplyr                * 0.8.3     2019-07-04 [1]
 edgeR                * 3.26.8    2019-09-01 [1]
 ellipsis               0.3.0     2019-09-20 [1]
 evaluate               0.14      2019-05-28 [1]
 fansi                  0.4.0     2018-10-05 [1]
 farver                 1.1.0     2018-11-20 [1]
 fdrtool                1.2.15    2015-07-08 [1]
 forcats              * 0.4.0     2019-02-17 [1]
 fs                     1.3.1     2019-05-06 [1]
 gdtools              * 0.2.0     2019-09-03 [1]
 generics               0.0.2     2018-11-29 [1]
 GenomeInfoDb         * 1.20.0    2019-05-02 [1]
 GenomeInfoDbData       1.2.1     2019-04-30 [1]
 GenomicAlignments      1.20.1    2019-06-18 [1]
 GenomicFeatures        1.36.4    2019-07-10 [1]
 GenomicRanges        * 1.36.1    2019-09-06 [1]
 ggbeeswarm             0.6.0     2017-08-07 [1]
 ggforce              * 0.3.1     2019-08-20 [1]
 ggplot2              * 3.2.1     2019-08-10 [1]
 ggrepel              * 0.8.1     2019-05-07 [1]
 ggridges             * 0.5.1     2018-09-27 [1]
 ggthemes             * 4.2.0     2019-05-13 [1]
 ggtree                 1.16.6    2019-08-26 [1]
 git2r                  0.26.1    2019-06-29 [1]
 glue                   1.3.1     2019-03-12 [1]
 gridExtra              2.3       2017-09-09 [1]
 gtable                 0.3.0     2019-03-25 [1]
 haven                  2.1.1     2019-07-04 [1]
 hms                    0.5.1     2019-08-23 [1]
 htmltools              0.3.6     2017-04-28 [1]
 httr                   1.4.1     2019-08-05 [1]
 IHW                  * 1.12.0    2019-05-02 [1]
 IRanges              * 2.18.3    2019-09-24 [1]
 irlba                  2.3.3     2019-02-05 [1]
 jsonlite               1.6       2018-12-07 [1]
 knitr                  1.25      2019-09-18 [1]
 labeling               0.3       2014-08-23 [1]
 lattice                0.20-38   2018-11-04 [4]
 lazyeval               0.2.2     2019-03-15 [1]
 lifecycle              0.1.0     2019-08-01 [1]
 limma                * 3.40.6    2019-07-26 [1]
 locfit                 1.5-9.1   2013-04-20 [1]
 lpsymphony             1.12.0    2019-05-02 [1]
 lubridate              1.7.4     2018-04-11 [1]
 magrittr               1.5       2014-11-22 [1]
 MASS                   7.3-51.1  2018-11-01 [4]
 Matrix                 1.2-17    2019-03-22 [1]
 matrixStats          * 0.55.0    2019-09-07 [1]
 memoise                1.1.0     2017-04-21 [1]
 modelr                 0.1.5     2019-08-08 [1]
 munsell                0.5.0     2018-06-12 [1]
 nlme                   3.1-139   2019-04-09 [4]
 org.Hs.eg.db         * 3.8.2     2019-05-01 [1]
 pheatmap               1.0.12    2019-01-04 [1]
 pillar                 1.4.2     2019-06-29 [1]
 pkgbuild               1.0.5     2019-08-26 [1]
 pkgconfig              2.0.3     2019-09-22 [1]
 pkgload                1.0.2     2018-10-29 [1]
 plyr                   1.8.4     2016-06-08 [1]
 polyclip               1.10-0    2019-03-14 [1]
 prettyunits            1.0.2     2015-07-13 [1]
 processx               3.4.1     2019-07-18 [1]
 progress               1.2.2     2019-05-16 [1]
 ps                     1.3.0     2018-12-21 [1]
 purrr                * 0.3.2     2019-03-15 [1]
 R6                     2.4.0     2019-02-14 [1]
 RColorBrewer         * 1.1-2     2014-12-07 [1]
 Rcpp                   1.0.2     2019-07-25 [1]
 RCurl                  1.95-4.12 2019-03-04 [1]
 readr                * 1.3.1     2018-12-21 [1]
 readxl                 1.3.1     2019-03-13 [1]
 remotes                2.1.0     2019-06-24 [1]
 rlang                * 0.4.0     2019-06-25 [1]
 rmarkdown              1.15      2019-08-21 [1]
 rprojroot              1.3-2     2018-01-03 [1]
 Rsamtools              2.0.1     2019-09-19 [1]
 RSQLite                2.1.2     2019-07-24 [1]
 rstudioapi             0.10      2019-03-19 [1]
 rsvd                   1.0.2     2019-07-29 [1]
 rtracklayer            1.44.4    2019-09-06 [1]
 rvcheck                0.1.3     2018-12-06 [1]
 rvest                  0.3.4     2019-05-15 [1]
 S4Vectors            * 0.22.1    2019-09-09 [1]
 scales                 1.0.0     2018-08-09 [1]
 scater               * 1.12.2    2019-05-24 [1]
 scico                  1.1.0     2018-11-20 [1]
 sessioninfo            1.1.1     2018-11-05 [1]
 SingleCellExperiment * 1.6.0     2019-05-02 [1]
 slam                   0.1-45    2019-02-26 [1]
 snpStats               1.34.0    2019-05-02 [1]
 stringi                1.4.3     2019-03-12 [1]
 stringr              * 1.4.0     2019-02-10 [1]
 SummarizedExperiment * 1.14.1    2019-07-31 [1]
 superheat            * 0.1.0     2017-02-04 [1]
 survival               2.43-3    2018-11-26 [4]
 svglite                1.2.2     2019-05-17 [1]
 systemfonts            0.1.1     2019-07-01 [1]
 testthat               2.2.1     2019-07-25 [1]
 tibble               * 2.1.3     2019-06-06 [1]
 tidyr                * 1.0.0     2019-09-11 [1]
 tidyselect             0.2.5     2018-10-11 [1]
 tidytree               0.2.7     2019-09-12 [1]
 tidyverse            * 1.2.1     2017-11-14 [1]
 treeio                 1.8.2     2019-08-21 [1]
 tweenr                 1.0.1     2018-12-14 [1]
 usethis                1.5.1     2019-07-04 [1]
 utf8                   1.1.4     2018-05-24 [1]
 VariantAnnotation      1.30.1    2019-05-19 [1]
 vctrs                  0.2.0     2019-07-05 [1]
 vipor                  0.4.5     2017-03-22 [1]
 viridis              * 0.5.1     2018-03-29 [1]
 viridisLite          * 0.3.0     2018-02-01 [1]
 whisker                0.4       2019-08-28 [1]
 withr                  2.1.2     2018-03-15 [1]
 workflowr              1.4.0     2019-06-08 [1]
 xfun                   0.9       2019-08-21 [1]
 XML                    3.98-1.20 2019-06-06 [1]
 xml2                   1.2.2     2019-08-09 [1]
 XVector                0.24.0    2019-05-02 [1]
 yaml                   2.2.0     2018-07-25 [1]
 zeallot                0.1.0     2018-01-28 [1]
 zlibbioc               1.30.0    2019-05-02 [1]
 source                          
 Bioconductor                    
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 Bioconductor                    
 Bioconductor                    
 Bioconductor                    
 Bioconductor                    
 Bioconductor                    
 Bioconductor                    
 Bioconductor                    
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 Bioconductor                    
 CRAN (R 3.6.0)                  
 Github (PMBio/cardelino@efd846f)
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 Bioconductor                    
 Bioconductor                    
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 Bioconductor                    
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 Bioconductor                    
 Bioconductor                    
 Bioconductor                    
 Bioconductor                    
 Bioconductor                    
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 Bioconductor                    
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 Bioconductor                    
 Bioconductor                    
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.5.1)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 Bioconductor                    
 CRAN (R 3.6.0)                  
 Bioconductor (R 3.6.0)          
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.5.1)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.5.3)                  
 Bioconductor                    
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 Bioconductor                    
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 Bioconductor                    
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 Bioconductor                    
 CRAN (R 3.6.0)                  
 Bioconductor                    
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 Bioconductor                    
 CRAN (R 3.6.0)                  
 Bioconductor                    
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 Bioconductor                    
 CRAN (R 3.6.0)                  
 CRAN (R 3.5.1)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 Bioconductor                    
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 Bioconductor                    
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 Bioconductor                    
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 Bioconductor                    

[1] /home/AD.SVI.EDU.AU/dmccarthy/R/x86_64-pc-linux-gnu-library/3.6
[2] /usr/local/lib/R/site-library
[3] /usr/lib/R/site-library
[4] /usr/lib/R/library</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
