<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Davis J. McCarthy" />


<title>Differential expression across all donors - cardelino-relax results</title>

<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/htmlwidgets-1.3/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.9/datatables.js"></script>
<link href="site_libs/dt-core-1.10.19/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.19/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.19/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fibroblast-clonality</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/davismcc/fibroblast-clonality">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Differential expression across all donors - cardelino-relax results</h1>
<h4 class="author">Davis J. McCarthy</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-10-30
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>fibroblast-clonality/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20180807code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20180807)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20180807code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20180807)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomdavismccfibroblastclonalitytreef2e01ef23334fbb86d5874dacc97ddb1493358e5targetblankf2e01efa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/davismcc/fibroblast-clonality/tree/f2e01ef23334fbb86d5874dacc97ddb1493358e5" target="_blank">f2e01ef</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomdavismccfibroblastclonalitytreef2e01ef23334fbb86d5874dacc97ddb1493358e5targetblankf2e01efa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .vscode/
    Ignored:    code/.DS_Store
    Ignored:    code/selection/.DS_Store
    Ignored:    code/selection/.Rhistory
    Ignored:    code/selection/figures/
    Ignored:    data/.DS_Store
    Ignored:    logs/
    Ignored:    src/.DS_Store
    Ignored:    src/Rmd/.Rhistory

Untracked files:
    Untracked:  .dockerignore
    Untracked:  .dropbox
    Untracked:  .snakemake/
    Untracked:  Rplots.pdf
    Untracked:  Snakefile_clonality
    Untracked:  Snakefile_somatic_calling
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/assess_mutect2_fibro-ipsc_variant_calls.ipynb
    Untracked:  analysis/cardelino_fig1b.R
    Untracked:  analysis/cardelino_fig2b.R
    Untracked:  code/analysis_for_garx.Rmd
    Untracked:  code/selection/data/
    Untracked:  code/selection/fit-dist.nb
    Untracked:  code/selection/result-figure.R
    Untracked:  code/yuanhua/
    Untracked:  data/Melanoma-RegevGarraway-DFCI-scRNA-Seq/
    Untracked:  data/PRJNA485423/
    Untracked:  data/canopy/
    Untracked:  data/cell_assignment/
    Untracked:  data/cnv/
    Untracked:  data/de_analysis_FTv62/
    Untracked:  data/donor_info_070818.txt
    Untracked:  data/donor_info_core.csv
    Untracked:  data/donor_neutrality.tsv
    Untracked:  data/exome-point-mutations/
    Untracked:  data/fdr10.annot.txt.gz
    Untracked:  data/human_H_v5p2.rdata
    Untracked:  data/human_c2_v5p2.rdata
    Untracked:  data/human_c6_v5p2.rdata
    Untracked:  data/neg-bin-rsquared-petr.csv
    Untracked:  data/neutralitytestr-petr.tsv
    Untracked:  data/raw/
    Untracked:  data/sce_merged_donors_cardelino_donorid_all_qc_filt.rds
    Untracked:  data/sce_merged_donors_cardelino_donorid_all_with_qc_labels.rds
    Untracked:  data/sce_merged_donors_cardelino_donorid_unstim_qc_filt.rds
    Untracked:  data/sces/
    Untracked:  data/selection/
    Untracked:  data/simulations/
    Untracked:  data/variance_components/
    Untracked:  docs/figure/analysis_for_joxm_cardelino-relax.Rmd/
    Untracked:  docs/figure/clone_prevalences_cardelino-relax.Rmd/
    Untracked:  figures/
    Untracked:  output/differential_expression/
    Untracked:  output/differential_expression_cardelino-relax/
    Untracked:  output/donor_specific/
    Untracked:  output/line_info.tsv
    Untracked:  output/nvars_by_category_by_donor.tsv
    Untracked:  output/nvars_by_category_by_line.tsv
    Untracked:  output/variance_components/
    Untracked:  qolg_BIC.pdf
    Untracked:  references/
    Untracked:  reports/
    Untracked:  src/Rmd/DE_pathways_FTv62_callset_clones_pairwise_vs_base.unst_cells.carderelax.Rmd
    Untracked:  src/Rmd/Rplots.pdf
    Untracked:  src/Rmd/cell_assignment_cardelino-relax_template.Rmd
    Untracked:  tree.txt

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/550176f57f3f33670ae89659f72e541ad8e8ba41/analysis/differential_expression_cardelino-relax.Rmd" target="_blank">550176f</a>
</td>
<td>
Davis McCarthy
</td>
<td>
2019-10-30
</td>
<td>
Updating analysis to reflect accepted ms
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Here, we will lok at differential expresion between clones across all lines ( i.e. donors) at the gene and gene set levels, using the clone-assignment results using the cardelino-relax method (which treats the Canopy clonal tree as a prior for the clonal structure rather than the definitive truth).</p>
<div id="load-libraries-data-and-de-results" class="section level2">
<h2>Load libraries, data and DE results</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1">knitr<span class="op">::</span>opts_chunk<span class="op">$</span><span class="kw">set</span>(<span class="dt">echo =</span> <span class="ot">TRUE</span>, <span class="dt">warning =</span> <span class="ot">FALSE</span>, <span class="dt">message =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw">library</span>(tidyverse)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw">library</span>(scater)</a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="kw">library</span>(ggridges)</a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="kw">library</span>(GenomicRanges)</a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="kw">library</span>(RColorBrewer)</a>
<a class="sourceLine" id="cb1-7" data-line-number="7"><span class="kw">library</span>(edgeR)</a>
<a class="sourceLine" id="cb1-8" data-line-number="8"><span class="kw">library</span>(ggrepel)</a>
<a class="sourceLine" id="cb1-9" data-line-number="9"><span class="kw">library</span>(ggcorrplot)</a>
<a class="sourceLine" id="cb1-10" data-line-number="10"><span class="kw">library</span>(rlang)</a>
<a class="sourceLine" id="cb1-11" data-line-number="11"><span class="kw">library</span>(limma)</a>
<a class="sourceLine" id="cb1-12" data-line-number="12"><span class="kw">library</span>(org.Hs.eg.db)</a>
<a class="sourceLine" id="cb1-13" data-line-number="13"><span class="kw">library</span>(ggforce)</a>
<a class="sourceLine" id="cb1-14" data-line-number="14"><span class="kw">library</span>(superheat)</a>
<a class="sourceLine" id="cb1-15" data-line-number="15"><span class="kw">library</span>(viridis)</a>
<a class="sourceLine" id="cb1-16" data-line-number="16"><span class="kw">library</span>(IHW)</a>
<a class="sourceLine" id="cb1-17" data-line-number="17"><span class="kw">library</span>(cowplot)</a>
<a class="sourceLine" id="cb1-18" data-line-number="18"><span class="kw">library</span>(broom)</a>
<a class="sourceLine" id="cb1-19" data-line-number="19"><span class="kw">library</span>(cardelino)</a>
<a class="sourceLine" id="cb1-20" data-line-number="20"><span class="kw">options</span>(<span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb1-21" data-line-number="21"><span class="kw">dir.create</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax&quot;</span>, <span class="dt">showWarnings =</span> <span class="ot">FALSE</span>, </a>
<a class="sourceLine" id="cb1-22" data-line-number="22">           <span class="dt">recursive =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<p>Load the genewise differential expression results produced with the edgeR quasi-likelihood F test and gene set enrichment results produced with camera.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">params &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb2-2" data-line-number="2">params<span class="op">$</span>callset &lt;-<span class="st"> &quot;filt_lenient.cell_coverage_sites&quot;</span></a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="kw">load</span>(<span class="kw">file.path</span>(<span class="st">&quot;data/human_c6_v5p2.rdata&quot;</span>))</a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="kw">load</span>(<span class="kw">file.path</span>(<span class="st">&quot;data/human_H_v5p2.rdata&quot;</span>))</a>
<a class="sourceLine" id="cb2-5" data-line-number="5"><span class="kw">load</span>(<span class="kw">file.path</span>(<span class="st">&quot;data/human_c2_v5p2.rdata&quot;</span>))</a>
<a class="sourceLine" id="cb2-6" data-line-number="6"></a>
<a class="sourceLine" id="cb2-7" data-line-number="7">de_res &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">paste0</span>(<span class="st">&quot;data/de_analysis_FTv62/carderelax.&quot;</span>,</a>
<a class="sourceLine" id="cb2-8" data-line-number="8">                         params<span class="op">$</span>callset, </a>
<a class="sourceLine" id="cb2-9" data-line-number="9">                         <span class="st">&quot;.de_results_unstimulated_cells.rds&quot;</span>))</a></code></pre></div>
<p>Load SingleCellExpression objects with data used for differential expression analyses.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">fls &lt;-<span class="st"> </span><span class="kw">list.files</span>(<span class="st">&quot;data/sces&quot;</span>)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2">fls &lt;-<span class="st"> </span>fls[<span class="kw">grepl</span>(<span class="kw">paste0</span>(<span class="st">&quot;carderelax.&quot;</span>, params<span class="op">$</span>callset), fls)]</a>
<a class="sourceLine" id="cb3-3" data-line-number="3">donors &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;.*ce_([a-z]+)_with_clone_assignments_carderelax.*&quot;</span>, <span class="st">&quot;</span><span class="ch">\\</span><span class="st">1&quot;</span>, fls)</a>
<a class="sourceLine" id="cb3-4" data-line-number="4"></a>
<a class="sourceLine" id="cb3-5" data-line-number="5">sce_unst_list &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb3-6" data-line-number="6"><span class="cf">for</span> (don <span class="cf">in</span> donors) {</a>
<a class="sourceLine" id="cb3-7" data-line-number="7">    sce_unst_list[[don]] &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">file.path</span>(<span class="st">&quot;data/sces&quot;</span>,</a>
<a class="sourceLine" id="cb3-8" data-line-number="8">        <span class="kw">paste0</span>(<span class="st">&quot;sce_&quot;</span>, don, <span class="st">&quot;_with_clone_assignments_carderelax.&quot;</span>, </a>
<a class="sourceLine" id="cb3-9" data-line-number="9">               params<span class="op">$</span>callset, <span class="st">&quot;.rds&quot;</span>)))</a>
<a class="sourceLine" id="cb3-10" data-line-number="10">    <span class="kw">cat</span>(<span class="kw">paste</span>(<span class="st">&quot;reading&quot;</span>, don, <span class="st">&quot;:   &quot;</span>, <span class="kw">ncol</span>(sce_unst_list[[don]]), </a>
<a class="sourceLine" id="cb3-11" data-line-number="11">              <span class="st">&quot;unstimulated cells.</span><span class="ch">\n</span><span class="st">&quot;</span>))</a>
<a class="sourceLine" id="cb3-12" data-line-number="12">}</a></code></pre></div>
<pre><code>reading euts :    79 unstimulated cells.
reading fawm :    53 unstimulated cells.
reading feec :    75 unstimulated cells.
reading fikt :    39 unstimulated cells.
reading garx :    70 unstimulated cells.
reading gesg :    105 unstimulated cells.
reading heja :    50 unstimulated cells.
reading hipn :    62 unstimulated cells.
reading ieki :    58 unstimulated cells.
reading joxm :    79 unstimulated cells.
reading kuco :    48 unstimulated cells.
reading laey :    55 unstimulated cells.
reading lexy :    63 unstimulated cells.
reading naju :    44 unstimulated cells.
reading nusw :    60 unstimulated cells.
reading oaaz :    38 unstimulated cells.
reading oilg :    90 unstimulated cells.
reading pipw :    107 unstimulated cells.
reading puie :    41 unstimulated cells.
reading qayj :    97 unstimulated cells.
reading qolg :    36 unstimulated cells.
reading qonc :    58 unstimulated cells.
reading rozh :    91 unstimulated cells.
reading sehl :    30 unstimulated cells.
reading ualf :    89 unstimulated cells.
reading vass :    37 unstimulated cells.
reading vils :    37 unstimulated cells.
reading vuna :    71 unstimulated cells.
reading wahn :    82 unstimulated cells.
reading wetu :    77 unstimulated cells.
reading xugn :    35 unstimulated cells.
reading zoxy :    88 unstimulated cells.</code></pre>
<p>The starting point for differential expression analysis was a set of 32 donors, of which 31 donors had enough cells assigned to clones to conduct DE testing.</p>
<p>Summarise cell assignment information.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">assignments_lst &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="cf">for</span> (don <span class="cf">in</span> donors) {</a>
<a class="sourceLine" id="cb5-3" data-line-number="3">    assignments_lst[[don]] &lt;-<span class="st"> </span><span class="kw">as_tibble</span>(</a>
<a class="sourceLine" id="cb5-4" data-line-number="4">        <span class="kw">colData</span>(sce_unst_list[[don]])[, </a>
<a class="sourceLine" id="cb5-5" data-line-number="5">                                      <span class="kw">c</span>(<span class="st">&quot;donor_short_id&quot;</span>, <span class="st">&quot;highest_prob&quot;</span>, </a>
<a class="sourceLine" id="cb5-6" data-line-number="6">                                        <span class="st">&quot;assigned&quot;</span>, <span class="st">&quot;total_features&quot;</span>,</a>
<a class="sourceLine" id="cb5-7" data-line-number="7">                                        <span class="st">&quot;total_counts_endogenous&quot;</span>, <span class="st">&quot;num_processed&quot;</span>)])</a>
<a class="sourceLine" id="cb5-8" data-line-number="8">}</a>
<a class="sourceLine" id="cb5-9" data-line-number="9">assignments &lt;-<span class="st"> </span><span class="kw">do.call</span>(<span class="st">&quot;rbind&quot;</span>, assignments_lst)</a></code></pre></div>
<p>88% of cells from these donors are assigned with confidence to a clone.</p>
<p>Load donor info including evidence for selection dynamics in donors.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">df_donor_info &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;data/donor_info_070818.txt&quot;</span>)</a></code></pre></div>
</div>
<div id="genewise-de-results" class="section level2">
<h2>Genewise DE results</h2>
<p>We first look at differential expression at the level of individual genes.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">fdr_thresh &lt;-<span class="st"> </span><span class="dv">1</span></a>
<a class="sourceLine" id="cb7-2" data-line-number="2">df_de_all_unst &lt;-<span class="st"> </span><span class="kw">tibble</span>()</a>
<a class="sourceLine" id="cb7-3" data-line-number="3"><span class="cf">for</span> (donor <span class="cf">in</span> <span class="kw">names</span>(de_res[[<span class="st">&quot;qlf_list&quot;</span>]])) {</a>
<a class="sourceLine" id="cb7-4" data-line-number="4">    tmp &lt;-<span class="st"> </span>de_res[[<span class="st">&quot;qlf_list&quot;</span>]][[donor]]<span class="op">$</span>table</a>
<a class="sourceLine" id="cb7-5" data-line-number="5">    tmp<span class="op">$</span>gene &lt;-<span class="st"> </span><span class="kw">rownames</span>(de_res[[<span class="st">&quot;qlf_list&quot;</span>]][[donor]]<span class="op">$</span>table)</a>
<a class="sourceLine" id="cb7-6" data-line-number="6">    ihw_res &lt;-<span class="st"> </span><span class="kw">ihw</span>(PValue <span class="op">~</span><span class="st"> </span>logCPM, <span class="dt">data =</span> tmp, <span class="dt">alpha =</span> <span class="fl">0.05</span>)</a>
<a class="sourceLine" id="cb7-7" data-line-number="7">    tmp<span class="op">$</span>FDR &lt;-<span class="st"> </span><span class="kw">adj_pvalues</span>(ihw_res)</a>
<a class="sourceLine" id="cb7-8" data-line-number="8">    tmp &lt;-<span class="st"> </span>tmp[tmp<span class="op">$</span>FDR <span class="op">&lt;=</span><span class="st"> </span>fdr_thresh,]</a>
<a class="sourceLine" id="cb7-9" data-line-number="9">    <span class="cf">if</span> (<span class="kw">nrow</span>(tmp) <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.5</span>) {</a>
<a class="sourceLine" id="cb7-10" data-line-number="10">        tmp[[<span class="st">&quot;donor&quot;</span>]] &lt;-<span class="st"> </span>donor</a>
<a class="sourceLine" id="cb7-11" data-line-number="11">        df_de_all_unst &lt;-<span class="st"> </span><span class="kw">bind_rows</span>(df_de_all_unst, tmp)</a>
<a class="sourceLine" id="cb7-12" data-line-number="12">    }</a>
<a class="sourceLine" id="cb7-13" data-line-number="13">}</a>
<a class="sourceLine" id="cb7-14" data-line-number="14"></a>
<a class="sourceLine" id="cb7-15" data-line-number="15">df_ncells_de &lt;-<span class="st"> </span>assignments <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(assigned <span class="op">!=</span><span class="st"> &quot;unassigned&quot;</span>, </a>
<a class="sourceLine" id="cb7-16" data-line-number="16">                              donor_short_id <span class="op">%in%</span><span class="st"> </span><span class="kw">names</span>(de_res<span class="op">$</span>qlf_list)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-17" data-line-number="17"><span class="st">    </span><span class="kw">group_by</span>(donor_short_id) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-18" data-line-number="18"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">n_cells =</span> <span class="kw">n</span>())</a>
<a class="sourceLine" id="cb7-19" data-line-number="19"><span class="kw">colnames</span>(df_ncells_de)[<span class="dv">1</span>] &lt;-<span class="st"> &quot;donor&quot;</span></a>
<a class="sourceLine" id="cb7-20" data-line-number="20"></a>
<a class="sourceLine" id="cb7-21" data-line-number="21">fdr_thresh &lt;-<span class="st"> </span><span class="fl">0.1</span></a>
<a class="sourceLine" id="cb7-22" data-line-number="22">df_de_sig_unst &lt;-<span class="st"> </span><span class="kw">tibble</span>()</a>
<a class="sourceLine" id="cb7-23" data-line-number="23"><span class="cf">for</span> (donor <span class="cf">in</span> <span class="kw">names</span>(de_res[[<span class="st">&quot;qlf_list&quot;</span>]])) {</a>
<a class="sourceLine" id="cb7-24" data-line-number="24">    tmp &lt;-<span class="st"> </span>de_res[[<span class="st">&quot;qlf_list&quot;</span>]][[donor]]<span class="op">$</span>table</a>
<a class="sourceLine" id="cb7-25" data-line-number="25">    tmp<span class="op">$</span>gene &lt;-<span class="st"> </span><span class="kw">rownames</span>(de_res[[<span class="st">&quot;qlf_list&quot;</span>]][[donor]]<span class="op">$</span>table)</a>
<a class="sourceLine" id="cb7-26" data-line-number="26">    ihw_res &lt;-<span class="st"> </span><span class="kw">ihw</span>(PValue <span class="op">~</span><span class="st"> </span>logCPM, <span class="dt">data =</span> tmp, <span class="dt">alpha =</span> <span class="fl">0.05</span>)</a>
<a class="sourceLine" id="cb7-27" data-line-number="27">    tmp<span class="op">$</span>FDR &lt;-<span class="st"> </span><span class="kw">adj_pvalues</span>(ihw_res)</a>
<a class="sourceLine" id="cb7-28" data-line-number="28">    tmp &lt;-<span class="st"> </span>tmp[tmp<span class="op">$</span>FDR <span class="op">&lt;</span><span class="st"> </span>fdr_thresh,]</a>
<a class="sourceLine" id="cb7-29" data-line-number="29">    <span class="cf">if</span> (<span class="kw">nrow</span>(tmp) <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.5</span>) {</a>
<a class="sourceLine" id="cb7-30" data-line-number="30">        tmp[[<span class="st">&quot;donor&quot;</span>]] &lt;-<span class="st"> </span>donor</a>
<a class="sourceLine" id="cb7-31" data-line-number="31">        df_de_sig_unst &lt;-<span class="st"> </span><span class="kw">bind_rows</span>(df_de_sig_unst, tmp)</a>
<a class="sourceLine" id="cb7-32" data-line-number="32">    }</a>
<a class="sourceLine" id="cb7-33" data-line-number="33">}</a>
<a class="sourceLine" id="cb7-34" data-line-number="34"></a>
<a class="sourceLine" id="cb7-35" data-line-number="35">df_de_sig_unst <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb7-36" data-line-number="36"><span class="st">    </span><span class="kw">group_by</span>(gene) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb7-37" data-line-number="37"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, gene)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(id, <span class="dt">.keep_all =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-38" data-line-number="38"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">n_donors =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(n_donors) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-39" data-line-number="39"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">count =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-40" data-line-number="40"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> n_donors, <span class="dt">y =</span> count)) <span class="op">+</span></a>
<a class="sourceLine" id="cb7-41" data-line-number="41"><span class="st">    </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">x =</span> n_donors, <span class="dt">xend =</span> n_donors, <span class="dt">y =</span> count, <span class="dt">yend =</span> <span class="fl">0.1</span>), </a>
<a class="sourceLine" id="cb7-42" data-line-number="42">                 <span class="dt">colour =</span> <span class="st">&quot;gray50&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb7-43" data-line-number="43"><span class="st">        </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb7-44" data-line-number="44"><span class="st">    </span><span class="kw">scale_y_log10</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">10</span>, <span class="dv">100</span>, <span class="dv">1000</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb7-45" data-line-number="45"><span class="st">    </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks =</span> <span class="dv">0</span><span class="op">:</span><span class="dv">15</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb7-46" data-line-number="46"><span class="st">    </span><span class="kw">coord_cartesian</span>(<span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2000</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb7-47" data-line-number="47"><span class="st">    </span><span class="kw">theme_classic</span>(<span class="dv">20</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb7-48" data-line-number="48"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Number of lines in which significant (FDR &lt; 10%)&quot;</span>)  <span class="op">+</span></a>
<a class="sourceLine" id="cb7-49" data-line-number="49"><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Number of genes&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb7-50" data-line-number="50"><span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;edgeR QL F test DE results&quot;</span>)</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/genewise-de-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/alldonors_de_n_sig_donors_n_sig_genes.png&quot;</span>, </a>
<a class="sourceLine" id="cb8-2" data-line-number="2">       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="dv">10</span>)</a>
<a class="sourceLine" id="cb8-3" data-line-number="3"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/alldonors_de_n_sig_donors_n_sig_genes.pdf&quot;</span>, </a>
<a class="sourceLine" id="cb8-4" data-line-number="4">       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="dv">10</span>)</a>
<a class="sourceLine" id="cb8-5" data-line-number="5"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/alldonors_de_n_sig_donors_n_sig_genes.svg&quot;</span>, </a>
<a class="sourceLine" id="cb8-6" data-line-number="6">       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="dv">10</span>)</a>
<a class="sourceLine" id="cb8-7" data-line-number="7">       </a>
<a class="sourceLine" id="cb8-8" data-line-number="8">p1 &lt;-<span class="st"> </span>df_de_sig_unst <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb8-9" data-line-number="9"><span class="st">    </span><span class="kw">group_by</span>(gene) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb8-10" data-line-number="10"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, gene)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(id, <span class="dt">.keep_all =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-11" data-line-number="11"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">n_donors =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(n_donors) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-12" data-line-number="12"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">count =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-13" data-line-number="13"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> n_donors, <span class="dt">y =</span> count)) <span class="op">+</span></a>
<a class="sourceLine" id="cb8-14" data-line-number="14"><span class="st">    </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">x =</span> n_donors, <span class="dt">xend =</span> n_donors, <span class="dt">y =</span> count, <span class="dt">yend =</span> <span class="fl">0.1</span>), </a>
<a class="sourceLine" id="cb8-15" data-line-number="15">                 <span class="dt">colour =</span> <span class="st">&quot;gray50&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb8-16" data-line-number="16"><span class="st">        </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb8-17" data-line-number="17"><span class="st">    </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks =</span> <span class="dv">0</span><span class="op">:</span><span class="dv">14</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb8-18" data-line-number="18"><span class="st">    </span><span class="co">#coord_cartesian(ylim = c(1, 2200)) +</span></a>
<a class="sourceLine" id="cb8-19" data-line-number="19"><span class="st">    </span><span class="kw">theme_classic</span>(<span class="dv">16</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb8-20" data-line-number="20"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Number of lines significant (FDR &lt; 10%)&quot;</span>)  <span class="op">+</span></a>
<a class="sourceLine" id="cb8-21" data-line-number="21"><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Number of genes&quot;</span>)</a>
<a class="sourceLine" id="cb8-22" data-line-number="22">p1 </a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/genewise-de-2.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/alldonors_de_n_sig_donors_n_sig_genes_linscale.png&quot;</span>, </a>
<a class="sourceLine" id="cb9-2" data-line-number="2">       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="fl">5.5</span>)</a>
<a class="sourceLine" id="cb9-3" data-line-number="3"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/alldonors_de_n_sig_donors_n_sig_genes_linscale.pdf&quot;</span>, </a>
<a class="sourceLine" id="cb9-4" data-line-number="4">       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="fl">5.5</span>)</a>
<a class="sourceLine" id="cb9-5" data-line-number="5"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/alldonors_de_n_sig_donors_n_sig_genes_linscale.svg&quot;</span>, </a>
<a class="sourceLine" id="cb9-6" data-line-number="6">       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="fl">5.5</span>)</a></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">p2 &lt;-<span class="st"> </span>df_de_sig_unst <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb10-2" data-line-number="2"><span class="st">    </span><span class="kw">group_by</span>(gene) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb10-3" data-line-number="3"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, gene)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(id, <span class="dt">.keep_all =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-4" data-line-number="4"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">n_donors =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(gene, n_donors) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-5" data-line-number="5"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">gene =</span> <span class="kw">gsub</span>(<span class="st">&quot;ENSG.*_&quot;</span>, <span class="st">&quot;&quot;</span>, gene)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-6" data-line-number="6"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(n_donors <span class="op">&gt;</span><span class="st"> </span><span class="fl">7.5</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-7" data-line-number="7"><span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">y =</span> n_donors, <span class="dt">x =</span> <span class="kw">reorder</span>(gene, n_donors, max))) <span class="op">+</span></a>
<a class="sourceLine" id="cb10-8" data-line-number="8"><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.7</span>, <span class="dt">size =</span> <span class="dv">4</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb10-9" data-line-number="9"><span class="st">    </span><span class="kw">scale_y_continuous</span>(<span class="dt">breaks =</span> <span class="dv">7</span><span class="op">:</span><span class="dv">11</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb10-10" data-line-number="10"><span class="st">    </span>ggthemes<span class="op">::</span><span class="kw">scale_colour_tableau</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb10-11" data-line-number="11"><span class="st">    </span><span class="kw">coord_flip</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb10-12" data-line-number="12"><span class="st">    </span><span class="kw">theme_bw</span>(<span class="dv">16</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb10-13" data-line-number="13"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Gene&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Number of lines significant&quot;</span>)</a>
<a class="sourceLine" id="cb10-14" data-line-number="14">p2</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/unnamed-chunk-1-1.png" width="1152" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/alldonors_de_n_sig_donors_topgenes.png&quot;</span>, </a>
<a class="sourceLine" id="cb11-2" data-line-number="2">       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="fl">5.5</span>)</a>
<a class="sourceLine" id="cb11-3" data-line-number="3"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/alldonors_de_n_sig_donors_topgenes.pdf&quot;</span>, </a>
<a class="sourceLine" id="cb11-4" data-line-number="4">       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="fl">5.5</span>)</a>
<a class="sourceLine" id="cb11-5" data-line-number="5"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/alldonors_de_n_sig_donors_topgenes.svg&quot;</span>, </a>
<a class="sourceLine" id="cb11-6" data-line-number="6">       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="fl">5.5</span>)</a>
<a class="sourceLine" id="cb11-7" data-line-number="7"></a>
<a class="sourceLine" id="cb11-8" data-line-number="8"></a>
<a class="sourceLine" id="cb11-9" data-line-number="9"><span class="co">#cowplot::plot_grid(p1, p2, rel_heights = c(0.4, 0.6))</span></a>
<a class="sourceLine" id="cb11-10" data-line-number="10"></a>
<a class="sourceLine" id="cb11-11" data-line-number="11"></a>
<a class="sourceLine" id="cb11-12" data-line-number="12">df_donor_n_de &lt;-<span class="st"> </span>df_de_sig_unst <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb11-13" data-line-number="13"><span class="st">    </span><span class="kw">group_by</span>(gene) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb11-14" data-line-number="14"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, gene)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(id, <span class="dt">.keep_all =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-15" data-line-number="15"><span class="st">    </span><span class="kw">group_by</span>(donor) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-16" data-line-number="16"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">count =</span> <span class="kw">n</span>())</a>
<a class="sourceLine" id="cb11-17" data-line-number="17">no_de_donor &lt;-<span class="st"> </span><span class="kw">unique</span>(df_de_all_unst[[<span class="st">&quot;donor&quot;</span>]])[<span class="op">!</span>(<span class="kw">unique</span>(df_de_all_unst[[<span class="st">&quot;donor&quot;</span>]]) <span class="op">%in%</span><span class="st"> </span>df_donor_n_de[[<span class="st">&quot;donor&quot;</span>]])]</a>
<a class="sourceLine" id="cb11-18" data-line-number="18">df_donor_n_de &lt;-<span class="st"> </span><span class="kw">rbind</span>(df_donor_n_de, <span class="kw">tibble</span>(<span class="dt">donor =</span> no_de_donor, <span class="dt">count =</span> <span class="dv">0</span>))</a></code></pre></div>
<p>Permute gene labels to get a null distribution.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1">df_de_nsig &lt;-<span class="st"> </span>df_de_all_unst <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(FDR <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.1</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb12-2" data-line-number="2"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, gene)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(id, <span class="dt">.keep_all =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb12-3" data-line-number="3"><span class="st">    </span><span class="kw">group_by</span>(donor) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb12-4" data-line-number="4"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">n_sig =</span> <span class="kw">n</span>())</a>
<a class="sourceLine" id="cb12-5" data-line-number="5">df_nsig_ncells_de &lt;-<span class="st"> </span><span class="kw">full_join</span>(df_ncells_de, df_de_nsig)</a>
<a class="sourceLine" id="cb12-6" data-line-number="6">df_nsig_ncells_de<span class="op">$</span>n_sig[<span class="kw">is.na</span>(df_nsig_ncells_de<span class="op">$</span>n_sig)] &lt;-<span class="st"> </span><span class="dv">0</span></a>
<a class="sourceLine" id="cb12-7" data-line-number="7"></a>
<a class="sourceLine" id="cb12-8" data-line-number="8">permute_gene_labels &lt;-<span class="st"> </span><span class="cf">function</span>(gene_names, n_de)  {</a>
<a class="sourceLine" id="cb12-9" data-line-number="9">    sampled_genes &lt;-<span class="st"> </span><span class="kw">c</span>()</a>
<a class="sourceLine" id="cb12-10" data-line-number="10">    <span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">seq_along</span>(n_de))</a>
<a class="sourceLine" id="cb12-11" data-line-number="11">        sampled_genes &lt;-<span class="st"> </span><span class="kw">c</span>(sampled_genes, <span class="kw">sample</span>(gene_names, <span class="dt">size =</span> n_de[i]))</a>
<a class="sourceLine" id="cb12-12" data-line-number="12">    tab &lt;-<span class="st"> </span><span class="kw">table</span>(<span class="kw">table</span>(sampled_genes))</a>
<a class="sourceLine" id="cb12-13" data-line-number="13">    df &lt;-<span class="st"> </span><span class="kw">tibble</span>(<span class="dt">n_donors =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">12</span>, <span class="dt">n_genes =</span> <span class="dv">0</span>)</a>
<a class="sourceLine" id="cb12-14" data-line-number="14">    df[<span class="kw">names</span>(tab), <span class="dv">2</span>] &lt;-<span class="st"> </span>tab</a>
<a class="sourceLine" id="cb12-15" data-line-number="15">    df</a>
<a class="sourceLine" id="cb12-16" data-line-number="16">}</a>
<a class="sourceLine" id="cb12-17" data-line-number="17"></a>
<a class="sourceLine" id="cb12-18" data-line-number="18"></a>
<a class="sourceLine" id="cb12-19" data-line-number="19">n_perm &lt;-<span class="st"> </span><span class="dv">1000</span></a>
<a class="sourceLine" id="cb12-20" data-line-number="20">df_perm &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb12-21" data-line-number="21"><span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">seq_len</span>(n_perm))</a>
<a class="sourceLine" id="cb12-22" data-line-number="22">    df_perm[[i]] &lt;-<span class="st"> </span><span class="kw">permute_gene_labels</span>(<span class="kw">rownames</span>(de_res<span class="op">$</span>qlf_list<span class="op">$</span>vass<span class="op">$</span>table), </a>
<a class="sourceLine" id="cb12-23" data-line-number="23">                                        df_nsig_ncells_de[[<span class="st">&quot;n_sig&quot;</span>]])</a>
<a class="sourceLine" id="cb12-24" data-line-number="24">df_perm &lt;-<span class="st"> </span><span class="kw">do.call</span>(<span class="st">&quot;rbind&quot;</span>, df_perm)</a>
<a class="sourceLine" id="cb12-25" data-line-number="25">df_perm &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(df_perm, <span class="dt">data_type =</span> <span class="st">&quot;permuted&quot;</span>)</a>
<a class="sourceLine" id="cb12-26" data-line-number="26"></a>
<a class="sourceLine" id="cb12-27" data-line-number="27">df_perm <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(n_donors) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">min =</span> <span class="kw">min</span>(n_genes), </a>
<a class="sourceLine" id="cb12-28" data-line-number="28">                                             <span class="dt">median =</span> <span class="kw">median</span>(n_genes),</a>
<a class="sourceLine" id="cb12-29" data-line-number="29">                                             <span class="dt">mean =</span> <span class="kw">mean</span>(n_genes), </a>
<a class="sourceLine" id="cb12-30" data-line-number="30">                                             <span class="dt">max =</span> <span class="kw">max</span>(n_genes))</a></code></pre></div>
<pre><code># A tibble: 12 x 5
   n_donors   min median     mean   max
      &lt;int&gt; &lt;dbl&gt;  &lt;dbl&gt;    &lt;dbl&gt; &lt;dbl&gt;
 1        1  3919   4085 4084.     4259
 2        2  2416   2534 2534.     2674
 3        3   896    986  986.     1068
 4        4   226    270  271.      316
 5        5    35     56   56.1      78
 6        6     0      9    9.03     19
 7        7     0      1    1.11      7
 8        8     0      0    0.123     2
 9        9     0      0    0.009     1
10       10     0      0    0         0
11       11     0      0    0         0
12       12     0      0    0         0</code></pre>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1">ppp &lt;-<span class="st"> </span>df_de_sig_unst <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb14-2" data-line-number="2"><span class="st">    </span><span class="kw">group_by</span>(gene) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb14-3" data-line-number="3"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, gene)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(id, <span class="dt">.keep_all =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb14-4" data-line-number="4"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">n_donors =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(n_donors) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb14-5" data-line-number="5"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">n_genes =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">data_type =</span> <span class="st">&quot;observed&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb14-6" data-line-number="6"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> n_donors, <span class="dt">y =</span> n_genes)) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-7" data-line-number="7"><span class="st">    </span><span class="co"># geom_segment(aes(x = n_donors, xend = n_donors, y = count, yend = 0.1), </span></a>
<a class="sourceLine" id="cb14-8" data-line-number="8"><span class="st">    </span><span class="co">#              colour = &quot;gray50&quot;) +</span></a>
<a class="sourceLine" id="cb14-9" data-line-number="9"><span class="co">#    geom_hline(yintercept = 0, linetype = 2) +</span></a>
<a class="sourceLine" id="cb14-10" data-line-number="10"><span class="st">    </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="dv">0</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-11" data-line-number="11"><span class="st">    </span><span class="kw">geom_boxplot</span>(<span class="kw">aes</span>(<span class="dt">group =</span> n_donors, <span class="dt">y =</span> n_genes, <span class="dt">colour =</span> data_type), </a>
<a class="sourceLine" id="cb14-12" data-line-number="12">                 <span class="dt">fill =</span> <span class="st">&quot;gray80&quot;</span>, <span class="dt">data =</span> df_perm, <span class="dt">show.legend =</span> <span class="ot">FALSE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-13" data-line-number="13"><span class="st">    </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">colour =</span> data_type), <span class="dt">shape =</span> <span class="dv">17</span>, <span class="dt">size =</span> <span class="dv">5</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-14" data-line-number="14"><span class="st">    </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks =</span> <span class="dv">0</span><span class="op">:</span><span class="dv">13</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-15" data-line-number="15"><span class="st">    </span><span class="kw">scale_y_sqrt</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">10</span>, <span class="dv">100</span>, <span class="dv">500</span>, <span class="dv">1000</span>, <span class="dv">2000</span>, <span class="dv">3000</span>),</a>
<a class="sourceLine" id="cb14-16" data-line-number="16">                 <span class="dt">labels =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">10</span>, <span class="dv">100</span>, <span class="dv">500</span>, <span class="dv">1000</span>, <span class="dv">2000</span>, <span class="dv">3000</span>),</a>
<a class="sourceLine" id="cb14-17" data-line-number="17">                 <span class="dt">limits =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">4500</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-18" data-line-number="18"><span class="st">    </span><span class="kw">scale_colour_manual</span>(<span class="dt">name =</span> <span class="st">&#39;&#39;</span>, </a>
<a class="sourceLine" id="cb14-19" data-line-number="19">                        <span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;observed&quot;</span> =<span class="st"> &quot;black&quot;</span>, <span class="st">&quot;permuted&quot;</span> =<span class="st"> &quot;gray50&quot;</span>), </a>
<a class="sourceLine" id="cb14-20" data-line-number="20">                        <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;observed&quot;</span>, <span class="st">&quot;permuted&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-21" data-line-number="21"><span class="st">    </span><span class="kw">coord_cartesian</span>(<span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">4500</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-22" data-line-number="22"><span class="st">    </span><span class="kw">theme_bw</span>(<span class="dv">18</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-23" data-line-number="23"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="kw">c</span>(<span class="fl">0.8</span>, <span class="fl">0.88</span>),</a>
<a class="sourceLine" id="cb14-24" data-line-number="24">          <span class="dt">panel.grid.major.x =</span> <span class="kw">element_blank</span>(), </a>
<a class="sourceLine" id="cb14-25" data-line-number="25">          <span class="dt">panel.grid.minor.x =</span> <span class="kw">element_blank</span>()) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-26" data-line-number="26"><span class="st">    </span><span class="kw">guides</span>(<span class="dt">colour =</span> <span class="kw">guide_legend</span>(<span class="dt">override.aes =</span> <span class="kw">list</span>(<span class="dt">shape =</span> <span class="kw">c</span>(<span class="dv">17</span>, <span class="dv">19</span>))),</a>
<a class="sourceLine" id="cb14-27" data-line-number="27">           <span class="dt">fill =</span> <span class="ot">FALSE</span>, <span class="dt">boxplot =</span> <span class="ot">FALSE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-28" data-line-number="28"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Number of lines significant (FDR &lt; 10%)&quot;</span>)  <span class="op">+</span></a>
<a class="sourceLine" id="cb14-29" data-line-number="29"><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Number of genes&quot;</span>)</a>
<a class="sourceLine" id="cb14-30" data-line-number="30"></a>
<a class="sourceLine" id="cb14-31" data-line-number="31"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/alldonors_de_n_sig_donors_n_sig_genes_sqrtscale_perm.png&quot;</span>, </a>
<a class="sourceLine" id="cb14-32" data-line-number="32">       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="dv">10</span>, <span class="dt">plot =</span> ppp)</a>
<a class="sourceLine" id="cb14-33" data-line-number="33"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/alldonors_de_n_sig_donors_n_sig_genes_sqrtscale_perm.pdf&quot;</span>, </a>
<a class="sourceLine" id="cb14-34" data-line-number="34">       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="dv">10</span>, <span class="dt">plot =</span> ppp)</a>
<a class="sourceLine" id="cb14-35" data-line-number="35"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/alldonors_de_n_sig_donors_n_sig_genes_sqrtscale_perm.svg&quot;</span>, </a>
<a class="sourceLine" id="cb14-36" data-line-number="36">       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="dv">10</span>, <span class="dt">plot =</span> ppp)</a>
<a class="sourceLine" id="cb14-37" data-line-number="37"></a>
<a class="sourceLine" id="cb14-38" data-line-number="38">df_de_sig_unst <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb14-39" data-line-number="39"><span class="st">    </span><span class="kw">group_by</span>(gene) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb14-40" data-line-number="40"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, gene)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(id, <span class="dt">.keep_all =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb14-41" data-line-number="41"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">n_donors =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(n_donors) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb14-42" data-line-number="42"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">n_genes =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">data_type =</span> <span class="st">&quot;observed&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb14-43" data-line-number="43"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> n_donors, <span class="dt">y =</span> n_genes)) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-44" data-line-number="44"><span class="st">    </span><span class="co"># geom_segment(aes(x = n_donors, xend = n_donors, y = count, yend = 0.1), </span></a>
<a class="sourceLine" id="cb14-45" data-line-number="45"><span class="st">    </span><span class="co">#              colour = &quot;gray50&quot;) +</span></a>
<a class="sourceLine" id="cb14-46" data-line-number="46"><span class="co">#    geom_hline(yintercept = 0, linetype = 2) +</span></a>
<a class="sourceLine" id="cb14-47" data-line-number="47"><span class="st">    </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="dv">0</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-48" data-line-number="48"><span class="st">    </span><span class="kw">geom_boxplot</span>(<span class="kw">aes</span>(<span class="dt">group =</span> n_donors, <span class="dt">y =</span> n_genes, <span class="dt">colour =</span> data_type), </a>
<a class="sourceLine" id="cb14-49" data-line-number="49">                 <span class="dt">fill =</span> <span class="st">&quot;gray80&quot;</span>, <span class="dt">data =</span> df_perm, <span class="dt">show.legend =</span> <span class="ot">FALSE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-50" data-line-number="50"><span class="st">    </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">colour =</span> data_type), <span class="dt">shape =</span> <span class="dv">17</span>, <span class="dt">size =</span> <span class="dv">5</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-51" data-line-number="51"><span class="st">    </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks =</span> <span class="dv">0</span><span class="op">:</span><span class="dv">13</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-52" data-line-number="52"><span class="st">    </span><span class="kw">scale_y_sqrt</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">10</span>, <span class="dv">100</span>, <span class="dv">500</span>, <span class="dv">1000</span>, <span class="dv">2000</span>, <span class="dv">4000</span>),</a>
<a class="sourceLine" id="cb14-53" data-line-number="53">                 <span class="dt">labels =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">10</span>, <span class="dv">100</span>, <span class="dv">500</span>, <span class="dv">1000</span>, <span class="dv">2000</span>, <span class="dv">4000</span>),</a>
<a class="sourceLine" id="cb14-54" data-line-number="54">                 <span class="dt">limits =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">4500</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-55" data-line-number="55"><span class="st">    </span><span class="kw">scale_colour_manual</span>(<span class="dt">name =</span> <span class="st">&#39;&#39;</span>, </a>
<a class="sourceLine" id="cb14-56" data-line-number="56">                        <span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;observed&quot;</span> =<span class="st"> &quot;black&quot;</span>, <span class="st">&quot;permuted&quot;</span> =<span class="st"> &quot;gray50&quot;</span>), </a>
<a class="sourceLine" id="cb14-57" data-line-number="57">                        <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;observed&quot;</span>, <span class="st">&quot;permuted&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-58" data-line-number="58"><span class="st">    </span><span class="kw">coord_cartesian</span>(<span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">4500</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-59" data-line-number="59"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="kw">c</span>(<span class="fl">0.7</span>, <span class="fl">0.88</span>),</a>
<a class="sourceLine" id="cb14-60" data-line-number="60">          <span class="dt">panel.grid.major.x =</span> <span class="kw">element_blank</span>(), </a>
<a class="sourceLine" id="cb14-61" data-line-number="61">          <span class="dt">panel.grid.minor.x =</span> <span class="kw">element_blank</span>()) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-62" data-line-number="62"><span class="st">    </span><span class="kw">guides</span>(<span class="dt">colour =</span> <span class="kw">guide_legend</span>(<span class="dt">override.aes =</span> <span class="kw">list</span>(<span class="dt">shape =</span> <span class="kw">c</span>(<span class="dv">17</span>, <span class="dv">19</span>))),</a>
<a class="sourceLine" id="cb14-63" data-line-number="63">           <span class="dt">fill =</span> <span class="ot">FALSE</span>, <span class="dt">boxplot =</span> <span class="ot">FALSE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-64" data-line-number="64"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Number of lines significant (FDR &lt; 10%)&quot;</span>)  <span class="op">+</span></a>
<a class="sourceLine" id="cb14-65" data-line-number="65"><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Number of genes&quot;</span>)</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/permute-de-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/alldonors_de_n_sig_donors_n_sig_genes_sqrtscale_perm_skinny.png&quot;</span>, </a>
<a class="sourceLine" id="cb15-2" data-line-number="2">       <span class="dt">height =</span> <span class="fl">5.5</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)</a>
<a class="sourceLine" id="cb15-3" data-line-number="3"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/alldonors_de_n_sig_donors_n_sig_genes_sqrtscale_perm_veryskinny.png&quot;</span>, </a>
<a class="sourceLine" id="cb15-4" data-line-number="4">       <span class="dt">height =</span> <span class="fl">7.5</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)</a></code></pre></div>
<p>Look at recurrently DE genes.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1">df_gene_n_de &lt;-<span class="st"> </span>df_de_sig_unst <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb16-2" data-line-number="2"><span class="st">    </span><span class="kw">group_by</span>(gene) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb16-3" data-line-number="3"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, gene)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(id, <span class="dt">.keep_all =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb16-4" data-line-number="4"><span class="st">    </span><span class="kw">group_by</span>(gene) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb16-5" data-line-number="5"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">count =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb16-6" data-line-number="6"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(<span class="kw">desc</span>(count)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb16-7" data-line-number="7"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">ensembl_gene_id =</span> <span class="kw">gsub</span>(<span class="st">&quot;_.*&quot;</span>, <span class="st">&quot;&quot;</span>, gene),</a>
<a class="sourceLine" id="cb16-8" data-line-number="8">                  <span class="dt">hgnc_symbol =</span> <span class="kw">gsub</span>(<span class="st">&quot;.*_&quot;</span>, <span class="st">&quot;&quot;</span>, gene))</a>
<a class="sourceLine" id="cb16-9" data-line-number="9">df_gene_n_de &lt;-<span class="st"> </span><span class="kw">left_join</span>(</a>
<a class="sourceLine" id="cb16-10" data-line-number="10">    df_gene_n_de,</a>
<a class="sourceLine" id="cb16-11" data-line-number="11">    dplyr<span class="op">::</span><span class="kw">select</span>(de_res<span class="op">$</span>qlf_pairwise<span class="op">$</span>joxm<span class="op">$</span>clone2_clone1<span class="op">$</span>table, </a>
<a class="sourceLine" id="cb16-12" data-line-number="12">                  ensembl_gene_id, hgnc_symbol, entrezid)</a>
<a class="sourceLine" id="cb16-13" data-line-number="13">    )</a>
<a class="sourceLine" id="cb16-14" data-line-number="14">df_gene_n_de &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(</a>
<a class="sourceLine" id="cb16-15" data-line-number="15">    df_gene_n_de,</a>
<a class="sourceLine" id="cb16-16" data-line-number="16">    <span class="dt">cell_cycle_growth =</span> (entrezid <span class="op">%in%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb16-17" data-line-number="17"><span class="st">                             </span><span class="kw">c</span>(Hs.H<span class="op">$</span>HALLMARK_G2M_CHECKPOINT,</a>
<a class="sourceLine" id="cb16-18" data-line-number="18">                               Hs.H<span class="op">$</span>HALLMARK_MITOTIC_SPINDLE,</a>
<a class="sourceLine" id="cb16-19" data-line-number="19">                               Hs.H<span class="op">$</span>HALLMARK_E2F_TARGETS)),</a>
<a class="sourceLine" id="cb16-20" data-line-number="20">    <span class="dt">myc =</span> (entrezid <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(Hs.H<span class="op">$</span>HALLMARK_MYC_TARGETS_V1,</a>
<a class="sourceLine" id="cb16-21" data-line-number="21">                           Hs.H<span class="op">$</span>HALLMARK_MYC_TARGETS_V2)),</a>
<a class="sourceLine" id="cb16-22" data-line-number="22">    <span class="dt">emt =</span> (entrezid <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(Hs.H<span class="op">$</span>HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION))</a>
<a class="sourceLine" id="cb16-23" data-line-number="23">)</a>
<a class="sourceLine" id="cb16-24" data-line-number="24">df_gene_n_de <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(count <span class="op">&gt;=</span><span class="st"> </span><span class="dv">8</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb16-25" data-line-number="25"><span class="st">  </span>DT<span class="op">::</span><span class="kw">datatable</span>(.)</a></code></pre></div>
<div id="htmlwidget-d47eaf2c5a5686268d9a" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-d47eaf2c5a5686268d9a">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131"],["ENSG00000159753_RLTPR","ENSG00000166321_NUDT13","ENSG00000106819_ASPN","ENSG00000113368_LMNB1","ENSG00000146374_RSPO3","ENSG00000255526_NEDD8-MDP1","ENSG00000259511_UBE2Q2L","ENSG00000006016_CRLF1","ENSG00000018869_ZNF582","ENSG00000065717_TLE2","ENSG00000077420_APBB1IP","ENSG00000104081_BMF","ENSG00000125398_SOX9","ENSG00000127083_OMD","ENSG00000128045_RASL11B","ENSG00000128573_FOXP2","ENSG00000135750_KCNK1","ENSG00000144362_PHOSPHO2","ENSG00000146592_CREB5","ENSG00000146826_C7orf43","ENSG00000153094_BCL2L11","ENSG00000170745_KCNS3","ENSG00000178852_EFCAB13","ENSG00000185666_SYN3","ENSG00000188290_HES4","ENSG00000197409_HIST1H3D","ENSG00000214188_ST7-OT4","ENSG00000243789_JMJD7","ENSG00000064763_FAR2","ENSG00000065809_FAM107B","ENSG00000072080_SPP2","ENSG00000099999_RNF215","ENSG00000109794_FAM149A","ENSG00000115138_POMC","ENSG00000116701_NCF2","ENSG00000117245_KIF17","ENSG00000118292_C1orf54","ENSG00000120658_ENOX1","ENSG00000129465_RIPK3","ENSG00000143882_ATP6V1C2","ENSG00000150394_CDH8","ENSG00000155666_KDM8","ENSG00000157399_ARSE","ENSG00000163739_CXCL1","ENSG00000166415_WDR72","ENSG00000166670_MMP10","ENSG00000180881_CAPS2","ENSG00000182580_EPHB3","ENSG00000183208_GDPGP1","ENSG00000184381_PLA2G6","ENSG00000185697_MYBL1","ENSG00000196440_ARMCX4","ENSG00000197826_C4orf22","ENSG00000206530_WDR52","ENSG00000214827_MTCP1","ENSG00000218336_TENM3","ENSG00000224470_ATXN1L","ENSG00000232472_EEF1B2P3","ENSG00000232888_RPS11P5","ENSG00000242265_PEG10","ENSG00000250959_GLUD1P3","ENSG00000006468_ETV1","ENSG00000067445_TRO","ENSG00000072952_MRVI1","ENSG00000076258_FMO4","ENSG00000079313_REXO1","ENSG00000095397_DFNB31","ENSG00000095739_BAMBI","ENSG00000100784_RPS6KA5","ENSG00000101057_MYBL2","ENSG00000101160_CTSZ","ENSG00000105852_PON3","ENSG00000106025_TSPAN12","ENSG00000108823_SGCA","ENSG00000110723_EXPH5","ENSG00000120832_MTERFD3","ENSG00000121316_PLBD1","ENSG00000122862_SRGN","ENSG00000124116_WFDC3","ENSG00000125384_PTGER2","ENSG00000126603_GLIS2","ENSG00000127586_CHTF18","ENSG00000128918_ALDH1A2","ENSG00000130193_THEM6","ENSG00000136379_ABHD17C","ENSG00000136870_ZNF189","ENSG00000137033_IL33","ENSG00000137463_MGARP","ENSG00000142627_EPHA2","ENSG00000143178_TBX19","ENSG00000144355_DLX1","ENSG00000153234_NR4A2","ENSG00000158813_EDA","ENSG00000159023_EPB41","ENSG00000161653_NAGS","ENSG00000162591_MEGF6","ENSG00000162630_B3GALT2","ENSG00000164099_PRSS12","ENSG00000164124_TMEM144","ENSG00000164185_ZNF474","ENSG00000166987_MBD6","ENSG00000168944_CEP120","ENSG00000170561_IRX2","ENSG00000171097_CCBL1","ENSG00000173269_MMRN2","ENSG00000173406_DAB1","ENSG00000174599_TRAM1L1","ENSG00000177191_B3GNT8","ENSG00000178150_ZNF114","ENSG00000181915_ADO","ENSG00000182325_FBXL6","ENSG00000184349_EFNA5","ENSG00000185033_SEMA4B","ENSG00000185345_PARK2","ENSG00000185507_IRF7","ENSG00000185716_C16orf52","ENSG00000186493_C5orf38","ENSG00000187323_DCC","ENSG00000187800_PEAR1","ENSG00000196793_ZNF239","ENSG00000197472_ZNF695","ENSG00000198597_ZNF536","ENSG00000198673_FAM19A2","ENSG00000198837_DENND4B","ENSG00000211445_GPX3","ENSG00000211448_DIO2","ENSG00000214050_FBXO16","ENSG00000214265_SNURF","ENSG00000243444_PALM2","ENSG00000253159_PCDHGA12","ENSG00000259728_LINC00933"],[12,12,11,11,11,11,11,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8],["ENSG00000159753","ENSG00000166321","ENSG00000106819","ENSG00000113368","ENSG00000146374","ENSG00000255526","ENSG00000259511","ENSG00000006016","ENSG00000018869","ENSG00000065717","ENSG00000077420","ENSG00000104081","ENSG00000125398","ENSG00000127083","ENSG00000128045","ENSG00000128573","ENSG00000135750","ENSG00000144362","ENSG00000146592","ENSG00000146826","ENSG00000153094","ENSG00000170745","ENSG00000178852","ENSG00000185666","ENSG00000188290","ENSG00000197409","ENSG00000214188","ENSG00000243789","ENSG00000064763","ENSG00000065809","ENSG00000072080","ENSG00000099999","ENSG00000109794","ENSG00000115138","ENSG00000116701","ENSG00000117245","ENSG00000118292","ENSG00000120658","ENSG00000129465","ENSG00000143882","ENSG00000150394","ENSG00000155666","ENSG00000157399","ENSG00000163739","ENSG00000166415","ENSG00000166670","ENSG00000180881","ENSG00000182580","ENSG00000183208","ENSG00000184381","ENSG00000185697","ENSG00000196440","ENSG00000197826","ENSG00000206530","ENSG00000214827","ENSG00000218336","ENSG00000224470","ENSG00000232472","ENSG00000232888","ENSG00000242265","ENSG00000250959","ENSG00000006468","ENSG00000067445","ENSG00000072952","ENSG00000076258","ENSG00000079313","ENSG00000095397","ENSG00000095739","ENSG00000100784","ENSG00000101057","ENSG00000101160","ENSG00000105852","ENSG00000106025","ENSG00000108823","ENSG00000110723","ENSG00000120832","ENSG00000121316","ENSG00000122862","ENSG00000124116","ENSG00000125384","ENSG00000126603","ENSG00000127586","ENSG00000128918","ENSG00000130193","ENSG00000136379","ENSG00000136870","ENSG00000137033","ENSG00000137463","ENSG00000142627","ENSG00000143178","ENSG00000144355","ENSG00000153234","ENSG00000158813","ENSG00000159023","ENSG00000161653","ENSG00000162591","ENSG00000162630","ENSG00000164099","ENSG00000164124","ENSG00000164185","ENSG00000166987","ENSG00000168944","ENSG00000170561","ENSG00000171097","ENSG00000173269","ENSG00000173406","ENSG00000174599","ENSG00000177191","ENSG00000178150","ENSG00000181915","ENSG00000182325","ENSG00000184349","ENSG00000185033","ENSG00000185345","ENSG00000185507","ENSG00000185716","ENSG00000186493","ENSG00000187323","ENSG00000187800","ENSG00000196793","ENSG00000197472","ENSG00000198597","ENSG00000198673","ENSG00000198837","ENSG00000211445","ENSG00000211448","ENSG00000214050","ENSG00000214265","ENSG00000243444","ENSG00000253159","ENSG00000259728"],["RLTPR","NUDT13","ASPN","LMNB1","RSPO3","NEDD8-MDP1","UBE2Q2L","CRLF1","ZNF582","TLE2","APBB1IP","BMF","SOX9","OMD","RASL11B","FOXP2","KCNK1","PHOSPHO2","CREB5","C7orf43","BCL2L11","KCNS3","EFCAB13","SYN3","HES4","HIST1H3D","ST7-OT4","JMJD7","FAR2","FAM107B","SPP2","RNF215","FAM149A","POMC","NCF2","KIF17","C1orf54","ENOX1","RIPK3","ATP6V1C2","CDH8","KDM8","ARSE","CXCL1","WDR72","MMP10","CAPS2","EPHB3","GDPGP1","PLA2G6","MYBL1","ARMCX4","C4orf22","WDR52","MTCP1","TENM3","ATXN1L","EEF1B2P3","RPS11P5","PEG10","GLUD1P3","ETV1","TRO","MRVI1","FMO4","REXO1","DFNB31","BAMBI","RPS6KA5","MYBL2","CTSZ","PON3","TSPAN12","SGCA","EXPH5","MTERFD3","PLBD1","SRGN","WFDC3","PTGER2","GLIS2","CHTF18","ALDH1A2","THEM6","ABHD17C","ZNF189","IL33","MGARP","EPHA2","TBX19","DLX1","NR4A2","EDA","EPB41","NAGS","MEGF6","B3GALT2","PRSS12","TMEM144","ZNF474","MBD6","CEP120","IRX2","CCBL1","MMRN2","DAB1","TRAM1L1","B3GNT8","ZNF114","ADO","FBXL6","EFNA5","SEMA4B","PARK2","IRF7","C16orf52","C5orf38","DCC","PEAR1","ZNF239","ZNF695","ZNF536","FAM19A2","DENND4B","GPX3","DIO2","FBXO16","SNURF","PALM2","PCDHGA12","LINC00933"],["146206","25961","54829","4001","84870","100528064","100505679","9244","147948","7089","54518","90427","6662","4958","65997","93986","3775","493911","9586","55262","10018","3790","124989","8224","57801","8351","338069","100137047","55711","83641","6694","200312","25854","5443","4688","57576","79630","55068","11035","245973","1006","79831","415","2919","256764","4319","84698","2049","390637","8398","4603","100131755","255119","55779","4515","55714","342371",null,null,"23089","2749","2115","7216","10335","2329","57455","25861","25805","9252","4605","1522","5446","23554","6442","23086","80298","79887","5552","140686","5732","84662","63922","8854","51337","58489","7743","90865","84709","1969","9095","1745","4929","1896","2035","162417","1953","8707","8492","55314","133923","114785","153241","153572","883","79812","1600","133022","374907","163071","84890","26233","1946","10509","5071","3665","730094","153571","1630","375033","8187","57116","9745","338811","9909","2878","1734","157574",null,"114299","26025","100506874"],[false,false,false,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,true,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false],[false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false],[false,false,false,false,false,false,false,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>gene<\/th>\n      <th>count<\/th>\n      <th>ensembl_gene_id<\/th>\n      <th>hgnc_symbol<\/th>\n      <th>entrezid<\/th>\n      <th>cell_cycle_growth<\/th>\n      <th>myc<\/th>\n      <th>emt<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":2},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<hr />
</div>
<div id="camera-results" class="section level2">
<h2>Camera results</h2>
<p>First, aggregate gene set enrichment results across all donors.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1">fdr_thresh &lt;-<span class="st"> </span><span class="dv">1</span></a>
<a class="sourceLine" id="cb17-2" data-line-number="2">df_camera_all_unst &lt;-<span class="st"> </span><span class="kw">tibble</span>()</a>
<a class="sourceLine" id="cb17-3" data-line-number="3"><span class="cf">for</span> (geneset <span class="cf">in</span> <span class="kw">names</span>(de_res[[<span class="st">&quot;camera&quot;</span>]])) {</a>
<a class="sourceLine" id="cb17-4" data-line-number="4">    <span class="cf">for</span> (donor <span class="cf">in</span> <span class="kw">names</span>(de_res[[<span class="st">&quot;camera&quot;</span>]][[geneset]])) {</a>
<a class="sourceLine" id="cb17-5" data-line-number="5">        <span class="cf">for</span> (coeff <span class="cf">in</span> <span class="kw">names</span>(de_res[[<span class="st">&quot;camera&quot;</span>]][[geneset]][[donor]])) {</a>
<a class="sourceLine" id="cb17-6" data-line-number="6">            <span class="cf">for</span> (stat <span class="cf">in</span> <span class="kw">names</span>(de_res[[<span class="st">&quot;camera&quot;</span>]][[geneset]][[donor]][[coeff]])) {</a>
<a class="sourceLine" id="cb17-7" data-line-number="7">                tmp &lt;-<span class="st"> </span>de_res[[<span class="st">&quot;camera&quot;</span>]][[geneset]][[donor]][[coeff]][[stat]]</a>
<a class="sourceLine" id="cb17-8" data-line-number="8">                tmp &lt;-<span class="st"> </span>tmp[tmp<span class="op">$</span>FDR <span class="op">&lt;=</span><span class="st"> </span>fdr_thresh,]</a>
<a class="sourceLine" id="cb17-9" data-line-number="9">                <span class="cf">if</span> (<span class="kw">nrow</span>(tmp) <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.5</span>) {</a>
<a class="sourceLine" id="cb17-10" data-line-number="10">                    tmp[[<span class="st">&quot;collection&quot;</span>]] &lt;-<span class="st"> </span>geneset</a>
<a class="sourceLine" id="cb17-11" data-line-number="11">                    tmp[[<span class="st">&quot;geneset&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">rownames</span>(tmp)</a>
<a class="sourceLine" id="cb17-12" data-line-number="12">                    tmp[[<span class="st">&quot;coeff&quot;</span>]] &lt;-<span class="st"> </span>coeff</a>
<a class="sourceLine" id="cb17-13" data-line-number="13">                    tmp[[<span class="st">&quot;donor&quot;</span>]] &lt;-<span class="st"> </span>donor</a>
<a class="sourceLine" id="cb17-14" data-line-number="14">                    tmp[[<span class="st">&quot;stat&quot;</span>]] &lt;-<span class="st"> </span>stat</a>
<a class="sourceLine" id="cb17-15" data-line-number="15">                    df_camera_all_unst &lt;-<span class="st"> </span><span class="kw">bind_rows</span>(df_camera_all_unst, tmp)</a>
<a class="sourceLine" id="cb17-16" data-line-number="16">                }</a>
<a class="sourceLine" id="cb17-17" data-line-number="17">            }</a>
<a class="sourceLine" id="cb17-18" data-line-number="18">        }</a>
<a class="sourceLine" id="cb17-19" data-line-number="19">    }</a>
<a class="sourceLine" id="cb17-20" data-line-number="20">}</a>
<a class="sourceLine" id="cb17-21" data-line-number="21"></a>
<a class="sourceLine" id="cb17-22" data-line-number="22">fdr_thresh &lt;-<span class="st"> </span><span class="fl">0.05</span></a>
<a class="sourceLine" id="cb17-23" data-line-number="23">df_camera_sig_unst &lt;-<span class="st"> </span><span class="kw">tibble</span>()</a>
<a class="sourceLine" id="cb17-24" data-line-number="24"><span class="cf">for</span> (geneset <span class="cf">in</span> <span class="kw">names</span>(de_res[[<span class="st">&quot;camera&quot;</span>]])) {</a>
<a class="sourceLine" id="cb17-25" data-line-number="25">    <span class="cf">for</span> (donor <span class="cf">in</span> <span class="kw">names</span>(de_res[[<span class="st">&quot;camera&quot;</span>]][[geneset]])) {</a>
<a class="sourceLine" id="cb17-26" data-line-number="26">        <span class="cf">for</span> (coeff <span class="cf">in</span> <span class="kw">names</span>(de_res[[<span class="st">&quot;camera&quot;</span>]][[geneset]][[donor]])) {</a>
<a class="sourceLine" id="cb17-27" data-line-number="27">            <span class="cf">for</span> (stat <span class="cf">in</span> <span class="kw">names</span>(de_res[[<span class="st">&quot;camera&quot;</span>]][[geneset]][[donor]][[coeff]])) {</a>
<a class="sourceLine" id="cb17-28" data-line-number="28">                tmp &lt;-<span class="st"> </span>de_res[[<span class="st">&quot;camera&quot;</span>]][[geneset]][[donor]][[coeff]][[stat]]</a>
<a class="sourceLine" id="cb17-29" data-line-number="29">                tmp &lt;-<span class="st"> </span>tmp[tmp<span class="op">$</span>FDR <span class="op">&lt;=</span><span class="st"> </span>fdr_thresh,]</a>
<a class="sourceLine" id="cb17-30" data-line-number="30">                <span class="cf">if</span> (<span class="kw">nrow</span>(tmp) <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.5</span>) {</a>
<a class="sourceLine" id="cb17-31" data-line-number="31">                    tmp[[<span class="st">&quot;collection&quot;</span>]] &lt;-<span class="st"> </span>geneset</a>
<a class="sourceLine" id="cb17-32" data-line-number="32">                    tmp[[<span class="st">&quot;geneset&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">rownames</span>(tmp)</a>
<a class="sourceLine" id="cb17-33" data-line-number="33">                    tmp[[<span class="st">&quot;coeff&quot;</span>]] &lt;-<span class="st"> </span>coeff</a>
<a class="sourceLine" id="cb17-34" data-line-number="34">                    tmp[[<span class="st">&quot;donor&quot;</span>]] &lt;-<span class="st"> </span>donor</a>
<a class="sourceLine" id="cb17-35" data-line-number="35">                    tmp[[<span class="st">&quot;stat&quot;</span>]] &lt;-<span class="st"> </span>stat</a>
<a class="sourceLine" id="cb17-36" data-line-number="36">                    df_camera_sig_unst &lt;-<span class="st"> </span><span class="kw">bind_rows</span>(df_camera_sig_unst, tmp)</a>
<a class="sourceLine" id="cb17-37" data-line-number="37">                }</a>
<a class="sourceLine" id="cb17-38" data-line-number="38">            }</a>
<a class="sourceLine" id="cb17-39" data-line-number="39">        }</a>
<a class="sourceLine" id="cb17-40" data-line-number="40">    }</a>
<a class="sourceLine" id="cb17-41" data-line-number="41">}</a>
<a class="sourceLine" id="cb17-42" data-line-number="42"></a>
<a class="sourceLine" id="cb17-43" data-line-number="43">df_camera_sig_unst &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(</a>
<a class="sourceLine" id="cb17-44" data-line-number="44">    df_camera_sig_unst,</a>
<a class="sourceLine" id="cb17-45" data-line-number="45">    <span class="dt">contrast =</span> <span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot; - &quot;</span>, coeff),</a>
<a class="sourceLine" id="cb17-46" data-line-number="46">    <span class="dt">msigdb_collection =</span> plyr<span class="op">::</span><span class="kw">mapvalues</span>(collection, <span class="dt">from =</span> <span class="kw">c</span>(<span class="st">&quot;c2&quot;</span>, <span class="st">&quot;c6&quot;</span>, <span class="st">&quot;H&quot;</span>), <span class="dt">to =</span> <span class="kw">c</span>(<span class="st">&quot;MSigDB curated (c2)&quot;</span>, <span class="st">&quot;MSigDB oncogenic (c6)&quot;</span>, <span class="st">&quot;MSigDB Hallmark&quot;</span>)))</a>
<a class="sourceLine" id="cb17-47" data-line-number="47"></a>
<a class="sourceLine" id="cb17-48" data-line-number="48">df_camera_all_unst &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(</a>
<a class="sourceLine" id="cb17-49" data-line-number="49">    df_camera_all_unst,</a>
<a class="sourceLine" id="cb17-50" data-line-number="50">    <span class="dt">contrast =</span> <span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot; - &quot;</span>, coeff),</a>
<a class="sourceLine" id="cb17-51" data-line-number="51">    <span class="dt">msigdb_collection =</span> plyr<span class="op">::</span><span class="kw">mapvalues</span>(collection, <span class="dt">from =</span> <span class="kw">c</span>(<span class="st">&quot;c2&quot;</span>, <span class="st">&quot;c6&quot;</span>, <span class="st">&quot;H&quot;</span>), <span class="dt">to =</span> <span class="kw">c</span>(<span class="st">&quot;MSigDB curated (c2)&quot;</span>, <span class="st">&quot;MSigDB oncogenic (c6)&quot;</span>, <span class="st">&quot;MSigDB Hallmark&quot;</span>)))</a></code></pre></div>
<p>We now have a dataframe for significant (FDR &lt;5%) results from the camera gene set enrichment results.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1"><span class="kw">head</span>(df_camera_sig_unst)</a></code></pre></div>
<pre><code># A tibble: 6 x 11
  NGenes Direction  PValue     FDR collection geneset coeff donor stat 
   &lt;dbl&gt; &lt;chr&gt;       &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;      &lt;chr&gt;   &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;
1      7 Down      9.11e-9 4.29e-5 c2         WANG_R… clon… euts  signF
2      5 Down      2.04e-8 4.80e-5 c2         GUTIER… clon… euts  signF
3      7 Down      1.89e-7 2.95e-4 c2         KUMAMO… clon… euts  signF
4      3 Down      3.27e-7 3.85e-4 c2         REACTO… clon… euts  signF
5      2 Down      7.99e-7 7.52e-4 c2         SMID_B… clon… euts  signF
6      7 Down      1.20e-6 8.46e-4 c2         THILLA… clon… euts  signF
# … with 2 more variables: contrast &lt;chr&gt;, msigdb_collection &lt;chr&gt;</code></pre>
<p>And a dataframe with all results.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1"><span class="kw">head</span>(df_camera_all_unst)</a></code></pre></div>
<pre><code># A tibble: 6 x 11
  NGenes Direction  PValue   FDR collection geneset coeff donor stat 
   &lt;dbl&gt; &lt;chr&gt;       &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;      &lt;chr&gt;   &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;
1      7 Down      1.62e-4 0.764 c2         WANG_R… clon… euts  logFC
2      7 Down      5.48e-4 0.862 c2         THILLA… clon… euts  logFC
3     55 Up        7.22e-4 0.862 c2         PUJANA… clon… euts  logFC
4     10 Down      7.65e-4 0.862 c2         BIOCAR… clon… euts  logFC
5      3 Down      9.17e-4 0.862 c2         REACTO… clon… euts  logFC
6      2 Down      1.30e-3 0.871 c2         ALONSO… clon… euts  logFC
# … with 2 more variables: contrast &lt;chr&gt;, msigdb_collection &lt;chr&gt;</code></pre>
<p>For now, focus on gene set enrichment results computed using log-fold change statistics for pairwise comparisons of clones estimated from the edgeR QL-F models.</p>
<p>We can look at all significant results summarised by donor, geneset and pairwise contrast of clones.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1">df_camera_sig_unst <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb22-2" data-line-number="2"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(stat <span class="op">==</span><span class="st"> &quot;logFC&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb22-3" data-line-number="3"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">donor =</span> <span class="kw">factor</span>(donor, <span class="dt">levels =</span> <span class="kw">rev</span>(<span class="kw">levels</span>(<span class="kw">factor</span>(donor))))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb22-4" data-line-number="4"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">y =</span> <span class="op">-</span><span class="kw">log10</span>(PValue), <span class="dt">x =</span> donor, <span class="dt">colour =</span> contrast)) <span class="op">+</span></a>
<a class="sourceLine" id="cb22-5" data-line-number="5"><span class="st">    </span>ggbeeswarm<span class="op">::</span><span class="kw">geom_quasirandom</span>(<span class="dt">alpha =</span> <span class="fl">0.7</span>,  <span class="dt">groupOnX =</span> <span class="ot">FALSE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb22-6" data-line-number="6"><span class="st">    </span><span class="kw">facet_grid</span>(contrast <span class="op">~</span><span class="st"> </span>msigdb_collection) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb22-7" data-line-number="7"><span class="st">    </span><span class="kw">scale_colour_brewer</span>(<span class="dt">palette =</span> <span class="st">&quot;Accent&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb22-8" data-line-number="8"><span class="st">    </span><span class="kw">coord_flip</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>()</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/camera-allcontr-allsets-sig-bydonor-pvals-1.png" width="1056" style="display: block; margin: auto;" /></p>
<p>Similarly, we can look at all results summarised by donor, geneset and pairwise contrast of clones.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1">df_camera_all_unst <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb23-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(stat <span class="op">==</span><span class="st"> &quot;logFC&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb23-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">donor =</span> <span class="kw">factor</span>(donor, <span class="dt">levels =</span> <span class="kw">rev</span>(<span class="kw">levels</span>(<span class="kw">factor</span>(donor))))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb23-4" data-line-number="4"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">y =</span> <span class="op">-</span><span class="kw">log10</span>(FDR), <span class="dt">x =</span> donor, <span class="dt">colour =</span> contrast)) <span class="op">+</span></a>
<a class="sourceLine" id="cb23-5" data-line-number="5"><span class="st">  </span>ggbeeswarm<span class="op">::</span><span class="kw">geom_quasirandom</span>(<span class="dt">alpha =</span> <span class="fl">0.7</span>,  <span class="dt">groupOnX =</span> <span class="ot">FALSE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb23-6" data-line-number="6"><span class="st">  </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="op">-</span><span class="kw">log10</span>(<span class="fl">0.05</span>), <span class="dt">linetype =</span> <span class="dv">2</span>, <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb23-7" data-line-number="7"><span class="st">  </span><span class="kw">facet_grid</span>(contrast <span class="op">~</span><span class="st"> </span>msigdb_collection, <span class="dt">scales =</span> <span class="st">&quot;free_x&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb23-8" data-line-number="8"><span class="st">  </span><span class="kw">scale_colour_brewer</span>(<span class="dt">palette =</span> <span class="st">&quot;Accent&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb23-9" data-line-number="9"><span class="st">  </span><span class="kw">coord_flip</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>()</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/camera-allcontr-allsets-all-bydonor-fdr-1.png" width="1056" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/alldonors_camera_enrichment_by_donor_all_results.png&quot;</span>, </a>
<a class="sourceLine" id="cb24-2" data-line-number="2">       <span class="dt">height =</span> <span class="dv">16</span>, <span class="dt">width =</span> <span class="dv">14</span>)</a>
<a class="sourceLine" id="cb24-3" data-line-number="3"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/alldonors_camera_enrichment_by_donor_all_results.pdf&quot;</span>, </a>
<a class="sourceLine" id="cb24-4" data-line-number="4">       <span class="dt">height =</span> <span class="dv">16</span>, <span class="dt">width =</span> <span class="dv">14</span>)</a>
<a class="sourceLine" id="cb24-5" data-line-number="5"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/alldonors_camera_enrichment_by_donor_all_results.svg&quot;</span>, </a>
<a class="sourceLine" id="cb24-6" data-line-number="6">       <span class="dt">height =</span> <span class="dv">16</span>, <span class="dt">width =</span> <span class="dv">14</span>)</a></code></pre></div>
<p>We can check the number of significant gene sets for each donor, for each MSigDB gene set collection.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" data-line-number="1">df_camera_sig_unst <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb25-2" data-line-number="2"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(stat <span class="op">==</span><span class="st"> &quot;logFC&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb25-3" data-line-number="3"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(FDR <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb25-4" data-line-number="4"><span class="st">    </span><span class="kw">group_by</span>(donor, msigdb_collection) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb25-5" data-line-number="5"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">n_sig =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">print</span>(<span class="dt">n =</span> <span class="ot">Inf</span>)</a></code></pre></div>
<pre><code># A tibble: 63 x 3
# Groups:   donor [27]
   donor msigdb_collection     n_sig
   &lt;chr&gt; &lt;chr&gt;                 &lt;int&gt;
 1 euts  MSigDB curated (c2)      29
 2 fawm  MSigDB curated (c2)     317
 3 fawm  MSigDB Hallmark           8
 4 fawm  MSigDB oncogenic (c6)     4
 5 feec  MSigDB curated (c2)     521
 6 feec  MSigDB Hallmark           9
 7 feec  MSigDB oncogenic (c6)     4
 8 fikt  MSigDB curated (c2)      84
 9 fikt  MSigDB Hallmark           4
10 garx  MSigDB curated (c2)     429
11 garx  MSigDB Hallmark           8
12 garx  MSigDB oncogenic (c6)     7
13 gesg  MSigDB curated (c2)       2
14 heja  MSigDB curated (c2)      18
15 hipn  MSigDB curated (c2)      77
16 hipn  MSigDB oncogenic (c6)     2
17 ieki  MSigDB curated (c2)     274
18 ieki  MSigDB Hallmark           4
19 ieki  MSigDB oncogenic (c6)     2
20 joxm  MSigDB curated (c2)     142
21 joxm  MSigDB Hallmark           2
22 joxm  MSigDB oncogenic (c6)     1
23 laey  MSigDB curated (c2)     509
24 laey  MSigDB Hallmark           9
25 laey  MSigDB oncogenic (c6)    20
26 lexy  MSigDB curated (c2)     262
27 lexy  MSigDB Hallmark           4
28 naju  MSigDB curated (c2)      97
29 naju  MSigDB Hallmark           3
30 naju  MSigDB oncogenic (c6)    10
31 nusw  MSigDB oncogenic (c6)     1
32 oilg  MSigDB curated (c2)     409
33 oilg  MSigDB Hallmark           5
34 oilg  MSigDB oncogenic (c6)     3
35 pipw  MSigDB curated (c2)       1
36 puie  MSigDB curated (c2)      38
37 puie  MSigDB Hallmark           3
38 qayj  MSigDB curated (c2)     466
39 qayj  MSigDB Hallmark           7
40 qayj  MSigDB oncogenic (c6)    20
41 qolg  MSigDB curated (c2)     324
42 qolg  MSigDB Hallmark           5
43 qolg  MSigDB oncogenic (c6)     7
44 rozh  MSigDB curated (c2)     799
45 rozh  MSigDB Hallmark          16
46 rozh  MSigDB oncogenic (c6)    15
47 sehl  MSigDB curated (c2)     608
48 sehl  MSigDB Hallmark          12
49 sehl  MSigDB oncogenic (c6)    11
50 ualf  MSigDB curated (c2)     719
51 ualf  MSigDB Hallmark          10
52 ualf  MSigDB oncogenic (c6)    18
53 vass  MSigDB curated (c2)     587
54 vass  MSigDB Hallmark          14
55 vass  MSigDB oncogenic (c6)    10
56 wahn  MSigDB curated (c2)      32
57 wetu  MSigDB curated (c2)      19
58 xugn  MSigDB curated (c2)     521
59 xugn  MSigDB Hallmark          10
60 xugn  MSigDB oncogenic (c6)     6
61 zoxy  MSigDB curated (c2)     692
62 zoxy  MSigDB Hallmark          11
63 zoxy  MSigDB oncogenic (c6)    32</code></pre>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" data-line-number="1">df_camera_sig_unst <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb27-2" data-line-number="2"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(stat <span class="op">==</span><span class="st"> &quot;logFC&quot;</span>, msigdb_collection <span class="op">==</span><span class="st"> &quot;MSigDB Hallmark&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb27-3" data-line-number="3"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(FDR <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb27-4" data-line-number="4"><span class="st">    </span><span class="kw">group_by</span>(donor, msigdb_collection) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb27-5" data-line-number="5"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">n_sig =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">print</span>(<span class="dt">n =</span> <span class="ot">Inf</span>)</a></code></pre></div>
<pre><code># A tibble: 19 x 3
# Groups:   donor [19]
   donor msigdb_collection n_sig
   &lt;chr&gt; &lt;chr&gt;             &lt;int&gt;
 1 fawm  MSigDB Hallmark       8
 2 feec  MSigDB Hallmark       9
 3 fikt  MSigDB Hallmark       4
 4 garx  MSigDB Hallmark       8
 5 ieki  MSigDB Hallmark       4
 6 joxm  MSigDB Hallmark       2
 7 laey  MSigDB Hallmark       9
 8 lexy  MSigDB Hallmark       4
 9 naju  MSigDB Hallmark       3
10 oilg  MSigDB Hallmark       5
11 puie  MSigDB Hallmark       3
12 qayj  MSigDB Hallmark       7
13 qolg  MSigDB Hallmark       5
14 rozh  MSigDB Hallmark      16
15 sehl  MSigDB Hallmark      12
16 ualf  MSigDB Hallmark      10
17 vass  MSigDB Hallmark      14
18 xugn  MSigDB Hallmark      10
19 zoxy  MSigDB Hallmark      11</code></pre>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb29-1" data-line-number="1">df_camera_sig_unst <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb29-2" data-line-number="2"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(stat <span class="op">==</span><span class="st"> &quot;logFC&quot;</span>, msigdb_collection <span class="op">==</span><span class="st"> &quot;MSigDB Hallmark&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb29-3" data-line-number="3"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(FDR <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb29-4" data-line-number="4"><span class="st">    </span><span class="kw">group_by</span>(geneset, msigdb_collection) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb29-5" data-line-number="5"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">n_sig =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">print</span>(<span class="dt">n =</span> <span class="ot">Inf</span>)</a></code></pre></div>
<pre><code># A tibble: 17 x 3
# Groups:   geneset [17]
   geneset                                    msigdb_collection n_sig
   &lt;chr&gt;                                      &lt;chr&gt;             &lt;int&gt;
 1 HALLMARK_ANGIOGENESIS                      MSigDB Hallmark       5
 2 HALLMARK_APICAL_JUNCTION                   MSigDB Hallmark       1
 3 HALLMARK_COAGULATION                       MSigDB Hallmark       5
 4 HALLMARK_E2F_TARGETS                       MSigDB Hallmark      40
 5 HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION MSigDB Hallmark       5
 6 HALLMARK_G2M_CHECKPOINT                    MSigDB Hallmark      40
 7 HALLMARK_INTERFERON_ALPHA_RESPONSE         MSigDB Hallmark       2
 8 HALLMARK_INTERFERON_GAMMA_RESPONSE         MSigDB Hallmark       2
 9 HALLMARK_KRAS_SIGNALING_DN                 MSigDB Hallmark       1
10 HALLMARK_KRAS_SIGNALING_UP                 MSigDB Hallmark       1
11 HALLMARK_MITOTIC_SPINDLE                   MSigDB Hallmark      17
12 HALLMARK_MYC_TARGETS_V1                    MSigDB Hallmark       7
13 HALLMARK_MYC_TARGETS_V2                    MSigDB Hallmark       1
14 HALLMARK_PANCREAS_BETA_CELLS               MSigDB Hallmark       1
15 HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY   MSigDB Hallmark       1
16 HALLMARK_SPERMATOGENESIS                   MSigDB Hallmark      14
17 HALLMARK_TNFA_SIGNALING_VIA_NFKB           MSigDB Hallmark       1</code></pre>
<p>We can look at the number of significant gene sets for each donor.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb31-1" data-line-number="1"><span class="co">## simpler version</span></a>
<a class="sourceLine" id="cb31-2" data-line-number="2">df_camera_all_unst <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb31-3" data-line-number="3"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(stat <span class="op">==</span><span class="st"> &quot;logFC&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb31-4" data-line-number="4"><span class="st">    </span><span class="kw">group_by</span>(donor, msigdb_collection) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb31-5" data-line-number="5"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">n_sig =</span> <span class="kw">sum</span>(FDR <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb31-6" data-line-number="6"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">donor =</span> <span class="kw">factor</span>(donor, <span class="dt">levels =</span> <span class="kw">rev</span>(<span class="kw">levels</span>(<span class="kw">factor</span>(donor))))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb31-7" data-line-number="7"><span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">y =</span> n_sig, <span class="dt">x =</span> donor)) <span class="op">+</span></a>
<a class="sourceLine" id="cb31-8" data-line-number="8"><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="dv">1</span>, <span class="dt">size =</span> <span class="dv">4</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb31-9" data-line-number="9"><span class="st">    </span><span class="kw">facet_wrap</span>(<span class="op">~</span><span class="st"> </span>msigdb_collection, <span class="dt">scales =</span> <span class="st">&quot;free_x&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb31-10" data-line-number="10"><span class="st">    </span><span class="kw">scale_fill_brewer</span>(<span class="dt">palette =</span> <span class="st">&quot;Accent&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb31-11" data-line-number="11"><span class="st">    </span><span class="kw">coord_flip</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb31-12" data-line-number="12"><span class="st">    </span><span class="kw">theme_bw</span>(<span class="dv">16</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb31-13" data-line-number="13"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Donor&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Number of significant genesets (FDR &lt; 5%)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb31-14" data-line-number="14"><span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;Camera MSigDB gene set enrichment by line&quot;</span>)</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/alldonors_camera_enrichment_by_donor_simple-1.png" width="768" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb32-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/alldonors_camera_enrichment_by_donor_simple.png&quot;</span>, </a>
<a class="sourceLine" id="cb32-2" data-line-number="2">       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="dv">10</span>)</a>
<a class="sourceLine" id="cb32-3" data-line-number="3"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/alldonors_camera_enrichment_by_donor_simple.pdf&quot;</span>, </a>
<a class="sourceLine" id="cb32-4" data-line-number="4">       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="dv">10</span>)</a>
<a class="sourceLine" id="cb32-5" data-line-number="5"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/alldonors_camera_enrichment_by_donor_simple.svg&quot;</span>, </a>
<a class="sourceLine" id="cb32-6" data-line-number="6">       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="dv">10</span>)</a></code></pre></div>
<p>We can look at the effect of the the number of cells for each donor on the DE results obtained.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb33-1" data-line-number="1">ncells_by_donor &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="ot">NA</span>, <span class="kw">length</span>(sce_unst_list))</a>
<a class="sourceLine" id="cb33-2" data-line-number="2"><span class="kw">names</span>(ncells_by_donor) &lt;-<span class="st"> </span><span class="kw">names</span>(sce_unst_list)</a>
<a class="sourceLine" id="cb33-3" data-line-number="3"><span class="cf">for</span> (don <span class="cf">in</span> <span class="kw">names</span>(sce_unst_list))</a>
<a class="sourceLine" id="cb33-4" data-line-number="4">    ncells_by_donor[don] &lt;-<span class="st"> </span><span class="kw">ncol</span>(sce_unst_list[[don]])</a>
<a class="sourceLine" id="cb33-5" data-line-number="5"></a>
<a class="sourceLine" id="cb33-6" data-line-number="6">df_camera_all_unst <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb33-7" data-line-number="7"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(stat <span class="op">==</span><span class="st"> &quot;logFC&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb33-8" data-line-number="8"><span class="st">    </span><span class="kw">group_by</span>(donor, msigdb_collection) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb33-9" data-line-number="9"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">n_sig =</span> <span class="kw">sum</span>(FDR <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() -&gt;<span class="st"> </span>df_to_plot</a>
<a class="sourceLine" id="cb33-10" data-line-number="10">df_to_plot &lt;-<span class="st"> </span><span class="kw">inner_join</span>(df_to_plot, </a>
<a class="sourceLine" id="cb33-11" data-line-number="11">                        <span class="kw">tibble</span>(<span class="dt">donor =</span> <span class="kw">names</span>(ncells_by_donor),</a>
<a class="sourceLine" id="cb33-12" data-line-number="12">                                   <span class="dt">ncells =</span> ncells_by_donor))</a>
<a class="sourceLine" id="cb33-13" data-line-number="13">df_to_plot <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb33-14" data-line-number="14"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">donor =</span> <span class="kw">factor</span>(donor, <span class="dt">levels =</span> <span class="kw">rev</span>(<span class="kw">levels</span>(<span class="kw">factor</span>(donor))))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb33-15" data-line-number="15"><span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">y =</span> n_sig, <span class="dt">x =</span> donor, <span class="dt">size =</span> ncells)) <span class="op">+</span></a>
<a class="sourceLine" id="cb33-16" data-line-number="16"><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="dv">1</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb33-17" data-line-number="17"><span class="st">    </span><span class="kw">facet_wrap</span>(<span class="op">~</span><span class="st"> </span>msigdb_collection, <span class="dt">scales =</span> <span class="st">&quot;free_x&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb33-18" data-line-number="18"><span class="st">    </span><span class="kw">scale_fill_brewer</span>(<span class="dt">palette =</span> <span class="st">&quot;Accent&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb33-19" data-line-number="19"><span class="st">    </span><span class="kw">coord_flip</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb33-20" data-line-number="20"><span class="st">    </span><span class="kw">theme_bw</span>(<span class="dv">16</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb33-21" data-line-number="21"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Donor&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Number of significant genesets (FDR &lt; 5%)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb33-22" data-line-number="22"><span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;Camera MSigDB gene set enrichment by line&quot;</span>)</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/alldonors_camera_enrichment_by_donor_simple_size_by_ncells-1.png" width="768" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb34-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/alldonors_camera_enrichment_by_donor_simple_size_by_ncells.png&quot;</span>, </a>
<a class="sourceLine" id="cb34-2" data-line-number="2">       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="dv">10</span>)</a>
<a class="sourceLine" id="cb34-3" data-line-number="3"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/alldonors_camera_enrichment_by_donor_simple_size_by_ncells.pdf&quot;</span>, </a>
<a class="sourceLine" id="cb34-4" data-line-number="4">       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="dv">10</span>)</a>
<a class="sourceLine" id="cb34-5" data-line-number="5"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/alldonors_camera_enrichment_by_donor_simple_size_by_ncells.svg&quot;</span>, </a>
<a class="sourceLine" id="cb34-6" data-line-number="6">       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="dv">10</span>)</a></code></pre></div>
<div id="hallmark-gene-set" class="section level3">
<h3>Hallmark gene set</h3>
<p>Focus now on looking at DE results for the MSigDB Hallmark gene set (50 of the best-characterised gene sets as determined by MSigDB).</p>
<p>Look at the gene sets that are found to be enriched in multiple donors.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb35-1" data-line-number="1"><span class="co">## Hallmark geneset</span></a>
<a class="sourceLine" id="cb35-2" data-line-number="2">df_camera_sig_unst <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(collection <span class="op">==</span><span class="st"> &quot;H&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb35-3" data-line-number="3"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(stat <span class="op">==</span><span class="st"> &quot;logFC&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb35-4" data-line-number="4"><span class="st">    </span><span class="kw">group_by</span>(geneset) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb35-5" data-line-number="5"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, geneset)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(id, <span class="dt">.keep_all =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb35-6" data-line-number="6"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">n_donors =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(geneset, n_donors) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb35-7" data-line-number="7"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">geneset =</span> <span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot; &quot;</span>, <span class="kw">gsub</span>(<span class="st">&quot;HALLMARK_&quot;</span>, <span class="st">&quot;&quot;</span>, geneset))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb35-8" data-line-number="8"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">y =</span> n_donors, <span class="dt">x =</span> <span class="kw">reorder</span>(geneset, n_donors, max))) <span class="op">+</span></a>
<a class="sourceLine" id="cb35-9" data-line-number="9"><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.7</span>, <span class="dt">size =</span> <span class="dv">4</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb35-10" data-line-number="10"><span class="st">    </span>ggthemes<span class="op">::</span><span class="kw">scale_colour_tableau</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb35-11" data-line-number="11"><span class="st">    </span><span class="kw">coord_flip</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb35-12" data-line-number="12"><span class="st">    </span><span class="kw">theme_bw</span>(<span class="dv">14</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb35-13" data-line-number="13"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Gene set&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Number of lines significant&quot;</span>)</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/alldonors_camera_enrichment_H_by_geneset-1.png" width="960" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb36-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/alldonors_camera_enrichment_H_by_geneset.png&quot;</span>, </a>
<a class="sourceLine" id="cb36-2" data-line-number="2">       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="fl">9.5</span>)</a>
<a class="sourceLine" id="cb36-3" data-line-number="3"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/alldonors_camera_enrichment_H_by_geneset.pdf&quot;</span>, </a>
<a class="sourceLine" id="cb36-4" data-line-number="4">       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="fl">9.5</span>)</a>
<a class="sourceLine" id="cb36-5" data-line-number="5"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/alldonors_camera_enrichment_H_by_geneset.svg&quot;</span>, </a>
<a class="sourceLine" id="cb36-6" data-line-number="6">       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="fl">9.5</span>)</a>
<a class="sourceLine" id="cb36-7" data-line-number="7"><span class="co">## number of donors with at least one significant geneset</span></a>
<a class="sourceLine" id="cb36-8" data-line-number="8">tmp &lt;-<span class="st"> </span>df_camera_sig_unst <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(collection <span class="op">==</span><span class="st"> &quot;H&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb36-9" data-line-number="9"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(stat <span class="op">==</span><span class="st"> &quot;logFC&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb36-10" data-line-number="10"><span class="st">    </span><span class="kw">group_by</span>(geneset)</a>
<a class="sourceLine" id="cb36-11" data-line-number="11"><span class="kw">unique</span>(tmp[[<span class="st">&quot;donor&quot;</span>]])</a></code></pre></div>
<pre><code> [1] &quot;fawm&quot; &quot;feec&quot; &quot;fikt&quot; &quot;garx&quot; &quot;ieki&quot; &quot;joxm&quot; &quot;laey&quot; &quot;lexy&quot; &quot;naju&quot; &quot;oilg&quot;
[11] &quot;puie&quot; &quot;qayj&quot; &quot;qolg&quot; &quot;rozh&quot; &quot;sehl&quot; &quot;ualf&quot; &quot;vass&quot; &quot;xugn&quot; &quot;zoxy&quot;</code></pre>
<p>19 donors have at least one significantly enriched Hallmark gene set.</p>
<p>Overall, we obtain very similar gene set enrichment results if we use the signed-F statistic rather than logFC as the ranking statistic for camera.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb38-1" data-line-number="1"><span class="co">## Hallmark geneset</span></a>
<a class="sourceLine" id="cb38-2" data-line-number="2">df_camera_sig_unst <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(collection <span class="op">==</span><span class="st"> &quot;H&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb38-3" data-line-number="3"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(stat <span class="op">==</span><span class="st"> &quot;signF&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb38-4" data-line-number="4"><span class="st">    </span><span class="kw">group_by</span>(geneset) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb38-5" data-line-number="5"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, geneset)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(id, <span class="dt">.keep_all =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb38-6" data-line-number="6"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">n_donors =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(geneset, n_donors) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb38-7" data-line-number="7"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">geneset =</span> <span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot; &quot;</span>, <span class="kw">gsub</span>(<span class="st">&quot;HALLMARK_&quot;</span>, <span class="st">&quot;&quot;</span>, geneset))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb38-8" data-line-number="8"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">y =</span> n_donors, <span class="dt">x =</span> <span class="kw">reorder</span>(geneset, n_donors, max))) <span class="op">+</span></a>
<a class="sourceLine" id="cb38-9" data-line-number="9"><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.7</span>, <span class="dt">size =</span> <span class="dv">4</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb38-10" data-line-number="10"><span class="st">    </span>ggthemes<span class="op">::</span><span class="kw">scale_colour_tableau</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb38-11" data-line-number="11"><span class="st">    </span><span class="kw">coord_flip</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb38-12" data-line-number="12"><span class="st">    </span><span class="kw">theme_bw</span>(<span class="dv">14</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb38-13" data-line-number="13"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Gene set&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Number of lines significant&quot;</span>)</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>For gene sets related directly to cell cycle and growth, we see contrasts being both up- and down- regulated, but for EMT, coagulation and angiogenesis pathways, we only see these down-regulated.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb39-1" data-line-number="1">df_camera_sig_unst <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(collection <span class="op">==</span><span class="st"> &quot;H&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb39-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(stat <span class="op">==</span><span class="st"> &quot;logFC&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb39-3" data-line-number="3"><span class="st">  </span><span class="kw">group_by</span>(geneset, Direction) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb39-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, geneset)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb39-5" data-line-number="5"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">n_donors =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(geneset, n_donors) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb39-6" data-line-number="6"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">y =</span> n_donors, <span class="dt">x =</span> <span class="kw">reorder</span>(geneset, n_donors, max),</a>
<a class="sourceLine" id="cb39-7" data-line-number="7">             <span class="dt">colour =</span> Direction)) <span class="op">+</span></a>
<a class="sourceLine" id="cb39-8" data-line-number="8"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.7</span>, <span class="dt">size =</span> <span class="dv">4</span>, <span class="dt">position =</span> <span class="kw">position_dodge</span>(<span class="dt">width =</span> <span class="fl">0.5</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb39-9" data-line-number="9"><span class="st">  </span>ggthemes<span class="op">::</span><span class="kw">scale_colour_tableau</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb39-10" data-line-number="10"><span class="st">  </span><span class="kw">coord_flip</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb39-11" data-line-number="11"><span class="st">  </span><span class="kw">theme_bw</span>(<span class="dv">16</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb39-12" data-line-number="12"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Gene set&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Number of lines significant&quot;</span>)  <span class="op">+</span></a>
<a class="sourceLine" id="cb39-13" data-line-number="13"><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;Camera MSigDB Hallmark gene set enrichment&quot;</span>)</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/alldonors_camera_enrichment_H_by_geneset_by_dir-1.png" width="1536" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="heatmap-of-results-for-camera-hallmark-geneset-testing" class="section level2">
<h2>Heatmap of results for camera Hallmark geneset testing</h2>
<p>We can get an overview of all the Hallmark gene set results by producing a heatmap, first showing just the significant (FDR &lt; 5%) results across all donors and pairwise contrasts of clones.</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb40-1" data-line-number="1">repeated_sig_H_genesets &lt;-<span class="st"> </span>df_camera_sig_unst <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb40-2" data-line-number="2"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(collection <span class="op">==</span><span class="st"> &quot;H&quot;</span>, stat <span class="op">==</span><span class="st"> &quot;logFC&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb40-3" data-line-number="3"><span class="st">    </span><span class="kw">group_by</span>(geneset) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb40-4" data-line-number="4"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, geneset)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(id, <span class="dt">.keep_all =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb40-5" data-line-number="5"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">n_donors =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(n_donors) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb40-6" data-line-number="6"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(n_donors <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.5</span>) </a>
<a class="sourceLine" id="cb40-7" data-line-number="7">repeated_sig_H_genesets_vec &lt;-<span class="st"> </span><span class="kw">unique</span>(repeated_sig_H_genesets[[<span class="st">&quot;geneset&quot;</span>]])</a>
<a class="sourceLine" id="cb40-8" data-line-number="8">repeated_sig_H_genesets_vec &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot; &quot;</span>, <span class="kw">gsub</span>(<span class="st">&quot;HALLMARK_&quot;</span>, <span class="st">&quot;&quot;</span>, </a>
<a class="sourceLine" id="cb40-9" data-line-number="9">                                                   repeated_sig_H_genesets_vec))</a>
<a class="sourceLine" id="cb40-10" data-line-number="10">     </a>
<a class="sourceLine" id="cb40-11" data-line-number="11">df_<span class="dv">4</span>_heatmap &lt;-<span class="st"> </span>df_camera_sig_unst <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb40-12" data-line-number="12"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(collection <span class="op">==</span><span class="st"> &quot;H&quot;</span>, stat <span class="op">==</span><span class="st"> &quot;logFC&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb40-13" data-line-number="13"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">geneset =</span> <span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot; &quot;</span>, <span class="kw">gsub</span>(<span class="st">&quot;HALLMARK_&quot;</span>, <span class="st">&quot;&quot;</span>, geneset))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb40-14" data-line-number="14"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">geneset =</span> </a>
<a class="sourceLine" id="cb40-15" data-line-number="15">                  <span class="kw">factor</span>(geneset, <span class="dt">levels =</span> repeated_sig_H_genesets_vec)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb40-16" data-line-number="16"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(geneset <span class="op">%in%</span><span class="st"> </span>repeated_sig_H_genesets_vec) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb40-17" data-line-number="17"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, <span class="st">&quot;: &quot;</span>, contrast))</a>
<a class="sourceLine" id="cb40-18" data-line-number="18">  </a>
<a class="sourceLine" id="cb40-19" data-line-number="19"></a>
<a class="sourceLine" id="cb40-20" data-line-number="20">div_lines &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;: c.*&quot;</span>, <span class="st">&quot;&quot;</span>,</a>
<a class="sourceLine" id="cb40-21" data-line-number="21">     <span class="kw">sort</span>(<span class="kw">unique</span>(<span class="kw">paste0</span>(df_<span class="dv">4</span>_heatmap[[<span class="st">&quot;donor&quot;</span>]], <span class="st">&quot;: &quot;</span>, </a>
<a class="sourceLine" id="cb40-22" data-line-number="22">                        df_<span class="dv">4</span>_heatmap[[<span class="st">&quot;contrast&quot;</span>]])))) <span class="op">%&gt;%</span><span class="st"> </span>table <span class="op">%&gt;%</span><span class="st"> </span>cumsum <span class="op">+</span><span class="st"> </span><span class="fl">0.5</span></a>
<a class="sourceLine" id="cb40-23" data-line-number="23">    </a>
<a class="sourceLine" id="cb40-24" data-line-number="24">df_<span class="dv">4</span>_heatmap <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb40-25" data-line-number="25"><span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> id, <span class="dt">y =</span> geneset, <span class="dt">fill =</span> Direction)) <span class="op">+</span></a>
<a class="sourceLine" id="cb40-26" data-line-number="26"><span class="st">    </span><span class="kw">geom_tile</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb40-27" data-line-number="27"><span class="st">    </span><span class="kw">geom_vline</span>(<span class="dt">xintercept =</span> div_lines, <span class="dt">colour =</span> <span class="st">&quot;gray70&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb40-28" data-line-number="28"><span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;lightgoldenrod1&quot;</span>, <span class="st">&quot;sienna1&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb40-29" data-line-number="29"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">60</span>, <span class="dt">hjust =</span> <span class="dv">1</span>))</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/top_genesets_H_direction_heatmap-1.png" width="1152" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb41-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/top_genesets_H_direction_heatmap.png&quot;</span>, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">width =</span> <span class="dv">12</span>)</a>
<a class="sourceLine" id="cb41-2" data-line-number="2"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/top_genesets_H_direction_heatmap.pdf&quot;</span>, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">width =</span> <span class="dv">12</span>)</a>
<a class="sourceLine" id="cb41-3" data-line-number="3"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/top_genesets_H_direction_heatmap.svg&quot;</span>, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">width =</span> <span class="dv">12</span>)</a></code></pre></div>
<p>We can do the same for all results for the gene sets that are significantly enriched in at least two donors.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb42-1" data-line-number="1">df_camera_all_unst <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb42-2" data-line-number="2"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">geneset =</span> <span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot; &quot;</span>, <span class="kw">gsub</span>(<span class="st">&quot;HALLMARK_&quot;</span>, <span class="st">&quot;&quot;</span>, geneset))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb42-3" data-line-number="3"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">geneset =</span> </a>
<a class="sourceLine" id="cb42-4" data-line-number="4">                      <span class="kw">factor</span>(geneset, <span class="dt">levels =</span> repeated_sig_H_genesets_vec)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb42-5" data-line-number="5"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(geneset <span class="op">%in%</span><span class="st"> </span>repeated_sig_H_genesets_vec) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb42-6" data-line-number="6"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, <span class="st">&quot;: &quot;</span>, contrast)) -&gt;</a>
<a class="sourceLine" id="cb42-7" data-line-number="7"><span class="st">    </span>df_<span class="dv">4</span>_heatmap_all</a>
<a class="sourceLine" id="cb42-8" data-line-number="8"></a>
<a class="sourceLine" id="cb42-9" data-line-number="9">df_<span class="dv">4</span>_heatmap_all &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(</a>
<a class="sourceLine" id="cb42-10" data-line-number="10">    df_<span class="dv">4</span>_heatmap_all,</a>
<a class="sourceLine" id="cb42-11" data-line-number="11">    <span class="dt">minlog10P =</span> <span class="kw">cut</span>(<span class="op">-</span><span class="kw">log10</span>(PValue), <span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">4</span>, <span class="dv">5</span>, <span class="dv">30</span>)))</a>
<a class="sourceLine" id="cb42-12" data-line-number="12"></a>
<a class="sourceLine" id="cb42-13" data-line-number="13">div_lines_all &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;: c.*&quot;</span>, <span class="st">&quot;&quot;</span>,</a>
<a class="sourceLine" id="cb42-14" data-line-number="14">     <span class="kw">sort</span>(<span class="kw">unique</span>(<span class="kw">paste0</span>(df_<span class="dv">4</span>_heatmap_all[[<span class="st">&quot;donor&quot;</span>]], <span class="st">&quot;: &quot;</span>, </a>
<a class="sourceLine" id="cb42-15" data-line-number="15">                        df_<span class="dv">4</span>_heatmap_all[[<span class="st">&quot;contrast&quot;</span>]])))) <span class="op">%&gt;%</span><span class="st"> </span>table <span class="op">%&gt;%</span><span class="st"> </span>cumsum <span class="op">+</span><span class="st"> </span><span class="fl">0.5</span></a>
<a class="sourceLine" id="cb42-16" data-line-number="16"></a>
<a class="sourceLine" id="cb42-17" data-line-number="17">pp &lt;-<span class="st"> </span>df_<span class="dv">4</span>_heatmap_all <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb42-18" data-line-number="18"><span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> id, <span class="dt">y =</span> geneset, <span class="dt">fill =</span> Direction, <span class="dt">alpha =</span> minlog10P)) <span class="op">+</span></a>
<a class="sourceLine" id="cb42-19" data-line-number="19"><span class="st">    </span><span class="kw">geom_tile</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb42-20" data-line-number="20"><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="dv">1</span>, <span class="dt">data =</span> df_<span class="dv">4</span>_heatmap, <span class="dt">pch =</span> <span class="dv">19</span>, <span class="dt">size =</span> <span class="fl">0.5</span>, <span class="dt">show.legend =</span> <span class="ot">FALSE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb42-21" data-line-number="21"><span class="st">    </span><span class="kw">geom_vline</span>(<span class="dt">xintercept =</span> div_lines_all, <span class="dt">colour =</span> <span class="st">&quot;gray70&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb42-22" data-line-number="22"><span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;lightgoldenrod1&quot;</span>, <span class="st">&quot;sienna1&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb42-23" data-line-number="23"><span class="st">    </span><span class="kw">scale_alpha_discrete</span>(<span class="dt">name =</span> <span class="st">&quot;-log10(P)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb42-24" data-line-number="24"><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Gene set&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb42-25" data-line-number="25"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Line and clone comparison&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb42-26" data-line-number="26"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">60</span>, <span class="dt">hjust =</span> <span class="dv">1</span>),</a>
<a class="sourceLine" id="cb42-27" data-line-number="27">          <span class="dt">legend.position =</span> <span class="st">&quot;right&quot;</span>)</a>
<a class="sourceLine" id="cb42-28" data-line-number="28">pp    </a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/top_genesets_H_direction_heatmap_all_contrasts-1.png" width="1728" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb43-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/top_genesets_H_direction_heatmap_all_contrasts.png&quot;</span>, <span class="dt">height =</span> <span class="dv">9</span>, <span class="dt">width =</span> <span class="dv">20</span>)</a>
<a class="sourceLine" id="cb43-2" data-line-number="2"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/top_genesets_H_direction_heatmap_all_contrasts.pdf&quot;</span>, <span class="dt">height =</span> <span class="dv">9</span>, <span class="dt">width =</span> <span class="dv">20</span>)</a>
<a class="sourceLine" id="cb43-3" data-line-number="3"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/top_genesets_H_direction_heatmap_all_contrasts.svg&quot;</span>, <span class="dt">height =</span> <span class="dv">9</span>, <span class="dt">width =</span> <span class="dv">20</span>)</a></code></pre></div>
<p>We can also add a panel to this figure showing the number of donors in which each of these gene sets is significantly enriched.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb44-1" data-line-number="1"><span class="co">## Hallmark geneset</span></a>
<a class="sourceLine" id="cb44-2" data-line-number="2">pp_nsig &lt;-<span class="st"> </span>df_camera_sig_unst <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(collection <span class="op">==</span><span class="st"> &quot;H&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb44-3" data-line-number="3"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(stat <span class="op">==</span><span class="st"> &quot;logFC&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb44-4" data-line-number="4"><span class="st">    </span><span class="kw">group_by</span>(geneset) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb44-5" data-line-number="5"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, geneset)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(id, <span class="dt">.keep_all =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb44-6" data-line-number="6"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">n_donors =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(geneset, n_donors) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb44-7" data-line-number="7"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">geneset =</span> <span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot; &quot;</span>, <span class="kw">gsub</span>(<span class="st">&quot;HALLMARK_&quot;</span>, <span class="st">&quot;&quot;</span>, geneset))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb44-8" data-line-number="8"><span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">y =</span> n_donors, <span class="dt">x =</span> <span class="kw">reorder</span>(geneset, n_donors, max))) <span class="op">+</span></a>
<a class="sourceLine" id="cb44-9" data-line-number="9"><span class="st">    </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="dv">0</span>, <span class="dt">colour =</span> <span class="st">&quot;gray50&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb44-10" data-line-number="10"><span class="st">    </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">xend =</span> <span class="kw">reorder</span>(geneset, n_donors, max), <span class="dt">yend =</span> <span class="dv">0</span>),</a>
<a class="sourceLine" id="cb44-11" data-line-number="11">                 <span class="dt">colour =</span> <span class="st">&quot;gray50&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb44-12" data-line-number="12"><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">4</span>, <span class="dt">colour =</span> <span class="st">&quot;gray30&quot;</span>, <span class="dt">alpha =</span> <span class="dv">1</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb44-13" data-line-number="13"><span class="st">    </span>ggthemes<span class="op">::</span><span class="kw">scale_colour_tableau</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb44-14" data-line-number="14"><span class="st">    </span><span class="kw">coord_flip</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb44-15" data-line-number="15"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Gene set&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Number of lines significant&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb44-16" data-line-number="16"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.title.y =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb44-17" data-line-number="17">          <span class="dt">axis.text.y =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb44-18" data-line-number="18">          <span class="dt">axis.ticks.y =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb44-19" data-line-number="19">          <span class="dt">axis.line.y =</span> <span class="kw">element_blank</span>())</a>
<a class="sourceLine" id="cb44-20" data-line-number="20"></a>
<a class="sourceLine" id="cb44-21" data-line-number="21">prow &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(pp <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>), </a>
<a class="sourceLine" id="cb44-22" data-line-number="22">                  pp_nsig, <span class="dt">align =</span> <span class="st">&#39;h&#39;</span>, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="dv">7</span>, <span class="dv">1</span>))</a>
<a class="sourceLine" id="cb44-23" data-line-number="23">lgnd &lt;-<span class="st"> </span><span class="kw">get_legend</span>(pp)</a>
<a class="sourceLine" id="cb44-24" data-line-number="24"><span class="kw">plot_grid</span>(prow, lgnd, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="dv">3</span>, <span class="fl">.3</span>))</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/top_genesets_H_direction_heatmap_all_contrasts_with_nsig_donors-1.png" width="1920" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb45-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/top_genesets_H_direction_heatmap_all_contrasts_with_nsig_donors.png&quot;</span>, <span class="dt">height =</span> <span class="dv">9</span>, <span class="dt">width =</span> <span class="dv">22</span>)</a>
<a class="sourceLine" id="cb45-2" data-line-number="2"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/top_genesets_H_direction_heatmap_all_contrasts_with_nsig_donors.pdf&quot;</span>, <span class="dt">height =</span> <span class="dv">9</span>, <span class="dt">width =</span> <span class="dv">22</span>)</a>
<a class="sourceLine" id="cb45-3" data-line-number="3"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/top_genesets_H_direction_heatmap_all_contrasts_with_nsig_donors.svg&quot;</span>, <span class="dt">height =</span> <span class="dv">9</span>, <span class="dt">width =</span> <span class="dv">22</span>)</a></code></pre></div>
<p>However, the plot above is very complicated, so we may want to focus just on the lines for which there are multiple clones that show differing behaviour amongst each other. To simplify, let us just look at the 19 donors that have significant geneset enrichment for at least one contrasts and just look at the 8 gene sets that are significant in at least three lines.</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb46-1" data-line-number="1">repeated_sig_H_genesets3 &lt;-<span class="st"> </span>df_camera_sig_unst <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb46-2" data-line-number="2"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(collection <span class="op">==</span><span class="st"> &quot;H&quot;</span>, stat <span class="op">==</span><span class="st"> &quot;logFC&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb46-3" data-line-number="3"><span class="st">    </span><span class="kw">group_by</span>(geneset) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb46-4" data-line-number="4"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, geneset)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(id, <span class="dt">.keep_all =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb46-5" data-line-number="5"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">n_donors =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(n_donors) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb46-6" data-line-number="6"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(n_donors <span class="op">&gt;</span><span class="st"> </span><span class="fl">2.5</span>) </a>
<a class="sourceLine" id="cb46-7" data-line-number="7">repeated_sig_H_genesets_vec3 &lt;-<span class="st"> </span><span class="kw">unique</span>(repeated_sig_H_genesets3[[<span class="st">&quot;geneset&quot;</span>]])</a>
<a class="sourceLine" id="cb46-8" data-line-number="8">repeated_sig_H_genesets_vec3 &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot; &quot;</span>, <span class="kw">gsub</span>(<span class="st">&quot;HALLMARK_&quot;</span>, <span class="st">&quot;&quot;</span>, </a>
<a class="sourceLine" id="cb46-9" data-line-number="9">                                                   repeated_sig_H_genesets_vec3))</a>
<a class="sourceLine" id="cb46-10" data-line-number="10"></a>
<a class="sourceLine" id="cb46-11" data-line-number="11">selected_donors &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;fawm&quot;</span>, <span class="st">&quot;feec&quot;</span>, <span class="st">&quot;fikt&quot;</span>, <span class="st">&quot;garx&quot;</span>, <span class="st">&quot;ieki&quot;</span>, <span class="st">&quot;joxm&quot;</span>, <span class="st">&quot;laey&quot;</span>, </a>
<a class="sourceLine" id="cb46-12" data-line-number="12">                     <span class="st">&quot;lexy&quot;</span>, <span class="st">&quot;naju&quot;</span>, <span class="st">&quot;oilg&quot;</span>, <span class="st">&quot;qayj&quot;</span>, <span class="st">&quot;qolg&quot;</span>, <span class="st">&quot;rozh&quot;</span>, </a>
<a class="sourceLine" id="cb46-13" data-line-number="13">                     <span class="st">&quot;sehl&quot;</span>, <span class="st">&quot;ualf&quot;</span>, <span class="st">&quot;vass&quot;</span>, <span class="st">&quot;wahn&quot;</span>, <span class="st">&quot;xugn&quot;</span>, <span class="st">&quot;zoxy&quot;</span>)</a>
<a class="sourceLine" id="cb46-14" data-line-number="14"></a>
<a class="sourceLine" id="cb46-15" data-line-number="15">df_<span class="dv">4</span>_heatmap_filt &lt;-<span class="st"> </span>df_camera_all_unst <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb46-16" data-line-number="16"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">geneset =</span> <span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot; &quot;</span>, <span class="kw">gsub</span>(<span class="st">&quot;HALLMARK_&quot;</span>, <span class="st">&quot;&quot;</span>, geneset))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb46-17" data-line-number="17"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">geneset =</span> </a>
<a class="sourceLine" id="cb46-18" data-line-number="18">                  <span class="kw">factor</span>(geneset, <span class="dt">levels =</span> repeated_sig_H_genesets_vec3)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb46-19" data-line-number="19"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(geneset <span class="op">%in%</span><span class="st"> </span>repeated_sig_H_genesets_vec3,</a>
<a class="sourceLine" id="cb46-20" data-line-number="20">                donor <span class="op">%in%</span><span class="st"> </span>selected_donors) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb46-21" data-line-number="21"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, <span class="st">&quot;: &quot;</span>, contrast))</a>
<a class="sourceLine" id="cb46-22" data-line-number="22"></a>
<a class="sourceLine" id="cb46-23" data-line-number="23">df_<span class="dv">4</span>_heatmap_filt_sig &lt;-<span class="st"> </span>df_camera_sig_unst <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb46-24" data-line-number="24"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(collection <span class="op">==</span><span class="st"> &quot;H&quot;</span>, stat <span class="op">==</span><span class="st"> &quot;logFC&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb46-25" data-line-number="25"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">geneset =</span> <span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot; &quot;</span>, <span class="kw">gsub</span>(<span class="st">&quot;HALLMARK_&quot;</span>, <span class="st">&quot;&quot;</span>, geneset))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb46-26" data-line-number="26"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">geneset =</span> </a>
<a class="sourceLine" id="cb46-27" data-line-number="27">                  <span class="kw">factor</span>(geneset, <span class="dt">levels =</span> repeated_sig_H_genesets_vec3)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb46-28" data-line-number="28"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(geneset <span class="op">%in%</span><span class="st"> </span>repeated_sig_H_genesets_vec3,</a>
<a class="sourceLine" id="cb46-29" data-line-number="29">                donor <span class="op">%in%</span><span class="st"> </span>selected_donors) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb46-30" data-line-number="30"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, <span class="st">&quot;: &quot;</span>, contrast))</a>
<a class="sourceLine" id="cb46-31" data-line-number="31"></a>
<a class="sourceLine" id="cb46-32" data-line-number="32">df_<span class="dv">4</span>_heatmap_filt &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(</a>
<a class="sourceLine" id="cb46-33" data-line-number="33">    df_<span class="dv">4</span>_heatmap_filt,</a>
<a class="sourceLine" id="cb46-34" data-line-number="34">    <span class="dt">minlog10P =</span> <span class="kw">cut</span>(<span class="op">-</span><span class="kw">log10</span>(PValue), <span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">4</span>, <span class="dv">5</span>, <span class="dv">30</span>)))</a>
<a class="sourceLine" id="cb46-35" data-line-number="35"></a>
<a class="sourceLine" id="cb46-36" data-line-number="36">div_lines_filt &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;: c.*&quot;</span>, <span class="st">&quot;&quot;</span>,</a>
<a class="sourceLine" id="cb46-37" data-line-number="37">                       <span class="kw">sort</span>(<span class="kw">unique</span>(<span class="kw">paste0</span>(df_<span class="dv">4</span>_heatmap_filt[[<span class="st">&quot;donor&quot;</span>]], <span class="st">&quot;: &quot;</span>, </a>
<a class="sourceLine" id="cb46-38" data-line-number="38">                                          df_<span class="dv">4</span>_heatmap_filt[[<span class="st">&quot;contrast&quot;</span>]])))) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb46-39" data-line-number="39"><span class="st">  </span>table <span class="op">%&gt;%</span><span class="st"> </span>cumsum <span class="op">+</span><span class="st"> </span><span class="fl">0.5</span></a>
<a class="sourceLine" id="cb46-40" data-line-number="40"></a>
<a class="sourceLine" id="cb46-41" data-line-number="41">pp_filt &lt;-<span class="st"> </span>df_<span class="dv">4</span>_heatmap_filt <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb46-42" data-line-number="42"><span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> id, <span class="dt">y =</span> geneset, <span class="dt">fill =</span> Direction, <span class="dt">alpha =</span> minlog10P)) <span class="op">+</span></a>
<a class="sourceLine" id="cb46-43" data-line-number="43"><span class="st">    </span><span class="kw">geom_tile</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb46-44" data-line-number="44"><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="dv">1</span>, <span class="dt">data =</span> df_<span class="dv">4</span>_heatmap_filt_sig, <span class="dt">pch =</span> <span class="dv">19</span>, <span class="dt">size =</span> <span class="fl">0.5</span>, </a>
<a class="sourceLine" id="cb46-45" data-line-number="45">               <span class="dt">show.legend =</span> <span class="ot">FALSE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb46-46" data-line-number="46"><span class="st">    </span><span class="kw">geom_vline</span>(<span class="dt">xintercept =</span> div_lines_filt, <span class="dt">colour =</span> <span class="st">&quot;gray70&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb46-47" data-line-number="47"><span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;lightgoldenrod1&quot;</span>, <span class="st">&quot;sienna1&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb46-48" data-line-number="48"><span class="st">    </span><span class="kw">scale_alpha_discrete</span>(<span class="dt">name =</span> <span class="st">&quot;-log10(P)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb46-49" data-line-number="49"><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Gene set&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb46-50" data-line-number="50"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Line and clone comparison&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb46-51" data-line-number="51"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">60</span>, <span class="dt">hjust =</span> <span class="dv">1</span>),</a>
<a class="sourceLine" id="cb46-52" data-line-number="52">          <span class="dt">legend.position =</span> <span class="st">&quot;right&quot;</span>)</a>
<a class="sourceLine" id="cb46-53" data-line-number="53">pp_filt    </a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/top_genesets_H_direction_heatmap_all_contrasts-simple-1.png" width="1728" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb47-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/top_genesets_H_direction_heatmap_filt_contrasts.png&quot;</span>, <span class="dt">plot =</span> pp_filt, <span class="dt">height =</span> <span class="dv">5</span>, <span class="dt">width =</span> <span class="dv">12</span>)</a>
<a class="sourceLine" id="cb47-2" data-line-number="2"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/top_genesets_H_direction_heatmap_filt_contrasts.pdf&quot;</span>, <span class="dt">plot =</span> pp_filt, <span class="dt">height =</span> <span class="dv">5</span>, <span class="dt">width =</span> <span class="dv">12</span>)</a>
<a class="sourceLine" id="cb47-3" data-line-number="3"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/top_genesets_H_direction_heatmap_filt_contrasts.svg&quot;</span>, <span class="dt">plot =</span> pp_filt, <span class="dt">height =</span> <span class="dv">5</span>, <span class="dt">width =</span> <span class="dv">12</span>)</a>
<a class="sourceLine" id="cb47-4" data-line-number="4"></a>
<a class="sourceLine" id="cb47-5" data-line-number="5"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/top_genesets_H_direction_heatmap_filt_contrasts_wide.png&quot;</span>, <span class="dt">plot =</span> pp_filt, <span class="dt">height =</span> <span class="dv">5</span>, <span class="dt">width =</span> <span class="dv">16</span>)</a>
<a class="sourceLine" id="cb47-6" data-line-number="6"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/top_genesets_H_direction_heatmap_filt_contrasts_wide.pdf&quot;</span>, <span class="dt">plot =</span> pp_filt, <span class="dt">height =</span> <span class="dv">5</span>, <span class="dt">width =</span> <span class="dv">16</span>)</a>
<a class="sourceLine" id="cb47-7" data-line-number="7"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/top_genesets_H_direction_heatmap_filt_contrasts_wide.svg&quot;</span>, <span class="dt">plot =</span> pp_filt, <span class="dt">height =</span> <span class="dv">5</span>, <span class="dt">width =</span> <span class="dv">16</span>)</a></code></pre></div>
</div>
<div id="correlation-of-gene-set-results-and-genes-contained" class="section level2">
<h2>Correlation of gene set results and genes contained</h2>
<p>Let’s look at the correlation between gene set results (Spearman correlation of signed -log10(P-values) from <em>camera</em> tests) and compare to the proportion of genes overlapping between pairs of gene sets.</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb48-1" data-line-number="1">repeated_sig_H_genesets_vec2 &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;HALLMARK_&quot;</span>, </a>
<a class="sourceLine" id="cb48-2" data-line-number="2">                                       <span class="kw">gsub</span>(<span class="st">&quot; &quot;</span>, <span class="st">&quot;_&quot;</span>, repeated_sig_H_genesets_vec))</a>
<a class="sourceLine" id="cb48-3" data-line-number="3"><span class="co">## all results</span></a>
<a class="sourceLine" id="cb48-4" data-line-number="4">df_H_pvals &lt;-<span class="st"> </span>df_camera_all_unst <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(collection <span class="op">==</span><span class="st"> &quot;H&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb48-5" data-line-number="5"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(stat <span class="op">==</span><span class="st"> &quot;logFC&quot;</span>, geneset <span class="op">%in%</span><span class="st"> </span>repeated_sig_H_genesets_vec2) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb48-6" data-line-number="6"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">donor_coeff =</span> <span class="kw">paste</span>(donor, coeff, <span class="dt">sep =</span> <span class="st">&quot;.&quot;</span>),</a>
<a class="sourceLine" id="cb48-7" data-line-number="7">                <span class="dt">sign =</span> <span class="kw">ifelse</span>(Direction <span class="op">==</span><span class="st"> &quot;Down&quot;</span>, <span class="dv">-1</span>, <span class="dv">1</span>),</a>
<a class="sourceLine" id="cb48-8" data-line-number="8">                <span class="dt">signed_P =</span> sign <span class="op">*</span><span class="st"> </span><span class="op">-</span><span class="kw">log10</span>(PValue)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb48-9" data-line-number="9"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(geneset, donor_coeff, signed_P) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb48-10" data-line-number="10"><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">spread</span>(<span class="dt">key =</span> donor_coeff, <span class="dt">value =</span> signed_P)</a>
<a class="sourceLine" id="cb48-11" data-line-number="11"></a>
<a class="sourceLine" id="cb48-12" data-line-number="12">mat_H_pvals &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(df_H_pvals[, <span class="dv">-1</span>])</a>
<a class="sourceLine" id="cb48-13" data-line-number="13"><span class="kw">rownames</span>(mat_H_pvals) &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot; &quot;</span>, <span class="kw">gsub</span>(<span class="st">&quot;HALLMARK_&quot;</span>, <span class="st">&quot;&quot;</span>, df_H_pvals[[<span class="dv">1</span>]]))</a>
<a class="sourceLine" id="cb48-14" data-line-number="14">cor_H_pvals &lt;-<span class="st"> </span><span class="kw">cor</span>(<span class="kw">t</span>(mat_H_pvals), <span class="dt">method =</span> <span class="st">&quot;spearman&quot;</span>)</a>
<a class="sourceLine" id="cb48-15" data-line-number="15">p.mat &lt;-<span class="st"> </span><span class="kw">cor_pmat</span>(<span class="kw">t</span>(mat_H_pvals))</a>
<a class="sourceLine" id="cb48-16" data-line-number="16"><span class="kw">ggcorrplot</span>(cor_H_pvals, <span class="dt">hc.order =</span> <span class="ot">TRUE</span>, <span class="dt">p.mat =</span> p.mat, <span class="dt">insig =</span> <span class="st">&quot;blank&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb48-17" data-line-number="17"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">panel.grid.major =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb48-18" data-line-number="18">        <span class="dt">panel.grid.minor =</span> <span class="kw">element_blank</span>())</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/corr-maps-1.png" width="1248" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb49-1" data-line-number="1">hclust_cor &lt;-<span class="st"> </span><span class="kw">hclust</span>(<span class="kw">as.dist</span>(<span class="dv">1</span> <span class="op">-</span><span class="st"> </span>cor_H_pvals))</a>
<a class="sourceLine" id="cb49-2" data-line-number="2">corrplot1 &lt;-<span class="st"> </span><span class="kw">ggcorrplot</span>(cor_H_pvals[hclust_cor<span class="op">$</span>order, hclust_cor<span class="op">$</span>order], </a>
<a class="sourceLine" id="cb49-3" data-line-number="3">           <span class="dt">p.mat =</span> p.mat[hclust_cor<span class="op">$</span>order, hclust_cor<span class="op">$</span>order], <span class="dt">insig =</span> <span class="st">&quot;blank&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb49-4" data-line-number="4"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">panel.grid.major =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb49-5" data-line-number="5">        <span class="dt">panel.grid.minor =</span> <span class="kw">element_blank</span>())</a>
<a class="sourceLine" id="cb49-6" data-line-number="6"></a>
<a class="sourceLine" id="cb49-7" data-line-number="7">mat_H_gene_overlap &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="dt">nrow =</span> <span class="kw">nrow</span>(cor_H_pvals), <span class="dt">ncol =</span> <span class="kw">ncol</span>(cor_H_pvals),</a>
<a class="sourceLine" id="cb49-8" data-line-number="8">                             <span class="dt">dimnames =</span> <span class="kw">dimnames</span>(cor_H_pvals))</a>
<a class="sourceLine" id="cb49-9" data-line-number="9"><span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">seq_along</span>(repeated_sig_H_genesets_vec2)) {</a>
<a class="sourceLine" id="cb49-10" data-line-number="10">  <span class="cf">for</span> (j <span class="cf">in</span> <span class="kw">seq_along</span>(repeated_sig_H_genesets_vec2)) {</a>
<a class="sourceLine" id="cb49-11" data-line-number="11">    gs1 &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;HALLMARK_&quot;</span>, <span class="kw">gsub</span>(<span class="st">&quot; &quot;</span>, <span class="st">&quot;_&quot;</span>, <span class="kw">rownames</span>(mat_H_gene_overlap)[i]))</a>
<a class="sourceLine" id="cb49-12" data-line-number="12">    gs2 &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;HALLMARK_&quot;</span>, <span class="kw">gsub</span>(<span class="st">&quot; &quot;</span>, <span class="st">&quot;_&quot;</span>, <span class="kw">rownames</span>(mat_H_gene_overlap)[j]))</a>
<a class="sourceLine" id="cb49-13" data-line-number="13">    mat_H_gene_overlap[i, j] &lt;-<span class="st"> </span><span class="kw">mean</span>(Hs.H[[gs1]] <span class="op">%in%</span><span class="st"> </span>Hs.H[[gs2]])</a>
<a class="sourceLine" id="cb49-14" data-line-number="14">  }</a>
<a class="sourceLine" id="cb49-15" data-line-number="15">}</a>
<a class="sourceLine" id="cb49-16" data-line-number="16"></a>
<a class="sourceLine" id="cb49-17" data-line-number="17">corrplot2 &lt;-<span class="st"> </span><span class="kw">ggcorrplot</span>(mat_H_gene_overlap[hclust_cor<span class="op">$</span>order, hclust_cor<span class="op">$</span>order]) <span class="op">+</span></a>
<a class="sourceLine" id="cb49-18" data-line-number="18"><span class="st">  </span><span class="kw">scale_fill_gradient</span>(<span class="dt">name =</span> <span class="st">&quot;Gene set</span><span class="ch">\n</span><span class="st">overlap&quot;</span>, <span class="dt">low =</span> <span class="st">&quot;white&quot;</span>, <span class="dt">high =</span> <span class="st">&quot;black&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb49-19" data-line-number="19"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">panel.grid.major =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb49-20" data-line-number="20">        <span class="dt">panel.grid.minor =</span> <span class="kw">element_blank</span>())</a>
<a class="sourceLine" id="cb49-21" data-line-number="21">corrplot2</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/corr-maps-2.png" width="1248" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb50-1" data-line-number="1">corrplot3 &lt;-<span class="st"> </span>corrplot2 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.text.y =</span> <span class="kw">element_blank</span>())</a></code></pre></div>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb51-1" data-line-number="1"><span class="kw">plot_grid</span>(corrplot1 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">plot.margin =</span> <span class="kw">unit</span>(<span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>), <span class="st">&quot;cm&quot;</span>)), </a>
<a class="sourceLine" id="cb51-2" data-line-number="2">          corrplot3 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">plot.margin =</span> <span class="kw">unit</span>(<span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>), <span class="st">&quot;cm&quot;</span>)), </a>
<a class="sourceLine" id="cb51-3" data-line-number="3">          <span class="dt">align =</span> <span class="st">&quot;h&quot;</span>, <span class="dt">axis =</span> <span class="st">&quot;b&quot;</span>, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="fl">0.58</span>, <span class="fl">0.42</span>))</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/corr-plot-combined-1.png" width="1920" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb52-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/top_genesets_H_corrplots.png&quot;</span>, <span class="dt">height =</span> <span class="dv">9</span>, <span class="dt">width =</span> <span class="dv">20</span>)</a>
<a class="sourceLine" id="cb52-2" data-line-number="2"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/top_genesets_H_corrplots.pdf&quot;</span>, <span class="dt">height =</span> <span class="dv">9</span>, <span class="dt">width =</span> <span class="dv">20</span>)</a>
<a class="sourceLine" id="cb52-3" data-line-number="3"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/top_genesets_H_corrplots.svg&quot;</span>, <span class="dt">height =</span> <span class="dv">9</span>, <span class="dt">width =</span> <span class="dv">20</span>)</a></code></pre></div>
<p>Plot gene set correlation with the number of donors in which each gene set is significant.</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb53-1" data-line-number="1">pp_nsig &lt;-<span class="st"> </span>df_camera_sig_unst <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(collection <span class="op">==</span><span class="st"> &quot;H&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb53-2" data-line-number="2"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(stat <span class="op">==</span><span class="st"> &quot;logFC&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb53-3" data-line-number="3"><span class="st">    </span><span class="kw">group_by</span>(geneset) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb53-4" data-line-number="4"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, geneset)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(id, <span class="dt">.keep_all =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb53-5" data-line-number="5"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">n_donors =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(geneset, n_donors) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb53-6" data-line-number="6"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">geneset =</span> <span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot; &quot;</span>, <span class="kw">gsub</span>(<span class="st">&quot;HALLMARK_&quot;</span>, <span class="st">&quot;&quot;</span>, geneset))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb53-7" data-line-number="7"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">geneset =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb53-8" data-line-number="8">      geneset, <span class="dt">levels =</span> <span class="kw">rownames</span>(mat_H_gene_overlap)[hclust_cor<span class="op">$</span>order])) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb53-9" data-line-number="9"><span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">y =</span> n_donors, <span class="dt">x =</span> geneset)) <span class="op">+</span></a>
<a class="sourceLine" id="cb53-10" data-line-number="10"><span class="st">    </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="dv">0</span>, <span class="dt">colour =</span> <span class="st">&quot;gray50&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb53-11" data-line-number="11"><span class="st">    </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">xend =</span> geneset, <span class="dt">yend =</span> <span class="dv">0</span>),</a>
<a class="sourceLine" id="cb53-12" data-line-number="12">                 <span class="dt">colour =</span> <span class="st">&quot;gray50&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb53-13" data-line-number="13"><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">4</span>, <span class="dt">colour =</span> <span class="st">&quot;gray30&quot;</span>, <span class="dt">alpha =</span> <span class="dv">1</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb53-14" data-line-number="14"><span class="st">    </span>ggthemes<span class="op">::</span><span class="kw">scale_colour_tableau</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb53-15" data-line-number="15"><span class="st">    </span><span class="kw">coord_flip</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb53-16" data-line-number="16"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Gene set&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Number of lines</span><span class="ch">\n</span><span class="st">significant&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb53-17" data-line-number="17"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.title.y =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb53-18" data-line-number="18">          <span class="dt">axis.text.y =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb53-19" data-line-number="19">          <span class="dt">axis.ticks.y =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb53-20" data-line-number="20">          <span class="dt">axis.line.y =</span> <span class="kw">element_blank</span>())</a>
<a class="sourceLine" id="cb53-21" data-line-number="21"></a>
<a class="sourceLine" id="cb53-22" data-line-number="22"><span class="kw">ggdraw</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb53-23" data-line-number="23"><span class="st">  </span><span class="kw">draw_plot</span>(corrplot1 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;top&quot;</span>), </a>
<a class="sourceLine" id="cb53-24" data-line-number="24">            <span class="dt">x =</span> <span class="dv">0</span>,  <span class="dt">y =</span> <span class="dv">0</span>, <span class="dt">width =</span> <span class="fl">0.8</span>, <span class="dt">scale =</span> <span class="dv">1</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb53-25" data-line-number="25"><span class="st">  </span><span class="kw">draw_plot</span>(pp_nsig, </a>
<a class="sourceLine" id="cb53-26" data-line-number="26">            <span class="dt">x =</span> <span class="fl">0.685</span>,  <span class="dt">y =</span> <span class="fl">0.25</span>, <span class="dt">width =</span> <span class="fl">0.25</span>, <span class="dt">height =</span> <span class="fl">0.6445</span>)</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/unnamed-chunk-5-1.png" width="1152" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb54-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/top_genesets_H_corrplot_with_nsig_donor.png&quot;</span>, </a>
<a class="sourceLine" id="cb54-2" data-line-number="2">       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="dv">12</span>)</a>
<a class="sourceLine" id="cb54-3" data-line-number="3"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/top_genesets_H_corrplot_with_nsig_donor.pdf&quot;</span>, </a>
<a class="sourceLine" id="cb54-4" data-line-number="4">       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="dv">12</span>)</a></code></pre></div>
</div>
<div id="number-of-de-genes-pathways-and-clones-vs-number-of-cells" class="section level2">
<h2>Number of DE genes, pathways and clones vs number of cells</h2>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb55-1" data-line-number="1">df_donor_info &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;data/donor_info_070818.txt&quot;</span>)</a>
<a class="sourceLine" id="cb55-2" data-line-number="2">df_donor_info &lt;-<span class="st"> </span><span class="kw">as_tibble</span>(df_donor_info)</a>
<a class="sourceLine" id="cb55-3" data-line-number="3">df_donor_info<span class="op">$</span>donor &lt;-<span class="st"> </span>df_donor_info<span class="op">$</span>donor_short</a>
<a class="sourceLine" id="cb55-4" data-line-number="4"></a>
<a class="sourceLine" id="cb55-5" data-line-number="5">df_ncells_de &lt;-<span class="st"> </span>assignments <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(assigned <span class="op">!=</span><span class="st"> &quot;unassigned&quot;</span>, </a>
<a class="sourceLine" id="cb55-6" data-line-number="6">                              donor_short_id <span class="op">%in%</span><span class="st"> </span><span class="kw">names</span>(de_res<span class="op">$</span>qlf_list)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb55-7" data-line-number="7"><span class="st">    </span><span class="kw">group_by</span>(donor_short_id) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb55-8" data-line-number="8"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">n_cells =</span> <span class="kw">n</span>())</a>
<a class="sourceLine" id="cb55-9" data-line-number="9"><span class="kw">colnames</span>(df_ncells_de)[<span class="dv">1</span>] &lt;-<span class="st"> &quot;donor&quot;</span></a>
<a class="sourceLine" id="cb55-10" data-line-number="10"></a>
<a class="sourceLine" id="cb55-11" data-line-number="11">df_prop_assigned &lt;-<span class="st"> </span>assignments <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb55-12" data-line-number="12"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(donor_short_id <span class="op">%in%</span><span class="st"> </span><span class="kw">names</span>(de_res<span class="op">$</span>qlf_list)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb55-13" data-line-number="13"><span class="st">    </span><span class="kw">group_by</span>(donor_short_id) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb55-14" data-line-number="14"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">prop_assigned =</span> <span class="kw">mean</span>(assigned <span class="op">!=</span><span class="st"> &quot;unassigned&quot;</span>))</a>
<a class="sourceLine" id="cb55-15" data-line-number="15"><span class="kw">colnames</span>(df_prop_assigned)[<span class="dv">1</span>] &lt;-<span class="st"> &quot;donor&quot;</span></a>
<a class="sourceLine" id="cb55-16" data-line-number="16"></a>
<a class="sourceLine" id="cb55-17" data-line-number="17">df_nvars_by_cat &lt;-<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_tsv</span>(<span class="st">&quot;output/nvars_by_category_by_donor.tsv&quot;</span>)</a>
<a class="sourceLine" id="cb55-18" data-line-number="18">df_nvars_by_cat_wd &lt;-<span class="st"> </span>tidyr<span class="op">::</span><span class="kw">spread</span>(</a>
<a class="sourceLine" id="cb55-19" data-line-number="19">  df_nvars_by_cat[, <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>], consequence, n_vars_all_genes)</a>
<a class="sourceLine" id="cb55-20" data-line-number="20">df_nvars_by_cat_wd &lt;-<span class="st"> </span><span class="kw">left_join</span>(</a>
<a class="sourceLine" id="cb55-21" data-line-number="21">  dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="kw">group_by</span>(df_nvars_by_cat, donor), <span class="dt">nvars_all =</span> <span class="kw">sum</span>(n_vars_all_genes)),</a>
<a class="sourceLine" id="cb55-22" data-line-number="22">  df_nvars_by_cat_wd</a>
<a class="sourceLine" id="cb55-23" data-line-number="23">)</a>
<a class="sourceLine" id="cb55-24" data-line-number="24">df_nvars_by_cat_wd &lt;-<span class="st"> </span>df_nvars_by_cat <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb55-25" data-line-number="25"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(consequence <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;missense&quot;</span>, <span class="st">&quot;splicing&quot;</span>, <span class="st">&quot;nonsense&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb55-26" data-line-number="26"><span class="st">  </span><span class="kw">group_by</span>(donor) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb55-27" data-line-number="27"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">nvars_misnonspli =</span> <span class="kw">sum</span>(n_vars_all_genes)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb55-28" data-line-number="28"><span class="st">  </span><span class="kw">left_join</span>(., df_nvars_by_cat_wd)</a>
<a class="sourceLine" id="cb55-29" data-line-number="29"></a>
<a class="sourceLine" id="cb55-30" data-line-number="30">df_donor_info &lt;-<span class="st"> </span><span class="kw">left_join</span>(df_ncells_de, df_donor_info)</a>
<a class="sourceLine" id="cb55-31" data-line-number="31">df_donor_info &lt;-<span class="st"> </span><span class="kw">left_join</span>(df_prop_assigned, df_donor_info)</a>
<a class="sourceLine" id="cb55-32" data-line-number="32">df_donor_info &lt;-<span class="st"> </span><span class="kw">left_join</span>(df_donor_n_de, df_donor_info)</a>
<a class="sourceLine" id="cb55-33" data-line-number="33">df_donor_info<span class="op">$</span>n_de_genes &lt;-<span class="st"> </span>df_donor_info<span class="op">$</span>count</a>
<a class="sourceLine" id="cb55-34" data-line-number="34">df_donor_info &lt;-<span class="st"> </span><span class="kw">left_join</span>(df_donor_info, df_nvars_by_cat_wd)</a>
<a class="sourceLine" id="cb55-35" data-line-number="35"></a>
<a class="sourceLine" id="cb55-36" data-line-number="36">nbglm_nde &lt;-<span class="st"> </span>MASS<span class="op">::</span><span class="kw">glm.nb</span>(n_de_genes <span class="op">~</span><span class="st"> </span>n_cells, <span class="dt">data =</span> df_donor_info)</a>
<a class="sourceLine" id="cb55-37" data-line-number="37">df_nbglm_nde &lt;-<span class="st"> </span>broom<span class="op">::</span><span class="kw">augment</span>(nbglm_nde) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb55-38" data-line-number="38"><span class="st">  </span><span class="kw">left_join</span>(df_donor_info)</a>
<a class="sourceLine" id="cb55-39" data-line-number="39"></a>
<a class="sourceLine" id="cb55-40" data-line-number="40"><span class="co">## N DE genes vs Number of cells</span></a>
<a class="sourceLine" id="cb55-41" data-line-number="41">df_nbglm_nde <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb55-42" data-line-number="42"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> n_cells, <span class="dt">y =</span> n_de_genes)) <span class="op">+</span></a>
<a class="sourceLine" id="cb55-43" data-line-number="43"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb55-44" data-line-number="44"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>, <span class="dt">fill =</span> <span class="st">&quot;gray50&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb55-45" data-line-number="45"><span class="st">  </span><span class="kw">geom_label_repel</span>(<span class="kw">aes</span>(<span class="dt">label =</span> donor)) <span class="op">+</span></a>
<a class="sourceLine" id="cb55-46" data-line-number="46"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb55-47" data-line-number="47"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Number of cells&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb55-48" data-line-number="48"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>))</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/nde-vs-ncels-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb56-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_de_genes_vs_n_cells_nogrouping.png&quot;</span>, </a>
<a class="sourceLine" id="cb56-2" data-line-number="2">       <span class="dt">height =</span> <span class="fl">6.5</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)</a>
<a class="sourceLine" id="cb56-3" data-line-number="3"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_de_genes_vs_n_cells_nogrouping.pdf&quot;</span>, </a>
<a class="sourceLine" id="cb56-4" data-line-number="4">       <span class="dt">height =</span> <span class="fl">6.5</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)</a>
<a class="sourceLine" id="cb56-5" data-line-number="5"></a>
<a class="sourceLine" id="cb56-6" data-line-number="6">p1 &lt;-<span class="st"> </span>df_nbglm_nde <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb56-7" data-line-number="7"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">num_clones =</span> <span class="kw">paste</span>(num_clones_total, <span class="st">&quot;clones&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb56-8" data-line-number="8"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> n_cells, <span class="dt">y =</span> n_de_genes)) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-9" data-line-number="9"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-10" data-line-number="10"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>, <span class="dt">fill =</span> <span class="st">&quot;gray50&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-11" data-line-number="11"><span class="st">  </span><span class="kw">geom_label_repel</span>(<span class="kw">aes</span>(<span class="dt">label =</span> donor)) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-12" data-line-number="12"><span class="st">  </span><span class="kw">facet_wrap</span>(<span class="op">~</span>num_clones) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-13" data-line-number="13"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-14" data-line-number="14"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Number of cells&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-15" data-line-number="15"><span class="st">  </span><span class="kw">coord_cartesian</span>(<span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1200</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-16" data-line-number="16"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>))</a>
<a class="sourceLine" id="cb56-17" data-line-number="17">p1</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/nde-vs-ncels-2.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb57-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_de_genes_vs_n_cells_facet_by_num_clones.png&quot;</span>, </a>
<a class="sourceLine" id="cb57-2" data-line-number="2">       <span class="dt">height =</span> <span class="fl">6.5</span>, <span class="dt">width =</span> <span class="fl">9.5</span>)</a>
<a class="sourceLine" id="cb57-3" data-line-number="3"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_de_genes_vs_n_cells_facet_by_num_clones.pdf&quot;</span>, </a>
<a class="sourceLine" id="cb57-4" data-line-number="4">       <span class="dt">height =</span> <span class="fl">6.5</span>, <span class="dt">width =</span> <span class="fl">9.5</span>)</a>
<a class="sourceLine" id="cb57-5" data-line-number="5"></a>
<a class="sourceLine" id="cb57-6" data-line-number="6"></a>
<a class="sourceLine" id="cb57-7" data-line-number="7"><span class="co">## Number of significant genesets vs number of cells</span></a>
<a class="sourceLine" id="cb57-8" data-line-number="8">df_nbglm_nde &lt;-<span class="st"> </span>df_camera_sig_unst <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(collection <span class="op">==</span><span class="st"> &quot;H&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb57-9" data-line-number="9"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(stat <span class="op">==</span><span class="st"> &quot;logFC&quot;</span>, FDR <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb57-10" data-line-number="10"><span class="st">  </span><span class="kw">group_by</span>(donor) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb57-11" data-line-number="11"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, geneset)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb57-12" data-line-number="12"><span class="st">  </span><span class="kw">distinct</span>(id, <span class="dt">.keep_all =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb57-13" data-line-number="13"><span class="st">  </span><span class="kw">summarise</span>(<span class="dt">n_sig_genesets =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb57-14" data-line-number="14"><span class="kw">left_join</span>(df_nbglm_nde, .)</a>
<a class="sourceLine" id="cb57-15" data-line-number="15"></a>
<a class="sourceLine" id="cb57-16" data-line-number="16">df_nbglm_nde[[<span class="st">&quot;n_sig_genesets&quot;</span>]][<span class="kw">is.na</span>(df_nbglm_nde[[<span class="st">&quot;n_sig_genesets&quot;</span>]])] &lt;-<span class="st"> </span><span class="dv">0</span></a>
<a class="sourceLine" id="cb57-17" data-line-number="17"></a>
<a class="sourceLine" id="cb57-18" data-line-number="18"><span class="kw">ggplot</span>(df_nbglm_nde, <span class="kw">aes</span>(<span class="dt">x =</span> n_cells, <span class="dt">y =</span> n_sig_genesets)) <span class="op">+</span></a>
<a class="sourceLine" id="cb57-19" data-line-number="19"><span class="st">  </span><span class="co">#geom_smooth(aes(group = 1), colour = &quot;firebrick&quot;, method = &quot;lm&quot;, level = 0.9) +</span></a>
<a class="sourceLine" id="cb57-20" data-line-number="20"><span class="st">  </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="dv">0</span>, <span class="dt">linetype =</span> <span class="dv">2</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb57-21" data-line-number="21"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>, <span class="dt">fill =</span> <span class="st">&quot;gray50&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb57-22" data-line-number="22"><span class="st">  </span><span class="kw">geom_label_repel</span>(<span class="kw">aes</span>(<span class="dt">label =</span> donor)) <span class="op">+</span></a>
<a class="sourceLine" id="cb57-23" data-line-number="23"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of significant Hallmark genesets&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb57-24" data-line-number="24"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Number of cells&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb57-25" data-line-number="25"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>))</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/nde-vs-ncels-3.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb58-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_sig_genesets_vs_n_cells_nogrouping.png&quot;</span>, </a>
<a class="sourceLine" id="cb58-2" data-line-number="2">       <span class="dt">height =</span> <span class="fl">6.5</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)</a>
<a class="sourceLine" id="cb58-3" data-line-number="3"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_sig_genesets_vs_n_cells_nogrouping.pdf&quot;</span>, </a>
<a class="sourceLine" id="cb58-4" data-line-number="4">       <span class="dt">height =</span> <span class="fl">6.5</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)</a>
<a class="sourceLine" id="cb58-5" data-line-number="5"></a>
<a class="sourceLine" id="cb58-6" data-line-number="6">p2 &lt;-<span class="st"> </span>df_nbglm_nde <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb58-7" data-line-number="7"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">num_clones =</span> <span class="kw">paste</span>(num_clones_total, <span class="st">&quot;clones&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb58-8" data-line-number="8"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> n_cells, <span class="dt">y =</span> n_sig_genesets)) <span class="op">+</span></a>
<a class="sourceLine" id="cb58-9" data-line-number="9"><span class="st">  </span><span class="co">#geom_smooth(aes(group = 1), colour = &quot;firebrick&quot;, method = &quot;lm&quot;, level = 0.9) +</span></a>
<a class="sourceLine" id="cb58-10" data-line-number="10"><span class="st">  </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="dv">0</span>, <span class="dt">linetype =</span> <span class="dv">2</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb58-11" data-line-number="11"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>, <span class="dt">fill =</span> <span class="st">&quot;gray50&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb58-12" data-line-number="12"><span class="st">  </span><span class="kw">geom_label_repel</span>(<span class="kw">aes</span>(<span class="dt">label =</span> donor)) <span class="op">+</span></a>
<a class="sourceLine" id="cb58-13" data-line-number="13"><span class="st">  </span><span class="kw">facet_wrap</span>(<span class="op">~</span>num_clones) <span class="op">+</span></a>
<a class="sourceLine" id="cb58-14" data-line-number="14"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of significant Hallmark genesets&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb58-15" data-line-number="15"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Number of cells&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb58-16" data-line-number="16"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>))</a>
<a class="sourceLine" id="cb58-17" data-line-number="17">p2</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/nde-vs-ncels-4.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb59-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_sig_genesets_vs_n_cells_facet_by_num_clones.png&quot;</span>, </a>
<a class="sourceLine" id="cb59-2" data-line-number="2">       <span class="dt">height =</span> <span class="fl">6.5</span>, <span class="dt">width =</span> <span class="fl">9.5</span>)</a>
<a class="sourceLine" id="cb59-3" data-line-number="3"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_sig_genesets_vs_n_cells_facet_by_num_clones.pdf&quot;</span>, </a>
<a class="sourceLine" id="cb59-4" data-line-number="4">       <span class="dt">height =</span> <span class="fl">6.5</span>, <span class="dt">width =</span> <span class="fl">9.5</span>)</a>
<a class="sourceLine" id="cb59-5" data-line-number="5"></a>
<a class="sourceLine" id="cb59-6" data-line-number="6"></a>
<a class="sourceLine" id="cb59-7" data-line-number="7">pp &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(p1, p2, <span class="dt">labels =</span> <span class="st">&quot;auto&quot;</span>, <span class="dt">nrow =</span> <span class="dv">2</span>)</a>
<a class="sourceLine" id="cb59-8" data-line-number="8"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_sig_genesets_N_de_genes_vs_n_cells_2panel_facet_by_num_clones.png&quot;</span>, </a>
<a class="sourceLine" id="cb59-9" data-line-number="9">       <span class="dt">height =</span> <span class="fl">10.5</span>, <span class="dt">width =</span> <span class="fl">9.5</span>, <span class="dt">plot =</span> pp)</a>
<a class="sourceLine" id="cb59-10" data-line-number="10"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_sig_genesets_N_de_genes_vs_n_cells_2panel_facet_by_num_clones.pdf&quot;</span>, </a>
<a class="sourceLine" id="cb59-11" data-line-number="11">       <span class="dt">height =</span> <span class="fl">10.5</span>, <span class="dt">width =</span> <span class="fl">9.5</span>, <span class="dt">plot =</span> pp)</a>
<a class="sourceLine" id="cb59-12" data-line-number="12"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_sig_genesets_N_de_genes_vs_n_cells_2panel_facet_by_num_clones.svg&quot;</span>, </a>
<a class="sourceLine" id="cb59-13" data-line-number="13">       <span class="dt">height =</span> <span class="fl">10.5</span>, <span class="dt">width =</span> <span class="fl">9.5</span>, <span class="dt">plot =</span> pp)</a>
<a class="sourceLine" id="cb59-14" data-line-number="14"></a>
<a class="sourceLine" id="cb59-15" data-line-number="15"></a>
<a class="sourceLine" id="cb59-16" data-line-number="16"><span class="co">## Number of clones vs number of cells</span></a>
<a class="sourceLine" id="cb59-17" data-line-number="17"><span class="kw">ggplot</span>(df_nbglm_nde, <span class="kw">aes</span>(<span class="dt">x =</span> n_cells, <span class="dt">y =</span> num_clones_total)) <span class="op">+</span></a>
<a class="sourceLine" id="cb59-18" data-line-number="18"><span class="st">  </span><span class="co">#geom_smooth(aes(group = 1), colour = &quot;firebrick&quot;, method = &quot;lm&quot;, level = 0.9) +</span></a>
<a class="sourceLine" id="cb59-19" data-line-number="19"><span class="st">  </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="dv">0</span>, <span class="dt">linetype =</span> <span class="dv">2</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb59-20" data-line-number="20"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>, <span class="dt">fill =</span> <span class="st">&quot;gray50&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb59-21" data-line-number="21"><span class="st">  </span><span class="kw">geom_label_repel</span>(<span class="kw">aes</span>(<span class="dt">label =</span> donor)) <span class="op">+</span></a>
<a class="sourceLine" id="cb59-22" data-line-number="22"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of inferred clones (total)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb59-23" data-line-number="23"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Number of cells&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb59-24" data-line-number="24"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>))</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/nde-vs-ncels-5.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb60-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_clones_total_vs_n_cells_nogrouping.png&quot;</span>, </a>
<a class="sourceLine" id="cb60-2" data-line-number="2">       <span class="dt">height =</span> <span class="fl">6.5</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)</a>
<a class="sourceLine" id="cb60-3" data-line-number="3"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_clones_total_vs_n_cells_nogrouping.pdf&quot;</span>, </a>
<a class="sourceLine" id="cb60-4" data-line-number="4">       <span class="dt">height =</span> <span class="fl">6.5</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)</a>
<a class="sourceLine" id="cb60-5" data-line-number="5"></a>
<a class="sourceLine" id="cb60-6" data-line-number="6"><span class="kw">ggplot</span>(df_nbglm_nde, <span class="kw">aes</span>(<span class="dt">x =</span> n_cells, <span class="dt">y =</span> num_clones_with_cells)) <span class="op">+</span></a>
<a class="sourceLine" id="cb60-7" data-line-number="7"><span class="st">  </span><span class="co">#geom_smooth(aes(group = 1), colour = &quot;firebrick&quot;, method = &quot;lm&quot;, level = 0.9) +</span></a>
<a class="sourceLine" id="cb60-8" data-line-number="8"><span class="st">  </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="dv">0</span>, <span class="dt">linetype =</span> <span class="dv">2</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb60-9" data-line-number="9"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>, <span class="dt">fill =</span> <span class="st">&quot;gray50&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb60-10" data-line-number="10"><span class="st">  </span><span class="kw">geom_label_repel</span>(<span class="kw">aes</span>(<span class="dt">label =</span> donor)) <span class="op">+</span></a>
<a class="sourceLine" id="cb60-11" data-line-number="11"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of inferred clones (with cells assigned)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb60-12" data-line-number="12"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Number of cells&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb60-13" data-line-number="13"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>))</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/nde-vs-ncels-6.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb61-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_clones_with_cells_vs_n_cells_nogrouping.png&quot;</span>, </a>
<a class="sourceLine" id="cb61-2" data-line-number="2">       <span class="dt">height =</span> <span class="fl">6.5</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)</a>
<a class="sourceLine" id="cb61-3" data-line-number="3"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_clones_with_cells_vs_n_cells_nogrouping.pdf&quot;</span>, </a>
<a class="sourceLine" id="cb61-4" data-line-number="4">       <span class="dt">height =</span> <span class="fl">6.5</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)</a>
<a class="sourceLine" id="cb61-5" data-line-number="5"></a>
<a class="sourceLine" id="cb61-6" data-line-number="6"><span class="co">## Number of DE genes vs number of clones</span></a>
<a class="sourceLine" id="cb61-7" data-line-number="7">df_nbglm_nde <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb61-8" data-line-number="8"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> num_clones_total, <span class="dt">y =</span> n_de_genes)) <span class="op">+</span></a>
<a class="sourceLine" id="cb61-9" data-line-number="9"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb61-10" data-line-number="10"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>, <span class="dt">fill =</span> <span class="st">&quot;gray50&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb61-11" data-line-number="11"><span class="st">  </span><span class="kw">geom_label_repel</span>(<span class="kw">aes</span>(<span class="dt">label =</span> donor)) <span class="op">+</span></a>
<a class="sourceLine" id="cb61-12" data-line-number="12"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb61-13" data-line-number="13"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Number of clones&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb61-14" data-line-number="14"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>))</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/nde-vs-ncels-7.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb62-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_de_genes_vs_n_clones_total.png&quot;</span>, </a>
<a class="sourceLine" id="cb62-2" data-line-number="2">       <span class="dt">height =</span> <span class="fl">6.5</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)</a>
<a class="sourceLine" id="cb62-3" data-line-number="3"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_de_genes_vs_n_clones_total.pdf&quot;</span>, </a>
<a class="sourceLine" id="cb62-4" data-line-number="4">       <span class="dt">height =</span> <span class="fl">6.5</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)</a>
<a class="sourceLine" id="cb62-5" data-line-number="5"></a>
<a class="sourceLine" id="cb62-6" data-line-number="6"><span class="co">## Number of sig genesets vs number of clones</span></a>
<a class="sourceLine" id="cb62-7" data-line-number="7">df_nbglm_nde <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb62-8" data-line-number="8"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> num_clones_total, <span class="dt">y =</span> n_sig_genesets)) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-9" data-line-number="9"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-10" data-line-number="10"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>, <span class="dt">fill =</span> <span class="st">&quot;gray50&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-11" data-line-number="11"><span class="st">  </span><span class="kw">geom_label_repel</span>(<span class="kw">aes</span>(<span class="dt">label =</span> donor)) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-12" data-line-number="12"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of significant Hallmark genesets&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-13" data-line-number="13"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Number of clones&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-14" data-line-number="14"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>))</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/nde-vs-ncels-8.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb63-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_sig_genesets_vs_n_clones_total.png&quot;</span>, </a>
<a class="sourceLine" id="cb63-2" data-line-number="2">       <span class="dt">height =</span> <span class="fl">6.5</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)</a>
<a class="sourceLine" id="cb63-3" data-line-number="3"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_sig_genesets_vs_n_clones_total.pdf&quot;</span>, </a>
<a class="sourceLine" id="cb63-4" data-line-number="4">       <span class="dt">height =</span> <span class="fl">6.5</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)</a></code></pre></div>
</div>
<div id="linking-de-to-selection" class="section level2">
<h2>Linking DE to selection</h2>
<div id="selection-defined-using-williams-et-al-2018-models" class="section level3">
<h3>Selection defined using Williams et al 2018 models</h3>
<p>The Williams et al 2018 model provides as output a “probability of selection”. We define lines with a probability of selection greater than 0.7 as “selected”, those with probability of selection less than 0.3 as “neutral” and the remaining lines as “undetermined”.</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb64-1" data-line-number="1">df_williams2018 &lt;-<span class="st"> </span><span class="kw">read_csv</span>(<span class="st">&quot;data/selection/p1-selection.csv&quot;</span>)</a>
<a class="sourceLine" id="cb64-2" data-line-number="2"><span class="kw">colnames</span>(df_williams2018) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;donor&quot;</span>, <span class="st">&quot;prob_select_williams2018&quot;</span>, </a>
<a class="sourceLine" id="cb64-3" data-line-number="3">                               <span class="st">&quot;selection_williams2018&quot;</span>)</a>
<a class="sourceLine" id="cb64-4" data-line-number="4">df_nbglm_nde &lt;-<span class="st"> </span><span class="kw">left_join</span>(df_nbglm_nde, df_williams2018)</a>
<a class="sourceLine" id="cb64-5" data-line-number="5">df_williams2018</a></code></pre></div>
<pre><code># A tibble: 32 x 3
   donor prob_select_williams2018 selection_williams2018
   &lt;chr&gt;                    &lt;dbl&gt; &lt;chr&gt;                 
 1 euts                     1     selected              
 2 fawm                     1     selected              
 3 feec                     1.000 selected              
 4 fikt                     1     selected              
 5 garx                     1     selected              
 6 gesg                     0.357 undetermined          
 7 heja                     0.470 undetermined          
 8 hipn                     0.318 undetermined          
 9 ieki                     0.994 selected              
10 joxm                     1     selected              
# … with 22 more rows</code></pre>
<p>We get confident results from the Williams et al 2018 model for lines with at least 100 somatic variants:</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb66-1" data-line-number="1">df_high_conf_selection &lt;-<span class="st"> </span>df_nbglm_nde <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb66-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(num_mutations <span class="op">&gt;=</span><span class="st"> </span><span class="dv">100</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb66-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(donor, num_mutations, n_cells, prob_select_williams2018, </a>
<a class="sourceLine" id="cb66-4" data-line-number="4">                selection_williams2018, n_de_genes, n_sig_genesets) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb66-5" data-line-number="5"><span class="st">  </span><span class="kw">print</span>(<span class="dt">n =</span> <span class="ot">Inf</span>)</a></code></pre></div>
<pre><code># A tibble: 20 x 7
   donor num_mutations n_cells prob_select_wil… selection_willi… n_de_genes
   &lt;chr&gt;         &lt;int&gt;   &lt;int&gt;            &lt;dbl&gt; &lt;chr&gt;                 &lt;int&gt;
 1 euts            292      79            1     selected                738
 2 fawm            101      49            1     selected                370
 3 feec            170      67            1.000 selected                756
 4 fikt            142      34            1     selected                599
 5 garx            592      70            1     selected                902
 6 gesg            157     100            0.357 undetermined           1094
 7 heja            192      50            0.470 undetermined            336
 8 joxm            612      78            1     selected               1133
 9 laey            278      55            0.665 undetermined            419
10 oilg            211      89            0.983 selected                396
11 pipw            233     107            1     selected                372
12 puie            117      41            1     selected                163
13 qolg            120      36            1     selected                131
14 qonc            131      40            0.414 undetermined            832
15 sehl            178      29            0.251 neutral                   2
16 ualf            325      89            0.871 selected                835
17 vass            412      37            1.000 selected                282
18 vuna            135      71            0.314 undetermined            685
19 wahn            496      78            0.984 selected               1119
20 xugn            124      35            0.171 neutral                  43
# … with 1 more variable: n_sig_genesets &lt;dbl&gt;</code></pre>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb68-1" data-line-number="1">dplyr<span class="op">::</span><span class="kw">filter</span>(df_high_conf_selection, selection_williams2018 <span class="op">==</span><span class="st"> &quot;selected&quot;</span>)</a></code></pre></div>
<pre><code># A tibble: 13 x 7
   donor num_mutations n_cells prob_select_wil… selection_willi… n_de_genes
   &lt;chr&gt;         &lt;int&gt;   &lt;int&gt;            &lt;dbl&gt; &lt;chr&gt;                 &lt;int&gt;
 1 euts            292      79            1     selected                738
 2 fawm            101      49            1     selected                370
 3 feec            170      67            1.000 selected                756
 4 fikt            142      34            1     selected                599
 5 garx            592      70            1     selected                902
 6 joxm            612      78            1     selected               1133
 7 oilg            211      89            0.983 selected                396
 8 pipw            233     107            1     selected                372
 9 puie            117      41            1     selected                163
10 qolg            120      36            1     selected                131
11 ualf            325      89            0.871 selected                835
12 vass            412      37            1.000 selected                282
13 wahn            496      78            0.984 selected               1119
# … with 1 more variable: n_sig_genesets &lt;dbl&gt;</code></pre>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb70-1" data-line-number="1">dplyr<span class="op">::</span><span class="kw">filter</span>(df_high_conf_selection, selection_williams2018 <span class="op">==</span><span class="st"> &quot;neutral&quot;</span>)</a></code></pre></div>
<pre><code># A tibble: 2 x 7
  donor num_mutations n_cells prob_select_wil… selection_willi… n_de_genes
  &lt;chr&gt;         &lt;int&gt;   &lt;int&gt;            &lt;dbl&gt; &lt;chr&gt;                 &lt;int&gt;
1 sehl            178      29            0.251 neutral                   2
2 xugn            124      35            0.171 neutral                  43
# … with 1 more variable: n_sig_genesets &lt;dbl&gt;</code></pre>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb72-1" data-line-number="1">dplyr<span class="op">::</span><span class="kw">filter</span>(df_high_conf_selection, selection_williams2018 <span class="op">==</span><span class="st"> &quot;undetermined&quot;</span>)</a></code></pre></div>
<pre><code># A tibble: 5 x 7
  donor num_mutations n_cells prob_select_wil… selection_willi… n_de_genes
  &lt;chr&gt;         &lt;int&gt;   &lt;int&gt;            &lt;dbl&gt; &lt;chr&gt;                 &lt;int&gt;
1 gesg            157     100            0.357 undetermined           1094
2 heja            192      50            0.470 undetermined            336
3 laey            278      55            0.665 undetermined            419
4 qonc            131      40            0.414 undetermined            832
5 vuna            135      71            0.314 undetermined            685
# … with 1 more variable: n_sig_genesets &lt;dbl&gt;</code></pre>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb74-1" data-line-number="1">selected_lines &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(df_high_conf_selection, </a>
<a class="sourceLine" id="cb74-2" data-line-number="2">                                selection_williams2018 <span class="op">==</span><span class="st"> &quot;selected&quot;</span>)[[<span class="st">&quot;donor&quot;</span>]]</a>
<a class="sourceLine" id="cb74-3" data-line-number="3">undetermined_lines &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(df_high_conf_selection, </a>
<a class="sourceLine" id="cb74-4" data-line-number="4">                                selection_williams2018 <span class="op">==</span><span class="st"> &quot;undetermined&quot;</span>)[[<span class="st">&quot;donor&quot;</span>]]</a>
<a class="sourceLine" id="cb74-5" data-line-number="5">neutral_lines &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(df_high_conf_selection, </a>
<a class="sourceLine" id="cb74-6" data-line-number="6">                                selection_williams2018 <span class="op">==</span><span class="st"> &quot;neutral&quot;</span>)[[<span class="st">&quot;donor&quot;</span>]]</a>
<a class="sourceLine" id="cb74-7" data-line-number="7"></a>
<a class="sourceLine" id="cb74-8" data-line-number="8">df_camera_sig_unst <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(collection <span class="op">==</span><span class="st"> &quot;H&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb74-9" data-line-number="9"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(stat <span class="op">==</span><span class="st"> &quot;logFC&quot;</span>, donor <span class="op">%in%</span><span class="st"> </span>neutral_lines) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb74-10" data-line-number="10"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(NGenes, Direction, PValue, FDR, geneset, contrast, donor) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb74-11" data-line-number="11"><span class="st">  </span><span class="kw">print</span>(<span class="dt">n =</span> <span class="ot">Inf</span>)</a></code></pre></div>
<pre><code># A tibble: 22 x 7
   NGenes Direction   PValue      FDR geneset             contrast    donor
    &lt;dbl&gt; &lt;chr&gt;        &lt;dbl&gt;    &lt;dbl&gt; &lt;chr&gt;               &lt;chr&gt;       &lt;chr&gt;
 1    190 Up        4.10e-11 2.05e- 9 HALLMARK_G2M_CHECK… clone2 - c… sehl 
 2    197 Up        1.45e-10 3.62e- 9 HALLMARK_E2F_TARGE… clone2 - c… sehl 
 3    181 Up        9.52e- 4 1.59e- 2 HALLMARK_MITOTIC_S… clone2 - c… sehl 
 4    197 Up        2.33e-16 1.16e-14 HALLMARK_E2F_TARGE… clone3 - c… sehl 
 5    190 Up        1.01e-14 2.53e-13 HALLMARK_G2M_CHECK… clone3 - c… sehl 
 6    181 Up        1.92e- 5 3.19e- 4 HALLMARK_MITOTIC_S… clone3 - c… sehl 
 7     65 Up        3.85e- 3 4.81e- 2 HALLMARK_SPERMATOG… clone3 - c… sehl 
 8     71 Down      4.85e- 3 4.85e- 2 HALLMARK_COAGULATI… clone3 - c… sehl 
 9    190 Up        1.50e- 6 7.52e- 5 HALLMARK_G2M_CHECK… clone4 - c… sehl 
10    197 Up        4.75e- 6 1.19e- 4 HALLMARK_E2F_TARGE… clone4 - c… sehl 
11    197 Down      9.95e- 5 4.98e- 3 HALLMARK_E2F_TARGE… clone4 - c… sehl 
12    190 Down      1.16e- 3 2.91e- 2 HALLMARK_G2M_CHECK… clone4 - c… sehl 
13    190 Up        3.13e-12 9.75e-11 HALLMARK_G2M_CHECK… clone2 - c… xugn 
14    197 Up        3.90e-12 9.75e-11 HALLMARK_E2F_TARGE… clone2 - c… xugn 
15     65 Up        1.90e- 4 3.17e- 3 HALLMARK_SPERMATOG… clone2 - c… xugn 
16    198 Up        6.40e- 4 8.00e- 3 HALLMARK_MYC_TARGE… clone2 - c… xugn 
17     71 Down      6.05e- 9 3.02e- 7 HALLMARK_INTERFERO… clone3 - c… xugn 
18    123 Down      2.20e- 6 5.49e- 5 HALLMARK_INTERFERO… clone3 - c… xugn 
19    197 Up        7.29e- 5 1.22e- 3 HALLMARK_E2F_TARGE… clone3 - c… xugn 
20    190 Up        2.25e- 4 2.81e- 3 HALLMARK_G2M_CHECK… clone3 - c… xugn 
21    190 Down      6.76e- 7 3.38e- 5 HALLMARK_G2M_CHECK… clone3 - c… xugn 
22    197 Down      1.65e- 6 4.13e- 5 HALLMARK_E2F_TARGE… clone3 - c… xugn </code></pre>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb76-1" data-line-number="1">df_camera_sig_unst <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(collection <span class="op">==</span><span class="st"> &quot;H&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb76-2" data-line-number="2"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(stat <span class="op">==</span><span class="st"> &quot;logFC&quot;</span>, donor <span class="op">%in%</span><span class="st"> </span>undetermined_lines) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb76-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(NGenes, Direction, PValue, FDR, geneset, contrast, donor) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb76-4" data-line-number="4"><span class="st">  </span><span class="kw">print</span>(<span class="dt">n =</span> <span class="ot">Inf</span>)</a></code></pre></div>
<pre><code># A tibble: 9 x 7
  NGenes Direction   PValue      FDR geneset             contrast     donor
   &lt;dbl&gt; &lt;chr&gt;        &lt;dbl&gt;    &lt;dbl&gt; &lt;chr&gt;               &lt;chr&gt;        &lt;chr&gt;
1    197 Up        3.14e- 9 1.57e- 7 HALLMARK_E2F_TARGE… clone2 - cl… laey 
2    190 Up        3.64e- 7 9.10e- 6 HALLMARK_G2M_CHECK… clone2 - cl… laey 
3     71 Down      1.30e- 3 2.17e- 2 HALLMARK_COAGULATI… clone2 - cl… laey 
4    198 Up        2.16e- 3 2.70e- 2 HALLMARK_MYC_TARGE… clone2 - cl… laey 
5    197 Up        9.30e-18 4.65e-16 HALLMARK_E2F_TARGE… clone3 - cl… laey 
6    190 Up        1.74e-13 4.34e-12 HALLMARK_G2M_CHECK… clone3 - cl… laey 
7    181 Up        8.57e- 4 1.16e- 2 HALLMARK_MITOTIC_S… clone3 - cl… laey 
8     65 Up        9.30e- 4 1.16e- 2 HALLMARK_SPERMATOG… clone3 - cl… laey 
9     23 Down      4.49e- 3 4.49e- 2 HALLMARK_ANGIOGENE… clone3 - cl… laey </code></pre>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb78-1" data-line-number="1">df_camera_sig_unst <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(collection <span class="op">==</span><span class="st"> &quot;H&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb78-2" data-line-number="2"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(stat <span class="op">==</span><span class="st"> &quot;logFC&quot;</span>, donor <span class="op">%in%</span><span class="st"> </span>selected_lines) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb78-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(NGenes, Direction, PValue, FDR, geneset, contrast, donor) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb78-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, geneset)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb78-5" data-line-number="5"><span class="st">  </span><span class="kw">distinct</span>(id, <span class="dt">.keep_all =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(geneset) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb78-6" data-line-number="6"><span class="st">  </span><span class="kw">summarise</span>(<span class="dt">n_sig =</span> <span class="kw">sum</span>(FDR <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">print</span>(<span class="dt">n =</span> <span class="ot">Inf</span>)</a></code></pre></div>
<pre><code># A tibble: 13 x 2
   geneset                                    n_sig
   &lt;chr&gt;                                      &lt;int&gt;
 1 HALLMARK_ANGIOGENESIS                          1
 2 HALLMARK_APICAL_JUNCTION                       1
 3 HALLMARK_COAGULATION                           1
 4 HALLMARK_E2F_TARGETS                          10
 5 HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION     3
 6 HALLMARK_G2M_CHECKPOINT                        9
 7 HALLMARK_INTERFERON_ALPHA_RESPONSE             1
 8 HALLMARK_INTERFERON_GAMMA_RESPONSE             1
 9 HALLMARK_MITOTIC_SPINDLE                       6
10 HALLMARK_MYC_TARGETS_V1                        2
11 HALLMARK_PANCREAS_BETA_CELLS                   1
12 HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY       1
13 HALLMARK_SPERMATOGENESIS                       7</code></pre>
<p>NB: * 10 of 13 selected lines have Hallmark genesets significantly enriched at FDR &lt; 5% - all of these are significant for E2F targets and 9 are significant for G2M checkpoint. * <code>euts</code> would be significantly enriched for E2F targets and G2M checkpoints at FDR &lt; 10% (down in clone3 vs clone2). * <code>wahn</code> would have have E2F targets significantly enriched at FDR &lt; 20% (Up in clone3 vs clone2) and G2M checkpoint significant at FDR &lt; 30% * 3 of 7 neutral and undetermined lines have significantly enriched Hallmark genesets at FDR &lt; 5%.</p>
<ul>
<li>all 13 selected lines have c2 genesets significantly enriched at FDR &lt; 5%</li>
<li>5 of 7 neutral and undetermined lines have c2 genesets significantly enriched at FDR &lt; 5%.</li>
</ul>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb80-1" data-line-number="1"><span class="co">## selection, n_de resid boxplot</span></a>
<a class="sourceLine" id="cb80-2" data-line-number="2">df_nbglm_nde <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb80-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb80-4" data-line-number="4">    selection_williams2018, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb80-5" data-line-number="5"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> selection, <span class="dt">y =</span> .resid)) <span class="op">+</span></a>
<a class="sourceLine" id="cb80-6" data-line-number="6"><span class="st">  </span><span class="kw">geom_violin</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> selection), <span class="dt">alpha =</span> <span class="fl">0.7</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb80-7" data-line-number="7"><span class="st">  </span><span class="kw">geom_boxplot</span>(<span class="dt">outlier.alpha =</span> <span class="dv">0</span>, <span class="dt">width =</span> <span class="fl">0.2</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb80-8" data-line-number="8"><span class="st">  </span>ggbeeswarm<span class="op">::</span><span class="kw">geom_quasirandom</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> selection), <span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb80-9" data-line-number="9"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes (residual from NB GLM)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb80-10" data-line-number="10"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Inferred selection status (Willams 2018 model)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb80-11" data-line-number="11"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb80-12" data-line-number="12"><span class="st">  </span><span class="kw">coord_flip</span>()</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/link-de-to-selection-williams2018-model-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb81-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_de_resid_selection_williams2018_boxplot.png&quot;</span>, </a>
<a class="sourceLine" id="cb81-2" data-line-number="2">       <span class="dt">height =</span> <span class="fl">4.5</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)</a>
<a class="sourceLine" id="cb81-3" data-line-number="3"></a>
<a class="sourceLine" id="cb81-4" data-line-number="4"><span class="kw">summary</span>(<span class="kw">lm</span>(.resid <span class="op">~</span><span class="st"> </span>selection_williams2018, <span class="dt">data =</span> df_nbglm_nde))</a></code></pre></div>
<pre><code>
Call:
lm(formula = .resid ~ selection_williams2018, data = df_nbglm_nde)

Residuals:
    Min      1Q  Median      3Q     Max 
-1.9742 -0.6067  0.1408  0.5894  2.1238 

Coefficients:
                                   Estimate Std. Error t value Pr(&gt;|t|)  
(Intercept)                         -0.8562     0.4205  -2.036   0.0513 .
selection_williams2018selected       0.7494     0.4930   1.520   0.1397  
selection_williams2018undetermined   0.5806     0.5428   1.070   0.2940  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 1.03 on 28 degrees of freedom
Multiple R-squared:  0.07645,   Adjusted R-squared:  0.01048 
F-statistic: 1.159 on 2 and 28 DF,  p-value: 0.3284</code></pre>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb83-1" data-line-number="1"><span class="co">## selection, n_de (sqrt scale) boxplot</span></a>
<a class="sourceLine" id="cb83-2" data-line-number="2">df_nbglm_nde <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb83-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb83-4" data-line-number="4">    selection_williams2018, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb83-5" data-line-number="5"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> selection, <span class="dt">y =</span> n_de_genes)) <span class="op">+</span></a>
<a class="sourceLine" id="cb83-6" data-line-number="6"><span class="st">  </span><span class="kw">geom_violin</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> selection), <span class="dt">alpha =</span> <span class="fl">0.7</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb83-7" data-line-number="7"><span class="st">  </span><span class="kw">geom_boxplot</span>(<span class="dt">outlier.alpha =</span> <span class="dv">0</span>, <span class="dt">width =</span> <span class="fl">0.2</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb83-8" data-line-number="8"><span class="st">  </span>ggbeeswarm<span class="op">::</span><span class="kw">geom_quasirandom</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> selection), <span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb83-9" data-line-number="9"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb83-10" data-line-number="10"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Inferred selection status (Williams 2018 model)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb83-11" data-line-number="11"><span class="st">  </span><span class="kw">scale_y_sqrt</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">100</span>, <span class="dv">500</span>, <span class="dv">1000</span>, <span class="dv">1500</span>, <span class="dv">2000</span>, <span class="dv">2500</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb83-12" data-line-number="12"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb83-13" data-line-number="13"><span class="st">  </span><span class="kw">coord_flip</span>()</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/link-de-to-selection-williams2018-model-2.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb84-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_de_sqrt_selection_williams2018_boxplot.png&quot;</span>, </a>
<a class="sourceLine" id="cb84-2" data-line-number="2">       <span class="dt">height =</span> <span class="fl">5.5</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)</a>
<a class="sourceLine" id="cb84-3" data-line-number="3"></a>
<a class="sourceLine" id="cb84-4" data-line-number="4"><span class="co">## n_de (resids) vs goodness of fit cumul. mutation model</span></a>
<a class="sourceLine" id="cb84-5" data-line-number="5">df_nbglm_nde <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb84-6" data-line-number="6"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb84-7" data-line-number="7">    selection_williams2018, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb84-8" data-line-number="8"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> prob_select_williams2018, <span class="dt">y =</span> .resid, <span class="dt">fill =</span> selection)) <span class="op">+</span></a>
<a class="sourceLine" id="cb84-9" data-line-number="9"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb84-10" data-line-number="10"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb84-11" data-line-number="11"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes (residual from NB GLM)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb84-12" data-line-number="12"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Probability of selection (Williams 2018 model)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb84-13" data-line-number="13"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>))</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/link-de-to-selection-williams2018-model-3.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb85-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_de_resid_vs_prob_selection_williams2018_model.png&quot;</span>, </a>
<a class="sourceLine" id="cb85-2" data-line-number="2">       <span class="dt">height =</span> <span class="fl">6.5</span>, <span class="dt">width =</span> <span class="fl">5.5</span>)</a>
<a class="sourceLine" id="cb85-3" data-line-number="3"></a>
<a class="sourceLine" id="cb85-4" data-line-number="4"></a>
<a class="sourceLine" id="cb85-5" data-line-number="5"><span class="co">## n_de (sqrt scale) vs goodness of fit cumul. mutation model</span></a>
<a class="sourceLine" id="cb85-6" data-line-number="6">df_nbglm_nde <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb85-7" data-line-number="7"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb85-8" data-line-number="8">    selection_williams2018, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb85-9" data-line-number="9"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> prob_select_williams2018, <span class="dt">y =</span> n_de_genes, <span class="dt">fill =</span> selection)) <span class="op">+</span></a>
<a class="sourceLine" id="cb85-10" data-line-number="10"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb85-11" data-line-number="11"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb85-12" data-line-number="12"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb85-13" data-line-number="13"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Probability of selection (Williams 2018 model)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb85-14" data-line-number="14"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>))</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/link-de-to-selection-williams2018-model-4.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb86-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_de_sqrt_vs_prob_selection_williams2018_model.png&quot;</span>, </a>
<a class="sourceLine" id="cb86-2" data-line-number="2">       <span class="dt">height =</span> <span class="fl">6.5</span>, <span class="dt">width =</span> <span class="fl">5.5</span>)</a>
<a class="sourceLine" id="cb86-3" data-line-number="3"></a>
<a class="sourceLine" id="cb86-4" data-line-number="4"><span class="co">## n_de (resids) vs mutational load (all)</span></a>
<a class="sourceLine" id="cb86-5" data-line-number="5">df_nbglm_nde <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb86-6" data-line-number="6"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb86-7" data-line-number="7">    selection_williams2018, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb86-8" data-line-number="8"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> nvars_all, <span class="dt">y =</span> .resid, <span class="dt">fill =</span> selection)) <span class="op">+</span></a>
<a class="sourceLine" id="cb86-9" data-line-number="9"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb86-10" data-line-number="10"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb86-11" data-line-number="11"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes (residual from NB GLM)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb86-12" data-line-number="12"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Number of somatic variants&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb86-13" data-line-number="13"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb86-14" data-line-number="14"><span class="st">  </span><span class="kw">scale_x_log10</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">50</span>, <span class="dv">100</span>, <span class="dv">500</span>))</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/link-de-to-selection-williams2018-model-5.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb87-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_de_resid_selection_williams2018_vs_n_somatic_vars_all.png&quot;</span>, </a>
<a class="sourceLine" id="cb87-2" data-line-number="2">       <span class="dt">height =</span> <span class="fl">6.5</span>, <span class="dt">width =</span> <span class="fl">5.5</span>)</a>
<a class="sourceLine" id="cb87-3" data-line-number="3"></a>
<a class="sourceLine" id="cb87-4" data-line-number="4"><span class="co">## n_de (sqrt scale) vs mutational load (all)</span></a>
<a class="sourceLine" id="cb87-5" data-line-number="5">df_nbglm_nde <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb87-6" data-line-number="6"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb87-7" data-line-number="7">    selection_williams2018, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb87-8" data-line-number="8"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> nvars_all, <span class="dt">y =</span> n_de_genes, <span class="dt">fill =</span> selection)) <span class="op">+</span></a>
<a class="sourceLine" id="cb87-9" data-line-number="9"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb87-10" data-line-number="10"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb87-11" data-line-number="11"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb87-12" data-line-number="12"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Number of somatic variants&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb87-13" data-line-number="13"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb87-14" data-line-number="14"><span class="st">  </span><span class="kw">scale_x_log10</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">50</span>, <span class="dv">100</span>, <span class="dv">500</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb87-15" data-line-number="15"><span class="st">  </span><span class="kw">scale_y_sqrt</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">10</span>, <span class="dv">100</span>, <span class="dv">500</span>, <span class="dv">1000</span>, <span class="dv">1500</span>, <span class="dv">2000</span>, <span class="dv">2500</span>))</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/link-de-to-selection-williams2018-model-6.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb88-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_de_sqrt_selection_vs_n_somatic_vars_all.png&quot;</span>, </a>
<a class="sourceLine" id="cb88-2" data-line-number="2">       <span class="dt">height =</span> <span class="fl">5.5</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)</a>
<a class="sourceLine" id="cb88-3" data-line-number="3"></a>
<a class="sourceLine" id="cb88-4" data-line-number="4">df_nbglm_nde <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb88-5" data-line-number="5"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb88-6" data-line-number="6">    selection_williams2018, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb88-7" data-line-number="7"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> nvars_all, <span class="dt">y =</span> prob_select_williams2018, <span class="dt">fill =</span> selection)) <span class="op">+</span></a>
<a class="sourceLine" id="cb88-8" data-line-number="8"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb88-9" data-line-number="9"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb88-10" data-line-number="10"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Probability of selection (Williams 2018 model)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb88-11" data-line-number="11"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Number of somatic variants&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb88-12" data-line-number="12"><span class="st">  </span><span class="kw">coord_cartesian</span>(<span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb88-13" data-line-number="13"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb88-14" data-line-number="14"><span class="st">  </span><span class="kw">scale_x_log10</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">50</span>, <span class="dv">100</span>, <span class="dv">500</span>))</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/link-de-to-selection-williams2018-model-7.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb89-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/prob_select_williams2018_model_vs_n_somatic_vars.png&quot;</span>, </a>
<a class="sourceLine" id="cb89-2" data-line-number="2">       <span class="dt">height =</span> <span class="fl">6.5</span>, <span class="dt">width =</span> <span class="fl">5.5</span>)</a></code></pre></div>
</div>
<div id="selection-defined-using-williams-et-al-2016-and-negative-binomial-models" class="section level3">
<h3>Selection defined using Williams et al 2016 and negative binomial models</h3>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb90-1" data-line-number="1"><span class="co">## n_de vs n_cells</span></a>
<a class="sourceLine" id="cb90-2" data-line-number="2">df_nbglm_nde <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb90-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb90-4" data-line-number="4">    selection, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb90-5" data-line-number="5"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> n_cells, <span class="dt">y =</span> n_de_genes, <span class="dt">fill =</span> selection)) <span class="op">+</span></a>
<a class="sourceLine" id="cb90-6" data-line-number="6"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb90-7" data-line-number="7"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb90-8" data-line-number="8"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb90-9" data-line-number="9"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Number of cells&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb90-10" data-line-number="10"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>))</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/link-de-to-selection-old-models-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb91-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_de_genes_vs_n_cells.png&quot;</span>, </a>
<a class="sourceLine" id="cb91-2" data-line-number="2">       <span class="dt">height =</span> <span class="fl">5.5</span>, <span class="dt">width =</span> <span class="fl">5.5</span>)</a>
<a class="sourceLine" id="cb91-3" data-line-number="3"></a>
<a class="sourceLine" id="cb91-4" data-line-number="4"></a>
<a class="sourceLine" id="cb91-5" data-line-number="5"><span class="co">## selection, n_de resid boxplot</span></a>
<a class="sourceLine" id="cb91-6" data-line-number="6">df_nbglm_nde <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb91-7" data-line-number="7"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb91-8" data-line-number="8">    selection, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb91-9" data-line-number="9"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> selection, <span class="dt">y =</span> .resid)) <span class="op">+</span></a>
<a class="sourceLine" id="cb91-10" data-line-number="10"><span class="st">  </span><span class="kw">geom_violin</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> selection), <span class="dt">alpha =</span> <span class="fl">0.7</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb91-11" data-line-number="11"><span class="st">  </span><span class="kw">geom_boxplot</span>(<span class="dt">outlier.alpha =</span> <span class="dv">0</span>, <span class="dt">width =</span> <span class="fl">0.2</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb91-12" data-line-number="12"><span class="st">  </span>ggbeeswarm<span class="op">::</span><span class="kw">geom_quasirandom</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> selection), <span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb91-13" data-line-number="13"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes (residual from NB GLM)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb91-14" data-line-number="14"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Inferred selection status&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb91-15" data-line-number="15"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb91-16" data-line-number="16"><span class="st">  </span><span class="kw">coord_flip</span>()</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/link-de-to-selection-old-models-2.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb92-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_de_resid_selection_boxplot.png&quot;</span>, </a>
<a class="sourceLine" id="cb92-2" data-line-number="2">       <span class="dt">height =</span> <span class="fl">4.5</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)</a>
<a class="sourceLine" id="cb92-3" data-line-number="3"></a>
<a class="sourceLine" id="cb92-4" data-line-number="4"><span class="kw">summary</span>(<span class="kw">lm</span>(.resid <span class="op">~</span><span class="st"> </span>selection, <span class="dt">data =</span> df_nbglm_nde))</a></code></pre></div>
<pre><code>
Call:
lm(formula = .resid ~ selection, data = df_nbglm_nde)

Residuals:
     Min       1Q   Median       3Q      Max 
-1.99630 -0.52453  0.06752  0.62506  2.68235 

Coefficients:
                      Estimate Std. Error t value Pr(&gt;|t|)  
(Intercept)            -0.8341     0.4198  -1.987   0.0568 .
selectionselected       0.8533     0.5554   1.536   0.1356  
selectionundetermined   0.5709     0.4883   1.169   0.2522  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 1.028 on 28 degrees of freedom
Multiple R-squared:  0.07928,   Adjusted R-squared:  0.01352 
F-statistic: 1.206 on 2 and 28 DF,  p-value: 0.3146</code></pre>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb94-1" data-line-number="1"><span class="co">## selection, n_de (sqrt scale) boxplot</span></a>
<a class="sourceLine" id="cb94-2" data-line-number="2">df_nbglm_nde <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb94-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb94-4" data-line-number="4">    selection, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb94-5" data-line-number="5"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> selection, <span class="dt">y =</span> n_de_genes)) <span class="op">+</span></a>
<a class="sourceLine" id="cb94-6" data-line-number="6"><span class="st">  </span><span class="kw">geom_violin</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> selection), <span class="dt">alpha =</span> <span class="fl">0.7</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb94-7" data-line-number="7"><span class="st">  </span><span class="kw">geom_boxplot</span>(<span class="dt">outlier.alpha =</span> <span class="dv">0</span>, <span class="dt">width =</span> <span class="fl">0.2</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb94-8" data-line-number="8"><span class="st">  </span>ggbeeswarm<span class="op">::</span><span class="kw">geom_quasirandom</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> selection), <span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb94-9" data-line-number="9"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb94-10" data-line-number="10"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Inferred selection status&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb94-11" data-line-number="11"><span class="st">  </span><span class="kw">scale_y_sqrt</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">100</span>, <span class="dv">500</span>, <span class="dv">1000</span>, <span class="dv">1500</span>, <span class="dv">2000</span>, <span class="dv">2500</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb94-12" data-line-number="12"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb94-13" data-line-number="13"><span class="st">  </span><span class="kw">coord_flip</span>()</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/link-de-to-selection-old-models-3.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb95-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_de_sqrt_selection_boxplot.png&quot;</span>, </a>
<a class="sourceLine" id="cb95-2" data-line-number="2">       <span class="dt">height =</span> <span class="fl">5.5</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)</a>
<a class="sourceLine" id="cb95-3" data-line-number="3"></a>
<a class="sourceLine" id="cb95-4" data-line-number="4"><span class="co">## n_de (resids) vs goodness of fit cumul. mutation model</span></a>
<a class="sourceLine" id="cb95-5" data-line-number="5">df_nbglm_nde <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb95-6" data-line-number="6"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb95-7" data-line-number="7">    selection, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb95-8" data-line-number="8"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> rsq_ntrtestr, <span class="dt">y =</span> .resid, <span class="dt">fill =</span> selection)) <span class="op">+</span></a>
<a class="sourceLine" id="cb95-9" data-line-number="9"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb95-10" data-line-number="10"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb95-11" data-line-number="11"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes (residual from NB GLM)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb95-12" data-line-number="12"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Goodness of fit: cumulative mutations&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb95-13" data-line-number="13"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>))</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/link-de-to-selection-old-models-4.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb96-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_de_resid_selection_vs_gof_cumul_mut_model.png&quot;</span>, </a>
<a class="sourceLine" id="cb96-2" data-line-number="2">       <span class="dt">height =</span> <span class="fl">6.5</span>, <span class="dt">width =</span> <span class="fl">5.5</span>)</a>
<a class="sourceLine" id="cb96-3" data-line-number="3"></a>
<a class="sourceLine" id="cb96-4" data-line-number="4"></a>
<a class="sourceLine" id="cb96-5" data-line-number="5"><span class="co">## n_de (sqrt scale) vs goodness of fit cumul. mutation model</span></a>
<a class="sourceLine" id="cb96-6" data-line-number="6">df_nbglm_nde <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb96-7" data-line-number="7"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb96-8" data-line-number="8">    selection, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb96-9" data-line-number="9"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> rsq_ntrtestr, <span class="dt">y =</span> n_de_genes, <span class="dt">fill =</span> selection)) <span class="op">+</span></a>
<a class="sourceLine" id="cb96-10" data-line-number="10"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb96-11" data-line-number="11"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb96-12" data-line-number="12"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb96-13" data-line-number="13"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Goodness of fit: cumulative mutations&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb96-14" data-line-number="14"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>))</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/link-de-to-selection-old-models-5.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb97-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_de_sqrt_selection_vs_gof_cumul_mut_model.png&quot;</span>, </a>
<a class="sourceLine" id="cb97-2" data-line-number="2">       <span class="dt">height =</span> <span class="fl">5.5</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)</a>
<a class="sourceLine" id="cb97-3" data-line-number="3"></a>
<a class="sourceLine" id="cb97-4" data-line-number="4"><span class="co">## n_de (resids) vs goodness of fit NB model</span></a>
<a class="sourceLine" id="cb97-5" data-line-number="5">df_nbglm_nde <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb97-6" data-line-number="6"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb97-7" data-line-number="7">    selection, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb97-8" data-line-number="8"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> rsq_negbinfit, <span class="dt">y =</span> .resid, <span class="dt">fill =</span> selection)) <span class="op">+</span></a>
<a class="sourceLine" id="cb97-9" data-line-number="9"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb97-10" data-line-number="10"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb97-11" data-line-number="11"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes (residual from NB GLM)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb97-12" data-line-number="12"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Goodness of fit: negative binomial distribution&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb97-13" data-line-number="13"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>))</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/link-de-to-selection-old-models-6.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb98-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_de_resid_selection_vs_gof_negbin_model.png&quot;</span>, </a>
<a class="sourceLine" id="cb98-2" data-line-number="2">       <span class="dt">height =</span> <span class="fl">5.5</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)</a>
<a class="sourceLine" id="cb98-3" data-line-number="3"></a>
<a class="sourceLine" id="cb98-4" data-line-number="4"><span class="co">## n_de (resids) vs ave goodness of fit from both models</span></a>
<a class="sourceLine" id="cb98-5" data-line-number="5">df_nbglm_nde <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb98-6" data-line-number="6"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb98-7" data-line-number="7">    selection, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb98-8" data-line-number="8"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> (rsq_negbinfit <span class="op">+</span><span class="st"> </span>rsq_ntrtestr) <span class="op">/</span><span class="st"> </span><span class="dv">2</span>, <span class="dt">y =</span> .resid, <span class="dt">fill =</span> selection)) <span class="op">+</span></a>
<a class="sourceLine" id="cb98-9" data-line-number="9"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb98-10" data-line-number="10"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb98-11" data-line-number="11"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes (residual from NB GLM)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb98-12" data-line-number="12"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Ave. goodness of fit:</span><span class="ch">\n</span><span class="st"> negative binomial &amp; cumulative mutations&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb98-13" data-line-number="13"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>))</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/link-de-to-selection-old-models-7.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb99-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_de_resid_selection_vs_ave_gof_2_models.png&quot;</span>, </a>
<a class="sourceLine" id="cb99-2" data-line-number="2">       <span class="dt">height =</span> <span class="fl">5.5</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)</a>
<a class="sourceLine" id="cb99-3" data-line-number="3"></a>
<a class="sourceLine" id="cb99-4" data-line-number="4"><span class="co">## n_de vs ave goodness of fit from both models</span></a>
<a class="sourceLine" id="cb99-5" data-line-number="5">df_nbglm_nde <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb99-6" data-line-number="6"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb99-7" data-line-number="7">    selection, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb99-8" data-line-number="8"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> (rsq_negbinfit <span class="op">+</span><span class="st"> </span>rsq_ntrtestr) <span class="op">/</span><span class="st"> </span><span class="dv">2</span>, <span class="dt">y =</span> n_de_genes, </a>
<a class="sourceLine" id="cb99-9" data-line-number="9">           <span class="dt">fill =</span> selection)) <span class="op">+</span></a>
<a class="sourceLine" id="cb99-10" data-line-number="10"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb99-11" data-line-number="11"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb99-12" data-line-number="12"><span class="st">  </span><span class="kw">scale_y_sqrt</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb99-13" data-line-number="13"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb99-14" data-line-number="14"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Ave. goodness of fit:</span><span class="ch">\n</span><span class="st"> negative binomial &amp; cumulative mutations&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb99-15" data-line-number="15"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>))</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/link-de-to-selection-old-models-8.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb100-1" data-line-number="1"><span class="co">## n_de (resids) vs mutational load (all)</span></a>
<a class="sourceLine" id="cb100-2" data-line-number="2">df_nbglm_nde <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb100-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb100-4" data-line-number="4">    selection, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb100-5" data-line-number="5"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> nvars_all, <span class="dt">y =</span> .resid, <span class="dt">fill =</span> selection)) <span class="op">+</span></a>
<a class="sourceLine" id="cb100-6" data-line-number="6"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb100-7" data-line-number="7"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb100-8" data-line-number="8"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes (residual from NB GLM)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb100-9" data-line-number="9"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Number of somatic variants&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb100-10" data-line-number="10"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb100-11" data-line-number="11"><span class="st">  </span><span class="kw">scale_x_log10</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">50</span>, <span class="dv">100</span>, <span class="dv">500</span>))</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/link-de-to-selection-old-models-9.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb101-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_de_resid_selection_vs_n_somatic_vars_all.png&quot;</span>, </a>
<a class="sourceLine" id="cb101-2" data-line-number="2">       <span class="dt">height =</span> <span class="fl">6.5</span>, <span class="dt">width =</span> <span class="fl">5.5</span>)</a>
<a class="sourceLine" id="cb101-3" data-line-number="3"></a>
<a class="sourceLine" id="cb101-4" data-line-number="4"><span class="co">## n_de (sqrt scale) vs mutational load (all)</span></a>
<a class="sourceLine" id="cb101-5" data-line-number="5">df_nbglm_nde <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb101-6" data-line-number="6"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb101-7" data-line-number="7">    selection, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb101-8" data-line-number="8"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> nvars_all, <span class="dt">y =</span> n_de_genes, <span class="dt">fill =</span> selection)) <span class="op">+</span></a>
<a class="sourceLine" id="cb101-9" data-line-number="9"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb101-10" data-line-number="10"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb101-11" data-line-number="11"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb101-12" data-line-number="12"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Number of somatic variants&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb101-13" data-line-number="13"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb101-14" data-line-number="14"><span class="st">  </span><span class="kw">scale_x_log10</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">50</span>, <span class="dv">100</span>, <span class="dv">500</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb101-15" data-line-number="15"><span class="st">  </span><span class="kw">scale_y_sqrt</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">10</span>, <span class="dv">100</span>, <span class="dv">500</span>, <span class="dv">1000</span>, <span class="dv">1500</span>, <span class="dv">2000</span>, <span class="dv">2500</span>))</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/link-de-to-selection-old-models-10.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb102-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_de_sqrt_selection_vs_n_somatic_vars_all.png&quot;</span>, </a>
<a class="sourceLine" id="cb102-2" data-line-number="2">       <span class="dt">height =</span> <span class="fl">5.5</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)</a>
<a class="sourceLine" id="cb102-3" data-line-number="3"></a>
<a class="sourceLine" id="cb102-4" data-line-number="4"><span class="co">## n_de (resids) vs mutational load (missense)</span></a>
<a class="sourceLine" id="cb102-5" data-line-number="5">df_nbglm_nde <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb102-6" data-line-number="6"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb102-7" data-line-number="7">    selection, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb102-8" data-line-number="8"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> missense, <span class="dt">y =</span> .resid, <span class="dt">fill =</span> selection)) <span class="op">+</span></a>
<a class="sourceLine" id="cb102-9" data-line-number="9"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb102-10" data-line-number="10"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb102-11" data-line-number="11"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes (residual from NB GLM)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb102-12" data-line-number="12"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Number of somatic missense variants&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb102-13" data-line-number="13"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb102-14" data-line-number="14"><span class="st">  </span><span class="kw">scale_x_log10</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">50</span>, <span class="dv">100</span>, <span class="dv">500</span>))</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/link-de-to-selection-old-models-11.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb103-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_de_resid_selection_vs_n_somatic_vars_missense.png&quot;</span>, </a>
<a class="sourceLine" id="cb103-2" data-line-number="2">       <span class="dt">height =</span> <span class="fl">5.5</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)</a>
<a class="sourceLine" id="cb103-3" data-line-number="3"></a>
<a class="sourceLine" id="cb103-4" data-line-number="4"><span class="co">## n_de (resids) vs mutational load (missense, nonsense, splicing)</span></a>
<a class="sourceLine" id="cb103-5" data-line-number="5">df_nbglm_nde <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb103-6" data-line-number="6"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb103-7" data-line-number="7">    selection, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb103-8" data-line-number="8"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> nvars_misnonspli, <span class="dt">y =</span> .resid, <span class="dt">fill =</span> selection)) <span class="op">+</span></a>
<a class="sourceLine" id="cb103-9" data-line-number="9"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb103-10" data-line-number="10"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb103-11" data-line-number="11"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes (residual from NB GLM)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb103-12" data-line-number="12"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Number of missense, nonsense &amp; splicing variants&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb103-13" data-line-number="13"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb103-14" data-line-number="14"><span class="st">  </span><span class="kw">scale_x_log10</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">50</span>, <span class="dv">100</span>, <span class="dv">500</span>))</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/link-de-to-selection-old-models-12.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb104-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_de_resid_selection_vs_n_somatic_vars_misnonspli.png&quot;</span>, </a>
<a class="sourceLine" id="cb104-2" data-line-number="2">       <span class="dt">height =</span> <span class="fl">6.5</span>, <span class="dt">width =</span> <span class="fl">5.5</span>)</a>
<a class="sourceLine" id="cb104-3" data-line-number="3"></a>
<a class="sourceLine" id="cb104-4" data-line-number="4">df_nbglm_nde <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb104-5" data-line-number="5"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb104-6" data-line-number="6">    selection, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb104-7" data-line-number="7"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> nvars_all, <span class="dt">y =</span> rsq_ntrtestr, <span class="dt">fill =</span> selection)) <span class="op">+</span></a>
<a class="sourceLine" id="cb104-8" data-line-number="8"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb104-9" data-line-number="9"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb104-10" data-line-number="10"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Goodness of fit: cumulative mutations &quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb104-11" data-line-number="11"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Number of somatic variants&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb104-12" data-line-number="12"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb104-13" data-line-number="13"><span class="st">  </span><span class="kw">scale_x_log10</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">50</span>, <span class="dv">100</span>, <span class="dv">500</span>))</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/link-de-to-selection-old-models-13.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb105-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/gof_cumul_mut_model_vs_n_somatic_vars.png&quot;</span>, </a>
<a class="sourceLine" id="cb105-2" data-line-number="2">       <span class="dt">height =</span> <span class="fl">6.5</span>, <span class="dt">width =</span> <span class="fl">5.5</span>)</a></code></pre></div>
<p>We can also check the relationship between the number of somatic variants and evidence for selection as immediately above, but flipping the relationship.</p>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb106-1" data-line-number="1">df_nbglm_nde <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb106-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb106-3" data-line-number="3">    selection, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb106-4" data-line-number="4"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> rsq_ntrtestr, <span class="dt">y =</span> nvars_all, <span class="dt">fill =</span> selection)) <span class="op">+</span></a>
<a class="sourceLine" id="cb106-5" data-line-number="5"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb106-6" data-line-number="6"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb106-7" data-line-number="7"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Goodness of fit: cumulative mutations &quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb106-8" data-line-number="8"><span class="st">  </span><span class="kw">ylab</span>(<span class="kw">bquote</span>(<span class="st">&quot;Number of somatic variants (log&quot;</span>[<span class="dv">10</span>]<span class="op">~</span><span class="st"> &quot;scale)&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb106-9" data-line-number="9"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb106-10" data-line-number="10"><span class="st">  </span><span class="kw">scale_y_log10</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">50</span>, <span class="dv">100</span>, <span class="dv">500</span>))</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/nvars-vs-gof-selection-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb107-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_somatic_vars_vs_gof_cumul_mut_model.png&quot;</span>, </a>
<a class="sourceLine" id="cb107-2" data-line-number="2">       <span class="dt">height =</span> <span class="fl">6.5</span>, <span class="dt">width =</span> <span class="fl">5.5</span>)</a>
<a class="sourceLine" id="cb107-3" data-line-number="3"></a>
<a class="sourceLine" id="cb107-4" data-line-number="4">df_nbglm_nde <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb107-5" data-line-number="5"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb107-6" data-line-number="6">    selection, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb107-7" data-line-number="7"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> rsq_negbinfit, <span class="dt">y =</span> nvars_all, <span class="dt">fill =</span> selection)) <span class="op">+</span></a>
<a class="sourceLine" id="cb107-8" data-line-number="8"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb107-9" data-line-number="9"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb107-10" data-line-number="10"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Goodness of fit: negative binomial &quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb107-11" data-line-number="11"><span class="st">  </span><span class="kw">ylab</span>(<span class="kw">bquote</span>(<span class="st">&quot;Number of somatic variants (log&quot;</span>[<span class="dv">10</span>]<span class="op">~</span><span class="st"> &quot;scale)&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb107-12" data-line-number="12"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb107-13" data-line-number="13"><span class="st">  </span><span class="kw">scale_y_log10</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">50</span>, <span class="dv">100</span>, <span class="dv">500</span>))</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/nvars-vs-gof-selection-2.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb108-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_somatic_vars_vs_gof_cumul_mut_model.png&quot;</span>, </a>
<a class="sourceLine" id="cb108-2" data-line-number="2">       <span class="dt">height =</span> <span class="fl">6.5</span>, <span class="dt">width =</span> <span class="fl">5.5</span>)</a>
<a class="sourceLine" id="cb108-3" data-line-number="3"></a>
<a class="sourceLine" id="cb108-4" data-line-number="4">df_nbglm_nde <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb108-5" data-line-number="5"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb108-6" data-line-number="6">    selection, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb108-7" data-line-number="7"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> (rsq_negbinfit <span class="op">+</span><span class="st"> </span>rsq_ntrtestr) <span class="op">/</span><span class="st"> </span><span class="dv">2</span>, <span class="dt">y =</span> nvars_all, </a>
<a class="sourceLine" id="cb108-8" data-line-number="8">           <span class="dt">fill =</span> selection)) <span class="op">+</span></a>
<a class="sourceLine" id="cb108-9" data-line-number="9"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb108-10" data-line-number="10"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb108-11" data-line-number="11"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Ave. goodness of fit:</span><span class="ch">\n</span><span class="st"> negative binomial &amp; cumulative mutations&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb108-12" data-line-number="12"><span class="st">  </span><span class="kw">ylab</span>(<span class="kw">bquote</span>(<span class="st">&quot;Number of somatic variants (log&quot;</span>[<span class="dv">10</span>]<span class="op">~</span><span class="st"> &quot;scale)&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb108-13" data-line-number="13"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb108-14" data-line-number="14"><span class="st">  </span><span class="kw">scale_y_log10</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">50</span>, <span class="dv">100</span>, <span class="dv">500</span>))</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/nvars-vs-gof-selection-3.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb109-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_somatic_vars_vs_ave_gof_two_models.png&quot;</span>, </a>
<a class="sourceLine" id="cb109-2" data-line-number="2">       <span class="dt">height =</span> <span class="fl">6.5</span>, <span class="dt">width =</span> <span class="fl">5.5</span>)</a></code></pre></div>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb110-1" data-line-number="1">p1 &lt;-<span class="st"> </span>df_nbglm_nde <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb110-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb110-3" data-line-number="3">    selection, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb110-4" data-line-number="4"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> rsq_ntrtestr, <span class="dt">y =</span> .resid, </a>
<a class="sourceLine" id="cb110-5" data-line-number="5">           <span class="dt">fill =</span> selection)) <span class="op">+</span></a>
<a class="sourceLine" id="cb110-6" data-line-number="6"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb110-7" data-line-number="7"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb110-8" data-line-number="8"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes (residual from NB GLM)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb110-9" data-line-number="9"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Goodness of fit: cumulative mutations &quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb110-10" data-line-number="10"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>))</a>
<a class="sourceLine" id="cb110-11" data-line-number="11"></a>
<a class="sourceLine" id="cb110-12" data-line-number="12">p2 &lt;-<span class="st"> </span>df_nbglm_nde <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb110-13" data-line-number="13"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb110-14" data-line-number="14">    selection, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb110-15" data-line-number="15"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> nvars_all, <span class="dt">y =</span> rsq_ntrtestr, </a>
<a class="sourceLine" id="cb110-16" data-line-number="16">           <span class="dt">fill =</span> selection)) <span class="op">+</span></a>
<a class="sourceLine" id="cb110-17" data-line-number="17"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb110-18" data-line-number="18"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb110-19" data-line-number="19"><span class="st">  </span><span class="kw">xlab</span>(<span class="kw">bquote</span>(<span class="st">&quot;Number of somatic variants (log&quot;</span>[<span class="dv">10</span>]<span class="op">~</span><span class="st"> &quot;scale)&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb110-20" data-line-number="20"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Goodness of fit: cumulative mutations &quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb110-21" data-line-number="21"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb110-22" data-line-number="22"><span class="st">  </span><span class="kw">scale_x_log10</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">50</span>, <span class="dv">100</span>, <span class="dv">500</span>))</a>
<a class="sourceLine" id="cb110-23" data-line-number="23"></a>
<a class="sourceLine" id="cb110-24" data-line-number="24">p3 &lt;-<span class="st"> </span>df_nbglm_nde <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb110-25" data-line-number="25"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb110-26" data-line-number="26">    selection, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb110-27" data-line-number="27"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> nvars_all, <span class="dt">y =</span> .resid,</a>
<a class="sourceLine" id="cb110-28" data-line-number="28">           <span class="dt">fill =</span> selection)) <span class="op">+</span></a>
<a class="sourceLine" id="cb110-29" data-line-number="29"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb110-30" data-line-number="30"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb110-31" data-line-number="31"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes (residual from NB GLM)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb110-32" data-line-number="32"><span class="st">  </span><span class="kw">xlab</span>(<span class="kw">bquote</span>(<span class="st">&quot;Number of somatic variants (log&quot;</span>[<span class="dv">10</span>]<span class="op">~</span><span class="st"> &quot;scale)&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb110-33" data-line-number="33"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb110-34" data-line-number="34"><span class="st">  </span><span class="kw">scale_x_log10</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">50</span>, <span class="dv">100</span>, <span class="dv">500</span>))</a>
<a class="sourceLine" id="cb110-35" data-line-number="35"></a>
<a class="sourceLine" id="cb110-36" data-line-number="36"><span class="kw">plot_grid</span>(p1, p2, p3, <span class="dt">nrow =</span> <span class="dv">1</span>, <span class="dt">labels =</span> <span class="st">&quot;auto&quot;</span>)</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/nde-plots-grid-gof-cummut-1.png" width="1152" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb111-1" data-line-number="1">p1 &lt;-<span class="st"> </span>df_nbglm_nde <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb111-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb111-3" data-line-number="3">    selection, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb111-4" data-line-number="4"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> (rsq_ntrtestr <span class="op">+</span><span class="st"> </span>rsq_negbinfit) <span class="op">/</span><span class="st"> </span><span class="dv">2</span>, <span class="dt">y =</span> .resid, </a>
<a class="sourceLine" id="cb111-5" data-line-number="5">           <span class="dt">fill =</span> selection)) <span class="op">+</span></a>
<a class="sourceLine" id="cb111-6" data-line-number="6"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb111-7" data-line-number="7"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb111-8" data-line-number="8"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes (residual from NB GLM)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb111-9" data-line-number="9"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Ave. goodness of fit:</span><span class="ch">\n</span><span class="st"> negative binomial &amp; cumulative mutations&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb111-10" data-line-number="10"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>))</a>
<a class="sourceLine" id="cb111-11" data-line-number="11"></a>
<a class="sourceLine" id="cb111-12" data-line-number="12">p2 &lt;-<span class="st"> </span>df_nbglm_nde <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb111-13" data-line-number="13"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb111-14" data-line-number="14">    selection, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb111-15" data-line-number="15"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> nvars_all, <span class="dt">y =</span> (rsq_ntrtestr <span class="op">+</span><span class="st"> </span>rsq_negbinfit) <span class="op">/</span><span class="st"> </span><span class="dv">2</span>, </a>
<a class="sourceLine" id="cb111-16" data-line-number="16">           <span class="dt">fill =</span> selection)) <span class="op">+</span></a>
<a class="sourceLine" id="cb111-17" data-line-number="17"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb111-18" data-line-number="18"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb111-19" data-line-number="19"><span class="st">  </span><span class="kw">xlab</span>(<span class="kw">bquote</span>(<span class="st">&quot;Number of somatic variants (log&quot;</span>[<span class="dv">10</span>]<span class="op">~</span><span class="st"> &quot;scale)&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb111-20" data-line-number="20"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Ave. goodness of fit:</span><span class="ch">\n</span><span class="st"> negative binomial &amp; cumulative mutations&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb111-21" data-line-number="21"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb111-22" data-line-number="22"><span class="st">  </span><span class="kw">scale_x_log10</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">50</span>, <span class="dv">100</span>, <span class="dv">500</span>))</a>
<a class="sourceLine" id="cb111-23" data-line-number="23"></a>
<a class="sourceLine" id="cb111-24" data-line-number="24">p3 &lt;-<span class="st"> </span>df_nbglm_nde <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb111-25" data-line-number="25"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb111-26" data-line-number="26">    selection, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb111-27" data-line-number="27"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> nvars_all, <span class="dt">y =</span> .resid,</a>
<a class="sourceLine" id="cb111-28" data-line-number="28">           <span class="dt">fill =</span> selection)) <span class="op">+</span></a>
<a class="sourceLine" id="cb111-29" data-line-number="29"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb111-30" data-line-number="30"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb111-31" data-line-number="31"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes (residual from NB GLM)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb111-32" data-line-number="32"><span class="st">  </span><span class="kw">xlab</span>(<span class="kw">bquote</span>(<span class="st">&quot;Number of somatic variants (log&quot;</span>[<span class="dv">10</span>]<span class="op">~</span><span class="st"> &quot;scale)&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb111-33" data-line-number="33"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb111-34" data-line-number="34"><span class="st">  </span><span class="kw">scale_x_log10</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">50</span>, <span class="dv">100</span>, <span class="dv">500</span>))</a>
<a class="sourceLine" id="cb111-35" data-line-number="35"></a>
<a class="sourceLine" id="cb111-36" data-line-number="36"><span class="kw">plot_grid</span>(p1, p2, p3, <span class="dt">nrow =</span> <span class="dv">1</span>, <span class="dt">labels =</span> <span class="st">&quot;auto&quot;</span>)</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/nde-plots-grid-gof-ave-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="linking-de-to-assignment-accuracy" class="section level2">
<h2>Linking DE to assignment accuracy</h2>
<p>Simulation results provide us with an estimate of the assignment accuracy for each line.</p>
<p>We load the simulation results for individual lines.</p>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb112-1" data-line-number="1">get_prob_label &lt;-<span class="st"> </span><span class="cf">function</span>(X){</a>
<a class="sourceLine" id="cb112-2" data-line-number="2">  max_idx &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="dv">0</span>, <span class="kw">nrow</span>(X))</a>
<a class="sourceLine" id="cb112-3" data-line-number="3">  <span class="cf">for</span> (j <span class="cf">in</span> <span class="kw">seq_len</span>(<span class="kw">nrow</span>(X))) {</a>
<a class="sourceLine" id="cb112-4" data-line-number="4">    max_idx[j] &lt;-<span class="st"> </span><span class="kw">which.max</span>(X[j, ])</a>
<a class="sourceLine" id="cb112-5" data-line-number="5">  }</a>
<a class="sourceLine" id="cb112-6" data-line-number="6">  max_idx</a>
<a class="sourceLine" id="cb112-7" data-line-number="7">}</a>
<a class="sourceLine" id="cb112-8" data-line-number="8"></a>
<a class="sourceLine" id="cb112-9" data-line-number="9">get_prob_value &lt;-<span class="st"> </span><span class="cf">function</span>(X, <span class="dt">mode =</span> <span class="st">&quot;best&quot;</span>) {</a>
<a class="sourceLine" id="cb112-10" data-line-number="10">  max_val &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="dv">0</span>, <span class="kw">nrow</span>(X))</a>
<a class="sourceLine" id="cb112-11" data-line-number="11">  <span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">seq_len</span>(<span class="kw">nrow</span>(X))) {</a>
<a class="sourceLine" id="cb112-12" data-line-number="12">    sorted_val &lt;-<span class="st"> </span><span class="kw">sort</span>(X[i,], <span class="dt">decreasing =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb112-13" data-line-number="13">    <span class="cf">if</span> (mode <span class="op">==</span><span class="st"> &quot;delta&quot;</span>) {</a>
<a class="sourceLine" id="cb112-14" data-line-number="14">      max_val[i] &lt;-<span class="st"> </span>sorted_val[<span class="dv">1</span>] <span class="op">-</span><span class="st"> </span>sorted_val[<span class="dv">2</span>]</a>
<a class="sourceLine" id="cb112-15" data-line-number="15">    } <span class="cf">else</span> <span class="cf">if</span> (mode <span class="op">==</span><span class="st"> &quot;second&quot;</span>) {</a>
<a class="sourceLine" id="cb112-16" data-line-number="16">      max_val[i] &lt;-<span class="st"> </span>sorted_val[<span class="dv">2</span>]</a>
<a class="sourceLine" id="cb112-17" data-line-number="17">    } <span class="cf">else</span> {<span class="co">#default mode: best</span></a>
<a class="sourceLine" id="cb112-18" data-line-number="18">      max_val[i] &lt;-<span class="st"> </span>sorted_val[<span class="dv">1</span>]</a>
<a class="sourceLine" id="cb112-19" data-line-number="19">    }</a>
<a class="sourceLine" id="cb112-20" data-line-number="20">  }</a>
<a class="sourceLine" id="cb112-21" data-line-number="21">  max_val</a>
<a class="sourceLine" id="cb112-22" data-line-number="22">}</a>
<a class="sourceLine" id="cb112-23" data-line-number="23"></a>
<a class="sourceLine" id="cb112-24" data-line-number="24">all_files &lt;-<span class="st"> </span><span class="kw">paste0</span>(donors, <span class="st">&quot;.simulate.rds&quot;</span>)</a>
<a class="sourceLine" id="cb112-25" data-line-number="25">assign_<span class="dv">0</span> &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="dv">0</span>, <span class="dt">nrow =</span> <span class="dv">500</span>, <span class="dt">ncol =</span> <span class="kw">length</span>(donors))</a>
<a class="sourceLine" id="cb112-26" data-line-number="26">assign_<span class="dv">1</span> &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="dv">0</span>, <span class="dt">nrow =</span> <span class="dv">500</span>, <span class="dt">ncol =</span> <span class="kw">length</span>(donors))</a>
<a class="sourceLine" id="cb112-27" data-line-number="27">prob_all &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="dv">0</span>, <span class="dt">nrow =</span> <span class="dv">500</span>, <span class="dt">ncol =</span> <span class="kw">length</span>(donors))</a>
<a class="sourceLine" id="cb112-28" data-line-number="28"></a>
<a class="sourceLine" id="cb112-29" data-line-number="29"><span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">seq_len</span>(<span class="kw">length</span>(all_files))) {</a>
<a class="sourceLine" id="cb112-30" data-line-number="30">  afile &lt;-<span class="st"> </span>all_files[i]</a>
<a class="sourceLine" id="cb112-31" data-line-number="31">  sim_dat &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">file.path</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;simulations&quot;</span>, afile))</a>
<a class="sourceLine" id="cb112-32" data-line-number="32">  assign_<span class="dv">0</span>[, i] &lt;-<span class="st"> </span><span class="kw">get_prob_label</span>(sim_dat<span class="op">$</span>I_sim)</a>
<a class="sourceLine" id="cb112-33" data-line-number="33">  assign_<span class="dv">1</span>[, i] &lt;-<span class="st"> </span><span class="kw">get_prob_label</span>(sim_dat<span class="op">$</span>prob_Gibbs)</a>
<a class="sourceLine" id="cb112-34" data-line-number="34">  prob_all[, i] &lt;-<span class="st"> </span><span class="kw">get_prob_value</span>(sim_dat<span class="op">$</span>prob_Gibbs, <span class="dt">mode =</span> <span class="st">&quot;best&quot;</span>)</a>
<a class="sourceLine" id="cb112-35" data-line-number="35">}</a>
<a class="sourceLine" id="cb112-36" data-line-number="36"></a>
<a class="sourceLine" id="cb112-37" data-line-number="37">precision_simu &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="dv">0</span>, <span class="kw">length</span>(donors))</a>
<a class="sourceLine" id="cb112-38" data-line-number="38"><span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">seq_len</span>(<span class="kw">length</span>(donors))) {</a>
<a class="sourceLine" id="cb112-39" data-line-number="39">  idx &lt;-<span class="st"> </span>prob_all[, i] <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.5</span></a>
<a class="sourceLine" id="cb112-40" data-line-number="40">  precision_simu[i] &lt;-<span class="st"> </span><span class="kw">mean</span>(assign_<span class="dv">0</span>[idx, i] <span class="op">==</span><span class="st"> </span>assign_<span class="dv">1</span>[idx, i])</a>
<a class="sourceLine" id="cb112-41" data-line-number="41">}</a></code></pre></div>
<p>Read results from real data as well and form a data frame for plotting.</p>
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb113-1" data-line-number="1">all_files &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;cardelino_results_carderelax.&quot;</span>, donors, </a>
<a class="sourceLine" id="cb113-2" data-line-number="2">                    <span class="st">&quot;.filt_lenient.cell_coverage_sites.rds&quot;</span>)</a>
<a class="sourceLine" id="cb113-3" data-line-number="3">n_sites &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="dv">0</span>, <span class="kw">length</span>(donors))</a>
<a class="sourceLine" id="cb113-4" data-line-number="4">n_clone &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="dv">0</span>, <span class="kw">length</span>(donors))</a>
<a class="sourceLine" id="cb113-5" data-line-number="5">recall_all &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="dv">0</span>, <span class="kw">length</span>(donors))</a>
<a class="sourceLine" id="cb113-6" data-line-number="6"><span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">seq_len</span>(<span class="kw">length</span>(all_files))) {</a>
<a class="sourceLine" id="cb113-7" data-line-number="7">  afile &lt;-<span class="st"> </span>all_files[i]</a>
<a class="sourceLine" id="cb113-8" data-line-number="8">  carde_dat &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">file.path</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;cell_assignment&quot;</span>, afile))</a>
<a class="sourceLine" id="cb113-9" data-line-number="9">  n_sites[i] &lt;-<span class="st"> </span><span class="kw">nrow</span>(carde_dat<span class="op">$</span>D)</a>
<a class="sourceLine" id="cb113-10" data-line-number="10">  n_clone[i] &lt;-<span class="st"> </span><span class="kw">ncol</span>(carde_dat<span class="op">$</span>prob_mat)</a>
<a class="sourceLine" id="cb113-11" data-line-number="11">  recall_all[i] &lt;-<span class="st"> </span><span class="kw">mean</span>(<span class="kw">get_prob_value</span>(carde_dat<span class="op">$</span>prob_mat, <span class="dt">mode =</span> <span class="st">&quot;best&quot;</span>) <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.5</span>)</a>
<a class="sourceLine" id="cb113-12" data-line-number="12">}</a>
<a class="sourceLine" id="cb113-13" data-line-number="13"></a>
<a class="sourceLine" id="cb113-14" data-line-number="14">df &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">line =</span> donors, <span class="dt">n_sites =</span> n_sites, <span class="dt">n_clone =</span> n_clone, </a>
<a class="sourceLine" id="cb113-15" data-line-number="15">                 <span class="dt">recall_real =</span> recall_all, <span class="dt">recall_simu =</span> <span class="kw">colMeans</span>(prob_all <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.5</span>),</a>
<a class="sourceLine" id="cb113-16" data-line-number="16">                 <span class="dt">precision_simu =</span> precision_simu)</a></code></pre></div>
<p>We can now look at the relationship between the number of DE genes detected per line and the assignment accuracy and assignment rate per line.</p>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb114-1" data-line-number="1">df &lt;-<span class="st"> </span><span class="kw">inner_join</span>(df, dplyr<span class="op">::</span><span class="kw">mutate</span>(df_nbglm_nde, <span class="dt">line =</span> donor))</a>
<a class="sourceLine" id="cb114-2" data-line-number="2"><span class="co">## n_de (resids) vs assignment accuracy</span></a>
<a class="sourceLine" id="cb114-3" data-line-number="3">p1 &lt;-<span class="st"> </span>df <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb114-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb114-5" data-line-number="5">    selection, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb114-6" data-line-number="6"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> precision_simu, <span class="dt">y =</span> .resid, <span class="dt">fill =</span> selection)) <span class="op">+</span></a>
<a class="sourceLine" id="cb114-7" data-line-number="7"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb114-8" data-line-number="8"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb114-9" data-line-number="9"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes (residual from NB GLM)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb114-10" data-line-number="10"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Assignment accuracy (precision) from simulations&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb114-11" data-line-number="11"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>))</a>
<a class="sourceLine" id="cb114-12" data-line-number="12"></a>
<a class="sourceLine" id="cb114-13" data-line-number="13"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_de_resid_selection_vs_sim_assign_acc.png&quot;</span>, </a>
<a class="sourceLine" id="cb114-14" data-line-number="14">       <span class="dt">plot =</span> p1, <span class="dt">height =</span> <span class="fl">5.5</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)</a>
<a class="sourceLine" id="cb114-15" data-line-number="15"></a>
<a class="sourceLine" id="cb114-16" data-line-number="16"><span class="co">## n_de (resids) vs assignment rate</span></a>
<a class="sourceLine" id="cb114-17" data-line-number="17">p2 &lt;-<span class="st"> </span>df <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb114-18" data-line-number="18"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(</a>
<a class="sourceLine" id="cb114-19" data-line-number="19">    selection, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb114-20" data-line-number="20"><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> recall_simu, <span class="dt">y =</span> .resid, <span class="dt">fill =</span> selection)) <span class="op">+</span></a>
<a class="sourceLine" id="cb114-21" data-line-number="21"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb114-22" data-line-number="22"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb114-23" data-line-number="23"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes (residual from NB GLM)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb114-24" data-line-number="24"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Assignment rate (recall) from simulations&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb114-25" data-line-number="25"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>))</a>
<a class="sourceLine" id="cb114-26" data-line-number="26"></a>
<a class="sourceLine" id="cb114-27" data-line-number="27"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_de_resid_selection_vs_sim_assign_rate.png&quot;</span>, </a>
<a class="sourceLine" id="cb114-28" data-line-number="28">       <span class="dt">plot =</span> p2, <span class="dt">height =</span> <span class="fl">5.5</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)</a>
<a class="sourceLine" id="cb114-29" data-line-number="29"></a>
<a class="sourceLine" id="cb114-30" data-line-number="30">p3 &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(p1, p2, <span class="dt">labels =</span> <span class="st">&quot;auto&quot;</span>, <span class="dt">nrow =</span> <span class="dv">2</span>)</a>
<a class="sourceLine" id="cb114-31" data-line-number="31"></a>
<a class="sourceLine" id="cb114-32" data-line-number="32"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression_cardelino-relax/n_de_resid_selection_vs_sim_assign_acc_rate_combined.png&quot;</span>, </a>
<a class="sourceLine" id="cb114-33" data-line-number="33">       <span class="dt">plot =</span> p3, <span class="dt">height =</span> <span class="dv">10</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)</a>
<a class="sourceLine" id="cb114-34" data-line-number="34"></a>
<a class="sourceLine" id="cb114-35" data-line-number="35">p3</a></code></pre></div>
<p><img src="figure/differential_expression_cardelino-relax.Rmd/plot-de-assignment-acc-1.png" width="624" style="display: block; margin: auto;" /></p>
</div>
<div id="linking-de-to-multiple-explanatory-factors" class="section level2">
<h2>Linking DE to multiple explanatory factors</h2>
<p>Let us fit a NB GLM to regress number of DE genes observed per line against number of cells, goodness of fit of selection models, number of somatic variants, assignment accuracy (from simulations) and assignment rate (observed) per line.</p>
<p>First we can look at fitting different combinations of these explanatory variables and do model selection using AIC (Akaike Information Criterion).</p>
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb115-1" data-line-number="1">broom<span class="op">::</span><span class="kw">glance</span>(MASS<span class="op">::</span><span class="kw">glm.nb</span>(</a>
<a class="sourceLine" id="cb115-2" data-line-number="2">  n_de_genes <span class="op">~</span><span class="st"> </span>n_cells <span class="op">+</span><span class="st"> </span>rsq_negbinfit <span class="op">+</span><span class="st"> </span>rsq_ntrtestr <span class="op">+</span><span class="st"> </span>nvars_all</a>
<a class="sourceLine" id="cb115-3" data-line-number="3">  <span class="op">+</span><span class="st"> </span>recall_real <span class="op">+</span><span class="st"> </span>precision_simu, <span class="dt">data =</span> df))</a></code></pre></div>
<pre><code># A tibble: 1 x 7
  null.deviance df.null logLik   AIC   BIC deviance df.residual
          &lt;dbl&gt;   &lt;int&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt;       &lt;int&gt;
1          51.8      30  -212.  441.  452.     34.7          24</code></pre>
<div class="sourceCode" id="cb117"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb117-1" data-line-number="1">broom<span class="op">::</span><span class="kw">glance</span>(MASS<span class="op">::</span><span class="kw">glm.nb</span>(</a>
<a class="sourceLine" id="cb117-2" data-line-number="2">  n_de_genes <span class="op">~</span><span class="st"> </span><span class="kw">log10</span>(n_cells) <span class="op">+</span><span class="st"> </span>rsq_negbinfit <span class="op">+</span><span class="st"> </span>rsq_ntrtestr <span class="op">+</span><span class="st"> </span>nvars_all</a>
<a class="sourceLine" id="cb117-3" data-line-number="3">  <span class="op">+</span><span class="st"> </span>recall_real <span class="op">+</span><span class="st"> </span>precision_simu, <span class="dt">data =</span> df))</a></code></pre></div>
<pre><code># A tibble: 1 x 7
  null.deviance df.null logLik   AIC   BIC deviance df.residual
          &lt;dbl&gt;   &lt;int&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt;       &lt;int&gt;
1          52.8      30  -212.  440.  452.     34.6          24</code></pre>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb119-1" data-line-number="1">broom<span class="op">::</span><span class="kw">glance</span>(MASS<span class="op">::</span><span class="kw">glm.nb</span>(</a>
<a class="sourceLine" id="cb119-2" data-line-number="2">  n_de_genes <span class="op">~</span><span class="st"> </span><span class="kw">sqrt</span>(n_cells) <span class="op">+</span><span class="st"> </span>rsq_negbinfit <span class="op">+</span><span class="st"> </span>rsq_ntrtestr <span class="op">+</span><span class="st"> </span>nvars_all</a>
<a class="sourceLine" id="cb119-3" data-line-number="3">  <span class="op">+</span><span class="st"> </span>recall_real <span class="op">+</span><span class="st"> </span>precision_simu, <span class="dt">data =</span> df))</a></code></pre></div>
<pre><code># A tibble: 1 x 7
  null.deviance df.null logLik   AIC   BIC deviance df.residual
          &lt;dbl&gt;   &lt;int&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt;       &lt;int&gt;
1          52.3      30  -212.  440.  452.     34.7          24</code></pre>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb121-1" data-line-number="1">broom<span class="op">::</span><span class="kw">glance</span>(MASS<span class="op">::</span><span class="kw">glm.nb</span>(</a>
<a class="sourceLine" id="cb121-2" data-line-number="2">  n_de_genes <span class="op">~</span><span class="st"> </span><span class="kw">log10</span>(n_cells) <span class="op">+</span><span class="st"> </span>rsq_negbinfit <span class="op">+</span><span class="st"> </span>rsq_ntrtestr <span class="op">+</span><span class="st"> </span><span class="kw">sqrt</span>(nvars_all)</a>
<a class="sourceLine" id="cb121-3" data-line-number="3">  <span class="op">+</span><span class="st"> </span>recall_real <span class="op">+</span><span class="st"> </span>precision_simu, <span class="dt">data =</span> df))</a></code></pre></div>
<pre><code># A tibble: 1 x 7
  null.deviance df.null logLik   AIC   BIC deviance df.residual
          &lt;dbl&gt;   &lt;int&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt;       &lt;int&gt;
1          53.1      30  -212.  440.  451.     34.6          24</code></pre>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb123-1" data-line-number="1">broom<span class="op">::</span><span class="kw">glance</span>(MASS<span class="op">::</span><span class="kw">glm.nb</span>(</a>
<a class="sourceLine" id="cb123-2" data-line-number="2">  n_de_genes <span class="op">~</span><span class="st"> </span><span class="kw">log10</span>(n_cells) <span class="op">+</span><span class="st"> </span>rsq_negbinfit <span class="op">+</span><span class="st"> </span>rsq_ntrtestr <span class="op">+</span><span class="st"> </span><span class="kw">log10</span>(nvars_all)</a>
<a class="sourceLine" id="cb123-3" data-line-number="3">  <span class="op">+</span><span class="st"> </span>recall_real <span class="op">+</span><span class="st"> </span>precision_simu, <span class="dt">data =</span> df))</a></code></pre></div>
<pre><code># A tibble: 1 x 7
  null.deviance df.null logLik   AIC   BIC deviance df.residual
          &lt;dbl&gt;   &lt;int&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt;       &lt;int&gt;
1          53.4      30  -212.  440.  451.     34.6          24</code></pre>
<div class="sourceCode" id="cb125"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb125-1" data-line-number="1">broom<span class="op">::</span><span class="kw">glance</span>(MASS<span class="op">::</span><span class="kw">glm.nb</span>(</a>
<a class="sourceLine" id="cb125-2" data-line-number="2">  n_de_genes <span class="op">~</span><span class="st"> </span><span class="kw">log10</span>(n_cells) <span class="op">+</span><span class="st"> </span>rsq_ntrtestr <span class="op">+</span><span class="st"> </span>nvars_all</a>
<a class="sourceLine" id="cb125-3" data-line-number="3">  <span class="op">+</span><span class="st"> </span>recall_real <span class="op">+</span><span class="st"> </span>precision_simu, <span class="dt">data =</span> df))</a></code></pre></div>
<pre><code># A tibble: 1 x 7
  null.deviance df.null logLik   AIC   BIC deviance df.residual
          &lt;dbl&gt;   &lt;int&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt;       &lt;int&gt;
1          52.8      30  -212.  438.  448.     34.6          25</code></pre>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb127-1" data-line-number="1">broom<span class="op">::</span><span class="kw">glance</span>(MASS<span class="op">::</span><span class="kw">glm.nb</span>(</a>
<a class="sourceLine" id="cb127-2" data-line-number="2">  n_de_genes <span class="op">~</span><span class="st"> </span><span class="kw">log10</span>(n_cells) <span class="op">+</span><span class="st"> </span>rsq_ntrtestr <span class="op">+</span><span class="st"> </span>nvars_all</a>
<a class="sourceLine" id="cb127-3" data-line-number="3">  <span class="op">+</span><span class="st"> </span>recall_real, <span class="dt">data =</span> df))</a></code></pre></div>
<pre><code># A tibble: 1 x 7
  null.deviance df.null logLik   AIC   BIC deviance df.residual
          &lt;dbl&gt;   &lt;int&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt;       &lt;int&gt;
1          51.6      30  -212.  437.  446.     34.7          26</code></pre>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb129-1" data-line-number="1">broom<span class="op">::</span><span class="kw">glance</span>(MASS<span class="op">::</span><span class="kw">glm.nb</span>(</a>
<a class="sourceLine" id="cb129-2" data-line-number="2">  n_de_genes <span class="op">~</span><span class="st"> </span><span class="kw">log10</span>(n_cells) <span class="op">+</span><span class="st"> </span>rsq_ntrtestr <span class="op">+</span><span class="st"> </span>nvars_all, <span class="dt">data =</span> df))</a></code></pre></div>
<pre><code># A tibble: 1 x 7
  null.deviance df.null logLik   AIC   BIC deviance df.residual
          &lt;dbl&gt;   &lt;int&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt;       &lt;int&gt;
1          51.6      30  -212.  435.  442.     34.7          27</code></pre>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb131-1" data-line-number="1">broom<span class="op">::</span><span class="kw">glance</span>(MASS<span class="op">::</span><span class="kw">glm.nb</span>(</a>
<a class="sourceLine" id="cb131-2" data-line-number="2">  n_de_genes <span class="op">~</span><span class="st"> </span><span class="kw">log10</span>(n_cells) <span class="op">+</span><span class="st"> </span>rsq_negbinfit <span class="op">+</span><span class="st"> </span>nvars_all, <span class="dt">data =</span> df))</a></code></pre></div>
<pre><code># A tibble: 1 x 7
  null.deviance df.null logLik   AIC   BIC deviance df.residual
          &lt;dbl&gt;   &lt;int&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt;       &lt;int&gt;
1          51.0      30  -213.  435.  443.     34.7          27</code></pre>
<div class="sourceCode" id="cb133"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb133-1" data-line-number="1">broom<span class="op">::</span><span class="kw">glance</span>(MASS<span class="op">::</span><span class="kw">glm.nb</span>(</a>
<a class="sourceLine" id="cb133-2" data-line-number="2">  n_de_genes <span class="op">~</span><span class="st"> </span><span class="kw">log10</span>(n_cells) <span class="op">+</span><span class="st"> </span>nvars_all, <span class="dt">data =</span> df))</a></code></pre></div>
<pre><code># A tibble: 1 x 7
  null.deviance df.null logLik   AIC   BIC deviance df.residual
          &lt;dbl&gt;   &lt;int&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt;       &lt;int&gt;
1          51.0      30  -213.  433.  439.     34.7          28</code></pre>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb135-1" data-line-number="1">broom<span class="op">::</span><span class="kw">glance</span>(MASS<span class="op">::</span><span class="kw">glm.nb</span>(</a>
<a class="sourceLine" id="cb135-2" data-line-number="2">  n_de_genes <span class="op">~</span><span class="st"> </span><span class="kw">log10</span>(n_cells), <span class="dt">data =</span> df))</a></code></pre></div>
<pre><code># A tibble: 1 x 7
  null.deviance df.null logLik   AIC   BIC deviance df.residual
          &lt;dbl&gt;   &lt;int&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt;       &lt;int&gt;
1          48.2      30  -214.  434.  438.     34.9          29</code></pre>
<p>The preferred model based on AIC values is one fitting log10-scale number of cells and number of somatic variants as predictor variables.</p>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb137-1" data-line-number="1">nbglm_nde_allfacts &lt;-<span class="st"> </span>MASS<span class="op">::</span><span class="kw">glm.nb</span>(</a>
<a class="sourceLine" id="cb137-2" data-line-number="2">  n_de_genes <span class="op">~</span><span class="st">  </span><span class="kw">log10</span>(n_cells) <span class="op">+</span><span class="st"> </span>rsq_negbinfit <span class="op">+</span><span class="st"> </span>rsq_ntrtestr <span class="op">+</span><span class="st"> </span>nvars_all</a>
<a class="sourceLine" id="cb137-3" data-line-number="3">  <span class="op">+</span><span class="st"> </span>recall_real <span class="op">+</span><span class="st"> </span>precision_simu, <span class="dt">data =</span> df)</a>
<a class="sourceLine" id="cb137-4" data-line-number="4">broom<span class="op">::</span><span class="kw">tidy</span>(nbglm_nde_allfacts)</a></code></pre></div>
<pre><code># A tibble: 7 x 5
  term            estimate std.error statistic p.value
  &lt;chr&gt;              &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt;
1 (Intercept)    -1.01       3.95     -0.255   0.799  
2 log10(n_cells)  3.33       1.08      3.08    0.00204
3 rsq_negbinfit  -0.000699   0.647    -0.00108 0.999  
4 rsq_ntrtestr   -1.58       2.00     -0.793   0.428  
5 nvars_all       0.00638    0.00575   1.11    0.267  
6 recall_real    -0.771      1.52     -0.507   0.612  
7 precision_simu  3.05       3.34      0.913   0.361  </code></pre>
<div class="sourceCode" id="cb139"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb139-1" data-line-number="1">nbglm_nde_selectfacts &lt;-<span class="st"> </span>MASS<span class="op">::</span><span class="kw">glm.nb</span>(</a>
<a class="sourceLine" id="cb139-2" data-line-number="2">  n_de_genes <span class="op">~</span><span class="st"> </span><span class="kw">log10</span>(n_cells) <span class="op">+</span><span class="st"> </span>nvars_all, <span class="dt">data =</span> df)</a>
<a class="sourceLine" id="cb139-3" data-line-number="3">broom<span class="op">::</span><span class="kw">tidy</span>(nbglm_nde_selectfacts)</a></code></pre></div>
<pre><code># A tibble: 3 x 5
  term           estimate std.error statistic p.value
  &lt;chr&gt;             &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt;
1 (Intercept)    -0.0343    1.78      -0.0193 0.985  
2 log10(n_cells)  3.25      1.06       3.07   0.00215
3 nvars_all       0.00773   0.00549    1.41   0.159  </code></pre>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb141-1" data-line-number="1">broom<span class="op">::</span><span class="kw">glance</span>(nbglm_nde_selectfacts)</a></code></pre></div>
<pre><code># A tibble: 1 x 7
  null.deviance df.null logLik   AIC   BIC deviance df.residual
          &lt;dbl&gt;   &lt;int&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt;       &lt;int&gt;
1          51.0      30  -213.  433.  439.     34.7          28</code></pre>
<div class="sourceCode" id="cb143"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb143-1" data-line-number="1">nbglm_nde_allfacts_tbl &lt;-<span class="st"> </span><span class="kw">bind_cols</span>(</a>
<a class="sourceLine" id="cb143-2" data-line-number="2">  broom<span class="op">::</span><span class="kw">tidy</span>(nbglm_nde_selectfacts),</a>
<a class="sourceLine" id="cb143-3" data-line-number="3">  broom<span class="op">::</span><span class="kw">confint_tidy</span>(nbglm_nde_selectfacts, <span class="dt">conf.level =</span> <span class="fl">0.9</span>))</a>
<a class="sourceLine" id="cb143-4" data-line-number="4"></a>
<a class="sourceLine" id="cb143-5" data-line-number="5"></a>
<a class="sourceLine" id="cb143-6" data-line-number="6">df_nbglm_nde &lt;-<span class="st"> </span>broom<span class="op">::</span><span class="kw">augment</span>(nbglm_nde) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb143-7" data-line-number="7"><span class="st">  </span><span class="kw">left_join</span>(df_donor_info)</a></code></pre></div>
</div>
<div id="linking-pathway-results-to-selection" class="section level2">
<h2>Linking pathway results to selection</h2>
<p>Not yet implemented</p>
</div>
<div id="compare-pathway-results-to-clone-prevalence" class="section level2">
<h2>Compare pathway results to clone prevalence</h2>
<p>Hallmark gene sets. Not yet implemented.</p>
</div>
<div id="write-de-and-pathway-results-to-file" class="section level2">
<h2>Write DE and pathway results to file</h2>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb144-1" data-line-number="1"><span class="co">## Write DE results to file:</span></a>
<a class="sourceLine" id="cb144-2" data-line-number="2"><span class="cf">for</span> (don <span class="cf">in</span> <span class="kw">names</span>(de_res<span class="op">$</span>qlf_list)) {</a>
<a class="sourceLine" id="cb144-3" data-line-number="3">  de_res<span class="op">$</span>qlf_list[[don]]<span class="op">$</span>table <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb144-4" data-line-number="4"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">gene =</span> <span class="kw">rownames</span>(.), <span class="dt">FDR =</span> IHW<span class="op">::</span><span class="kw">adj_pvalues</span>(IHW<span class="op">::</span><span class="kw">ihw</span>(PValue, logCPM, <span class="dt">alpha =</span> <span class="fl">0.1</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb144-5" data-line-number="5"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(FDR) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">write_tsv</span>(</a>
<a class="sourceLine" id="cb144-6" data-line-number="6">      <span class="kw">paste0</span>(<span class="st">&quot;output/differential_expression_cardelino-relax/&quot;</span>, don, <span class="st">&quot;_qlf_de_results.tsv&quot;</span>))</a>
<a class="sourceLine" id="cb144-7" data-line-number="7">}</a>
<a class="sourceLine" id="cb144-8" data-line-number="8"></a>
<a class="sourceLine" id="cb144-9" data-line-number="9"><span class="cf">for</span> (don <span class="cf">in</span> <span class="kw">names</span>(de_res<span class="op">$</span>camera<span class="op">$</span>H)) {</a>
<a class="sourceLine" id="cb144-10" data-line-number="10">  <span class="cf">for</span> (cntrst <span class="cf">in</span> <span class="kw">names</span>(de_res<span class="op">$</span>camera<span class="op">$</span>H[[don]])) {</a>
<a class="sourceLine" id="cb144-11" data-line-number="11">    de_res<span class="op">$</span>camera<span class="op">$</span>H[[don]][[cntrst]]<span class="op">$</span>logFC <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb144-12" data-line-number="12"><span class="st">      </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">geneset =</span> <span class="kw">rownames</span>(.)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb144-13" data-line-number="13"><span class="st">      </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(FDR) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">write_tsv</span>(</a>
<a class="sourceLine" id="cb144-14" data-line-number="14">        <span class="kw">paste0</span>(<span class="st">&quot;output/differential_expression_cardelino-relax/&quot;</span>, don, <span class="st">&quot;_camera_hallmark_geneset_results_&quot;</span>, cntrst, <span class="st">&quot;.tsv&quot;</span>))    </a>
<a class="sourceLine" id="cb144-15" data-line-number="15">  }</a>
<a class="sourceLine" id="cb144-16" data-line-number="16">}</a></code></pre></div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb145"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb145-1" data-line-number="1">devtools<span class="op">::</span><span class="kw">session_info</span>()</a></code></pre></div>
<pre><code>─ Session info ──────────────────────────────────────────────────────────
 setting  value                       
 version  R version 3.6.0 (2019-04-26)
 os       Ubuntu 18.04.3 LTS          
 system   x86_64, linux-gnu           
 ui       X11                         
 language (EN)                        
 collate  en_AU.UTF-8                 
 ctype    en_AU.UTF-8                 
 tz       Australia/Melbourne         
 date     2019-10-30                  

─ Packages ──────────────────────────────────────────────────────────────
 package              * version   date       lib
 AnnotationDbi        * 1.46.1    2019-08-20 [1]
 ape                    5.3       2019-03-17 [1]
 assertthat             0.2.1     2019-03-21 [1]
 backports              1.1.4     2019-04-10 [1]
 beeswarm               0.2.3     2016-04-25 [1]
 Biobase              * 2.44.0    2019-05-02 [1]
 BiocGenerics         * 0.30.0    2019-05-02 [1]
 BiocNeighbors          1.2.0     2019-05-02 [1]
 BiocParallel         * 1.18.1    2019-08-06 [1]
 BiocSingular           1.0.0     2019-05-02 [1]
 biomaRt                2.40.4    2019-08-19 [1]
 Biostrings             2.52.0    2019-05-02 [1]
 bit                    1.1-14    2018-05-29 [1]
 bit64                  0.9-7     2017-05-08 [1]
 bitops                 1.0-6     2013-08-17 [1]
 blob                   1.2.0     2019-07-09 [1]
 broom                * 0.5.2     2019-04-07 [1]
 BSgenome               1.52.0    2019-05-02 [1]
 callr                  3.3.2     2019-09-22 [1]
 cardelino            * 0.6.4     2019-08-21 [1]
 cellranger             1.1.0     2016-07-27 [1]
 cli                    1.1.0     2019-03-19 [1]
 colorspace             1.4-1     2019-03-18 [1]
 cowplot              * 1.0.0     2019-07-11 [1]
 crayon                 1.3.4     2017-09-16 [1]
 crosstalk              1.0.0     2016-12-21 [1]
 DBI                    1.0.0     2018-05-02 [1]
 DelayedArray         * 0.10.0    2019-05-02 [1]
 DelayedMatrixStats     1.6.1     2019-09-08 [1]
 desc                   1.2.0     2018-05-01 [1]
 devtools               2.2.1     2019-09-24 [1]
 digest                 0.6.21    2019-09-20 [1]
 dplyr                * 0.8.3     2019-07-04 [1]
 DT                     0.9       2019-09-17 [1]
 edgeR                * 3.26.8    2019-09-01 [1]
 ellipsis               0.3.0     2019-09-20 [1]
 evaluate               0.14      2019-05-28 [1]
 fansi                  0.4.0     2018-10-05 [1]
 farver                 1.1.0     2018-11-20 [1]
 fdrtool                1.2.15    2015-07-08 [1]
 forcats              * 0.4.0     2019-02-17 [1]
 fs                     1.3.1     2019-05-06 [1]
 gdtools              * 0.2.0     2019-09-03 [1]
 generics               0.0.2     2018-11-29 [1]
 GenomeInfoDb         * 1.20.0    2019-05-02 [1]
 GenomeInfoDbData       1.2.1     2019-04-30 [1]
 GenomicAlignments      1.20.1    2019-06-18 [1]
 GenomicFeatures        1.36.4    2019-07-10 [1]
 GenomicRanges        * 1.36.1    2019-09-06 [1]
 ggbeeswarm             0.6.0     2017-08-07 [1]
 ggcorrplot           * 0.1.3     2019-05-19 [1]
 ggforce              * 0.3.1     2019-08-20 [1]
 ggplot2              * 3.2.1     2019-08-10 [1]
 ggrepel              * 0.8.1     2019-05-07 [1]
 ggridges             * 0.5.1     2018-09-27 [1]
 ggthemes               4.2.0     2019-05-13 [1]
 ggtree                 1.16.6    2019-08-26 [1]
 git2r                  0.26.1    2019-06-29 [1]
 glue                   1.3.1     2019-03-12 [1]
 gridExtra              2.3       2017-09-09 [1]
 gtable                 0.3.0     2019-03-25 [1]
 haven                  2.1.1     2019-07-04 [1]
 hms                    0.5.1     2019-08-23 [1]
 htmltools              0.3.6     2017-04-28 [1]
 htmlwidgets            1.3       2018-09-30 [1]
 httpuv                 1.5.2     2019-09-11 [1]
 httr                   1.4.1     2019-08-05 [1]
 IHW                  * 1.12.0    2019-05-02 [1]
 IRanges              * 2.18.3    2019-09-24 [1]
 irlba                  2.3.3     2019-02-05 [1]
 jsonlite               1.6       2018-12-07 [1]
 knitr                  1.25      2019-09-18 [1]
 labeling               0.3       2014-08-23 [1]
 later                  0.8.0     2019-02-11 [1]
 lattice                0.20-38   2018-11-04 [4]
 lazyeval               0.2.2     2019-03-15 [1]
 lifecycle              0.1.0     2019-08-01 [1]
 limma                * 3.40.6    2019-07-26 [1]
 locfit                 1.5-9.1   2013-04-20 [1]
 lpsymphony             1.12.0    2019-05-02 [1]
 lubridate              1.7.4     2018-04-11 [1]
 magrittr               1.5       2014-11-22 [1]
 MASS                   7.3-51.1  2018-11-01 [4]
 Matrix                 1.2-17    2019-03-22 [1]
 matrixStats          * 0.55.0    2019-09-07 [1]
 memoise                1.1.0     2017-04-21 [1]
 mime                   0.7       2019-06-11 [1]
 modelr                 0.1.5     2019-08-08 [1]
 munsell                0.5.0     2018-06-12 [1]
 nlme                   3.1-139   2019-04-09 [4]
 org.Hs.eg.db         * 3.8.2     2019-05-01 [1]
 pheatmap               1.0.12    2019-01-04 [1]
 pillar                 1.4.2     2019-06-29 [1]
 pkgbuild               1.0.5     2019-08-26 [1]
 pkgconfig              2.0.3     2019-09-22 [1]
 pkgload                1.0.2     2018-10-29 [1]
 plyr                   1.8.4     2016-06-08 [1]
 polyclip               1.10-0    2019-03-14 [1]
 prettyunits            1.0.2     2015-07-13 [1]
 processx               3.4.1     2019-07-18 [1]
 progress               1.2.2     2019-05-16 [1]
 promises               1.0.1     2018-04-13 [1]
 ps                     1.3.0     2018-12-21 [1]
 purrr                * 0.3.2     2019-03-15 [1]
 R6                     2.4.0     2019-02-14 [1]
 RColorBrewer         * 1.1-2     2014-12-07 [1]
 Rcpp                   1.0.2     2019-07-25 [1]
 RCurl                  1.95-4.12 2019-03-04 [1]
 readr                * 1.3.1     2018-12-21 [1]
 readxl                 1.3.1     2019-03-13 [1]
 remotes                2.1.0     2019-06-24 [1]
 reshape2               1.4.3     2017-12-11 [1]
 rlang                * 0.4.0     2019-06-25 [1]
 rmarkdown              1.15      2019-08-21 [1]
 rprojroot              1.3-2     2018-01-03 [1]
 Rsamtools              2.0.1     2019-09-19 [1]
 RSQLite                2.1.2     2019-07-24 [1]
 rstudioapi             0.10      2019-03-19 [1]
 rsvd                   1.0.2     2019-07-29 [1]
 rtracklayer            1.44.4    2019-09-06 [1]
 rvcheck                0.1.3     2018-12-06 [1]
 rvest                  0.3.4     2019-05-15 [1]
 S4Vectors            * 0.22.1    2019-09-09 [1]
 scales                 1.0.0     2018-08-09 [1]
 scater               * 1.12.2    2019-05-24 [1]
 sessioninfo            1.1.1     2018-11-05 [1]
 shiny                  1.3.2     2019-04-22 [1]
 SingleCellExperiment * 1.6.0     2019-05-02 [1]
 slam                   0.1-45    2019-02-26 [1]
 snpStats               1.34.0    2019-05-02 [1]
 stringi                1.4.3     2019-03-12 [1]
 stringr              * 1.4.0     2019-02-10 [1]
 SummarizedExperiment * 1.14.1    2019-07-31 [1]
 superheat            * 0.1.0     2017-02-04 [1]
 survival               2.43-3    2018-11-26 [4]
 svglite                1.2.2     2019-05-17 [1]
 systemfonts            0.1.1     2019-07-01 [1]
 testthat               2.2.1     2019-07-25 [1]
 tibble               * 2.1.3     2019-06-06 [1]
 tidyr                * 1.0.0     2019-09-11 [1]
 tidyselect             0.2.5     2018-10-11 [1]
 tidytree               0.2.7     2019-09-12 [1]
 tidyverse            * 1.2.1     2017-11-14 [1]
 treeio                 1.8.2     2019-08-21 [1]
 tweenr                 1.0.1     2018-12-14 [1]
 usethis                1.5.1     2019-07-04 [1]
 utf8                   1.1.4     2018-05-24 [1]
 VariantAnnotation      1.30.1    2019-05-19 [1]
 vctrs                  0.2.0     2019-07-05 [1]
 vipor                  0.4.5     2017-03-22 [1]
 viridis              * 0.5.1     2018-03-29 [1]
 viridisLite          * 0.3.0     2018-02-01 [1]
 whisker                0.4       2019-08-28 [1]
 withr                  2.1.2     2018-03-15 [1]
 workflowr              1.4.0     2019-06-08 [1]
 xfun                   0.9       2019-08-21 [1]
 XML                    3.98-1.20 2019-06-06 [1]
 xml2                   1.2.2     2019-08-09 [1]
 xtable                 1.8-4     2019-04-21 [1]
 XVector                0.24.0    2019-05-02 [1]
 yaml                   2.2.0     2018-07-25 [1]
 zeallot                0.1.0     2018-01-28 [1]
 zlibbioc               1.30.0    2019-05-02 [1]
 source                          
 Bioconductor                    
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 Bioconductor                    
 Bioconductor                    
 Bioconductor                    
 Bioconductor                    
 Bioconductor                    
 Bioconductor                    
 Bioconductor                    
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 Bioconductor                    
 CRAN (R 3.6.0)                  
 Github (PMBio/cardelino@efd846f)
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 Bioconductor                    
 Bioconductor                    
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 Bioconductor                    
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 Bioconductor                    
 Bioconductor                    
 Bioconductor                    
 Bioconductor                    
 Bioconductor                    
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 Bioconductor                    
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 Bioconductor                    
 Bioconductor                    
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.5.1)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 Bioconductor                    
 CRAN (R 3.6.0)                  
 Bioconductor (R 3.6.0)          
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.5.1)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.5.3)                  
 Bioconductor                    
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 Bioconductor                    
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 Bioconductor                    
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 Bioconductor                    
 CRAN (R 3.6.0)                  
 Bioconductor                    
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 Bioconductor                    
 CRAN (R 3.6.0)                  
 Bioconductor                    
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 Bioconductor                    
 CRAN (R 3.6.0)                  
 CRAN (R 3.5.1)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 Bioconductor                    
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 Bioconductor                    
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 Bioconductor                    
 CRAN (R 3.6.0)                  
 CRAN (R 3.6.0)                  
 Bioconductor                    

[1] /home/AD.SVI.EDU.AU/dmccarthy/R/x86_64-pc-linux-gnu-library/3.6
[2] /usr/local/lib/R/site-library
[3] /usr/lib/R/site-library
[4] /usr/lib/R/library</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
