<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Raghd Rostom, Daniel J. Kunz &amp; Davis J. McCarthy" />


<title>Overview of lines</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fibroblast-clonality</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/davismcc/fibroblast-clonality">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Overview of lines</h1>
<h4 class="author">Raghd Rostom, Daniel J. Kunz &amp; Davis J. McCarthy</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-10-30
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>fibroblast-clonality/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20180807code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20180807)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20180807code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20180807)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomdavismccfibroblastclonalitytreecf8eb558a00bd2e5f605690acb9ecc94c4b40576targetblankcf8eb55a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/davismcc/fibroblast-clonality/tree/cf8eb558a00bd2e5f605690acb9ecc94c4b40576" target="_blank">cf8eb55</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomdavismccfibroblastclonalitytreecf8eb558a00bd2e5f605690acb9ecc94c4b40576targetblankcf8eb55a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .vscode/
    Ignored:    analysis/figure/
    Ignored:    code/.DS_Store
    Ignored:    code/selection/.DS_Store
    Ignored:    code/selection/.Rhistory
    Ignored:    code/selection/figures/
    Ignored:    data/.DS_Store
    Ignored:    logs/
    Ignored:    src/.DS_Store
    Ignored:    src/Rmd/.Rhistory

Untracked files:
    Untracked:  .dockerignore
    Untracked:  .dropbox
    Untracked:  .snakemake/
    Untracked:  Rplots.pdf
    Untracked:  Snakefile_clonality
    Untracked:  Snakefile_somatic_calling
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/assess_mutect2_fibro-ipsc_variant_calls.ipynb
    Untracked:  analysis/cardelino_fig1b.R
    Untracked:  analysis/cardelino_fig2b.R
    Untracked:  code/analysis_for_garx.Rmd
    Untracked:  code/selection/data/
    Untracked:  code/selection/fit-dist.nb
    Untracked:  code/selection/result-figure.R
    Untracked:  code/yuanhua/
    Untracked:  data/Melanoma-RegevGarraway-DFCI-scRNA-Seq/
    Untracked:  data/PRJNA485423/
    Untracked:  data/canopy/
    Untracked:  data/cell_assignment/
    Untracked:  data/cnv/
    Untracked:  data/de_analysis_FTv62/
    Untracked:  data/donor_info_070818.txt
    Untracked:  data/donor_info_core.csv
    Untracked:  data/donor_neutrality.tsv
    Untracked:  data/exome-point-mutations/
    Untracked:  data/fdr10.annot.txt.gz
    Untracked:  data/human_H_v5p2.rdata
    Untracked:  data/human_c2_v5p2.rdata
    Untracked:  data/human_c6_v5p2.rdata
    Untracked:  data/neg-bin-rsquared-petr.csv
    Untracked:  data/neutralitytestr-petr.tsv
    Untracked:  data/raw/
    Untracked:  data/sce_merged_donors_cardelino_donorid_all_qc_filt.rds
    Untracked:  data/sce_merged_donors_cardelino_donorid_all_with_qc_labels.rds
    Untracked:  data/sce_merged_donors_cardelino_donorid_unstim_qc_filt.rds
    Untracked:  data/sces/
    Untracked:  data/selection/
    Untracked:  data/simulations/
    Untracked:  data/variance_components/
    Untracked:  figures/
    Untracked:  output/differential_expression/
    Untracked:  output/differential_expression_cardelino-relax/
    Untracked:  output/donor_specific/
    Untracked:  output/line_info.tsv
    Untracked:  output/nvars_by_category_by_donor.tsv
    Untracked:  output/nvars_by_category_by_line.tsv
    Untracked:  output/variance_components/
    Untracked:  qolg_BIC.pdf
    Untracked:  references/
    Untracked:  reports/
    Untracked:  src/Rmd/DE_pathways_FTv62_callset_clones_pairwise_vs_base.unst_cells.carderelax.Rmd
    Untracked:  src/Rmd/Rplots.pdf
    Untracked:  src/Rmd/cell_assignment_cardelino-relax_template.Rmd
    Untracked:  tree.txt

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/cf8eb558a00bd2e5f605690acb9ecc94c4b40576/analysis/overview_lines.Rmd" target="_blank">cf8eb55</a>
</td>
<td>
Davis McCarthy
</td>
<td>
2019-10-30
</td>
<td>
Fixing bugs in overview_lines
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/ff037358ad0fe7d3252338c4b3a356ea37bc3bef/analysis/overview_lines.Rmd" target="_blank">ff03735</a>
</td>
<td>
Davis McCarthy
</td>
<td>
2019-10-30
</td>
<td>
Merge branch ‘master’ of <a href="https://github.com/davismcc/fibroblast-clonality" class="uri">https://github.com/davismcc/fibroblast-clonality</a>
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/550176f57f3f33670ae89659f72e541ad8e8ba41/analysis/overview_lines.Rmd" target="_blank">550176f</a>
</td>
<td>
Davis McCarthy
</td>
<td>
2019-10-30
</td>
<td>
Updating analysis to reflect accepted ms
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/davismcc/fibroblast-clonality/8729e022b6ba38c1b948735439f39be0977dcd35/docs/overview_lines.html" target="_blank">8729e02</a>
</td>
<td>
davismcc
</td>
<td>
2018-11-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/3c39fc25215efe119970ae77c4c8cc9493ceaab6/analysis/overview_lines.Rmd" target="_blank">3c39fc2</a>
</td>
<td>
d-j-k
</td>
<td>
2018-11-01
</td>
<td>
updated selection analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/davismcc/fibroblast-clonality/0540cdbc554ab43e0ff60557b7ceb82fb13f3d75/docs/overview_lines.html" target="_blank">0540cdb</a>
</td>
<td>
davismcc
</td>
<td>
2018-09-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/f5a463100f4795ec78443584fee5abb57019897c/analysis/overview_lines.Rmd" target="_blank">f5a4631</a>
</td>
<td>
davismcc
</td>
<td>
2018-09-02
</td>
<td>
Final plot tweaks
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/davismcc/fibroblast-clonality/f0ed980029a115234bc2edff312e5e52056d4eed/docs/overview_lines.html" target="_blank">f0ed980</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/1310c939b2a3803d152ab0c3f885b905e7286dd4/analysis/overview_lines.Rmd" target="_blank">1310c93</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-30
</td>
<td>
Tweaking plots
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/846dec4503c53c25d2e0b662ea3bfab57f083a26/analysis/overview_lines.Rmd" target="_blank">846dec4</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-29
</td>
<td>
Some small tweaks/additions to analyses
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/davismcc/fibroblast-clonality/ca3438f9274dbc7adf4aacc25b0fb017bc2d41fe/docs/overview_lines.html" target="_blank">ca3438f</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/dc78a9541488651a4b9bcda20e016a0b849d0b94/analysis/overview_lines.Rmd" target="_blank">dc78a95</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-29
</td>
<td>
Minor updates to analyses.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/davismcc/fibroblast-clonality/e573f2fcd4c0769acde01e24a2bb0c7fc3066ae9/docs/overview_lines.html" target="_blank">e573f2f</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/davismcc/fibroblast-clonality/9ec2a595a73ee48db72d6c6d860dc45fb0192a36/docs/overview_lines.html" target="_blank">9ec2a59</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/davismcc/fibroblast-clonality/36acf15b30f110282dd56b004c5d478d560b75e1/docs/overview_lines.html" target="_blank">36acf15</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/d618fe52b90fd1d6e4f9613c8d7bbe3a3afde245/analysis/overview_lines.Rmd" target="_blank">d618fe5</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-25
</td>
<td>
Updating analyses
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/davismcc/fibroblast-clonality/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/overview_lines.html" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/davismcc/fibroblast-clonality/8f884aeebf92a83e9b56ccfa6244bc26ec03d815/docs/overview_lines.html" target="_blank">8f884ae</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
<td>
Adding data pre-processing and line overview html files
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/5aa174b67899683cacd0e0e84870a1f824b46639/analysis/overview_lines.Rmd" target="_blank">5aa174b</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-20
</td>
<td>
Tidying up and saving dataframe to output file
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/fc582db56cff1220f322835fb1de45272be5d656/analysis/overview_lines.Rmd" target="_blank">fc582db</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-20
</td>
<td>
Fixing up small bug.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/3856e54eb1dafd6f968bea533cef4215ad4cebbe/analysis/overview_lines.Rmd" target="_blank">3856e54</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-20
</td>
<td>
Adding overview analysis of lines
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>This document provides overview information for 32 healthy human fibroblast cell lines used in this project. Note that each cell line was each derived from a distinct donor, so we use the terms “line” and “donor” interchangeably.</p>
<div id="load-libraries-and-data" class="section level2">
<h2>Load libraries and data</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">library</span>(readr)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw">library</span>(dplyr)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw">library</span>(scran)</a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="kw">library</span>(scater)</a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="kw">library</span>(viridis)</a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="kw">library</span>(ggplot2)</a>
<a class="sourceLine" id="cb1-7" data-line-number="7"><span class="kw">library</span>(ggforce)</a>
<a class="sourceLine" id="cb1-8" data-line-number="8"><span class="kw">library</span>(ggridges)</a>
<a class="sourceLine" id="cb1-9" data-line-number="9"><span class="kw">library</span>(SingleCellExperiment)</a>
<a class="sourceLine" id="cb1-10" data-line-number="10"><span class="kw">library</span>(edgeR)</a>
<a class="sourceLine" id="cb1-11" data-line-number="11"><span class="kw">library</span>(limma)</a>
<a class="sourceLine" id="cb1-12" data-line-number="12"><span class="kw">library</span>(org.Hs.eg.db)</a>
<a class="sourceLine" id="cb1-13" data-line-number="13"><span class="kw">library</span>(cowplot)</a>
<a class="sourceLine" id="cb1-14" data-line-number="14"><span class="kw">library</span>(gplots)</a>
<a class="sourceLine" id="cb1-15" data-line-number="15"><span class="kw">library</span>(ggrepel)</a>
<a class="sourceLine" id="cb1-16" data-line-number="16"><span class="kw">library</span>(sigfit)</a>
<a class="sourceLine" id="cb1-17" data-line-number="17"><span class="kw">library</span>(Rcpp)</a>
<a class="sourceLine" id="cb1-18" data-line-number="18"><span class="kw">library</span>(deconstructSigs)</a>
<a class="sourceLine" id="cb1-19" data-line-number="19"><span class="kw">options</span>(<span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)</a></code></pre></div>
<p>Load donor level information.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">donor_info &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">read_csv</span>(<span class="st">&quot;data/donor_info_core.csv&quot;</span>))</a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="co"># merge age bins</span></a>
<a class="sourceLine" id="cb2-3" data-line-number="3">donor_info<span class="op">$</span>age_decade &lt;-<span class="st"> &quot;&quot;</span></a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(donor_info)) {</a>
<a class="sourceLine" id="cb2-5" data-line-number="5">  <span class="cf">if</span> (donor_info<span class="op">$</span>age[i] <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;30-34&quot;</span>, <span class="st">&quot;35-39&quot;</span>)) </a>
<a class="sourceLine" id="cb2-6" data-line-number="6">    donor_info<span class="op">$</span>age_decade[i] &lt;-<span class="st"> &quot;30-39&quot;</span></a>
<a class="sourceLine" id="cb2-7" data-line-number="7">  <span class="cf">if</span> (donor_info<span class="op">$</span>age[i] <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;40-44&quot;</span>, <span class="st">&quot;45-49&quot;</span>)) </a>
<a class="sourceLine" id="cb2-8" data-line-number="8">    donor_info<span class="op">$</span>age_decade[i] &lt;-<span class="st"> &quot;40-49&quot;</span></a>
<a class="sourceLine" id="cb2-9" data-line-number="9">  <span class="cf">if</span> (donor_info<span class="op">$</span>age[i] <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;50-54&quot;</span>, <span class="st">&quot;55-59&quot;</span>)) </a>
<a class="sourceLine" id="cb2-10" data-line-number="10">    donor_info<span class="op">$</span>age_decade[i] &lt;-<span class="st"> &quot;50-59&quot;</span></a>
<a class="sourceLine" id="cb2-11" data-line-number="11">  <span class="cf">if</span> (donor_info<span class="op">$</span>age[i] <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;60-64&quot;</span>, <span class="st">&quot;65-69&quot;</span>)) </a>
<a class="sourceLine" id="cb2-12" data-line-number="12">    donor_info<span class="op">$</span>age_decade[i] &lt;-<span class="st"> &quot;60-69&quot;</span></a>
<a class="sourceLine" id="cb2-13" data-line-number="13">  <span class="cf">if</span> (donor_info<span class="op">$</span>age[i] <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;70-74&quot;</span>, <span class="st">&quot;75-79&quot;</span>)) </a>
<a class="sourceLine" id="cb2-14" data-line-number="14">    donor_info<span class="op">$</span>age_decade[i] &lt;-<span class="st"> &quot;70-79&quot;</span></a>
<a class="sourceLine" id="cb2-15" data-line-number="15">}</a></code></pre></div>
<p>Load exome variant sites.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">exome_sites &lt;-<span class="st"> </span><span class="kw">read_tsv</span>(<span class="st">&quot;data/exome-point-mutations/high-vs-low-exomes.v62.ft.filt_lenient-alldonors.txt.gz&quot;</span>,</a>
<a class="sourceLine" id="cb3-2" data-line-number="2">                        <span class="dt">col_types =</span> <span class="st">&quot;ciccdcciiiiccccccccddcdcll&quot;</span>, <span class="dt">comment =</span> <span class="st">&quot;#&quot;</span>,</a>
<a class="sourceLine" id="cb3-3" data-line-number="3">                        <span class="dt">col_names =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb3-4" data-line-number="4">exome_sites &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(</a>
<a class="sourceLine" id="cb3-5" data-line-number="5">  exome_sites,</a>
<a class="sourceLine" id="cb3-6" data-line-number="6">  <span class="dt">chrom =</span> <span class="kw">paste0</span>(<span class="st">&quot;chr&quot;</span>, <span class="kw">gsub</span>(<span class="st">&quot;chr&quot;</span>, <span class="st">&quot;&quot;</span>, chrom)),</a>
<a class="sourceLine" id="cb3-7" data-line-number="7">  <span class="dt">var_id =</span> <span class="kw">paste0</span>(chrom, <span class="st">&quot;:&quot;</span>, pos, <span class="st">&quot;_&quot;</span>, ref, <span class="st">&quot;_&quot;</span>, alt),</a>
<a class="sourceLine" id="cb3-8" data-line-number="8">  <span class="dt">chr_pos =</span> <span class="kw">paste0</span>(chrom, <span class="st">&quot;_&quot;</span>, pos))</a>
<a class="sourceLine" id="cb3-9" data-line-number="9"></a>
<a class="sourceLine" id="cb3-10" data-line-number="10">exome_sites &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(exome_sites)</a>
<a class="sourceLine" id="cb3-11" data-line-number="11"><span class="co">## deduplicate sites list</span></a>
<a class="sourceLine" id="cb3-12" data-line-number="12">exome_sites &lt;-<span class="st"> </span>exome_sites[<span class="op">!</span><span class="kw">duplicated</span>(exome_sites[[<span class="st">&quot;var_id&quot;</span>]]),]</a>
<a class="sourceLine" id="cb3-13" data-line-number="13"><span class="co">## calculate coverage at sites for each donor</span></a>
<a class="sourceLine" id="cb3-14" data-line-number="14">donor_vars_coverage &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb3-15" data-line-number="15"><span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">unique</span>(exome_sites<span class="op">$</span>donor_short_id)) {</a>
<a class="sourceLine" id="cb3-16" data-line-number="16">  exome_sites_subset &lt;-<span class="st"> </span>exome_sites[exome_sites<span class="op">$</span>donor_short_id <span class="op">==</span><span class="st"> </span>i, ]</a>
<a class="sourceLine" id="cb3-17" data-line-number="17">  donor_vars_coverage[[i]] &lt;-<span class="st"> </span>exome_sites_subset<span class="op">$</span>nREF_fibro <span class="op">+</span><span class="st"> </span>exome_sites_subset<span class="op">$</span>nALT_fibro</a>
<a class="sourceLine" id="cb3-18" data-line-number="18">}</a></code></pre></div>
<p>Load VEP annotations and show table with number of variants assigned to each functional annotation category.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">vep_best &lt;-<span class="st"> </span><span class="kw">read_tsv</span>(<span class="st">&quot;data/exome-point-mutations/high-vs-low-exomes.v62.ft.alldonors-filt_lenient.all_filt_sites.vep_most_severe_csq.txt&quot;</span>)</a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="kw">colnames</span>(vep_best)[<span class="dv">1</span>] &lt;-<span class="st"> &quot;Uploaded_variation&quot;</span></a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="co">## deduplicate dataframe</span></a>
<a class="sourceLine" id="cb4-4" data-line-number="4">vep_best &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(vep_best[<span class="op">!</span><span class="kw">duplicated</span>(vep_best[[<span class="st">&quot;Uploaded_variation&quot;</span>]]),])</a>
<a class="sourceLine" id="cb4-5" data-line-number="5"><span class="kw">as.data.frame</span>(<span class="kw">table</span>(vep_best[[<span class="st">&quot;Consequence&quot;</span>]]))</a></code></pre></div>
<pre><code>                                 Var1 Freq
1                 3_prime_UTR_variant  181
2                 5_prime_UTR_variant  121
3             downstream_gene_variant   13
4                  intergenic_variant   13
5                      intron_variant 1539
6                mature_miRNA_variant    3
7                    missense_variant 3648
8  non_coding_transcript_exon_variant  428
9           regulatory_region_variant    1
10            splice_acceptor_variant   41
11               splice_donor_variant   24
12              splice_region_variant  291
13                         start_lost    9
14                        stop_gained  227
15                          stop_lost    2
16              stop_retained_variant    5
17                 synonymous_variant 1923
18              upstream_gene_variant   34</code></pre>
<p>Add consequences to exome sites.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">vep_best[[<span class="st">&quot;var_id&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;chr&quot;</span>, vep_best[[<span class="st">&quot;Uploaded_variation&quot;</span>]])</a>
<a class="sourceLine" id="cb6-2" data-line-number="2">exome_sites &lt;-<span class="st"> </span><span class="kw">inner_join</span>(exome_sites, </a>
<a class="sourceLine" id="cb6-3" data-line-number="3">                          vep_best[, <span class="kw">c</span>(<span class="st">&quot;var_id&quot;</span>, <span class="st">&quot;Location&quot;</span>, <span class="st">&quot;Consequence&quot;</span>)], </a>
<a class="sourceLine" id="cb6-4" data-line-number="4">                          <span class="dt">by =</span> <span class="st">&quot;var_id&quot;</span>)</a></code></pre></div>
<p>Add donor level mutation information (aggregate across impacts) Not used in manuscript, but still calculated to store in <code>donor_info</code> table.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">impactful_csq &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;stop_lost&quot;</span>, <span class="st">&quot;start_lost&quot;</span>, <span class="st">&quot;stop_gained&quot;</span>,</a>
<a class="sourceLine" id="cb7-2" data-line-number="2">                   <span class="st">&quot;splice_donor_variant&quot;</span>, <span class="st">&quot;splice_acceptor_variant&quot;</span>,</a>
<a class="sourceLine" id="cb7-3" data-line-number="3">                   <span class="st">&quot;splice_region_variant&quot;</span>, <span class="st">&quot;missense_variant&quot;</span>)</a>
<a class="sourceLine" id="cb7-4" data-line-number="4">donor_info<span class="op">$</span>num_mutations &lt;-<span class="st"> </span><span class="ot">NA</span></a>
<a class="sourceLine" id="cb7-5" data-line-number="5">donor_info<span class="op">$</span>num_synonymous &lt;-<span class="st"> </span><span class="ot">NA</span></a>
<a class="sourceLine" id="cb7-6" data-line-number="6">donor_info<span class="op">$</span>num_missense &lt;-<span class="st"> </span><span class="ot">NA</span></a>
<a class="sourceLine" id="cb7-7" data-line-number="7">donor_info<span class="op">$</span>num_splice_region &lt;-<span class="st"> </span><span class="ot">NA</span></a>
<a class="sourceLine" id="cb7-8" data-line-number="8">donor_info<span class="op">$</span>num_splice_acceptor &lt;-<span class="st"> </span><span class="ot">NA</span></a>
<a class="sourceLine" id="cb7-9" data-line-number="9">donor_info<span class="op">$</span>num_splice_donor &lt;-<span class="st"> </span><span class="ot">NA</span></a>
<a class="sourceLine" id="cb7-10" data-line-number="10">donor_info<span class="op">$</span>num_stop_gained &lt;-<span class="st"> </span><span class="ot">NA</span></a>
<a class="sourceLine" id="cb7-11" data-line-number="11">donor_info<span class="op">$</span>num_start_lost &lt;-<span class="st"> </span><span class="ot">NA</span></a>
<a class="sourceLine" id="cb7-12" data-line-number="12">donor_info<span class="op">$</span>num_stop_lost &lt;-<span class="st"> </span><span class="ot">NA</span></a>
<a class="sourceLine" id="cb7-13" data-line-number="13"><span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">unique</span>(donor_info<span class="op">$</span>donor_short)) {</a>
<a class="sourceLine" id="cb7-14" data-line-number="14">  <span class="cf">if</span> (i <span class="op">%in%</span><span class="st"> </span><span class="kw">unique</span>(exome_sites<span class="op">$</span>donor_short_id)) {</a>
<a class="sourceLine" id="cb7-15" data-line-number="15">    exome_sites_subset &lt;-<span class="st"> </span>exome_sites[exome_sites<span class="op">$</span>donor_short_id <span class="op">==</span><span class="st"> </span>i, ]</a>
<a class="sourceLine" id="cb7-16" data-line-number="16">    donor_info<span class="op">$</span>num_mutations[donor_info<span class="op">$</span>donor_short <span class="op">==</span><span class="st"> </span>i] &lt;-<span class="st"> </span><span class="kw">length</span>(exome_sites_subset<span class="op">$</span>Consequence)</a>
<a class="sourceLine" id="cb7-17" data-line-number="17">    donor_info<span class="op">$</span>num_synonymous[donor_info<span class="op">$</span>donor_short <span class="op">==</span><span class="st"> </span>i] &lt;-<span class="st"> </span><span class="kw">sum</span>(exome_sites_subset<span class="op">$</span>Consequence <span class="op">==</span><span class="st"> &quot;synonymous_variant&quot;</span>)</a>
<a class="sourceLine" id="cb7-18" data-line-number="18">    donor_info<span class="op">$</span>num_missense[donor_info<span class="op">$</span>donor_short <span class="op">==</span><span class="st"> </span>i] &lt;-<span class="st"> </span><span class="kw">sum</span>(exome_sites_subset<span class="op">$</span>Consequence <span class="op">==</span><span class="st"> &quot;missense_variant&quot;</span>)</a>
<a class="sourceLine" id="cb7-19" data-line-number="19">    donor_info<span class="op">$</span>num_splice_region[donor_info<span class="op">$</span>donor_short <span class="op">==</span><span class="st"> </span>i] &lt;-<span class="st"> </span><span class="kw">sum</span>(exome_sites_subset<span class="op">$</span>Consequence <span class="op">==</span><span class="st"> &quot;splice_region_variant&quot;</span>)</a>
<a class="sourceLine" id="cb7-20" data-line-number="20">    donor_info<span class="op">$</span>num_splice_acceptor[donor_info<span class="op">$</span>donor_short <span class="op">==</span><span class="st"> </span>i] &lt;-<span class="st"> </span><span class="kw">sum</span>(exome_sites_subset<span class="op">$</span>Consequence <span class="op">==</span><span class="st"> &quot;splice_acceptor_variant&quot;</span>)</a>
<a class="sourceLine" id="cb7-21" data-line-number="21">    donor_info<span class="op">$</span>num_splice_donor[donor_info<span class="op">$</span>donor_short <span class="op">==</span><span class="st"> </span>i] &lt;-<span class="st"> </span><span class="kw">sum</span>(exome_sites_subset<span class="op">$</span>Consequence <span class="op">==</span><span class="st"> &quot;splice_donor_variant&quot;</span>)</a>
<a class="sourceLine" id="cb7-22" data-line-number="22">    donor_info<span class="op">$</span>num_stop_gained[donor_info<span class="op">$</span>donor_short <span class="op">==</span><span class="st"> </span>i] &lt;-<span class="st"> </span><span class="kw">sum</span>(exome_sites_subset<span class="op">$</span>Consequence <span class="op">==</span><span class="st"> &quot;stop_gained&quot;</span>)</a>
<a class="sourceLine" id="cb7-23" data-line-number="23">    donor_info<span class="op">$</span>num_start_lost[donor_info<span class="op">$</span>donor_short <span class="op">==</span><span class="st"> </span>i] &lt;-<span class="st"> </span><span class="kw">sum</span>(exome_sites_subset<span class="op">$</span>Consequence <span class="op">==</span><span class="st"> &quot;start_lost&quot;</span>)</a>
<a class="sourceLine" id="cb7-24" data-line-number="24">    donor_info<span class="op">$</span>num_stop_lost[donor_info<span class="op">$</span>donor_short <span class="op">==</span><span class="st"> </span>i] &lt;-<span class="st"> </span><span class="kw">sum</span>(exome_sites_subset<span class="op">$</span>Consequence <span class="op">==</span><span class="st"> &quot;stop_lost&quot;</span>)</a>
<a class="sourceLine" id="cb7-25" data-line-number="25">  }</a>
<a class="sourceLine" id="cb7-26" data-line-number="26">}</a></code></pre></div>
</div>
<div id="mutational-signatures" class="section level2">
<h2>Mutational signatures</h2>
<p>Add donor level mutation signature exposures, using the <code>sigfit</code> package and 30 COSMIC signatures. We load the filtered exome variant sites and calculate the tri-nucleotide context for each variant (required for computing signature exposures), using a function from the <code>deconstructSigs</code> package.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="kw">data</span>(<span class="st">&quot;cosmic_signatures&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;sigfit&quot;</span>)</a>
<a class="sourceLine" id="cb8-2" data-line-number="2"></a>
<a class="sourceLine" id="cb8-3" data-line-number="3"><span class="co">##new input </span></a>
<a class="sourceLine" id="cb8-4" data-line-number="4">mutation_list &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;data/exome-point-mutations/high-vs-low-exomes.v62.ft.filt_lenient-alldonors.txt.gz&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb8-5" data-line-number="5">mutation_list<span class="op">$</span>chr_pos &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;chr&quot;</span>, mutation_list<span class="op">$</span>chr, <span class="st">&quot;_&quot;</span>, mutation_list<span class="op">$</span>pos)</a>
<a class="sourceLine" id="cb8-6" data-line-number="6"></a>
<a class="sourceLine" id="cb8-7" data-line-number="7">mutation_donors &lt;-<span class="st"> </span><span class="kw">unique</span>(mutation_list<span class="op">$</span>donor_short_id)</a>
<a class="sourceLine" id="cb8-8" data-line-number="8">mutation_list_donors &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb8-9" data-line-number="9"><span class="cf">for</span> (i <span class="cf">in</span> mutation_donors) {</a>
<a class="sourceLine" id="cb8-10" data-line-number="10">  <span class="kw">cat</span>(<span class="st">&quot;....reading &quot;</span>, i, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)</a>
<a class="sourceLine" id="cb8-11" data-line-number="11">  mutation_list_donors[[i]] &lt;-<span class="st"> </span>mutation_list[<span class="kw">which</span>(mutation_list<span class="op">$</span>donor_short_id <span class="op">==</span><span class="st"> </span>i),]</a>
<a class="sourceLine" id="cb8-12" data-line-number="12">  mutation_list_donors[[i]]<span class="op">$</span>chr &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;chr&quot;</span>, mutation_list_donors[[i]]<span class="op">$</span>chrom)</a>
<a class="sourceLine" id="cb8-13" data-line-number="13">  mutation_list_donors[[i]]<span class="op">$</span>chr_pos =<span class="st"> </span><span class="kw">paste0</span>(mutation_list_donors[[i]]<span class="op">$</span>chr, <span class="st">&quot;_&quot;</span>, mutation_list_donors[[i]]<span class="op">$</span>pos)</a>
<a class="sourceLine" id="cb8-14" data-line-number="14">}</a>
<a class="sourceLine" id="cb8-15" data-line-number="15"></a>
<a class="sourceLine" id="cb8-16" data-line-number="16"><span class="co">## Calculate triNucleotide contexts for mutations using deconstructSigs command</span></a>
<a class="sourceLine" id="cb8-17" data-line-number="17">mut_triNs &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb8-18" data-line-number="18"><span class="cf">for</span> (i <span class="cf">in</span> mutation_donors) {</a>
<a class="sourceLine" id="cb8-19" data-line-number="19">  <span class="kw">cat</span>(<span class="st">&quot;....processing &quot;</span>, i, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)</a>
<a class="sourceLine" id="cb8-20" data-line-number="20">  mut_triNs[[i]] &lt;-<span class="st"> </span><span class="kw">mut.to.sigs.input</span>(mutation_list_donors[[i]], <span class="dt">sample.id =</span> <span class="st">&quot;donor_short_id&quot;</span>, </a>
<a class="sourceLine" id="cb8-21" data-line-number="21">                                      <span class="dt">chr =</span> <span class="st">&quot;chr&quot;</span>, <span class="dt">pos =</span> <span class="st">&quot;pos&quot;</span>, <span class="dt">ref =</span> <span class="st">&quot;ref&quot;</span>, <span class="dt">alt =</span> <span class="st">&quot;alt&quot;</span>)</a>
<a class="sourceLine" id="cb8-22" data-line-number="22">}</a>
<a class="sourceLine" id="cb8-23" data-line-number="23"></a>
<a class="sourceLine" id="cb8-24" data-line-number="24"><span class="co">## Convert to correct format</span></a>
<a class="sourceLine" id="cb8-25" data-line-number="25">sig_triNs &lt;-<span class="st"> </span><span class="kw">character</span>()</a>
<a class="sourceLine" id="cb8-26" data-line-number="26"><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="dv">96</span>) {</a>
<a class="sourceLine" id="cb8-27" data-line-number="27">  c1 &lt;-<span class="st"> </span><span class="kw">substr</span>(<span class="kw">colnames</span>(mut_triNs[[<span class="dv">1</span>]])[j], <span class="dv">1</span>,<span class="dv">1</span>)</a>
<a class="sourceLine" id="cb8-28" data-line-number="28">  ref &lt;-<span class="st"> </span><span class="kw">substr</span>(<span class="kw">colnames</span>(mut_triNs[[<span class="dv">1</span>]])[j], <span class="dv">3</span>,<span class="dv">3</span>)</a>
<a class="sourceLine" id="cb8-29" data-line-number="29">  alt &lt;-<span class="st"> </span><span class="kw">substr</span>(<span class="kw">colnames</span>(mut_triNs[[<span class="dv">1</span>]])[j], <span class="dv">5</span>,<span class="dv">5</span>)</a>
<a class="sourceLine" id="cb8-30" data-line-number="30">  c3 &lt;-<span class="st"> </span><span class="kw">substr</span>(<span class="kw">colnames</span>(mut_triNs[[<span class="dv">1</span>]])[j], <span class="dv">7</span>,<span class="dv">7</span>)</a>
<a class="sourceLine" id="cb8-31" data-line-number="31">  triN_sigfit &lt;-<span class="st"> </span><span class="kw">paste0</span>(c1,ref,c3,<span class="st">&quot;&gt;&quot;</span>,c1,alt,c3)</a>
<a class="sourceLine" id="cb8-32" data-line-number="32">  sig_triNs[j] &lt;-<span class="st"> </span>triN_sigfit</a>
<a class="sourceLine" id="cb8-33" data-line-number="33">}</a>
<a class="sourceLine" id="cb8-34" data-line-number="34"></a>
<a class="sourceLine" id="cb8-35" data-line-number="35"><span class="cf">for</span> (i <span class="cf">in</span> mutation_donors) {</a>
<a class="sourceLine" id="cb8-36" data-line-number="36">  <span class="kw">colnames</span>(mut_triNs[[i]]) &lt;-<span class="st"> </span>sig_triNs</a>
<a class="sourceLine" id="cb8-37" data-line-number="37">}</a>
<a class="sourceLine" id="cb8-38" data-line-number="38"></a>
<a class="sourceLine" id="cb8-39" data-line-number="39"><span class="co">## Fit signatures using sigfit</span></a>
<a class="sourceLine" id="cb8-40" data-line-number="40">mcmc_samples_fit &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb8-41" data-line-number="41"><span class="kw">set.seed</span>(<span class="dv">1234</span>)</a>
<a class="sourceLine" id="cb8-42" data-line-number="42"><span class="cf">for</span> (i <span class="cf">in</span> mutation_donors) {</a>
<a class="sourceLine" id="cb8-43" data-line-number="43">  mcmc_samples_fit[[i]] &lt;-<span class="st"> </span>sigfit<span class="op">::</span><span class="kw">fit_signatures</span>(</a>
<a class="sourceLine" id="cb8-44" data-line-number="44">    <span class="dt">counts =</span> mut_triNs[[i]], <span class="dt">signatures =</span> cosmic_signatures,</a>
<a class="sourceLine" id="cb8-45" data-line-number="45">    <span class="dt">iter =</span> <span class="dv">2000</span>, <span class="dt">warmup =</span> <span class="dv">1000</span>, <span class="dt">chains =</span> <span class="dv">1</span>, <span class="dt">seed =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb8-46" data-line-number="46">}</a>
<a class="sourceLine" id="cb8-47" data-line-number="47"></a>
<a class="sourceLine" id="cb8-48" data-line-number="48"><span class="co">## Estimate exposures using sigfit</span></a>
<a class="sourceLine" id="cb8-49" data-line-number="49">exposures &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb8-50" data-line-number="50"><span class="cf">for</span> (i <span class="cf">in</span> mutation_donors) {</a>
<a class="sourceLine" id="cb8-51" data-line-number="51">  exposures[[i]] &lt;-<span class="st"> </span>sigfit<span class="op">::</span><span class="kw">retrieve_pars</span>(</a>
<a class="sourceLine" id="cb8-52" data-line-number="52">    mcmc_samples_fit[[i]], <span class="dt">par =</span> <span class="st">&quot;exposures&quot;</span>, <span class="dt">hpd_prob =</span> <span class="fl">0.90</span>)</a>
<a class="sourceLine" id="cb8-53" data-line-number="53">}</a></code></pre></div>
<p>Plot an exposure barchart for each line.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="co">## Plot exposure bar charts</span></a>
<a class="sourceLine" id="cb9-2" data-line-number="2">donors &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;euts&quot;</span>, <span class="st">&quot;fawm&quot;</span>, <span class="st">&quot;feec&quot;</span>, <span class="st">&quot;fikt&quot;</span>, <span class="st">&quot;garx&quot;</span>, <span class="st">&quot;gesg&quot;</span>, <span class="st">&quot;heja&quot;</span>, <span class="st">&quot;hipn&quot;</span>, </a>
<a class="sourceLine" id="cb9-3" data-line-number="3">            <span class="st">&quot;ieki&quot;</span>, <span class="st">&quot;joxm&quot;</span>, <span class="st">&quot;kuco&quot;</span>, <span class="st">&quot;laey&quot;</span>, <span class="st">&quot;lexy&quot;</span>, <span class="st">&quot;naju&quot;</span>, <span class="st">&quot;nusw&quot;</span>, <span class="st">&quot;oaaz&quot;</span>, </a>
<a class="sourceLine" id="cb9-4" data-line-number="4">            <span class="st">&quot;oilg&quot;</span>, <span class="st">&quot;pipw&quot;</span>, <span class="st">&quot;puie&quot;</span>, <span class="st">&quot;qayj&quot;</span>, <span class="st">&quot;qolg&quot;</span>, <span class="st">&quot;qonc&quot;</span>, <span class="st">&quot;rozh&quot;</span>, <span class="st">&quot;sehl&quot;</span>, </a>
<a class="sourceLine" id="cb9-5" data-line-number="5">            <span class="st">&quot;ualf&quot;</span>, <span class="st">&quot;vass&quot;</span>, <span class="st">&quot;vils&quot;</span>, <span class="st">&quot;vuna&quot;</span>, <span class="st">&quot;wahn&quot;</span>, <span class="st">&quot;wetu&quot;</span>, <span class="st">&quot;xugn&quot;</span>, <span class="st">&quot;zoxy&quot;</span>)</a>
<a class="sourceLine" id="cb9-6" data-line-number="6"></a>
<a class="sourceLine" id="cb9-7" data-line-number="7">signature_names &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;1&quot;</span>, <span class="st">&quot;2&quot;</span>, <span class="st">&quot;3&quot;</span>, <span class="st">&quot;4&quot;</span>, <span class="st">&quot;5&quot;</span>, <span class="st">&quot;6&quot;</span>, <span class="st">&quot;7&quot;</span>, <span class="st">&quot;8&quot;</span>, <span class="st">&quot;9&quot;</span>, <span class="st">&quot;10&quot;</span>, <span class="st">&quot;11&quot;</span>, </a>
<a class="sourceLine" id="cb9-8" data-line-number="8">                     <span class="st">&quot;12&quot;</span>, <span class="st">&quot;13&quot;</span>, <span class="st">&quot;14&quot;</span>, <span class="st">&quot;15&quot;</span>, <span class="st">&quot;16&quot;</span>, <span class="st">&quot;17&quot;</span>, <span class="st">&quot;18&quot;</span>, <span class="st">&quot;19&quot;</span>, <span class="st">&quot;20&quot;</span>, <span class="st">&quot;21&quot;</span>, </a>
<a class="sourceLine" id="cb9-9" data-line-number="9">                     <span class="st">&quot;22&quot;</span>, <span class="st">&quot;23&quot;</span>, <span class="st">&quot;24&quot;</span>, <span class="st">&quot;25&quot;</span>, <span class="st">&quot;26&quot;</span>, <span class="st">&quot;27&quot;</span>, <span class="st">&quot;28&quot;</span>, <span class="st">&quot;29&quot;</span>, <span class="st">&quot;30&quot;</span>)</a>
<a class="sourceLine" id="cb9-10" data-line-number="10"></a>
<a class="sourceLine" id="cb9-11" data-line-number="11"><span class="cf">for</span> (j <span class="cf">in</span> donors) {</a>
<a class="sourceLine" id="cb9-12" data-line-number="12">  <span class="kw">cat</span>(<span class="st">&quot;....plotting &quot;</span>, j, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)</a>
<a class="sourceLine" id="cb9-13" data-line-number="13">   sigfit<span class="op">::</span><span class="kw">plot_exposures</span>(mcmc_samples_fit[[j]], </a>
<a class="sourceLine" id="cb9-14" data-line-number="14">                          <span class="dt">signature_names =</span> signature_names)</a>
<a class="sourceLine" id="cb9-15" data-line-number="15">   <span class="kw">png</span>(<span class="kw">paste0</span>(<span class="st">&quot;figures/overview_lines/mutational_signatures/exposure_barchart_&quot;</span>, </a>
<a class="sourceLine" id="cb9-16" data-line-number="16">             j, <span class="st">&quot;.png&quot;</span>),</a>
<a class="sourceLine" id="cb9-17" data-line-number="17">      <span class="dt">units =</span> <span class="st">&quot;in&quot;</span>, <span class="dt">width =</span> <span class="dv">12</span>, <span class="dt">height =</span> <span class="dv">10</span>, <span class="dt">res =</span> <span class="dv">500</span>)</a>
<a class="sourceLine" id="cb9-18" data-line-number="18">  sigfit<span class="op">::</span><span class="kw">plot_exposures</span>(mcmc_samples_fit[[j]], </a>
<a class="sourceLine" id="cb9-19" data-line-number="19">                         <span class="dt">signature_names =</span> signature_names)</a>
<a class="sourceLine" id="cb9-20" data-line-number="20">  <span class="kw">dev.off</span>()</a>
<a class="sourceLine" id="cb9-21" data-line-number="21"> </a>
<a class="sourceLine" id="cb9-22" data-line-number="22">}</a></code></pre></div>
<pre><code>....plotting  euts </code></pre>
<p><img src="figure/overview_lines.Rmd/plot-exposures-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-exposures-1">
Past versions of plot-exposures-1.png
</button>
</p>
<div id="fig-plot-exposures-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/figure/overview_lines.Rmd/plot-exposures-1.png" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>....plotting  fawm </code></pre>
<p><img src="figure/overview_lines.Rmd/plot-exposures-2.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-exposures-2">
Past versions of plot-exposures-2.png
</button>
</p>
<div id="fig-plot-exposures-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/figure/overview_lines.Rmd/plot-exposures-2.png" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>....plotting  feec </code></pre>
<p><img src="figure/overview_lines.Rmd/plot-exposures-3.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-exposures-3">
Past versions of plot-exposures-3.png
</button>
</p>
<div id="fig-plot-exposures-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/figure/overview_lines.Rmd/plot-exposures-3.png" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>....plotting  fikt </code></pre>
<p><img src="figure/overview_lines.Rmd/plot-exposures-4.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-exposures-4">
Past versions of plot-exposures-4.png
</button>
</p>
<div id="fig-plot-exposures-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/figure/overview_lines.Rmd/plot-exposures-4.png" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>....plotting  garx </code></pre>
<p><img src="figure/overview_lines.Rmd/plot-exposures-5.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-exposures-5">
Past versions of plot-exposures-5.png
</button>
</p>
<div id="fig-plot-exposures-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/figure/overview_lines.Rmd/plot-exposures-5.png" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>....plotting  gesg </code></pre>
<p><img src="figure/overview_lines.Rmd/plot-exposures-6.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-exposures-6">
Past versions of plot-exposures-6.png
</button>
</p>
<div id="fig-plot-exposures-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/figure/overview_lines.Rmd/plot-exposures-6.png" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>....plotting  heja </code></pre>
<p><img src="figure/overview_lines.Rmd/plot-exposures-7.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-exposures-7">
Past versions of plot-exposures-7.png
</button>
</p>
<div id="fig-plot-exposures-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/figure/overview_lines.Rmd/plot-exposures-7.png" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>....plotting  hipn </code></pre>
<p><img src="figure/overview_lines.Rmd/plot-exposures-8.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-exposures-8">
Past versions of plot-exposures-8.png
</button>
</p>
<div id="fig-plot-exposures-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/figure/overview_lines.Rmd/plot-exposures-8.png" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>....plotting  ieki </code></pre>
<p><img src="figure/overview_lines.Rmd/plot-exposures-9.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-exposures-9">
Past versions of plot-exposures-9.png
</button>
</p>
<div id="fig-plot-exposures-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/figure/overview_lines.Rmd/plot-exposures-9.png" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>....plotting  joxm </code></pre>
<p><img src="figure/overview_lines.Rmd/plot-exposures-10.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-exposures-10">
Past versions of plot-exposures-10.png
</button>
</p>
<div id="fig-plot-exposures-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/figure/overview_lines.Rmd/plot-exposures-10.png" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>....plotting  kuco </code></pre>
<p><img src="figure/overview_lines.Rmd/plot-exposures-11.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-exposures-11">
Past versions of plot-exposures-11.png
</button>
</p>
<div id="fig-plot-exposures-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/figure/overview_lines.Rmd/plot-exposures-11.png" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>....plotting  laey </code></pre>
<p><img src="figure/overview_lines.Rmd/plot-exposures-12.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-exposures-12">
Past versions of plot-exposures-12.png
</button>
</p>
<div id="fig-plot-exposures-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/figure/overview_lines.Rmd/plot-exposures-12.png" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>....plotting  lexy </code></pre>
<p><img src="figure/overview_lines.Rmd/plot-exposures-13.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-exposures-13">
Past versions of plot-exposures-13.png
</button>
</p>
<div id="fig-plot-exposures-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/figure/overview_lines.Rmd/plot-exposures-13.png" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>....plotting  naju </code></pre>
<p><img src="figure/overview_lines.Rmd/plot-exposures-14.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-exposures-14">
Past versions of plot-exposures-14.png
</button>
</p>
<div id="fig-plot-exposures-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/figure/overview_lines.Rmd/plot-exposures-14.png" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>....plotting  nusw </code></pre>
<p><img src="figure/overview_lines.Rmd/plot-exposures-15.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-exposures-15">
Past versions of plot-exposures-15.png
</button>
</p>
<div id="fig-plot-exposures-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/figure/overview_lines.Rmd/plot-exposures-15.png" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>....plotting  oaaz </code></pre>
<p><img src="figure/overview_lines.Rmd/plot-exposures-16.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-exposures-16">
Past versions of plot-exposures-16.png
</button>
</p>
<div id="fig-plot-exposures-16" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/figure/overview_lines.Rmd/plot-exposures-16.png" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>....plotting  oilg </code></pre>
<p><img src="figure/overview_lines.Rmd/plot-exposures-17.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-exposures-17">
Past versions of plot-exposures-17.png
</button>
</p>
<div id="fig-plot-exposures-17" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/figure/overview_lines.Rmd/plot-exposures-17.png" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>....plotting  pipw </code></pre>
<p><img src="figure/overview_lines.Rmd/plot-exposures-18.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-exposures-18">
Past versions of plot-exposures-18.png
</button>
</p>
<div id="fig-plot-exposures-18" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/figure/overview_lines.Rmd/plot-exposures-18.png" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>....plotting  puie </code></pre>
<p><img src="figure/overview_lines.Rmd/plot-exposures-19.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-exposures-19">
Past versions of plot-exposures-19.png
</button>
</p>
<div id="fig-plot-exposures-19" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/figure/overview_lines.Rmd/plot-exposures-19.png" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>....plotting  qayj </code></pre>
<p><img src="figure/overview_lines.Rmd/plot-exposures-20.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-exposures-20">
Past versions of plot-exposures-20.png
</button>
</p>
<div id="fig-plot-exposures-20" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/figure/overview_lines.Rmd/plot-exposures-20.png" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>....plotting  qolg </code></pre>
<p><img src="figure/overview_lines.Rmd/plot-exposures-21.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-exposures-21">
Past versions of plot-exposures-21.png
</button>
</p>
<div id="fig-plot-exposures-21" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/figure/overview_lines.Rmd/plot-exposures-21.png" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>....plotting  qonc </code></pre>
<p><img src="figure/overview_lines.Rmd/plot-exposures-22.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-exposures-22">
Past versions of plot-exposures-22.png
</button>
</p>
<div id="fig-plot-exposures-22" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/figure/overview_lines.Rmd/plot-exposures-22.png" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>....plotting  rozh </code></pre>
<p><img src="figure/overview_lines.Rmd/plot-exposures-23.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-exposures-23">
Past versions of plot-exposures-23.png
</button>
</p>
<div id="fig-plot-exposures-23" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/figure/overview_lines.Rmd/plot-exposures-23.png" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>....plotting  sehl </code></pre>
<p><img src="figure/overview_lines.Rmd/plot-exposures-24.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-exposures-24">
Past versions of plot-exposures-24.png
</button>
</p>
<div id="fig-plot-exposures-24" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/figure/overview_lines.Rmd/plot-exposures-24.png" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>....plotting  ualf </code></pre>
<p><img src="figure/overview_lines.Rmd/plot-exposures-25.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-exposures-25">
Past versions of plot-exposures-25.png
</button>
</p>
<div id="fig-plot-exposures-25" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/figure/overview_lines.Rmd/plot-exposures-25.png" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>....plotting  vass </code></pre>
<p><img src="figure/overview_lines.Rmd/plot-exposures-26.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-exposures-26">
Past versions of plot-exposures-26.png
</button>
</p>
<div id="fig-plot-exposures-26" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/figure/overview_lines.Rmd/plot-exposures-26.png" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>....plotting  vils </code></pre>
<p><img src="figure/overview_lines.Rmd/plot-exposures-27.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-exposures-27">
Past versions of plot-exposures-27.png
</button>
</p>
<div id="fig-plot-exposures-27" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/figure/overview_lines.Rmd/plot-exposures-27.png" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>....plotting  vuna </code></pre>
<p><img src="figure/overview_lines.Rmd/plot-exposures-28.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-exposures-28">
Past versions of plot-exposures-28.png
</button>
</p>
<div id="fig-plot-exposures-28" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/figure/overview_lines.Rmd/plot-exposures-28.png" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>....plotting  wahn </code></pre>
<p><img src="figure/overview_lines.Rmd/plot-exposures-29.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-exposures-29">
Past versions of plot-exposures-29.png
</button>
</p>
<div id="fig-plot-exposures-29" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/figure/overview_lines.Rmd/plot-exposures-29.png" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>....plotting  wetu </code></pre>
<p><img src="figure/overview_lines.Rmd/plot-exposures-30.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-exposures-30">
Past versions of plot-exposures-30.png
</button>
</p>
<div id="fig-plot-exposures-30" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/figure/overview_lines.Rmd/plot-exposures-30.png" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>....plotting  xugn </code></pre>
<p><img src="figure/overview_lines.Rmd/plot-exposures-31.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-exposures-31">
Past versions of plot-exposures-31.png
</button>
</p>
<div id="fig-plot-exposures-31" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/figure/overview_lines.Rmd/plot-exposures-31.png" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>....plotting  zoxy </code></pre>
<p><img src="figure/overview_lines.Rmd/plot-exposures-32.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-exposures-32">
Past versions of plot-exposures-32.png
</button>
</p>
<div id="fig-plot-exposures-32" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/figure/overview_lines.Rmd/plot-exposures-32.png" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Retrieve exposures for a given signal. Specifically, we will look an highest posterior density (HPD) intervals and mean exposures for Signature 7 (UV) and Signature 11, the only two that are significant across multiple lines. We add this information to the <code>donor_info</code> dataframe.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb42-1" data-line-number="1">get_signature_df &lt;-<span class="st"> </span><span class="cf">function</span>(exposures, samples, signature) {</a>
<a class="sourceLine" id="cb42-2" data-line-number="2">  signature_mat_mean &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="ot">NA</span>, <span class="dt">nrow =</span> <span class="kw">length</span>(samples), <span class="dv">30</span>)</a>
<a class="sourceLine" id="cb42-3" data-line-number="3">  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(samples)) {</a>
<a class="sourceLine" id="cb42-4" data-line-number="4">    signature_mat_mean[i,] &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(exposures[[i]]<span class="op">$</span>mean)</a>
<a class="sourceLine" id="cb42-5" data-line-number="5">  }</a>
<a class="sourceLine" id="cb42-6" data-line-number="6">  <span class="kw">rownames</span>(signature_mat_mean) &lt;-<span class="st"> </span>samples</a>
<a class="sourceLine" id="cb42-7" data-line-number="7">  <span class="kw">colnames</span>(signature_mat_mean) &lt;-<span class="st"> </span><span class="kw">colnames</span>(exposures[[i]]<span class="op">$</span>mean)</a>
<a class="sourceLine" id="cb42-8" data-line-number="8">  signature_mat_lower90 &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="ot">NA</span>, <span class="dt">nrow =</span> <span class="kw">length</span>(samples), <span class="dv">30</span>)</a>
<a class="sourceLine" id="cb42-9" data-line-number="9">  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(samples)) {</a>
<a class="sourceLine" id="cb42-10" data-line-number="10">    signature_mat_lower90[i,] &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(exposures[[i]]<span class="op">$</span>lower_<span class="dv">90</span>)</a>
<a class="sourceLine" id="cb42-11" data-line-number="11">  }</a>
<a class="sourceLine" id="cb42-12" data-line-number="12">  <span class="kw">rownames</span>(signature_mat_lower90) &lt;-<span class="st"> </span>samples</a>
<a class="sourceLine" id="cb42-13" data-line-number="13">  <span class="kw">colnames</span>(signature_mat_lower90) &lt;-<span class="st"> </span><span class="kw">colnames</span>(exposures[[i]]<span class="op">$</span>lower_<span class="dv">90</span>)</a>
<a class="sourceLine" id="cb42-14" data-line-number="14">  signature_mat_upper90 &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="ot">NA</span>, <span class="dt">nrow =</span> <span class="kw">length</span>(samples), <span class="dv">30</span>)</a>
<a class="sourceLine" id="cb42-15" data-line-number="15">  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(samples)) {</a>
<a class="sourceLine" id="cb42-16" data-line-number="16">    signature_mat_upper90[i,] &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(exposures[[i]]<span class="op">$</span>upper_<span class="dv">90</span>)</a>
<a class="sourceLine" id="cb42-17" data-line-number="17">  }</a>
<a class="sourceLine" id="cb42-18" data-line-number="18">  <span class="kw">rownames</span>(signature_mat_upper90) &lt;-<span class="st"> </span>samples</a>
<a class="sourceLine" id="cb42-19" data-line-number="19">  <span class="kw">colnames</span>(signature_mat_upper90) &lt;-<span class="st"> </span><span class="kw">colnames</span>(exposures[[i]]<span class="op">$</span>upper_<span class="dv">90</span>)</a>
<a class="sourceLine" id="cb42-20" data-line-number="20">  signature_df &lt;-<span class="st"> </span><span class="kw">cbind</span>(</a>
<a class="sourceLine" id="cb42-21" data-line-number="21">    <span class="kw">as.data.frame</span>(signature_mat_mean[,signature]),</a>
<a class="sourceLine" id="cb42-22" data-line-number="22">    <span class="kw">as.data.frame</span>(signature_mat_lower90[,signature]),</a>
<a class="sourceLine" id="cb42-23" data-line-number="23">    <span class="kw">as.data.frame</span>(signature_mat_upper90[,signature]))</a>
<a class="sourceLine" id="cb42-24" data-line-number="24">  <span class="kw">colnames</span>(signature_df) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="kw">paste0</span>(<span class="st">&quot;Sig&quot;</span>,signature,<span class="st">&quot;_mean&quot;</span>),</a>
<a class="sourceLine" id="cb42-25" data-line-number="25">                              <span class="kw">paste0</span>(<span class="st">&quot;Sig&quot;</span>,signature,<span class="st">&quot;_lower&quot;</span>),</a>
<a class="sourceLine" id="cb42-26" data-line-number="26">                              <span class="kw">paste0</span>(<span class="st">&quot;Sig&quot;</span>,signature,<span class="st">&quot;_upper&quot;</span>))</a>
<a class="sourceLine" id="cb42-27" data-line-number="27">  signature_df<span class="op">$</span>donor &lt;-<span class="st"> </span><span class="kw">rownames</span>(signature_df)</a>
<a class="sourceLine" id="cb42-28" data-line-number="28">  signature_df</a>
<a class="sourceLine" id="cb42-29" data-line-number="29">}</a>
<a class="sourceLine" id="cb42-30" data-line-number="30"></a>
<a class="sourceLine" id="cb42-31" data-line-number="31"><span class="co">## Get lower, mean and upper values for signatures 7 &amp; 11</span></a>
<a class="sourceLine" id="cb42-32" data-line-number="32">sig7_df &lt;-<span class="st"> </span><span class="kw">get_signature_df</span>(exposures, mutation_donors, <span class="dv">7</span>)</a>
<a class="sourceLine" id="cb42-33" data-line-number="33">sig11_df &lt;-<span class="st"> </span><span class="kw">get_signature_df</span>(exposures, mutation_donors, <span class="dv">11</span>)</a>
<a class="sourceLine" id="cb42-34" data-line-number="34"></a>
<a class="sourceLine" id="cb42-35" data-line-number="35"><span class="co">## Add Sigs 7/11 to donor info table</span></a>
<a class="sourceLine" id="cb42-36" data-line-number="36">sig_subset_df &lt;-<span class="st"> </span><span class="kw">merge</span>(sig7_df,sig11_df, <span class="dt">by =</span> <span class="st">&quot;donor&quot;</span>)</a>
<a class="sourceLine" id="cb42-37" data-line-number="37">sig_subset_df_means &lt;-<span class="st"> </span>sig_subset_df[,<span class="kw">c</span>(<span class="dv">1</span>,<span class="kw">grep</span>(<span class="st">&quot;mean&quot;</span>, <span class="kw">colnames</span>(sig_subset_df)))]</a>
<a class="sourceLine" id="cb42-38" data-line-number="38">sig_subset_df_means_melt &lt;-<span class="st"> </span>reshape2<span class="op">::</span><span class="kw">melt</span>(sig_subset_df_means)</a>
<a class="sourceLine" id="cb42-39" data-line-number="39">sig_subset_df_means_melt<span class="op">$</span>signature &lt;-<span class="st"> </span><span class="kw">substr</span>(sig_subset_df_means_melt<span class="op">$</span>variable, <span class="dv">1</span>, <span class="dv">5</span>)</a>
<a class="sourceLine" id="cb42-40" data-line-number="40">sig_subset_df_means_melt &lt;-<span class="st"> </span>sig_subset_df_means_melt[,<span class="kw">c</span>(<span class="st">&quot;donor&quot;</span>, <span class="st">&quot;signature&quot;</span>, <span class="st">&quot;value&quot;</span>)]</a>
<a class="sourceLine" id="cb42-41" data-line-number="41"></a>
<a class="sourceLine" id="cb42-42" data-line-number="42"><span class="kw">colnames</span>(sig_subset_df)[<span class="dv">1</span>] &lt;-<span class="st"> &quot;donor_short&quot;</span> </a>
<a class="sourceLine" id="cb42-43" data-line-number="43">donor_info &lt;-<span class="st"> </span><span class="kw">merge</span>(donor_info, sig_subset_df, <span class="dt">by =</span> <span class="st">&quot;donor_short&quot;</span>)</a></code></pre></div>
</div>
<div id="expression-data-and-cell-assignments" class="section level2">
<h2>Expression data and cell assignments</h2>
<div id="original-cardelino-cell-assignments" class="section level3">
<h3>Original Cardelino cell assignments</h3>
<p>Read in annotated SingleCellExperiment (SCE) objects and create a list of SCE objects containing all cells used for analysis and their assignment (using <code>cardelino</code>) to clones identified with Canopy from whole-exome sequencing data.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb43-1" data-line-number="1">params &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb43-2" data-line-number="2">params<span class="op">$</span>callset &lt;-<span class="st"> &quot;filt_lenient.cell_coverage_sites&quot;</span></a>
<a class="sourceLine" id="cb43-3" data-line-number="3">fls &lt;-<span class="st"> </span><span class="kw">list.files</span>(<span class="st">&quot;data/sces&quot;</span>)</a>
<a class="sourceLine" id="cb43-4" data-line-number="4">fls &lt;-<span class="st"> </span>fls[<span class="kw">grepl</span>(<span class="kw">paste0</span>(<span class="st">&quot;assignments.&quot;</span>, params<span class="op">$</span>callset), fls)]</a>
<a class="sourceLine" id="cb43-5" data-line-number="5">donors &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;.*ce_([a-z]+)_.*assignments.filt.*&quot;</span>, <span class="st">&quot;</span><span class="ch">\\</span><span class="st">1&quot;</span>, fls)</a>
<a class="sourceLine" id="cb43-6" data-line-number="6">sce_unst_list &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb43-7" data-line-number="7"><span class="cf">for</span> (don <span class="cf">in</span> donors) {</a>
<a class="sourceLine" id="cb43-8" data-line-number="8">    sce_unst_list[[don]] &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">file.path</span>(<span class="st">&quot;data/sces&quot;</span>, </a>
<a class="sourceLine" id="cb43-9" data-line-number="9">        <span class="kw">paste0</span>(<span class="st">&quot;sce_&quot;</span>, don, <span class="st">&quot;_with_clone_assignments.&quot;</span>, params<span class="op">$</span>callset, <span class="st">&quot;.rds&quot;</span>)))</a>
<a class="sourceLine" id="cb43-10" data-line-number="10">    <span class="kw">cat</span>(<span class="kw">paste</span>(<span class="st">&quot;reading&quot;</span>, don, <span class="st">&quot;:   &quot;</span>, <span class="kw">ncol</span>(sce_unst_list[[don]]), <span class="st">&quot;cells.</span><span class="ch">\n</span><span class="st">&quot;</span>))</a>
<a class="sourceLine" id="cb43-11" data-line-number="11">}</a></code></pre></div>
<pre><code>reading euts :    79 cells.
reading fawm :    53 cells.
reading feec :    75 cells.
reading fikt :    39 cells.
reading garx :    70 cells.
reading gesg :    105 cells.
reading heja :    50 cells.
reading hipn :    62 cells.
reading ieki :    58 cells.
reading joxm :    79 cells.
reading kuco :    48 cells.
reading laey :    55 cells.
reading lexy :    63 cells.
reading naju :    44 cells.
reading nusw :    60 cells.
reading oaaz :    38 cells.
reading oilg :    90 cells.
reading pipw :    107 cells.
reading puie :    41 cells.
reading qayj :    97 cells.
reading qolg :    36 cells.
reading qonc :    58 cells.
reading rozh :    91 cells.
reading sehl :    30 cells.
reading ualf :    89 cells.
reading vass :    37 cells.
reading vils :    37 cells.
reading vuna :    71 cells.
reading wahn :    82 cells.
reading wetu :    77 cells.
reading xugn :    35 cells.
reading zoxy :    88 cells.</code></pre>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb45-1" data-line-number="1"><span class="co">## Calculate single cell data metrics</span></a>
<a class="sourceLine" id="cb45-2" data-line-number="2">sc_metrics_summary &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb45-3" data-line-number="3">sc_metrics_summary_df &lt;-<span class="st"> </span><span class="kw">data.frame</span>()</a>
<a class="sourceLine" id="cb45-4" data-line-number="4"><span class="cf">for</span> (don <span class="cf">in</span> donors) {</a>
<a class="sourceLine" id="cb45-5" data-line-number="5">  sc_metrics_summary[[don]]<span class="op">$</span>num_unst_cells &lt;-<span class="st"> </span><span class="kw">ncol</span>(sce_unst_list[[don]])</a>
<a class="sourceLine" id="cb45-6" data-line-number="6">  sc_metrics_summary[[don]]<span class="op">$</span>num_assignable &lt;-<span class="st"> </span><span class="kw">sum</span>(sce_unst_list[[don]]<span class="op">$</span>assignable)</a>
<a class="sourceLine" id="cb45-7" data-line-number="7">  sc_metrics_summary[[don]]<span class="op">$</span>num_unassignable &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb45-8" data-line-number="8"><span class="st">    </span>(sc_metrics_summary[[don]]<span class="op">$</span>num_unst_cells <span class="op">-</span></a>
<a class="sourceLine" id="cb45-9" data-line-number="9"><span class="st">       </span>sc_metrics_summary[[don]]<span class="op">$</span>num_assignable)</a>
<a class="sourceLine" id="cb45-10" data-line-number="10">  sc_metrics_summary[[don]]<span class="op">$</span>num_clones_with_cells &lt;-</a>
<a class="sourceLine" id="cb45-11" data-line-number="11"><span class="st">    </span><span class="kw">length</span>(<span class="kw">unique</span>(sce_unst_list[[don]]<span class="op">$</span>assigned[</a>
<a class="sourceLine" id="cb45-12" data-line-number="12">      <span class="kw">which</span>(sce_unst_list[[don]]<span class="op">$</span>assigned <span class="op">!=</span><span class="st"> &quot;unassigned&quot;</span>)]))</a>
<a class="sourceLine" id="cb45-13" data-line-number="13">  sc_metrics_summary[[don]]<span class="op">$</span>donor &lt;-<span class="st"> </span>don</a>
<a class="sourceLine" id="cb45-14" data-line-number="14">  sc_metrics_summary_df &lt;-</a>
<a class="sourceLine" id="cb45-15" data-line-number="15"><span class="st">    </span><span class="kw">rbind</span>(sc_metrics_summary_df, <span class="kw">as.data.frame</span>(sc_metrics_summary[[don]]))</a>
<a class="sourceLine" id="cb45-16" data-line-number="16">}</a>
<a class="sourceLine" id="cb45-17" data-line-number="17"></a>
<a class="sourceLine" id="cb45-18" data-line-number="18"><span class="kw">colnames</span>(sc_metrics_summary_df) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;total_unst_cells&quot;</span>, <span class="st">&quot;assigned_unst_cells&quot;</span>,</a>
<a class="sourceLine" id="cb45-19" data-line-number="19">                                     <span class="st">&quot;unassigned_unst_cells&quot;</span>, </a>
<a class="sourceLine" id="cb45-20" data-line-number="20">                                     <span class="st">&quot;num_clones_with_cells&quot;</span>, <span class="st">&quot;donor_short&quot;</span>)</a>
<a class="sourceLine" id="cb45-21" data-line-number="21"></a>
<a class="sourceLine" id="cb45-22" data-line-number="22"><span class="co">## Merge with donor info table</span></a>
<a class="sourceLine" id="cb45-23" data-line-number="23">donor_info &lt;-<span class="st"> </span><span class="kw">merge</span>(donor_info, sc_metrics_summary_df, <span class="dt">by =</span> <span class="st">&quot;donor_short&quot;</span>, </a>
<a class="sourceLine" id="cb45-24" data-line-number="24">                    <span class="dt">all.x =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb45-25" data-line-number="25">donor_info<span class="op">$</span>percent_assigned_cells &lt;-</a>
<a class="sourceLine" id="cb45-26" data-line-number="26"><span class="st">  </span>donor_info<span class="op">$</span>assigned_unst_cells <span class="op">/</span><span class="st"> </span>donor_info<span class="op">$</span>total_unst_cells</a></code></pre></div>
</div>
</div>
<div id="canopy-clone-inference-information" class="section level2">
<h2>Canopy clone inference information</h2>
<p>First, we read in the canopy output for each line we analyse.</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb46-1" data-line-number="1">canopy_files &lt;-<span class="st"> </span><span class="kw">list.files</span>(<span class="st">&quot;data/canopy&quot;</span>)</a>
<a class="sourceLine" id="cb46-2" data-line-number="2">canopy_files &lt;-<span class="st"> </span>canopy_files[<span class="kw">grepl</span>(params<span class="op">$</span>callset, canopy_files)]</a>
<a class="sourceLine" id="cb46-3" data-line-number="3">canopy_list &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb46-4" data-line-number="4"><span class="cf">for</span> (don <span class="cf">in</span> donors) {</a>
<a class="sourceLine" id="cb46-5" data-line-number="5">  canopy_list[[don]] &lt;-<span class="st"> </span><span class="kw">readRDS</span>(</a>
<a class="sourceLine" id="cb46-6" data-line-number="6">    <span class="kw">file.path</span>(<span class="st">&quot;data/canopy&quot;</span>, </a>
<a class="sourceLine" id="cb46-7" data-line-number="7">              <span class="kw">paste0</span>(<span class="st">&quot;canopy_results.&quot;</span>, don, <span class="st">&quot;.&quot;</span>, params<span class="op">$</span>callset, <span class="st">&quot;.rds&quot;</span>)))</a>
<a class="sourceLine" id="cb46-8" data-line-number="8">}</a></code></pre></div>
<p>Second, summarise the number of mutations for each clone, for each line. Form these results into a dataframe.</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb47-1" data-line-number="1">clone_mut_list &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb47-2" data-line-number="2"><span class="cf">for</span> (don <span class="cf">in</span> donors) {</a>
<a class="sourceLine" id="cb47-3" data-line-number="3">  <span class="kw">cat</span>(<span class="kw">paste</span>(<span class="st">&quot;summarising clones and mutations for&quot;</span>, don, <span class="st">&#39;</span><span class="ch">\n</span><span class="st">&#39;</span>))</a>
<a class="sourceLine" id="cb47-4" data-line-number="4">  clone_mut_list[[don]] &lt;-<span class="st"> </span><span class="kw">colSums</span>(canopy_list[[don]]<span class="op">$</span>tree<span class="op">$</span>Z)</a>
<a class="sourceLine" id="cb47-5" data-line-number="5">  <span class="kw">cat</span>(<span class="kw">colSums</span>(canopy_list[[don]]<span class="op">$</span>tree<span class="op">$</span>Z))</a>
<a class="sourceLine" id="cb47-6" data-line-number="6">  <span class="kw">cat</span>(<span class="st">&#39;</span><span class="ch">\n</span><span class="st">&#39;</span>)</a>
<a class="sourceLine" id="cb47-7" data-line-number="7">}</a></code></pre></div>
<pre><code>summarising clones and mutations for euts 
0 39 29
summarising clones and mutations for fawm 
0 17 16
summarising clones and mutations for feec 
0 19 9 6
summarising clones and mutations for fikt 
0 13 23
summarising clones and mutations for garx 
0 79 57
summarising clones and mutations for gesg 
0 37 23
summarising clones and mutations for heja 
0 16 20
summarising clones and mutations for hipn 
0 8 8
summarising clones and mutations for ieki 
0 7 10
summarising clones and mutations for joxm 
0 41 98
summarising clones and mutations for kuco 
0 9
summarising clones and mutations for laey 
0 45 49
summarising clones and mutations for lexy 
0 9 6
summarising clones and mutations for naju 
0 13
summarising clones and mutations for nusw 
0 3 13
summarising clones and mutations for oaaz 
0 17 19
summarising clones and mutations for oilg 
0 2 37
summarising clones and mutations for pipw 
0 34 36
summarising clones and mutations for puie 
0 13 15
summarising clones and mutations for qayj 
0 11 7
summarising clones and mutations for qolg 
0 23
summarising clones and mutations for qonc 
0 17 7
summarising clones and mutations for rozh 
0 11 10 12
summarising clones and mutations for sehl 
0 2 11 23
summarising clones and mutations for ualf 
0 29 39
summarising clones and mutations for vass 
0 98 35
summarising clones and mutations for vils 
0 1 2 8
summarising clones and mutations for vuna 
0 33
summarising clones and mutations for wahn 
0 52 114
summarising clones and mutations for wetu 
0 8 17
summarising clones and mutations for xugn 
0 16 12
summarising clones and mutations for zoxy 
0 14 8</code></pre>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb49-1" data-line-number="1">clone_mut_df &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">clone1 =</span> <span class="kw">numeric</span>(), <span class="dt">clone2 =</span> <span class="kw">numeric</span>(), </a>
<a class="sourceLine" id="cb49-2" data-line-number="2">                           <span class="dt">clone3 =</span> <span class="kw">numeric</span>(), <span class="dt">clone4 =</span> <span class="kw">numeric</span>())</a>
<a class="sourceLine" id="cb49-3" data-line-number="3"><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(donors)) {</a>
<a class="sourceLine" id="cb49-4" data-line-number="4">  num_clones &lt;-<span class="st"> </span><span class="kw">length</span>(clone_mut_list[[i]])</a>
<a class="sourceLine" id="cb49-5" data-line-number="5">  num_NAs &lt;-<span class="st"> </span><span class="dv">4</span> <span class="op">-</span><span class="st"> </span>num_clones</a>
<a class="sourceLine" id="cb49-6" data-line-number="6">  temp_row  &lt;-<span class="st"> </span><span class="kw">c</span>(clone_mut_list[[i]], <span class="kw">rep</span>(<span class="ot">NA</span>, num_NAs))</a>
<a class="sourceLine" id="cb49-7" data-line-number="7">  clone_mut_df[i,] &lt;-<span class="st"> </span>temp_row</a>
<a class="sourceLine" id="cb49-8" data-line-number="8">}</a>
<a class="sourceLine" id="cb49-9" data-line-number="9"><span class="kw">rownames</span>(clone_mut_df) &lt;-<span class="st"> </span>donors</a>
<a class="sourceLine" id="cb49-10" data-line-number="10">clone_mut_df<span class="op">$</span>donor_short &lt;-<span class="st"> </span>donors</a></code></pre></div>
<p>Next, summarise the number of <em>unique</em> mutations tagging each clone identified by Canopy (that is, the number of variants for each clone that distinguish it from other clones in the line). Produce a dataframe with this information as well.</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb50-1" data-line-number="1">clone_mut_unique_list &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb50-2" data-line-number="2"><span class="cf">for</span> (don <span class="cf">in</span> donors) {</a>
<a class="sourceLine" id="cb50-3" data-line-number="3">        <span class="kw">cat</span>(<span class="kw">paste</span>(<span class="st">&quot;summarising clones and mutations for&quot;</span>, don, <span class="st">&#39;</span><span class="ch">\n</span><span class="st">&#39;</span>))</a>
<a class="sourceLine" id="cb50-4" data-line-number="4">        clone_mut_unique_list[[don]] &lt;-</a>
<a class="sourceLine" id="cb50-5" data-line-number="5"><span class="st">          </span><span class="kw">colSums</span>(canopy_list[[don]]<span class="op">$</span>tree<span class="op">$</span>Z[</a>
<a class="sourceLine" id="cb50-6" data-line-number="6">            (<span class="kw">rowSums</span>(canopy_list[[don]]<span class="op">$</span>tree<span class="op">$</span>Z) <span class="op">==</span><span class="st"> </span><span class="dv">1</span>),])</a>
<a class="sourceLine" id="cb50-7" data-line-number="7">        <span class="kw">cat</span>(<span class="kw">colSums</span>(canopy_list[[don]]<span class="op">$</span>tree<span class="op">$</span>Z[</a>
<a class="sourceLine" id="cb50-8" data-line-number="8">          (<span class="kw">rowSums</span>(canopy_list[[don]]<span class="op">$</span>tree<span class="op">$</span>Z) <span class="op">==</span><span class="st"> </span><span class="dv">1</span>),]))</a>
<a class="sourceLine" id="cb50-9" data-line-number="9">        <span class="kw">cat</span>(<span class="st">&#39;</span><span class="ch">\n</span><span class="st">&#39;</span>)</a>
<a class="sourceLine" id="cb50-10" data-line-number="10">}</a></code></pre></div>
<pre><code>summarising clones and mutations for euts 
0 20 10
summarising clones and mutations for fawm 
0 3 2
summarising clones and mutations for feec 
0 16 4 1
summarising clones and mutations for fikt 
0 3 13
summarising clones and mutations for garx 
0 41 19
summarising clones and mutations for gesg 
0 20 6
summarising clones and mutations for heja 
0 8 12
summarising clones and mutations for hipn 
0 6 6
summarising clones and mutations for ieki 
0 6 9
summarising clones and mutations for joxm 
0 14 71
summarising clones and mutations for kuco 
0 9
summarising clones and mutations for laey 
0 16 20
summarising clones and mutations for lexy 
0 5 2
summarising clones and mutations for naju 
0 13
summarising clones and mutations for nusw 
0 0 10
summarising clones and mutations for oaaz 
0 8 10
summarising clones and mutations for oilg 
0 2 37
summarising clones and mutations for pipw 
0 17 19
summarising clones and mutations for puie 
0 4 6
summarising clones and mutations for qayj 
0 7 3
summarising clones and mutations for qolg 
0 23
summarising clones and mutations for qonc 
0 13 3
summarising clones and mutations for rozh 
0 7 0 2
summarising clones and mutations for sehl 
0 0 3 15
summarising clones and mutations for ualf 
0 15 25
summarising clones and mutations for vass 
0 65 2
summarising clones and mutations for vils 
0 0 1 7
summarising clones and mutations for vuna 
0 33
summarising clones and mutations for wahn 
0 1 63
summarising clones and mutations for wetu 
0 2 11
summarising clones and mutations for xugn 
0 6 2
summarising clones and mutations for zoxy 
0 7 1</code></pre>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb52-1" data-line-number="1">clone_mut_unique_df &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">clone1 =</span> <span class="kw">numeric</span>(), <span class="dt">clone2 =</span> <span class="kw">numeric</span>(), </a>
<a class="sourceLine" id="cb52-2" data-line-number="2">                                  <span class="dt">clone3 =</span> <span class="kw">numeric</span>(),</a>
<a class="sourceLine" id="cb52-3" data-line-number="3">                           <span class="dt">clone4 =</span> <span class="kw">numeric</span>(), <span class="dt">min_unique_muts =</span> <span class="kw">numeric</span>())</a>
<a class="sourceLine" id="cb52-4" data-line-number="4"><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(donors)) {</a>
<a class="sourceLine" id="cb52-5" data-line-number="5">  num_clones &lt;-<span class="st"> </span><span class="kw">length</span>(clone_mut_unique_list[[i]])</a>
<a class="sourceLine" id="cb52-6" data-line-number="6">  num_NAs &lt;-<span class="st"> </span><span class="dv">4</span> <span class="op">-</span><span class="st"> </span>num_clones</a>
<a class="sourceLine" id="cb52-7" data-line-number="7">  temp_row  &lt;-<span class="st"> </span><span class="kw">c</span>(clone_mut_unique_list[[i]], <span class="kw">rep</span>(<span class="ot">NA</span>, num_NAs), </a>
<a class="sourceLine" id="cb52-8" data-line-number="8">                 <span class="kw">min</span>(clone_mut_unique_list[[i]][<span class="dv">2</span><span class="op">:</span>num_clones]))</a>
<a class="sourceLine" id="cb52-9" data-line-number="9">  clone_mut_unique_df[i,] &lt;-<span class="st"> </span>temp_row</a>
<a class="sourceLine" id="cb52-10" data-line-number="10">}</a>
<a class="sourceLine" id="cb52-11" data-line-number="11"><span class="kw">rownames</span>(clone_mut_df) &lt;-<span class="st"> </span>donors</a>
<a class="sourceLine" id="cb52-12" data-line-number="12">clone_mut_df<span class="op">$</span>donor_short &lt;-<span class="st"> </span>donors</a>
<a class="sourceLine" id="cb52-13" data-line-number="13"><span class="kw">rownames</span>(clone_mut_unique_df) &lt;-<span class="st"> </span>donors</a>
<a class="sourceLine" id="cb52-14" data-line-number="14">clone_mut_unique_df<span class="op">$</span>donor_short &lt;-<span class="st"> </span>donors</a>
<a class="sourceLine" id="cb52-15" data-line-number="15">donor_info &lt;-<span class="st"> </span><span class="kw">merge</span>(donor_info, clone_mut_unique_df, <span class="dt">by =</span> <span class="st">&quot;donor_short&quot;</span>, <span class="dt">all.x =</span> T)</a></code></pre></div>
<p>Finally, calculate the minimum Hamming distance between pairs of clones for each line. In general, assignment of cells to clones will be easier/more successful for lines with larger numbers of variants distinguishes between clones (that is, a high minimum Hamming distance).</p>
<p>We add all of this information to the <code>donor_info</code> dataframe and then have the data prepared to make some overview plots across lines.</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb53-1" data-line-number="1"><span class="co">## Calculate Hamming distance</span></a>
<a class="sourceLine" id="cb53-2" data-line-number="2">clone_mut_list_hamming &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb53-3" data-line-number="3"><span class="cf">for</span> (don <span class="cf">in</span> donors) {</a>
<a class="sourceLine" id="cb53-4" data-line-number="4">  Config &lt;-<span class="st"> </span>canopy_list[[don]]<span class="op">$</span>tree<span class="op">$</span>Z</a>
<a class="sourceLine" id="cb53-5" data-line-number="5">  unique_sites_paired &lt;-<span class="st"> </span><span class="kw">c</span>()</a>
<a class="sourceLine" id="cb53-6" data-line-number="6">  <span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">seq_len</span>(<span class="kw">ncol</span>(Config) <span class="op">-</span><span class="st"> </span><span class="dv">1</span>)) {</a>
<a class="sourceLine" id="cb53-7" data-line-number="7">    <span class="cf">for</span> (j <span class="cf">in</span> <span class="kw">seq</span>(i <span class="op">+</span><span class="st"> </span><span class="dv">1</span>, <span class="kw">ncol</span>(Config))) {</a>
<a class="sourceLine" id="cb53-8" data-line-number="8">      n_sites &lt;-<span class="st"> </span><span class="kw">sum</span>(<span class="kw">rowSums</span>(Config[, <span class="kw">c</span>(i,j)]) <span class="op">==</span><span class="st"> </span><span class="dv">1</span>)</a>
<a class="sourceLine" id="cb53-9" data-line-number="9">      unique_sites_paired &lt;-<span class="st"> </span><span class="kw">c</span>(unique_sites_paired, n_sites)</a>
<a class="sourceLine" id="cb53-10" data-line-number="10">    }</a>
<a class="sourceLine" id="cb53-11" data-line-number="11">  }</a>
<a class="sourceLine" id="cb53-12" data-line-number="12">  clone_mut_list_hamming[[don]] &lt;-<span class="st"> </span>unique_sites_paired  </a>
<a class="sourceLine" id="cb53-13" data-line-number="13">  <span class="kw">cat</span>(<span class="st">&quot;....hamming distances for &quot;</span>, don, <span class="st">&quot;: &quot;</span>, unique_sites_paired, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>) </a>
<a class="sourceLine" id="cb53-14" data-line-number="14">}</a></code></pre></div>
<pre><code>....hamming distances for  euts :  39 29 30 
....hamming distances for  fawm :  17 16 5 
....hamming distances for  feec :  19 9 6 22 19 5 
....hamming distances for  fikt :  13 23 16 
....hamming distances for  garx :  79 57 60 
....hamming distances for  gesg :  37 23 26 
....hamming distances for  heja :  16 20 20 
....hamming distances for  hipn :  8 8 12 
....hamming distances for  ieki :  7 10 15 
....hamming distances for  joxm :  41 98 85 
....hamming distances for  kuco :  9 
....hamming distances for  laey :  45 49 36 
....hamming distances for  lexy :  9 6 7 
....hamming distances for  naju :  13 
....hamming distances for  nusw :  3 13 10 
....hamming distances for  oaaz :  17 19 18 
....hamming distances for  oilg :  2 37 39 
....hamming distances for  pipw :  34 36 36 
....hamming distances for  puie :  13 15 10 
....hamming distances for  qayj :  11 7 10 
....hamming distances for  qolg :  23 
....hamming distances for  qonc :  17 7 16 
....hamming distances for  rozh :  11 10 12 13 15 2 
....hamming distances for  sehl :  2 11 23 9 21 18 
....hamming distances for  ualf :  29 39 40 
....hamming distances for  vass :  98 35 67 
....hamming distances for  vils :  1 2 8 1 7 8 
....hamming distances for  vuna :  33 
....hamming distances for  wahn :  52 114 64 
....hamming distances for  wetu :  8 17 13 
....hamming distances for  xugn :  16 12 8 
....hamming distances for  zoxy :  14 8 8 </code></pre>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb55-1" data-line-number="1">min_hamming_distance &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="st">&quot;donor_short&quot;</span> =<span class="st"> </span>donors,</a>
<a class="sourceLine" id="cb55-2" data-line-number="2">                                   <span class="st">&quot;min_hamming_dist&quot;</span> =<span class="st"> </span><span class="dv">0</span>)</a>
<a class="sourceLine" id="cb55-3" data-line-number="3"><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(donors)) {</a>
<a class="sourceLine" id="cb55-4" data-line-number="4">  min_hamming_distance<span class="op">$</span>min_hamming_dist[i] &lt;-<span class="st"> </span><span class="kw">min</span>(clone_mut_list_hamming[[i]])</a>
<a class="sourceLine" id="cb55-5" data-line-number="5">}</a>
<a class="sourceLine" id="cb55-6" data-line-number="6"></a>
<a class="sourceLine" id="cb55-7" data-line-number="7">donor_info &lt;-<span class="st"> </span><span class="kw">merge</span>(donor_info, min_hamming_distance, <span class="dt">by =</span> <span class="st">&quot;donor_short&quot;</span>, </a>
<a class="sourceLine" id="cb55-8" data-line-number="8">                    <span class="dt">all.x =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb55-9" data-line-number="9"></a>
<a class="sourceLine" id="cb55-10" data-line-number="10"><span class="co">## Number of clones</span></a>
<a class="sourceLine" id="cb55-11" data-line-number="11">donor_info<span class="op">$</span>num_clones_total &lt;-<span class="st"> </span><span class="dv">0</span></a>
<a class="sourceLine" id="cb55-12" data-line-number="12"><span class="cf">for</span> (i <span class="cf">in</span> donors) {</a>
<a class="sourceLine" id="cb55-13" data-line-number="13">  num_clones &lt;-<span class="st"> </span><span class="kw">length</span>(clone_mut_unique_list[[i]])</a>
<a class="sourceLine" id="cb55-14" data-line-number="14">  donor_info<span class="op">$</span>num_clones_total[<span class="kw">which</span>(donor_info<span class="op">$</span>donor_short <span class="op">==</span><span class="st"> </span>i)] &lt;-<span class="st"> </span>num_clones</a>
<a class="sourceLine" id="cb55-15" data-line-number="15">}</a></code></pre></div>
</div>
<div id="plot-line-metrics" class="section level2">
<h2>Plot line metrics</h2>
<p>We make a large combined plot stitching together individual plots showing:</p>
<ul>
<li>donor age;</li>
<li>number of somatic mutations;</li>
<li>signature 7 (UV) exposure;</li>
<li>probablility of selection (Williams et al, 2018); and</li>
<li>number of mutations per clone</li>
</ul>
<p>for each line.</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb56-1" data-line-number="1">donor_info_filt &lt;-<span class="st"> </span>donor_info[<span class="kw">which</span>(donor_info<span class="op">$</span>donor_short <span class="op">%in%</span><span class="st"> </span>donors), ]</a>
<a class="sourceLine" id="cb56-2" data-line-number="2">donor_filt_order &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(donor_info_filt, <span class="kw">desc</span>(num_mutations))</a>
<a class="sourceLine" id="cb56-3" data-line-number="3">donor_filt_order &lt;-<span class="st"> </span>donor_filt_order<span class="op">$</span>donor_short</a>
<a class="sourceLine" id="cb56-4" data-line-number="4"><span class="kw">rownames</span>(donor_info_filt) &lt;-<span class="st"> </span>donor_info_filt<span class="op">$</span>donor_short</a>
<a class="sourceLine" id="cb56-5" data-line-number="5"></a>
<a class="sourceLine" id="cb56-6" data-line-number="6"><span class="co"># Plot age</span></a>
<a class="sourceLine" id="cb56-7" data-line-number="7">selected_vars &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;donor_short&quot;</span>, <span class="st">&quot;age_decade&quot;</span>)</a>
<a class="sourceLine" id="cb56-8" data-line-number="8">donor_info_filt_melt &lt;-<span class="st"> </span>reshape2<span class="op">::</span><span class="kw">melt</span>(donor_info_filt[,selected_vars],</a>
<a class="sourceLine" id="cb56-9" data-line-number="9">                                       <span class="st">&quot;donor_short&quot;</span>)</a>
<a class="sourceLine" id="cb56-10" data-line-number="10">donor_info_filt_melt &lt;-<span class="st"> </span>donor_info_filt[,selected_vars]</a>
<a class="sourceLine" id="cb56-11" data-line-number="11"></a>
<a class="sourceLine" id="cb56-12" data-line-number="12">age_plot_filt &lt;-<span class="st"> </span><span class="kw">ggplot</span>(</a>
<a class="sourceLine" id="cb56-13" data-line-number="13">  donor_info_filt_melt,</a>
<a class="sourceLine" id="cb56-14" data-line-number="14">  <span class="kw">aes</span>(<span class="dt">x =</span> <span class="st">&quot;Age&quot;</span>, <span class="dt">y =</span> <span class="kw">factor</span>(donor_short, <span class="dt">levels =</span> <span class="kw">rev</span>(donor_filt_order)),</a>
<a class="sourceLine" id="cb56-15" data-line-number="15">      <span class="dt">fill =</span> age_decade)) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-16" data-line-number="16"><span class="st">  </span><span class="kw">geom_tile</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb56-17" data-line-number="17"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">x =</span> <span class="st">&quot;&quot;</span>, <span class="dt">y =</span> <span class="st">&quot;Donor&quot;</span>, <span class="dt">fill =</span> <span class="st">&quot;&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-18" data-line-number="18"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">rev</span>(<span class="kw">c</span>(<span class="kw">magma</span>(<span class="dv">8</span>)[<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">8</span>)])))  <span class="op">+</span></a>
<a class="sourceLine" id="cb56-19" data-line-number="19"><span class="st">  </span><span class="kw">guides</span>(<span class="dt">fill =</span> <span class="kw">guide_legend</span>(<span class="dt">nrow =</span> <span class="dv">3</span>,<span class="dt">byrow =</span> <span class="ot">TRUE</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-20" data-line-number="20"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">32</span>, <span class="dt">face =</span> <span class="st">&quot;plain&quot;</span>),</a>
<a class="sourceLine" id="cb56-21" data-line-number="21">        <span class="dt">axis.line =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb56-22" data-line-number="22">        <span class="dt">legend.position =</span> <span class="st">&quot;top&quot;</span>, <span class="dt">legend.direction =</span> <span class="st">&quot;horizontal&quot;</span>,</a>
<a class="sourceLine" id="cb56-23" data-line-number="23">        <span class="dt">legend.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">28</span>, <span class="dt">face =</span> <span class="st">&quot;plain&quot;</span>),</a>
<a class="sourceLine" id="cb56-24" data-line-number="24">        <span class="dt">legend.key.size =</span> <span class="kw">unit</span>(<span class="fl">0.3</span>,<span class="st">&quot;in&quot;</span>),</a>
<a class="sourceLine" id="cb56-25" data-line-number="25">        <span class="dt">axis.ticks.x =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb56-26" data-line-number="26">        <span class="dt">legend.margin =</span> <span class="kw">margin</span>(<span class="kw">unit</span>(<span class="fl">0.01</span>, <span class="st">&quot;cm&quot;</span>)),</a>
<a class="sourceLine" id="cb56-27" data-line-number="27">        <span class="dt">axis.text.y =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">30</span>, <span class="dt">face =</span> <span class="st">&quot;plain&quot;</span>),</a>
<a class="sourceLine" id="cb56-28" data-line-number="28">        <span class="dt">axis.title.y =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">36</span>, <span class="dt">face =</span> <span class="st">&quot;plain&quot;</span>),</a>
<a class="sourceLine" id="cb56-29" data-line-number="29">        <span class="dt">axis.title.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">36</span>, <span class="dt">face =</span> <span class="st">&quot;plain&quot;</span>))</a>
<a class="sourceLine" id="cb56-30" data-line-number="30"></a>
<a class="sourceLine" id="cb56-31" data-line-number="31"><span class="co"># Plot number of mutations</span></a>
<a class="sourceLine" id="cb56-32" data-line-number="32">selected_vars &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;donor_short&quot;</span>, <span class="st">&quot;num_mutations&quot;</span>)</a>
<a class="sourceLine" id="cb56-33" data-line-number="33">donor_info_filt_melt &lt;-<span class="st"> </span>reshape2<span class="op">::</span><span class="kw">melt</span>(donor_info_filt[,selected_vars],</a>
<a class="sourceLine" id="cb56-34" data-line-number="34">                                       <span class="st">&quot;donor_short&quot;</span>)</a>
<a class="sourceLine" id="cb56-35" data-line-number="35">num_mut_plot_filt &lt;-<span class="st"> </span><span class="kw">ggplot</span>(</a>
<a class="sourceLine" id="cb56-36" data-line-number="36">  donor_info_filt_melt,</a>
<a class="sourceLine" id="cb56-37" data-line-number="37">  <span class="kw">aes</span>(<span class="dt">x =</span> value, <span class="dt">y =</span> <span class="kw">factor</span>(donor_short, <span class="dt">levels =</span> <span class="kw">rev</span>(donor_filt_order)))) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-38" data-line-number="38"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">4</span>, <span class="dt">alpha =</span> <span class="fl">0.7</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-39" data-line-number="39"><span class="st">  </span><span class="kw">scale_shape_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="dv">4</span>, <span class="dv">19</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-40" data-line-number="40"><span class="st">  </span><span class="kw">scale_x_log10</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">50</span>, <span class="dv">100</span>, <span class="dv">500</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-41" data-line-number="41"><span class="st">  </span><span class="kw">theme_bw</span>(<span class="dv">16</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-42" data-line-number="42"><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot; &quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-43" data-line-number="43"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">x =</span> <span class="st">&quot;Number of mutations&quot;</span>, <span class="dt">y =</span> <span class="st">&quot;&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-44" data-line-number="44"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.y =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb56-45" data-line-number="45">        <span class="dt">axis.ticks.y =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb56-46" data-line-number="46">        <span class="dt">axis.line =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb56-47" data-line-number="47">        <span class="dt">legend.position =</span> <span class="st">&quot;top&quot;</span>,</a>
<a class="sourceLine" id="cb56-48" data-line-number="48">        <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">32</span>, <span class="dt">face =</span> <span class="st">&quot;plain&quot;</span>),</a>
<a class="sourceLine" id="cb56-49" data-line-number="49">        <span class="dt">title =</span> <span class="kw">element_text</span>(<span class="dt">colour=</span><span class="st">&quot;black&quot;</span>,<span class="dt">size =</span> <span class="dv">30</span>, <span class="dt">face =</span> <span class="st">&quot;plain&quot;</span>),</a>
<a class="sourceLine" id="cb56-50" data-line-number="50">        <span class="dt">axis.title.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">36</span>, <span class="dt">face =</span> <span class="st">&quot;plain&quot;</span>))</a>
<a class="sourceLine" id="cb56-51" data-line-number="51"></a>
<a class="sourceLine" id="cb56-52" data-line-number="52"><span class="co"># Plot signature 7</span></a>
<a class="sourceLine" id="cb56-53" data-line-number="53">donor_info_filt_melt &lt;-<span class="st"> </span>reshape2<span class="op">::</span><span class="kw">melt</span>(</a>
<a class="sourceLine" id="cb56-54" data-line-number="54">  donor_info_filt[,<span class="kw">c</span>(<span class="dv">1</span>,<span class="kw">grep</span>(<span class="st">&quot;Sig7_mean&quot;</span>,<span class="kw">colnames</span>(donor_info_filt)))])</a>
<a class="sourceLine" id="cb56-55" data-line-number="55">donor_info_filt_melt<span class="op">$</span>signature &lt;-<span class="st"> </span><span class="kw">substr</span>(donor_info_filt_melt<span class="op">$</span>variable, <span class="dv">1</span>, <span class="dv">5</span>)</a>
<a class="sourceLine" id="cb56-56" data-line-number="56">donor_info_filt_melt &lt;-</a>
<a class="sourceLine" id="cb56-57" data-line-number="57"><span class="st">  </span>donor_info_filt_melt[, <span class="kw">c</span>(<span class="st">&quot;donor_short&quot;</span>, <span class="st">&quot;signature&quot;</span>, <span class="st">&quot;value&quot;</span>)]</a>
<a class="sourceLine" id="cb56-58" data-line-number="58">sig_decomp_plot_filt &lt;-<span class="st"> </span><span class="kw">ggplot</span>(</a>
<a class="sourceLine" id="cb56-59" data-line-number="59">  donor_info_filt_melt,</a>
<a class="sourceLine" id="cb56-60" data-line-number="60">  <span class="kw">aes</span>(<span class="dt">y =</span> value, <span class="dt">x =</span> <span class="kw">factor</span>(donor_short, <span class="dt">levels =</span> <span class="kw">rev</span>(donor_filt_order)),</a>
<a class="sourceLine" id="cb56-61" data-line-number="61">      <span class="dt">fill =</span> <span class="kw">factor</span>(signature))) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-62" data-line-number="62"><span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">stat =</span> <span class="st">&quot;identity&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-63" data-line-number="63"><span class="st">  </span><span class="kw">coord_flip</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb56-64" data-line-number="64"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">rev</span>(<span class="kw">c</span>(<span class="kw">magma</span>(<span class="dv">10</span>)[<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">10</span>)]))) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-65" data-line-number="65"><span class="st">  </span><span class="kw">theme_bw</span>(<span class="dv">16</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-66" data-line-number="66"><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot; &quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-67" data-line-number="67"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">x =</span> <span class="st">&quot;&quot;</span>,<span class="dt">y =</span> <span class="st">&quot;Signature 7 exposure&quot;</span>, <span class="dt">fill =</span> <span class="st">&quot;&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-68" data-line-number="68"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.y =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb56-69" data-line-number="69">        <span class="dt">axis.ticks.y =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb56-70" data-line-number="70">        <span class="dt">axis.line =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb56-71" data-line-number="71">        <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">32</span>, <span class="dt">face =</span> <span class="st">&quot;plain&quot;</span>),</a>
<a class="sourceLine" id="cb56-72" data-line-number="72">        <span class="dt">title =</span> <span class="kw">element_text</span>(<span class="dt">colour =</span> <span class="st">&quot;black&quot;</span>, <span class="dt">size =</span> <span class="dv">30</span>, <span class="dt">face =</span> <span class="st">&quot;plain&quot;</span>),</a>
<a class="sourceLine" id="cb56-73" data-line-number="73">        <span class="dt">axis.title.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">36</span>, <span class="dt">face =</span> <span class="st">&quot;plain&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-74" data-line-number="74"><span class="st">  </span><span class="kw">guides</span>(<span class="dt">fill =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb56-75" data-line-number="75"></a>
<a class="sourceLine" id="cb56-76" data-line-number="76"></a>
<a class="sourceLine" id="cb56-77" data-line-number="77"></a>
<a class="sourceLine" id="cb56-78" data-line-number="78"><span class="co"># Plot selection results</span></a>
<a class="sourceLine" id="cb56-79" data-line-number="79">donor_selection_info &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/p1-selection.csv&quot;</span>, <span class="dt">row.names=</span><span class="dv">1</span>)</a>
<a class="sourceLine" id="cb56-80" data-line-number="80">donor_selection_info &lt;-<span class="st"> </span><span class="kw">read_csv</span>(<span class="st">&quot;data/p1-selection.csv&quot;</span>)</a>
<a class="sourceLine" id="cb56-81" data-line-number="81"></a>
<a class="sourceLine" id="cb56-82" data-line-number="82"><span class="co"># add missing donor (simulation failed)</span></a>
<a class="sourceLine" id="cb56-83" data-line-number="83">donor_selection_info<span class="op">$</span>ps1 &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(donor_selection_info<span class="op">$</span>ps1)</a>
<a class="sourceLine" id="cb56-84" data-line-number="84">donor_selection_info<span class="op">$</span>selection &lt;-<span class="st"> </span><span class="kw">factor</span>(donor_selection_info<span class="op">$</span>selection, <span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))</a>
<a class="sourceLine" id="cb56-85" data-line-number="85"></a>
<a class="sourceLine" id="cb56-86" data-line-number="86">selection_plot &lt;-<span class="st"> </span><span class="kw">ggplot</span>(donor_selection_info, <span class="kw">aes</span>(<span class="dt">x=</span>ps1, <span class="dt">y=</span><span class="kw">factor</span>(donor, <span class="dt">levels=</span><span class="kw">rev</span>(donor_filt_order)), <span class="dt">colour=</span>selection)) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-87" data-line-number="87"><span class="st">                </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">4</span>, <span class="dt">alpha =</span> <span class="fl">0.7</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-88" data-line-number="88"><span class="st">                </span><span class="kw">coord_cartesian</span>(<span class="dt">xlim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-89" data-line-number="89"><span class="st">                </span><span class="kw">scale_colour_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="dt">neutral=</span><span class="st">&quot;#1283FF&quot;</span>, <span class="dt">selected=</span><span class="st">&quot;#144E7B&quot;</span>, <span class="dt">undetermined=</span><span class="st">&quot;#CACACA&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-90" data-line-number="90"><span class="st">                </span><span class="kw">geom_vline</span>(<span class="dt">xintercept=</span><span class="kw">c</span>(<span class="fl">0.3</span>, <span class="fl">0.7</span>), <span class="dt">colour=</span><span class="st">&quot;#808080&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-91" data-line-number="91"><span class="st">                </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.3</span>, <span class="fl">0.5</span>, <span class="fl">0.7</span>, <span class="fl">1.0</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-92" data-line-number="92"><span class="st">                </span><span class="kw">theme_bw</span>(<span class="dv">16</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-93" data-line-number="93"><span class="st">        </span><span class="kw">labs</span>(<span class="dt">x =</span> <span class="st">&quot;Probability of Selection&quot;</span>, <span class="dt">y =</span> <span class="st">&quot;&quot;</span>, <span class="dt">colour =</span> <span class="st">&quot;Selection&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-94" data-line-number="94"><span class="st">    </span><span class="co"># guides(colour=guide_legend(title.position=&quot;top&quot;,</span></a>
<a class="sourceLine" id="cb56-95" data-line-number="95"><span class="st">    </span><span class="co">#                                  title.hjust =0.5)) +</span></a>
<a class="sourceLine" id="cb56-96" data-line-number="96"><span class="st">    </span><span class="kw">guides</span>(<span class="dt">colour =</span> <span class="kw">guide_legend</span>(<span class="dt">ncol =</span> <span class="dv">1</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-97" data-line-number="97"><span class="st">        </span><span class="kw">theme</span>(<span class="dt">axis.ticks.y =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb56-98" data-line-number="98">        <span class="dt">axis.line =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb56-99" data-line-number="99">            <span class="dt">legend.position =</span> <span class="st">&quot;top&quot;</span>,</a>
<a class="sourceLine" id="cb56-100" data-line-number="100">            <span class="dt">legend.key.size =</span> <span class="kw">unit</span>(<span class="fl">0.3</span>,<span class="st">&quot;in&quot;</span>),</a>
<a class="sourceLine" id="cb56-101" data-line-number="101">            <span class="dt">axis.title.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">36</span>, <span class="dt">face =</span> <span class="st">&quot;plain&quot;</span>),</a>
<a class="sourceLine" id="cb56-102" data-line-number="102">            <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">32</span>, <span class="dt">face =</span> <span class="st">&quot;plain&quot;</span>),</a>
<a class="sourceLine" id="cb56-103" data-line-number="103">            <span class="dt">axis.text.y =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb56-104" data-line-number="104">            <span class="dt">legend.justification =</span> <span class="st">&quot;center&quot;</span>,</a>
<a class="sourceLine" id="cb56-105" data-line-number="105">            <span class="dt">legend.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">28</span>, <span class="dt">face =</span> <span class="st">&quot;plain&quot;</span>),</a>
<a class="sourceLine" id="cb56-106" data-line-number="106">            <span class="dt">legend.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">30</span>, <span class="dt">face =</span> <span class="st">&quot;plain&quot;</span>))</a>
<a class="sourceLine" id="cb56-107" data-line-number="107"></a>
<a class="sourceLine" id="cb56-108" data-line-number="108"></a>
<a class="sourceLine" id="cb56-109" data-line-number="109"><span class="co"># Plot total number of mutations per clone</span></a>
<a class="sourceLine" id="cb56-110" data-line-number="110">donor_info_filt_subset &lt;-</a>
<a class="sourceLine" id="cb56-111" data-line-number="111"><span class="st">  </span>donor_info_filt[,<span class="kw">c</span>(<span class="dv">1</span>,<span class="kw">grep</span>(<span class="st">&quot;clone&quot;</span>,<span class="kw">colnames</span>(donor_info_filt)))]</a>
<a class="sourceLine" id="cb56-112" data-line-number="112"><span class="co"># Remove columns that do not relate to number of mutations per clone:</span></a>
<a class="sourceLine" id="cb56-113" data-line-number="113">donor_info_filt_subset &lt;-<span class="st"> </span>donor_info_filt_subset[,<span class="kw">c</span>(<span class="op">-</span><span class="dv">2</span>,<span class="op">-</span><span class="dv">7</span>)]</a>
<a class="sourceLine" id="cb56-114" data-line-number="114">donor_info_filt_melt &lt;-<span class="st"> </span>reshape2<span class="op">::</span><span class="kw">melt</span>(donor_info_filt_subset, <span class="st">&quot;donor_short&quot;</span>)</a>
<a class="sourceLine" id="cb56-115" data-line-number="115">donor_info_filt_melt<span class="op">$</span>variable &lt;-<span class="st"> </span><span class="kw">substr</span>(donor_info_filt_melt<span class="op">$</span>variable, <span class="dv">6</span>, <span class="dv">6</span>)</a>
<a class="sourceLine" id="cb56-116" data-line-number="116">num_clones_plot_filt &lt;-<span class="st"> </span><span class="kw">ggplot</span>(</a>
<a class="sourceLine" id="cb56-117" data-line-number="117">  donor_info_filt_melt,</a>
<a class="sourceLine" id="cb56-118" data-line-number="118">  <span class="kw">aes</span>(variable, <span class="kw">factor</span>(donor_short, <span class="dt">levels =</span> <span class="kw">rev</span>(donor_filt_order)))) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-119" data-line-number="119"><span class="st">  </span><span class="kw">geom_tile</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> value), <span class="dt">colour =</span> <span class="st">&quot;white&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-120" data-line-number="120"><span class="st">  </span><span class="kw">scale_fill_distiller</span>(<span class="dt">palette =</span> <span class="st">&quot;BuPu&quot;</span>, <span class="dt">values =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="fl">0.05</span>,<span class="fl">0.1</span>,<span class="fl">0.15</span>,<span class="fl">0.25</span>,<span class="fl">0.5</span>,<span class="dv">1</span>),</a>
<a class="sourceLine" id="cb56-121" data-line-number="121">                       <span class="dt">na.value =</span> <span class="st">&quot;white&quot;</span>, <span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">25</span>,<span class="dv">50</span>,<span class="dv">75</span>),</a>
<a class="sourceLine" id="cb56-122" data-line-number="122">                       <span class="dt">direction =</span> <span class="dv">1</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-123" data-line-number="123"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">x =</span> <span class="st">&quot;Clone&quot;</span>, <span class="dt">y =</span> <span class="st">&quot;&quot;</span>, <span class="dt">fill =</span> <span class="st">&quot;Number of mutations&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-124" data-line-number="124"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.ticks.y =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb56-125" data-line-number="125">        <span class="dt">axis.line =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb56-126" data-line-number="126">        <span class="dt">legend.position =</span> <span class="st">&quot;top&quot;</span>,</a>
<a class="sourceLine" id="cb56-127" data-line-number="127">        <span class="dt">legend.key.size =</span> <span class="kw">unit</span>(<span class="fl">0.3</span>,<span class="st">&quot;in&quot;</span>),</a>
<a class="sourceLine" id="cb56-128" data-line-number="128">        <span class="dt">axis.title.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">36</span>, <span class="dt">face =</span> <span class="st">&quot;plain&quot;</span>),</a>
<a class="sourceLine" id="cb56-129" data-line-number="129">        <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">32</span>, <span class="dt">face =</span> <span class="st">&quot;plain&quot;</span>),</a>
<a class="sourceLine" id="cb56-130" data-line-number="130">        <span class="dt">axis.text.y =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb56-131" data-line-number="131">        <span class="dt">legend.justification =</span> <span class="st">&quot;center&quot;</span>,</a>
<a class="sourceLine" id="cb56-132" data-line-number="132">        <span class="dt">legend.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">28</span>, <span class="dt">face =</span> <span class="st">&quot;plain&quot;</span>),</a>
<a class="sourceLine" id="cb56-133" data-line-number="133">        <span class="dt">legend.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">30</span>, <span class="dt">face =</span> <span class="st">&quot;plain&quot;</span>))</a>
<a class="sourceLine" id="cb56-134" data-line-number="134"></a>
<a class="sourceLine" id="cb56-135" data-line-number="135"><span class="co">## Combine above plots into Fig 2a</span></a>
<a class="sourceLine" id="cb56-136" data-line-number="136">fig_2a &lt;-<span class="st"> </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(age_plot_filt, num_mut_plot_filt,</a>
<a class="sourceLine" id="cb56-137" data-line-number="137">                             sig_decomp_plot_filt, selection_plot, num_clones_plot_filt,</a>
<a class="sourceLine" id="cb56-138" data-line-number="138">                             <span class="dt">nrow =</span> <span class="dv">1</span>, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="dv">3</span>, <span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">4</span>), <span class="dt">align =</span> <span class="st">&quot;h&quot;</span>,</a>
<a class="sourceLine" id="cb56-139" data-line-number="139">                             <span class="dt">axis =</span> <span class="st">&quot;t&quot;</span>, <span class="dt">scale =</span>  <span class="kw">c</span>(<span class="dv">1</span>, <span class="fl">0.988</span>, <span class="fl">0.988</span>, <span class="fl">0.988</span>))</a>
<a class="sourceLine" id="cb56-140" data-line-number="140"><span class="kw">ggsave</span>(<span class="st">&quot;figures/overview_lines/overview_lines_BuPu.png&quot;</span>,</a>
<a class="sourceLine" id="cb56-141" data-line-number="141">       <span class="dt">plot =</span> fig_2a, <span class="dt">width =</span> <span class="dv">30</span>, <span class="dt">height =</span> <span class="dv">14</span>)</a>
<a class="sourceLine" id="cb56-142" data-line-number="142"></a>
<a class="sourceLine" id="cb56-143" data-line-number="143">fig_2a</a></code></pre></div>
<p><img src="figure/overview_lines.Rmd/donor-info-summary-plot-1.png" width="2880" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-donor-info-summary-plot-1">
Past versions of donor-info-summary-plot-1.png
</button>
</p>
<div id="fig-donor-info-summary-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/0540cdbc554ab43e0ff60557b7ceb82fb13f3d75/docs/figure/overview_lines.Rmd/donor-info-summary-plot-1.png" target="_blank">0540cdb</a>
</td>
<td>
davismcc
</td>
<td>
2018-09-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/f0ed980029a115234bc2edff312e5e52056d4eed/docs/figure/overview_lines.Rmd/donor-info-summary-plot-1.png" target="_blank">f0ed980</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-31
</td>
</tr>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/figure/overview_lines.Rmd/donor-info-summary-plot-1.png" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="plot-cell-assignment-rate-vs-hamming-distance" class="section level2">
<h2>Plot cell assignment rate vs Hamming Distance</h2>
<p>Finally, we plot the cell assignment rate (from <code>cardelino</code>) against the minimum Hamming distance (minimum number of variants distinguishing a pair of clones) for each line.</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb57-1" data-line-number="1">fig_2c_no_size &lt;-<span class="st"> </span><span class="kw">ggplot</span>(</a>
<a class="sourceLine" id="cb57-2" data-line-number="2">  donor_info_filt, </a>
<a class="sourceLine" id="cb57-3" data-line-number="3">  <span class="kw">aes</span>(<span class="dt">y =</span> <span class="kw">as.numeric</span>(percent_assigned_cells), <span class="dt">x =</span> <span class="kw">as.numeric</span>(min_hamming_dist), </a>
<a class="sourceLine" id="cb57-4" data-line-number="4">      <span class="dt">fill =</span> total_unst_cells)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb57-5" data-line-number="5"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">pch =</span> <span class="dv">21</span>, <span class="dt">colour =</span> <span class="st">&quot;gray50&quot;</span>, <span class="dt">size =</span> <span class="dv">4</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb57-6" data-line-number="6"><span class="st">  </span><span class="kw">scale_shape_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="dv">4</span>, <span class="dv">19</span>)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb57-7" data-line-number="7"><span class="st">  </span><span class="kw">ylim</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb57-8" data-line-number="8"><span class="st">  </span><span class="kw">scale_x_log10</span>() <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb57-9" data-line-number="9"><span class="st">  </span><span class="kw">scale_fill_viridis</span>(<span class="dt">option =</span> <span class="st">&quot;magma&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb57-10" data-line-number="10"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Proportion cells assigned&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb57-11" data-line-number="11"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Minimum number of variants distinguishing clones&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb57-12" data-line-number="12"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">title =</span> <span class="st">&quot;&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb57-13" data-line-number="13"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">fill =</span> <span class="st">&quot;Number of cells&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb57-14" data-line-number="14"><span class="st">  </span><span class="kw">theme_bw</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb57-15" data-line-number="15"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">9</span>,<span class="dt">face =</span> <span class="st">&quot;bold&quot;</span>), </a>
<a class="sourceLine" id="cb57-16" data-line-number="16">        <span class="dt">axis.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">9</span>, <span class="dt">face =</span> <span class="st">&quot;bold&quot;</span>),</a>
<a class="sourceLine" id="cb57-17" data-line-number="17">        <span class="dt">axis.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">10</span>, <span class="dt">face =</span> <span class="st">&quot;bold&quot;</span>),</a>
<a class="sourceLine" id="cb57-18" data-line-number="18">        <span class="dt">plot.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">9</span>, <span class="dt">hjust =</span> <span class="fl">0.5</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb57-19" data-line-number="19"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">strip.background =</span> <span class="kw">element_blank</span>()) <span class="op">+</span></a>
<a class="sourceLine" id="cb57-20" data-line-number="20"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.justification =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">0</span>), </a>
<a class="sourceLine" id="cb57-21" data-line-number="21">        <span class="dt">legend.position =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">0</span>), </a>
<a class="sourceLine" id="cb57-22" data-line-number="22">        <span class="dt">legend.direction =</span> <span class="st">&quot;horizontal&quot;</span>,</a>
<a class="sourceLine" id="cb57-23" data-line-number="23">        <span class="dt">legend.background =</span> <span class="kw">element_rect</span>(<span class="dt">fill =</span> <span class="st">&quot;white&quot;</span>, <span class="dt">linetype =</span> <span class="dv">1</span>, </a>
<a class="sourceLine" id="cb57-24" data-line-number="24">                                         <span class="dt">colour =</span> <span class="st">&quot;black&quot;</span>, <span class="dt">size =</span> <span class="fl">0.2</span>),</a>
<a class="sourceLine" id="cb57-25" data-line-number="25">        <span class="dt">legend.key.size =</span> <span class="kw">unit</span>(<span class="fl">0.25</span>, <span class="st">&quot;cm&quot;</span>))</a>
<a class="sourceLine" id="cb57-26" data-line-number="26"></a>
<a class="sourceLine" id="cb57-27" data-line-number="27"><span class="kw">ggsave</span>(<span class="st">&quot;figures/overview_lines/cell_assignment_vs_min_hamming_dist.png&quot;</span>,</a>
<a class="sourceLine" id="cb57-28" data-line-number="28">       <span class="dt">plot =</span> fig_2c_no_size, <span class="dt">width =</span> <span class="dv">12</span>, <span class="dt">height =</span> <span class="dv">12</span>, <span class="dt">dpi =</span> <span class="dv">300</span>, <span class="dt">units =</span> <span class="st">&quot;cm&quot;</span>)</a>
<a class="sourceLine" id="cb57-29" data-line-number="29">fig_2c_no_size</a></code></pre></div>
<p><img src="figure/overview_lines.Rmd/plot-cell-assignment-vs-hamming-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-cell-assignment-vs-hamming-1">
Past versions of plot-cell-assignment-vs-hamming-1.png
</button>
</p>
<div id="fig-plot-cell-assignment-vs-hamming-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/figure/overview_lines.Rmd/plot-cell-assignment-vs-hamming-1.png" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="save-data-to-file" class="section level2">
<h2>Save data to file</h2>
<p>Save the donor info dataframe to <code>output/line_info.tsv</code>.</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb58-1" data-line-number="1"><span class="kw">write_tsv</span>(donor_info_filt, <span class="dt">path =</span> <span class="st">&quot;output/line_info.tsv&quot;</span>, <span class="dt">col_names =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb59-1" data-line-number="1">devtools<span class="op">::</span><span class="kw">session_info</span>()</a></code></pre></div>
<pre><code>─ Session info ──────────────────────────────────────────────────────────
 setting  value                       
 version  R version 3.6.0 (2019-04-26)
 os       Ubuntu 18.04.3 LTS          
 system   x86_64, linux-gnu           
 ui       X11                         
 language (EN)                        
 collate  en_AU.UTF-8                 
 ctype    en_AU.UTF-8                 
 tz       Australia/Melbourne         
 date     2019-10-30                  

─ Packages ──────────────────────────────────────────────────────────────
 package                     * version   date       lib
 AnnotationDbi               * 1.46.1    2019-08-20 [1]
 assertthat                    0.2.1     2019-03-21 [1]
 backports                     1.1.4     2019-04-10 [1]
 beeswarm                      0.2.3     2016-04-25 [1]
 Biobase                     * 2.44.0    2019-05-02 [1]
 BiocGenerics                * 0.30.0    2019-05-02 [1]
 BiocNeighbors                 1.2.0     2019-05-02 [1]
 BiocParallel                * 1.18.1    2019-08-06 [1]
 BiocSingular                  1.0.0     2019-05-02 [1]
 Biostrings                    2.52.0    2019-05-02 [1]
 bit                           1.1-14    2018-05-29 [1]
 bit64                         0.9-7     2017-05-08 [1]
 bitops                        1.0-6     2013-08-17 [1]
 blob                          1.2.0     2019-07-09 [1]
 BSgenome                      1.52.0    2019-05-02 [1]
 BSgenome.Hsapiens.UCSC.hg19   1.4.0     2019-06-28 [1]
 callr                         3.3.2     2019-09-22 [1]
 caTools                       1.17.1.2  2019-03-06 [1]
 cli                           1.1.0     2019-03-19 [1]
 clue                          0.3-57    2019-02-25 [1]
 cluster                       2.1.0     2019-06-19 [1]
 coda                          0.19-3    2019-07-05 [1]
 codetools                     0.2-16    2018-12-24 [4]
 colorspace                    1.4-1     2019-03-18 [1]
 cowplot                     * 1.0.0     2019-07-11 [1]
 crayon                        1.3.4     2017-09-16 [1]
 DBI                           1.0.0     2018-05-02 [1]
 deconstructSigs             * 1.8.0     2016-07-29 [1]
 DelayedArray                * 0.10.0    2019-05-02 [1]
 DelayedMatrixStats            1.6.1     2019-09-08 [1]
 desc                          1.2.0     2018-05-01 [1]
 devtools                      2.2.1     2019-09-24 [1]
 digest                        0.6.21    2019-09-20 [1]
 dplyr                       * 0.8.3     2019-07-04 [1]
 dqrng                         0.2.1     2019-05-17 [1]
 dynamicTreeCut                1.63-1    2016-03-11 [1]
 edgeR                       * 3.26.8    2019-09-01 [1]
 ellipsis                      0.3.0     2019-09-20 [1]
 evaluate                      0.14      2019-05-28 [1]
 farver                        1.1.0     2018-11-20 [1]
 fs                            1.3.1     2019-05-06 [1]
 gdata                         2.18.0    2017-06-06 [1]
 GenomeInfoDb                * 1.20.0    2019-05-02 [1]
 GenomeInfoDbData              1.2.1     2019-04-30 [1]
 GenomicAlignments             1.20.1    2019-06-18 [1]
 GenomicRanges               * 1.36.1    2019-09-06 [1]
 ggbeeswarm                    0.6.0     2017-08-07 [1]
 ggforce                     * 0.3.1     2019-08-20 [1]
 ggplot2                     * 3.2.1     2019-08-10 [1]
 ggrepel                     * 0.8.1     2019-05-07 [1]
 ggridges                    * 0.5.1     2018-09-27 [1]
 git2r                         0.26.1    2019-06-29 [1]
 glue                          1.3.1     2019-03-12 [1]
 gplots                      * 3.0.1.1   2019-01-27 [1]
 gridExtra                     2.3       2017-09-09 [1]
 gtable                        0.3.0     2019-03-25 [1]
 gtools                        3.8.1     2018-06-26 [1]
 hms                           0.5.1     2019-08-23 [1]
 htmltools                     0.3.6     2017-04-28 [1]
 igraph                        1.2.4.1   2019-04-22 [1]
 inline                        0.3.15    2018-05-18 [1]
 IRanges                     * 2.18.3    2019-09-24 [1]
 irlba                         2.3.3     2019-02-05 [1]
 KernSmooth                    2.23-15   2015-06-29 [1]
 knitr                         1.25      2019-09-18 [1]
 labeling                      0.3       2014-08-23 [1]
 lattice                       0.20-38   2018-11-04 [4]
 lazyeval                      0.2.2     2019-03-15 [1]
 limma                       * 3.40.6    2019-07-26 [1]
 locfit                        1.5-9.1   2013-04-20 [1]
 loo                           2.1.0     2019-03-13 [1]
 magrittr                      1.5       2014-11-22 [1]
 MASS                          7.3-51.1  2018-11-01 [4]
 Matrix                        1.2-17    2019-03-22 [1]
 matrixStats                 * 0.55.0    2019-09-07 [1]
 memoise                       1.1.0     2017-04-21 [1]
 munsell                       0.5.0     2018-06-12 [1]
 org.Hs.eg.db                * 3.8.2     2019-05-01 [1]
 pillar                        1.4.2     2019-06-29 [1]
 pkgbuild                      1.0.5     2019-08-26 [1]
 pkgconfig                     2.0.3     2019-09-22 [1]
 pkgload                       1.0.2     2018-10-29 [1]
 plyr                          1.8.4     2016-06-08 [1]
 polyclip                      1.10-0    2019-03-14 [1]
 prettyunits                   1.0.2     2015-07-13 [1]
 processx                      3.4.1     2019-07-18 [1]
 ps                            1.3.0     2018-12-21 [1]
 purrr                         0.3.2     2019-03-15 [1]
 R6                            2.4.0     2019-02-14 [1]
 RColorBrewer                  1.1-2     2014-12-07 [1]
 Rcpp                        * 1.0.2     2019-07-25 [1]
 RCurl                         1.95-4.12 2019-03-04 [1]
 readr                       * 1.3.1     2018-12-21 [1]
 remotes                       2.1.0     2019-06-24 [1]
 reshape2                      1.4.3     2017-12-11 [1]
 rlang                         0.4.0     2019-06-25 [1]
 rmarkdown                     1.15      2019-08-21 [1]
 rprojroot                     1.3-2     2018-01-03 [1]
 Rsamtools                     2.0.1     2019-09-19 [1]
 RSQLite                       2.1.2     2019-07-24 [1]
 rstan                         2.19.2    2019-07-09 [1]
 rsvd                          1.0.2     2019-07-29 [1]
 rtracklayer                   1.44.4    2019-09-06 [1]
 S4Vectors                   * 0.22.1    2019-09-09 [1]
 scales                        1.0.0     2018-08-09 [1]
 scater                      * 1.12.2    2019-05-24 [1]
 scran                       * 1.12.1    2019-05-27 [1]
 sessioninfo                   1.1.1     2018-11-05 [1]
 sigfit                      * 1.3.2     2019-10-30 [1]
 SingleCellExperiment        * 1.6.0     2019-05-02 [1]
 StanHeaders                   2.19.0    2019-09-07 [1]
 statmod                       1.4.32    2019-05-29 [1]
 stringi                       1.4.3     2019-03-12 [1]
 stringr                       1.4.0     2019-02-10 [1]
 SummarizedExperiment        * 1.14.1    2019-07-31 [1]
 testthat                      2.2.1     2019-07-25 [1]
 tibble                        2.1.3     2019-06-06 [1]
 tidyselect                    0.2.5     2018-10-11 [1]
 tweenr                        1.0.1     2018-12-14 [1]
 usethis                       1.5.1     2019-07-04 [1]
 vctrs                         0.2.0     2019-07-05 [1]
 vipor                         0.4.5     2017-03-22 [1]
 viridis                     * 0.5.1     2018-03-29 [1]
 viridisLite                 * 0.3.0     2018-02-01 [1]
 whisker                       0.4       2019-08-28 [1]
 withr                         2.1.2     2018-03-15 [1]
 workflowr                     1.4.0     2019-06-08 [1]
 xfun                          0.9       2019-08-21 [1]
 XML                           3.98-1.20 2019-06-06 [1]
 XVector                       0.24.0    2019-05-02 [1]
 yaml                          2.2.0     2018-07-25 [1]
 zeallot                       0.1.0     2018-01-28 [1]
 zlibbioc                      1.30.0    2019-05-02 [1]
 source                       
 Bioconductor                 
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 Bioconductor                 
 Bioconductor                 
 Bioconductor                 
 Bioconductor                 
 Bioconductor                 
 Bioconductor                 
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 Bioconductor                 
 Bioconductor                 
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.5.2)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 Bioconductor                 
 Bioconductor                 
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 Bioconductor                 
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 Bioconductor                 
 Bioconductor                 
 Bioconductor                 
 Bioconductor                 
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 Bioconductor                 
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.5.1)               
 CRAN (R 3.6.0)               
 Bioconductor                 
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.5.1)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 Bioconductor                 
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 Bioconductor                 
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 Bioconductor                 
 Bioconductor                 
 CRAN (R 3.6.0)               
 Bioconductor                 
 Bioconductor                 
 CRAN (R 3.6.0)               
 Github (kgori/sigfit@2c16353)
 Bioconductor                 
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 Bioconductor                 
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 Bioconductor                 
 CRAN (R 3.6.0)               
 CRAN (R 3.6.0)               
 Bioconductor                 

[1] /home/AD.SVI.EDU.AU/dmccarthy/R/x86_64-pc-linux-gnu-library/3.6
[2] /usr/local/lib/R/site-library
[3] /usr/lib/R/site-library
[4] /usr/lib/R/library</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
