<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Yuanhua Huang &amp; Davis J. McCarthy" />


<title>Variance components analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
div.sourceCode {
  overflow-x: visible;
}
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fibroblast-clonality</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/davismcc/fibroblast-clonality">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Variance components analysis</h1>
<h4 class="author"><em>Yuanhua Huang &amp; Davis J. McCarthy</em></h4>

</div>


<p><strong>Last updated:</strong> 2018-08-27</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>R Markdown file:</strong> up-to-date </summary></p>
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(20180807)</code> </summary></p>
<p>The command <code>set.seed(20180807)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/davismcc/fibroblast-clonality/tree/678546d87f50bae3c1430f12e82c511a6eecdf6f" target="_blank">678546d</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .vscode/
    Ignored:    code/.DS_Store
    Ignored:    data/raw/
    Ignored:    src/.DS_Store
    Ignored:    src/.ipynb_checkpoints/
    Ignored:    src/Rmd/.Rhistory

Untracked files:
    Untracked:  Snakefile_clonality
    Untracked:  Snakefile_somatic_calling
    Untracked:  code/analysis_for_garx.Rmd
    Untracked:  code/selection/
    Untracked:  code/yuanhua/
    Untracked:  data/canopy/
    Untracked:  data/cell_assignment/
    Untracked:  data/de_analysis_FTv62/
    Untracked:  data/donor_info_070818.txt
    Untracked:  data/donor_info_core.csv
    Untracked:  data/donor_neutrality.tsv
    Untracked:  data/exome-point-mutations/
    Untracked:  data/fdr10.annot.txt.gz
    Untracked:  data/human_H_v5p2.rdata
    Untracked:  data/human_c2_v5p2.rdata
    Untracked:  data/human_c6_v5p2.rdata
    Untracked:  data/neg-bin-rsquared-petr.csv
    Untracked:  data/neutralitytestr-petr.tsv
    Untracked:  data/sce_merged_donors_cardelino_donorid_all_qc_filt.rds
    Untracked:  data/sce_merged_donors_cardelino_donorid_all_with_qc_labels.rds
    Untracked:  data/sce_merged_donors_cardelino_donorid_unstim_qc_filt.rds
    Untracked:  data/sces/
    Untracked:  data/selection/
    Untracked:  data/simulations/
    Untracked:  data/variance_components/
    Untracked:  figures/
    Untracked:  output/differential_expression/
    Untracked:  output/donor_specific/
    Untracked:  output/line_info.tsv
    Untracked:  output/nvars_by_category_by_donor.tsv
    Untracked:  output/nvars_by_category_by_line.tsv
    Untracked:  output/variance_components/
    Untracked:  references/
    Untracked:  tree.txt

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/davismcc/fibroblast-clonality/9ec2a595a73ee48db72d6c6d860dc45fb0192a36/docs/variance_components.html" target="_blank">9ec2a59</a>
</td>
<td style="text-align:left;">
davismcc
</td>
<td style="text-align:left;">
2018-08-26
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/davismcc/fibroblast-clonality/36acf15b30f110282dd56b004c5d478d560b75e1/docs/variance_components.html" target="_blank">36acf15</a>
</td>
<td style="text-align:left;">
davismcc
</td>
<td style="text-align:left;">
2018-08-25
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/davismcc/fibroblast-clonality/blob/d618fe52b90fd1d6e4f9613c8d7bbe3a3afde245/analysis/variance_components.Rmd" target="_blank">d618fe5</a>
</td>
<td style="text-align:left;">
davismcc
</td>
<td style="text-align:left;">
2018-08-25
</td>
<td style="text-align:left;">
Updating analyses
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/davismcc/fibroblast-clonality/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/variance_components.html" target="_blank">090c1b9</a>
</td>
<td style="text-align:left;">
davismcc
</td>
<td style="text-align:left;">
2018-08-24
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/davismcc/fibroblast-clonality/d2e8b3145e4600ec75d0d0df9ad1e210ecb1bdd3/docs/variance_components.html" target="_blank">d2e8b31</a>
</td>
<td style="text-align:left;">
davismcc
</td>
<td style="text-align:left;">
2018-08-19
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/davismcc/fibroblast-clonality/1489d32ff1b8d5073a4170baec953cdbe4fcc14a/docs/variance_components.html" target="_blank">1489d32</a>
</td>
<td style="text-align:left;">
davismcc
</td>
<td style="text-align:left;">
2018-08-17
</td>
<td style="text-align:left;">
Add html files
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/davismcc/fibroblast-clonality/blob/c5c1416614dc3fbaf1d84af87b3b00004cd22e38/analysis/variance_components.Rmd" target="_blank">c5c1416</a>
</td>
<td style="text-align:left;">
davismcc
</td>
<td style="text-align:left;">
2018-08-17
</td>
<td style="text-align:left;">
Adding variance components analysis.
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<div id="load-libraries" class="section level2">
<h2>Load libraries</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(viridis)
<span class="kw">library</span>(tidyverse)
<span class="kw">library</span>(ggpubr)
<span class="kw">library</span>(ggrepel)
<span class="kw">library</span>(cowplot)
<span class="kw">library</span>(reshape2)
<span class="kw">dir.create</span>(<span class="st">&quot;figures/variance_components&quot;</span>, <span class="dt">showWarnings =</span> <span class="ot">FALSE</span>, 
           <span class="dt">recursive =</span> <span class="ot">TRUE</span>)</code></pre></div>
</div>
<div id="results-aggregated-across-all-lines" class="section level2">
<h2>Results aggregated across all lines</h2>
<p>Previously, we conducted variance components analysis combining scRNA-seq data across all lines. We used the <code>variancePartion</code> package, fitting assigned clone, line, plate (on which cells were processed) and cellular detection rate (cdr; proportion of genes per cell with non-zero expression) as random effects.</p>
<p>The output of the model gives us the proportion of variance explained, for each gene, by clone, line, plate, cdr and residuals.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">varPart.df &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(
  <span class="kw">read.table</span>(<span class="st">&quot;data/variance_components/fit_per_gene_highVar.csv&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;,&quot;</span>))
<span class="kw">colnames</span>(varPart.df) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;clone&quot;</span>, <span class="st">&quot;line&quot;</span>, <span class="st">&quot;plate&quot;</span>, <span class="st">&quot;cdr&quot;</span>, <span class="st">&quot;residuals&quot;</span>)
varPart.df<span class="op">$</span>gene_id &lt;-<span class="st"> </span><span class="kw">rownames</span>(varPart.df)
varPart.df &lt;-<span class="st"> </span><span class="kw">as_data_frame</span>(varPart.df)
<span class="kw">head</span>(varPart.df)</code></pre></div>
<pre><code># A tibble: 6 x 6
     clone    line   plate      cdr residuals gene_id               
     &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;    &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;                 
1 3.54e- 3 0.0179  0.0216  0.0124       0.945 ENSG00000001167_NFYA  
2 0.       0.167   0.0215  0.0354       0.776 ENSG00000001626_CFTR  
3 0.       0.00580 0.0143  0.000248     0.980 ENSG00000001629_ANKIB1
4 5.35e- 3 0.0194  0.0215  0.0152       0.939 ENSG00000001631_KRIT1 
5 9.73e-15 0.369   0.00582 0.0336       0.592 ENSG00000002016_RAD52 
6 0.       0.125   0.0827  0.0143       0.778 ENSG00000002330_BAD   </code></pre>
<p>We ran the analysis for 4998 highly variable genes (as identified using methods in the <code>scran</code> package).</p>
<p>We detect 194 genes with more than 5% of variance explained by clone and 1034 genes for which variance explained by clone is both greater than 1% and also greater than the variance explained by plate (a good proxy for technical effects on variability in gene expression between cells).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">varPart.mel &lt;-<span class="st"> </span>tidyr<span class="op">::</span><span class="kw">gather</span>(varPart.df, <span class="dt">key =</span> <span class="st">&quot;variable&quot;</span>, <span class="dt">value =</span> <span class="st">&quot;value&quot;</span>, 
                             <span class="op">-</span>gene_id)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dat_dir &lt;-<span class="st"> &quot;data/variance_components/donorVar/&quot;</span>

fig.violin &lt;-<span class="st"> </span>varPart.mel <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">variable =</span> <span class="kw">replace</span>(variable, variable <span class="op">==</span><span class="st"> &quot;cdr&quot;</span>, 
                                   <span class="st">&quot;cell. det. rate&quot;</span>)) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">variable =</span> <span class="kw">factor</span>(variable, 
                                  <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;residuals&quot;</span>, <span class="st">&quot;cell. det. rate&quot;</span>,
                                             <span class="st">&quot;plate&quot;</span>, <span class="st">&quot;line&quot;</span>, <span class="st">&quot;clone&quot;</span>))) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> variable, <span class="dt">y =</span> <span class="dv">100</span> <span class="op">*</span><span class="st"> </span>value)) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">geom_violin</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> variable), <span class="dt">scale =</span> <span class="st">&quot;width&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">geom_boxplot</span>(<span class="dt">width =</span> <span class="fl">0.07</span>, <span class="dt">fill =</span> <span class="st">&quot;grey&quot;</span>, <span class="dt">outlier.colour =</span> <span class="st">&quot;black&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Variance explained (%)&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;gray70&quot;</span>, <span class="st">&quot;#f7efe2&quot;</span>, <span class="st">&quot;#f9a603&quot;</span>, <span class="st">&quot;#f25c00&quot;</span>,
                               <span class="st">&quot;#f70025&quot;</span>)) <span class="op">+</span>
<span class="st">  </span><span class="kw">coord_flip</span>() <span class="op">+</span>
<span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">theme</span>(<span class="dt">plot.title =</span> <span class="kw">element_text</span>(<span class="dt">hjust =</span> <span class="fl">0.5</span>))

<span class="kw">ggsave</span>(<span class="st">&quot;figures/variance_components/varpart_violin_alllines.png&quot;</span>, 
       fig.violin, <span class="dt">height =</span> <span class="dv">4</span>, <span class="dt">width =</span> <span class="fl">6.5</span>, <span class="dt">dpi =</span> <span class="dv">300</span>)
<span class="kw">ggsave</span>(<span class="st">&quot;figures/variance_components/varpart_violin_alllines.pdf&quot;</span>, 
       fig.violin, <span class="dt">height =</span> <span class="dv">4</span>, <span class="dt">width =</span> <span class="fl">6.5</span>, <span class="dt">dpi =</span> <span class="dv">300</span>)
<span class="kw">ggsave</span>(<span class="st">&quot;figures/variance_components/varpart_violin_alllines_skinny.png&quot;</span>, 
       fig.violin, <span class="dt">height =</span> <span class="dv">4</span>, <span class="dt">width =</span> <span class="fl">4.5</span>, <span class="dt">dpi =</span> <span class="dv">300</span>)
<span class="kw">ggsave</span>(<span class="st">&quot;figures/variance_components/varpart_violin_alllines_skinny.pdf&quot;</span>, 
       fig.violin, <span class="dt">height =</span> <span class="dv">4</span>, <span class="dt">width =</span> <span class="fl">4.5</span>, <span class="dt">dpi =</span> <span class="dv">300</span>)

fig.violin</code></pre></div>
<p><img src="figure/variance_components.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-4-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/davismcc/fibroblast-clonality/blob/d2e8b3145e4600ec75d0d0df9ad1e210ecb1bdd3/docs/figure/variance_components.Rmd/unnamed-chunk-4-1.png" target="_blank">d2e8b31</a>
</td>
<td style="text-align:left;">
davismcc
</td>
<td style="text-align:left;">
2018-08-19
</td>
</tr>
</tbody>
</table>
<p></details></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># plotVarPart( varPart_mat )</span>
<span class="co"># idx_order &lt;- order(varPart_mat$clone, decreasing = TRUE)</span>
<span class="co"># plotPercentBars( varPart_mat[idx_order[1:10],] )</span></code></pre></div>
<div id="plot-line-component-against-clone-component" class="section level3">
<h3>Plot line component against clone component</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">varPart.df <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> <span class="dv">100</span> <span class="op">*</span><span class="st"> </span>line, <span class="dt">y =</span> <span class="dv">100</span> <span class="op">*</span><span class="st"> </span>clone)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.5</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_smooth</span>(<span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.7</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">coord_cartesian</span>(<span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">25</span>)) <span class="op">+</span>
<span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Variance explained by line (%)&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Variance explained by clone (%)&quot;</span>)</code></pre></div>
<p><img src="figure/variance_components.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-6-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/davismcc/fibroblast-clonality/blob/d2e8b3145e4600ec75d0d0df9ad1e210ecb1bdd3/docs/figure/variance_components.Rmd/unnamed-chunk-6-1.png" target="_blank">d2e8b31</a>
</td>
<td style="text-align:left;">
davismcc
</td>
<td style="text-align:left;">
2018-08-19
</td>
</tr>
</tbody>
</table>
<p></details></p>
</div>
</div>
<div id="results-for-individual-lines" class="section level2">
<h2>Results for individual lines</h2>
<div id="load-individual-lines" class="section level3">
<h3>Load individual lines</h3>
<p>We load a SingleCellExperiment object for each line containing expression data, cell metadata and gene metadata.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">params &lt;-<span class="st"> </span><span class="kw">list</span>()
params<span class="op">$</span>callset &lt;-<span class="st"> &quot;filt_lenient.cell_coverage_sites&quot;</span> 
lines &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;euts&quot;</span>, <span class="st">&quot;fawm&quot;</span>, <span class="st">&quot;feec&quot;</span>, <span class="st">&quot;fikt&quot;</span>, <span class="st">&quot;garx&quot;</span>, <span class="st">&quot;gesg&quot;</span>, <span class="st">&quot;heja&quot;</span>, <span class="st">&quot;hipn&quot;</span>, 
            <span class="st">&quot;ieki&quot;</span>, <span class="st">&quot;joxm&quot;</span>, <span class="st">&quot;kuco&quot;</span>, <span class="st">&quot;laey&quot;</span>, <span class="st">&quot;lexy&quot;</span>, <span class="st">&quot;naju&quot;</span>, <span class="st">&quot;nusw&quot;</span>, <span class="st">&quot;oaaz&quot;</span>, 
            <span class="st">&quot;oilg&quot;</span>, <span class="st">&quot;pipw&quot;</span>, <span class="st">&quot;puie&quot;</span>, <span class="st">&quot;qayj&quot;</span>, <span class="st">&quot;qolg&quot;</span>, <span class="st">&quot;qonc&quot;</span>, <span class="st">&quot;rozh&quot;</span>, <span class="st">&quot;sehl&quot;</span>, 
            <span class="st">&quot;ualf&quot;</span>, <span class="st">&quot;vass&quot;</span>, <span class="st">&quot;vuna&quot;</span>, <span class="st">&quot;wahn&quot;</span>, <span class="st">&quot;wetu&quot;</span>, <span class="st">&quot;xugn&quot;</span>, <span class="st">&quot;zoxy&quot;</span>)
## Load SCE objects
sce_unst_list &lt;-<span class="st"> </span><span class="kw">list</span>()
<span class="cf">for</span> (don <span class="cf">in</span> lines) {
    sce_unst_list[[don]] &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">file.path</span>(<span class="st">&quot;data/sces/&quot;</span>, 
        <span class="kw">paste0</span>(<span class="st">&quot;sce_&quot;</span>, don, <span class="st">&quot;_with_clone_assignments.&quot;</span>, params<span class="op">$</span>callset, <span class="st">&quot;.rds&quot;</span>)))
    <span class="kw">cat</span>(<span class="kw">paste</span>(<span class="st">&quot;reading&quot;</span>, don, <span class="st">&quot;:   &quot;</span>, <span class="kw">ncol</span>(sce_unst_list[[don]]), <span class="st">&quot;cells</span><span class="ch">\n</span><span class="st">&quot;</span>))
}</code></pre></div>
<pre><code>reading euts :    79 cells
reading fawm :    53 cells
reading feec :    75 cells
reading fikt :    39 cells
reading garx :    70 cells
reading gesg :    105 cells
reading heja :    50 cells
reading hipn :    62 cells
reading ieki :    58 cells
reading joxm :    79 cells
reading kuco :    48 cells
reading laey :    55 cells
reading lexy :    63 cells
reading naju :    44 cells
reading nusw :    60 cells
reading oaaz :    38 cells
reading oilg :    90 cells
reading pipw :    107 cells
reading puie :    41 cells
reading qayj :    97 cells
reading qolg :    36 cells
reading qonc :    58 cells
reading rozh :    91 cells
reading sehl :    30 cells
reading ualf :    89 cells
reading vass :    37 cells
reading vuna :    71 cells
reading wahn :    82 cells
reading wetu :    77 cells
reading xugn :    35 cells
reading zoxy :    88 cells</code></pre>
<p>We load variance component analysis results for each line.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">varPart_list &lt;-<span class="st"> </span><span class="kw">list</span>()
<span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">seq_along</span>(lines)) {
  df &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste0</span>(<span class="st">&quot;data/variance_components/donorVar/&quot;</span>,
                        lines[i], <span class="st">&quot;.var_part.var1.csv&quot;</span>))
    <span class="kw">colnames</span>(df) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;clone&quot;</span>, <span class="st">&quot;plate&quot;</span>, <span class="st">&quot;cell. det. rate&quot;</span>, <span class="st">&quot;residuals&quot;</span>)
    df<span class="op">$</span>gene_id &lt;-<span class="st"> </span><span class="kw">rownames</span>(df)
    df<span class="op">$</span>line &lt;-<span class="st"> </span>lines[i]
    varPart_list[[i]] &lt;-<span class="st"> </span><span class="kw">as_data_frame</span>(df)
}
<span class="kw">names</span>(varPart_list) &lt;-<span class="st"> </span>lines</code></pre></div>
<p>For each line we thus have a data.frame with the variance components results.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(varPart_list[[<span class="dv">1</span>]])</code></pre></div>
<pre><code># A tibble: 6 x 6
   clone   plate `cell. det. rate` residuals gene_id                  line 
   &lt;dbl&gt;   &lt;dbl&gt;             &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;                    &lt;chr&gt;
1 0      0.0727           0.0155       0.912 ENSG00000137804_NUSAP1   euts 
2 0.165  0                0.0439       0.791 ENSG00000115318_LOXL3    euts 
3 0      0.0311           0.000381     0.969 ENSG00000161888_SPC24    euts 
4 0.0274 0.00157          0.0518       0.919 ENSG00000171345_KRT19    euts 
5 0      0.0357           0.0222       0.942 ENSG00000089685_BIRC5    euts 
6 0      0                0.00906      0.991 ENSG00000197061_HIST1H4C euts </code></pre>
</div>
<div id="number-of-genes-with-substantial-clone-component" class="section level3">
<h3>Number of genes with substantial clone component</h3>
<p>We observe larger clone effects if we fit the linear mixed model for each line separately. A large number of genes for each line have &gt;5% of variance explained by clone (median: 825).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sort</span>(<span class="kw">sapply</span>(varPart_list, <span class="cf">function</span>(x) <span class="kw">sum</span>(x<span class="op">$</span>clone <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.05</span>)))</code></pre></div>
<pre><code>pipw vuna gesg euts oilg heja puie kuco wetu zoxy naju wahn qayj joxm qolg 
 331  347  382  478  532  548  559  561  566  584  609  635  651  741  760 
hipn xugn fawm ualf nusw garx oaaz lexy rozh vass feec fikt ieki laey sehl 
 825  844  856  881  901  920  948  962 1036 1096 1102 1314 1356 1392 1651 
qonc 
2162 </code></pre>
</div>
<div id="plot-results-for-individual-lines" class="section level3">
<h3>Plot results for individual lines</h3>
<p>For each line we will plot the variance explained by clone for the to 400 genes with most variance explained by clone (for that line).</p>
<p>First, we need to process the raw variance component results from each line.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">n_top &lt;-<span class="st"> </span><span class="dv">400</span>
n_lines &lt;-<span class="st"> </span><span class="kw">length</span>(varPart_list)

n_hv_genes &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="ot">NA</span>, n_lines)
n_cells_use &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="ot">NA</span>, n_lines)
n_genes_use &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="ot">NA</span>, n_lines)
var_top_mean &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="ot">NA</span>, n_lines)
var_top_list &lt;-<span class="st"> </span><span class="kw">list</span>()
<span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">seq_len</span>(<span class="kw">length</span>(varPart_list))) {
    sort_idx &lt;-<span class="st"> </span><span class="kw">order</span>(varPart_list[[i]]<span class="op">$</span>clone, <span class="dt">decreasing =</span> <span class="ot">TRUE</span>)
    var_top_list[[i]] &lt;-<span class="st"> </span>varPart_list[[i]][sort_idx[<span class="dv">1</span><span class="op">:</span>n_top], <span class="st">&quot;clone&quot;</span>, 
                                           drop =<span class="st"> </span><span class="ot">FALSE</span>]
    var_top_mean[i] &lt;-<span class="st"> </span><span class="kw">mean</span>(var_top_list[[i]]<span class="op">$</span>clone, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)
    n_hv_genes[i] &lt;-<span class="st"> </span><span class="kw">sum</span>(varPart_list[[i]]<span class="op">$</span>clone <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.25</span>, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)
    min_var &lt;-<span class="st"> </span><span class="fl">1.0</span>
    cell_idx &lt;-<span class="st"> </span><span class="kw">which</span>(sce_unst_list[[lines[i]]]<span class="op">$</span>assigned <span class="op">!=</span><span class="st"> &quot;unassigned&quot;</span>)
    n_cells_use[i] &lt;-<span class="st"> </span><span class="kw">length</span>(cell_idx)
    n_genes_use[i] &lt;-<span class="st"> </span><span class="kw">nrow</span>(varPart_list[[i]])
}
don_sort_idx &lt;-<span class="st"> </span><span class="kw">order</span>(var_top_mean, <span class="dt">decreasing =</span> <span class="ot">TRUE</span>)

df.line &lt;-<span class="st"> </span><span class="kw">data_frame</span>(<span class="dt">line_id =</span> lines, 
                       <span class="dt">clone_cells =</span> n_cells_use, 
                       <span class="dt">n_genes_use =</span> n_genes_use, 
                       <span class="dt">n_hv_genes =</span> n_hv_genes, 
                       <span class="dt">var_top_mean =</span> var_top_mean)[don_sort_idx, ]
df.line &lt;-<span class="st"> </span><span class="kw">cbind</span>(<span class="dt">rank =</span> <span class="kw">seq_len</span>(<span class="kw">nrow</span>(df.line)), df.line)

var_full &lt;-<span class="st"> </span><span class="kw">c</span>()
idx_full &lt;-<span class="st"> </span><span class="kw">c</span>()
don_full &lt;-<span class="st"> </span><span class="kw">c</span>()
rank_full &lt;-<span class="st"> </span><span class="kw">c</span>()
<span class="cf">for</span> (ii <span class="cf">in</span> don_sort_idx) {        
    var_full &lt;-<span class="st"> </span><span class="kw">c</span>(var_full, var_top_list[[ii]]<span class="op">$</span>clone)
    idx_full &lt;-<span class="st"> </span><span class="kw">c</span>(idx_full, <span class="kw">seq_len</span>(n_top))
    don_full &lt;-<span class="st"> </span><span class="kw">c</span>(don_full, <span class="kw">rep</span>(lines[ii], n_top))
    rank_full &lt;-<span class="st"> </span><span class="kw">c</span>(rank_full, <span class="kw">rep</span>(ii, n_top))
}

line_info &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="st">&quot;other&quot;</span>, <span class="kw">length</span>(don_full))
median_line &lt;-<span class="st"> </span>lines[don_sort_idx[<span class="kw">round</span>(<span class="kw">length</span>(don_sort_idx)<span class="op">/</span><span class="dv">2</span>)]]
<span class="co"># line_info[don_full == median_line] &lt;- &quot;median&quot;</span>
line_info[don_full <span class="op">==</span><span class="st"> &quot;joxm&quot;</span>] &lt;-<span class="st"> &quot;joxm&quot;</span>
<span class="co"># print(median_line)</span>
<span class="co"># summary(n_hv_genes)</span>

df &lt;-<span class="st"> </span><span class="kw">data_frame</span>(<span class="dt">varPart =</span> var_full, <span class="dt">gene_rank =</span> idx_full, 
                 <span class="dt">lines =</span> don_full, <span class="dt">line_rank =</span> rank_full,
                 <span class="dt">line_info =</span> line_info)
<span class="kw">rm</span>(line_info)

df &lt;-<span class="st"> </span>df <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">group_by</span>(gene_rank) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">summarise</span>(<span class="dt">varPart =</span> <span class="kw">median</span>(varPart)) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">lines =</span> <span class="st">&quot;median&quot;</span>, <span class="dt">line_info =</span> <span class="st">&quot;median&quot;</span>, <span class="dt">line_rank =</span> <span class="ot">NA</span>) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">bind_rows</span>(., df)</code></pre></div>
<p>The table below provides a summary of the per-line variance component results. We can read off the mean proportion of variance explained by clone for the top 400 genes for each line (<code>var_top_mean</code>), as well as other information for each line.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df.line</code></pre></div>
<pre><code>   rank line_id clone_cells n_genes_use n_hv_genes var_top_mean
1     1    feec          64        8243        342   0.50769558
2     2    xugn          34        7480        260   0.41529069
3     3    qonc          43        9882        420   0.35658891
4     4    fikt          36        9762        219   0.29630805
5     5    sehl          24        8792        198   0.27796718
6     6    ieki          26        8324        166   0.25419512
7     7    vass          37        7231        129   0.23747536
8     8    lexy          63       10154        123   0.23580397
9     9    oaaz          37        9946        105   0.22820218
10   10    nusw          20        9544        110   0.22397105
11   11    kuco          48       10021        121   0.22329573
12   12    euts          78       10210        128   0.21836347
13   13    wahn          77       10241        114   0.21728504
14   14    laey          55        7402         89   0.20944535
15   15    wetu          66        7177         79   0.20065196
16   16    hipn          49        7971         86   0.19522467
17   17    rozh          42        7819         57   0.18304904
18   18    heja          50        9315         57   0.16430667
19   19    qolg          36        8640         41   0.16297743
20   20    joxm          77       10095         45   0.15834547
21   21    garx          69        9571         25   0.15202524
22   22    pipw         107       10260         73   0.15192220
23   23    oilg          57        7875         44   0.14300906
24   24    qayj          59        6844         30   0.14024325
25   25    fawm          47        8537         17   0.13702877
26   26    ualf          88        7001         16   0.13288015
27   27    zoxy          82        6968         27   0.12433830
28   28    puie          41        8478         12   0.11222310
29   29    naju          44        9089          5   0.10986147
30   30    gesg         101       10036          5   0.08542351
31   31    vuna          71       10000          2   0.07668159</code></pre>
<p>Now we can plot a curve for each line showing the variance explained by clone for the top 400 genes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df_labs &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(df, gene_rank <span class="op">==</span><span class="st"> </span><span class="dv">200</span>) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">labs =</span> <span class="kw">ifelse</span>(varPart <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.33</span>, lines, <span class="st">&quot;&quot;</span>))

fig.curve &lt;-<span class="st"> </span><span class="kw">ggplot</span>(df, <span class="kw">aes</span>(<span class="dt">colour =</span> line_info)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_line</span>(<span class="kw">aes</span>(<span class="dt">x =</span> gene_rank, <span class="dt">y =</span> <span class="dv">100</span> <span class="op">*</span><span class="st"> </span>varPart, <span class="dt">color =</span> line_info, 
                <span class="dt">size =</span> line_info, <span class="dt">group =</span> lines)) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;firebrick&quot;</span>, <span class="st">&quot;black&quot;</span>, <span class="st">&quot;gray60&quot;</span>),
                     <span class="dt">name =</span> <span class="st">&quot;line&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_size_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="fl">0.5</span>, <span class="dv">1</span>, <span class="fl">0.5</span>), <span class="dt">name =</span> <span class="st">&quot;line&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_line</span>(<span class="dt">data =</span> df[df<span class="op">$</span>lines <span class="op">==</span><span class="st"> &quot;joxm&quot;</span>, ],
            <span class="kw">aes</span>(<span class="dt">x =</span> gene_rank, <span class="dt">y =</span> <span class="dv">100</span> <span class="op">*</span><span class="st"> </span>varPart), <span class="dt">size =</span> <span class="fl">0.5</span>, <span class="dt">color =</span> <span class="st">&quot;firebrick&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_line</span>(<span class="dt">data =</span> df[df<span class="op">$</span>lines <span class="op">==</span><span class="st"> &quot;median&quot;</span>, ],
            <span class="kw">aes</span>(<span class="dt">x =</span> gene_rank, <span class="dt">y =</span> <span class="dv">100</span> <span class="op">*</span><span class="st"> </span>varPart), <span class="dt">size =</span> <span class="dv">1</span>, <span class="dt">color =</span> <span class="st">&quot;black&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">x =</span> gene_rank <span class="op">-</span><span class="st"> </span><span class="dv">15</span>, <span class="dt">y =</span> <span class="dv">100</span> <span class="op">*</span><span class="st"> </span>varPart, <span class="dt">label =</span> labs),
                   <span class="dt">nudge_y =</span> <span class="dv">0</span>, <span class="dt">nudge_x =</span> <span class="dv">0</span>, 
                   <span class="dt">data =</span> df_labs, <span class="dt">colour =</span> <span class="st">&quot;gray40&quot;</span>, <span class="dt">size =</span> <span class="dv">2</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Top genes&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Variance explained by clone (%)&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlim</span>(<span class="dv">0</span>, <span class="dv">400</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="kw">c</span>(<span class="fl">0.8</span>, <span class="fl">0.8</span>))

<span class="kw">ggsave</span>(<span class="st">&quot;figures/variance_components/varpart_curve_perline.png&quot;</span>, 
       fig.curve, <span class="dt">height =</span> <span class="dv">4</span>, <span class="dt">width =</span> <span class="fl">6.5</span>, <span class="dt">dpi =</span> <span class="dv">300</span>)
<span class="kw">ggsave</span>(<span class="st">&quot;figures/variance_components/varpart_cruve_perline.pdf&quot;</span>, 
       fig.curve, <span class="dt">height =</span> <span class="dv">4</span>, <span class="dt">width =</span> <span class="fl">6.5</span>, <span class="dt">dpi =</span> <span class="dv">300</span>)
<span class="kw">ggsave</span>(<span class="st">&quot;figures/variance_components/varpart_curve_perline_skinny.png&quot;</span>, 
       fig.curve <span class="op">+</span><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="kw">c</span>(<span class="fl">0.7</span>, <span class="fl">0.8</span>)),
       <span class="dt">height =</span> <span class="dv">4</span>, <span class="dt">width =</span> <span class="fl">4.5</span>, <span class="dt">dpi =</span> <span class="dv">300</span>)
<span class="kw">ggsave</span>(<span class="st">&quot;figures/variance_components/varpart_cruve_perline_skinny.pdf&quot;</span>, 
       fig.curve <span class="op">+</span><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="kw">c</span>(<span class="fl">0.7</span>, <span class="fl">0.8</span>)), 
       <span class="dt">height =</span> <span class="dv">4</span>, <span class="dt">width =</span> <span class="fl">4.5</span>, <span class="dt">dpi =</span> <span class="dv">300</span>)

fig.curve</code></pre></div>
<p><img src="figure/variance_components.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-13-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/davismcc/fibroblast-clonality/blob/d2e8b3145e4600ec75d0d0df9ad1e210ecb1bdd3/docs/figure/variance_components.Rmd/unnamed-chunk-13-1.png" target="_blank">d2e8b31</a>
</td>
<td style="text-align:left;">
davismcc
</td>
<td style="text-align:left;">
2018-08-19
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>Each line has a substantial number of genes with a substantial proportion of variance explained by clone. For example, the number of genes with more than 25% of variance explained by line across lines:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sapply</span>(varPart_list, <span class="cf">function</span>(x) <span class="kw">sum</span>(x<span class="op">$</span>clone <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.25</span>))</code></pre></div>
<pre><code>euts fawm feec fikt garx gesg heja hipn ieki joxm kuco laey lexy naju nusw 
 128   17  342  219   25    5   57   86  166   45  121   89  123    5  110 
oaaz oilg pipw puie qayj qolg qonc rozh sehl ualf vass vuna wahn wetu xugn 
 105   44   73   12   30   41  420   57  198   16  129    2  114   79  260 
zoxy 
  27 </code></pre>
<p>There are a large number of genes in each line with more than 5% of variance explained by clone:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sapply</span>(varPart_list, <span class="cf">function</span>(x) <span class="kw">sum</span>(x<span class="op">$</span>clone <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.05</span>))</code></pre></div>
<pre><code>euts fawm feec fikt garx gesg heja hipn ieki joxm kuco laey lexy naju nusw 
 478  856 1102 1314  920  382  548  825 1356  741  561 1392  962  609  901 
oaaz oilg pipw puie qayj qolg qonc rozh sehl ualf vass vuna wahn wetu xugn 
 948  532  331  559  651  760 2162 1036 1651  881 1096  347  635  566  844 
zoxy 
 584 </code></pre>
</div>
</div>
<div id="plots-including-line-selection-information" class="section level2">
<h2>Plots including line selection information</h2>
<p>We have inferred selection dynamics from somatic variants detected from whole-exome sequencing data. We can overlay selection status on the plot above to look for any relationship between selection and variance explained by clone in top genes by line.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">line_selected &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;euts&quot;</span>, <span class="st">&quot;fawm&quot;</span>, <span class="st">&quot;fikt&quot;</span>, <span class="st">&quot;garx&quot;</span>, <span class="st">&quot;pipw&quot;</span>, <span class="st">&quot;puie&quot;</span>, <span class="st">&quot;qolg&quot;</span>, <span class="st">&quot;rozh&quot;</span>)
line_neutral &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;hipn&quot;</span>, <span class="st">&quot;nusw&quot;</span>, <span class="st">&quot;qonc&quot;</span>, <span class="st">&quot;sehl&quot;</span>, <span class="st">&quot;ualf&quot;</span>, <span class="st">&quot;xugn&quot;</span>)
line_undetermined &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;feec&quot;</span>, <span class="st">&quot;gesg&quot;</span>, <span class="st">&quot;heja&quot;</span>, <span class="st">&quot;ieki&quot;</span>, <span class="st">&quot;joxm&quot;</span>, <span class="st">&quot;kuco&quot;</span>, <span class="st">&quot;laey&quot;</span>, 
                        <span class="st">&quot;lexy&quot;</span>, <span class="st">&quot;naju&quot;</span>, <span class="st">&quot;oaaz&quot;</span>, 
                        <span class="st">&quot;oilg&quot;</span>, <span class="st">&quot;qayj&quot;</span>, <span class="st">&quot;vass&quot;</span>, <span class="st">&quot;vuna&quot;</span>, <span class="st">&quot;wahn&quot;</span>, <span class="st">&quot;wetu&quot;</span>, <span class="st">&quot;zoxy&quot;</span>)
df<span class="op">$</span>selection &lt;-<span class="st"> &quot;undetermined&quot;</span>
df<span class="op">$</span>selection[df<span class="op">$</span>lines <span class="op">%in%</span><span class="st"> </span>line_neutral] &lt;-<span class="st"> &quot;neutral&quot;</span>
df<span class="op">$</span>selection[df<span class="op">$</span>lines <span class="op">%in%</span><span class="st"> </span>line_selected] &lt;-<span class="st"> &quot;selection&quot;</span>
df<span class="op">$</span>selection[df<span class="op">$</span>lines <span class="op">%in%</span><span class="st"> &quot;median&quot;</span>] &lt;-<span class="st"> &quot;median&quot;</span>
df<span class="op">$</span>selection &lt;-<span class="st"> </span><span class="kw">factor</span>(df<span class="op">$</span>selection, 
                       <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selection&quot;</span>, 
                                  <span class="st">&quot;median&quot;</span>))</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df_labs &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(df, gene_rank <span class="op">==</span><span class="st"> </span><span class="dv">200</span>) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">labs =</span> <span class="kw">ifelse</span>(varPart <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.33</span>, lines, <span class="st">&quot;&quot;</span>))

fig.curve &lt;-<span class="st"> </span><span class="kw">ggplot</span>(df, <span class="kw">aes</span>(<span class="dt">colour =</span> selection)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_line</span>(<span class="kw">aes</span>(<span class="dt">x =</span> gene_rank, <span class="dt">y =</span> <span class="dv">100</span> <span class="op">*</span><span class="st"> </span>varPart, <span class="dt">color =</span> selection, 
                <span class="dt">size =</span> selection, <span class="dt">group =</span> lines)) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span>  <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>, 
                                 <span class="st">&quot;black&quot;</span>), <span class="dt">name =</span> <span class="st">&quot;selection</span><span class="ch">\n</span><span class="st">dynamics&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_size_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="fl">0.5</span>, <span class="fl">0.5</span>, <span class="fl">0.5</span>, <span class="dv">1</span>), <span class="dt">name =</span> <span class="st">&quot;selection</span><span class="ch">\n</span><span class="st">dynamics&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_line</span>(<span class="dt">data =</span> df[df<span class="op">$</span>lines <span class="op">==</span><span class="st"> &quot;median&quot;</span>, ],
            <span class="kw">aes</span>(<span class="dt">x =</span> gene_rank, <span class="dt">y =</span> <span class="dv">100</span> <span class="op">*</span><span class="st"> </span>varPart), <span class="dt">size =</span> <span class="dv">1</span>, <span class="dt">color =</span> <span class="st">&quot;black&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">x =</span> gene_rank <span class="op">-</span><span class="st"> </span><span class="dv">15</span>, <span class="dt">y =</span> <span class="dv">100</span> <span class="op">*</span><span class="st"> </span>varPart, <span class="dt">label =</span> labs),
                   <span class="dt">nudge_y =</span> <span class="dv">0</span>, <span class="dt">nudge_x =</span> <span class="dv">0</span>, 
                   <span class="dt">data =</span> df_labs, <span class="dt">colour =</span> <span class="st">&quot;gray40&quot;</span>, <span class="dt">size =</span> <span class="dv">2</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Top genes&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Variance explained by clone (%)&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlim</span>(<span class="dv">0</span>, <span class="dv">400</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="kw">c</span>(<span class="fl">0.8</span>, <span class="fl">0.8</span>))

fig.curve</code></pre></div>
<p><img src="figure/variance_components.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-17-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/davismcc/fibroblast-clonality/blob/d2e8b3145e4600ec75d0d0df9ad1e210ecb1bdd3/docs/figure/variance_components.Rmd/unnamed-chunk-17-1.png" target="_blank">d2e8b31</a>
</td>
<td style="text-align:left;">
davismcc
</td>
<td style="text-align:left;">
2018-08-19
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>There is not any obvious relationship between selection dynamics and variance explained by clone here.</p>
</div>
<div id="write-mean-variance-explained-to-file" class="section level2">
<h2>Write mean variance explained to file</h2>
<p>Write a table with the mean fraction of variance explained by clone for the top 400 genes per line out to file.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">idx &lt;-<span class="st"> </span><span class="kw">order</span>(var_top_mean, <span class="dt">decreasing =</span> <span class="ot">TRUE</span>)
line_var_df &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">line =</span> lines[idx], 
                           <span class="dt">meanFraction400 =</span> var_top_mean[idx])
<span class="kw">write.csv</span>(line_var_df, <span class="st">&quot;output/variance_components/line_var_top400.csv&quot;</span>)</code></pre></div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">devtools<span class="op">::</span><span class="kw">session_info</span>()</code></pre></div>
<pre><code>Session info -------------------------------------------------------------</code></pre>
<pre><code> setting  value                       
 version  R version 3.5.1 (2018-07-02)
 system   x86_64, darwin15.6.0        
 ui       X11                         
 language (EN)                        
 collate  en_GB.UTF-8                 
 tz       Europe/London               
 date     2018-08-27                  </code></pre>
<pre><code>Packages -----------------------------------------------------------------</code></pre>
<pre><code> package              * version   date       source        
 assertthat             0.2.0     2017-04-11 CRAN (R 3.5.0)
 backports              1.1.2     2017-12-13 CRAN (R 3.5.0)
 base                 * 3.5.1     2018-07-05 local         
 bindr                  0.1.1     2018-03-13 CRAN (R 3.5.0)
 bindrcpp             * 0.2.2     2018-03-29 CRAN (R 3.5.0)
 Biobase              * 2.40.0    2018-05-01 Bioconductor  
 BiocGenerics         * 0.26.0    2018-05-01 Bioconductor  
 BiocParallel         * 1.14.2    2018-07-08 Bioconductor  
 bitops                 1.0-6     2013-08-17 CRAN (R 3.5.0)
 broom                  0.5.0     2018-07-17 CRAN (R 3.5.0)
 cellranger             1.1.0     2016-07-27 CRAN (R 3.5.0)
 cli                    1.0.0     2017-11-05 CRAN (R 3.5.0)
 colorspace             1.3-2     2016-12-14 CRAN (R 3.5.0)
 compiler               3.5.1     2018-07-05 local         
 cowplot              * 0.9.3     2018-07-15 CRAN (R 3.5.0)
 crayon                 1.3.4     2017-09-16 CRAN (R 3.5.0)
 datasets             * 3.5.1     2018-07-05 local         
 DelayedArray         * 0.6.5     2018-08-15 Bioconductor  
 devtools               1.13.6    2018-06-27 CRAN (R 3.5.0)
 digest                 0.6.16    2018-08-22 CRAN (R 3.5.0)
 dplyr                * 0.7.6     2018-06-29 CRAN (R 3.5.1)
 evaluate               0.11      2018-07-17 CRAN (R 3.5.0)
 fansi                  0.3.0     2018-08-13 CRAN (R 3.5.0)
 forcats              * 0.3.0     2018-02-19 CRAN (R 3.5.0)
 GenomeInfoDb         * 1.16.0    2018-05-01 Bioconductor  
 GenomeInfoDbData       1.1.0     2018-04-25 Bioconductor  
 GenomicRanges        * 1.32.6    2018-07-20 Bioconductor  
 ggplot2              * 3.0.0     2018-07-03 CRAN (R 3.5.0)
 ggpubr               * 0.1.7     2018-06-23 CRAN (R 3.5.0)
 ggrepel              * 0.8.0     2018-05-09 CRAN (R 3.5.0)
 git2r                  0.23.0    2018-07-17 CRAN (R 3.5.0)
 glue                   1.3.0     2018-07-17 CRAN (R 3.5.0)
 graphics             * 3.5.1     2018-07-05 local         
 grDevices            * 3.5.1     2018-07-05 local         
 grid                   3.5.1     2018-07-05 local         
 gridExtra              2.3       2017-09-09 CRAN (R 3.5.0)
 gtable                 0.2.0     2016-02-26 CRAN (R 3.5.0)
 haven                  1.1.2     2018-06-27 CRAN (R 3.5.0)
 hms                    0.4.2     2018-03-10 CRAN (R 3.5.0)
 htmltools              0.3.6     2017-04-28 CRAN (R 3.5.0)
 httr                   1.3.1     2017-08-20 CRAN (R 3.5.0)
 IRanges              * 2.14.11   2018-08-24 Bioconductor  
 jsonlite               1.5       2017-06-01 CRAN (R 3.5.0)
 knitr                  1.20      2018-02-20 CRAN (R 3.5.0)
 labeling               0.3       2014-08-23 CRAN (R 3.5.0)
 lattice                0.20-35   2017-03-25 CRAN (R 3.5.1)
 lazyeval               0.2.1     2017-10-29 CRAN (R 3.5.0)
 lubridate              1.7.4     2018-04-11 CRAN (R 3.5.0)
 magrittr             * 1.5       2014-11-22 CRAN (R 3.5.0)
 Matrix                 1.2-14    2018-04-13 CRAN (R 3.5.1)
 matrixStats          * 0.54.0    2018-07-23 CRAN (R 3.5.0)
 memoise                1.1.0     2017-04-21 CRAN (R 3.5.0)
 methods              * 3.5.1     2018-07-05 local         
 mgcv                   1.8-24    2018-06-23 CRAN (R 3.5.1)
 modelr                 0.1.2     2018-05-11 CRAN (R 3.5.0)
 munsell                0.5.0     2018-06-12 CRAN (R 3.5.0)
 nlme                   3.1-137   2018-04-07 CRAN (R 3.5.1)
 parallel             * 3.5.1     2018-07-05 local         
 pillar                 1.3.0     2018-07-14 CRAN (R 3.5.0)
 pkgconfig              2.0.2     2018-08-16 CRAN (R 3.5.0)
 plyr                   1.8.4     2016-06-08 CRAN (R 3.5.0)
 purrr                * 0.2.5     2018-05-29 CRAN (R 3.5.0)
 R.methodsS3            1.7.1     2016-02-16 CRAN (R 3.5.0)
 R.oo                   1.22.0    2018-04-22 CRAN (R 3.5.0)
 R.utils                2.6.0     2017-11-05 CRAN (R 3.5.0)
 R6                     2.2.2     2017-06-17 CRAN (R 3.5.0)
 Rcpp                   0.12.18   2018-07-23 CRAN (R 3.5.0)
 RCurl                  1.95-4.11 2018-07-15 CRAN (R 3.5.0)
 readr                * 1.1.1     2017-05-16 CRAN (R 3.5.0)
 readxl                 1.1.0     2018-04-20 CRAN (R 3.5.0)
 reshape2             * 1.4.3     2017-12-11 CRAN (R 3.5.0)
 rlang                  0.2.2     2018-08-16 CRAN (R 3.5.0)
 rmarkdown              1.10      2018-06-11 CRAN (R 3.5.0)
 rprojroot              1.3-2     2018-01-03 CRAN (R 3.5.0)
 rstudioapi             0.7       2017-09-07 CRAN (R 3.5.0)
 rvest                  0.3.2     2016-06-17 CRAN (R 3.5.0)
 S4Vectors            * 0.18.3    2018-06-08 Bioconductor  
 scales                 1.0.0     2018-08-09 CRAN (R 3.5.0)
 SingleCellExperiment * 1.2.0     2018-05-01 Bioconductor  
 stats                * 3.5.1     2018-07-05 local         
 stats4               * 3.5.1     2018-07-05 local         
 stringi                1.2.4     2018-07-20 CRAN (R 3.5.0)
 stringr              * 1.3.1     2018-05-10 CRAN (R 3.5.0)
 SummarizedExperiment * 1.10.1    2018-05-11 Bioconductor  
 tibble               * 1.4.2     2018-01-22 CRAN (R 3.5.0)
 tidyr                * 0.8.1     2018-05-18 CRAN (R 3.5.0)
 tidyselect             0.2.4     2018-02-26 CRAN (R 3.5.0)
 tidyverse            * 1.2.1     2017-11-14 CRAN (R 3.5.0)
 tools                  3.5.1     2018-07-05 local         
 utf8                   1.1.4     2018-05-24 CRAN (R 3.5.0)
 utils                * 3.5.1     2018-07-05 local         
 viridis              * 0.5.1     2018-03-29 CRAN (R 3.5.0)
 viridisLite          * 0.3.0     2018-02-01 CRAN (R 3.5.0)
 whisker                0.3-2     2013-04-28 CRAN (R 3.5.0)
 withr                  2.1.2     2018-03-15 CRAN (R 3.5.0)
 workflowr              1.1.1     2018-07-06 CRAN (R 3.5.0)
 xml2                   1.2.0     2018-01-24 CRAN (R 3.5.0)
 XVector                0.20.0    2018-05-01 Bioconductor  
 yaml                   2.2.0     2018-07-25 CRAN (R 3.5.1)
 zlibbioc               1.26.0    2018-05-01 Bioconductor  </code></pre>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
