<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Yuanhua Huang &amp; Davis J. McCarthy" />


<title>Variance components analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fibroblast-clonality</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/davismcc/fibroblast-clonality">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Variance components analysis</h1>
<h4 class="author">Yuanhua Huang &amp; Davis J. McCarthy</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-10-30
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>fibroblast-clonality/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20180807code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20180807)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20180807code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20180807)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomdavismccfibroblastclonalitytreef2e01ef23334fbb86d5874dacc97ddb1493358e5targetblankf2e01efa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/davismcc/fibroblast-clonality/tree/f2e01ef23334fbb86d5874dacc97ddb1493358e5" target="_blank">f2e01ef</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomdavismccfibroblastclonalitytreef2e01ef23334fbb86d5874dacc97ddb1493358e5targetblankf2e01efa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .vscode/
    Ignored:    code/.DS_Store
    Ignored:    code/selection/.DS_Store
    Ignored:    code/selection/.Rhistory
    Ignored:    code/selection/figures/
    Ignored:    data/.DS_Store
    Ignored:    logs/
    Ignored:    src/.DS_Store
    Ignored:    src/Rmd/.Rhistory

Untracked files:
    Untracked:  .dockerignore
    Untracked:  .dropbox
    Untracked:  .snakemake/
    Untracked:  Rplots.pdf
    Untracked:  Snakefile_clonality
    Untracked:  Snakefile_somatic_calling
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/assess_mutect2_fibro-ipsc_variant_calls.ipynb
    Untracked:  analysis/cardelino_fig1b.R
    Untracked:  analysis/cardelino_fig2b.R
    Untracked:  code/analysis_for_garx.Rmd
    Untracked:  code/selection/data/
    Untracked:  code/selection/fit-dist.nb
    Untracked:  code/selection/result-figure.R
    Untracked:  code/yuanhua/
    Untracked:  data/Melanoma-RegevGarraway-DFCI-scRNA-Seq/
    Untracked:  data/PRJNA485423/
    Untracked:  data/canopy/
    Untracked:  data/cell_assignment/
    Untracked:  data/cnv/
    Untracked:  data/de_analysis_FTv62/
    Untracked:  data/donor_info_070818.txt
    Untracked:  data/donor_info_core.csv
    Untracked:  data/donor_neutrality.tsv
    Untracked:  data/exome-point-mutations/
    Untracked:  data/fdr10.annot.txt.gz
    Untracked:  data/human_H_v5p2.rdata
    Untracked:  data/human_c2_v5p2.rdata
    Untracked:  data/human_c6_v5p2.rdata
    Untracked:  data/neg-bin-rsquared-petr.csv
    Untracked:  data/neutralitytestr-petr.tsv
    Untracked:  data/raw/
    Untracked:  data/sce_merged_donors_cardelino_donorid_all_qc_filt.rds
    Untracked:  data/sce_merged_donors_cardelino_donorid_all_with_qc_labels.rds
    Untracked:  data/sce_merged_donors_cardelino_donorid_unstim_qc_filt.rds
    Untracked:  data/sces/
    Untracked:  data/selection/
    Untracked:  data/simulations/
    Untracked:  data/variance_components/
    Untracked:  docs/figure/analysis_for_joxm_cardelino-relax.Rmd/
    Untracked:  docs/figure/clone_prevalences_cardelino-relax.Rmd/
    Untracked:  figures/
    Untracked:  output/differential_expression/
    Untracked:  output/differential_expression_cardelino-relax/
    Untracked:  output/donor_specific/
    Untracked:  output/line_info.tsv
    Untracked:  output/nvars_by_category_by_donor.tsv
    Untracked:  output/nvars_by_category_by_line.tsv
    Untracked:  output/variance_components/
    Untracked:  qolg_BIC.pdf
    Untracked:  references/
    Untracked:  reports/
    Untracked:  src/Rmd/DE_pathways_FTv62_callset_clones_pairwise_vs_base.unst_cells.carderelax.Rmd
    Untracked:  src/Rmd/Rplots.pdf
    Untracked:  src/Rmd/cell_assignment_cardelino-relax_template.Rmd
    Untracked:  tree.txt

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/davismcc/fibroblast-clonality/8729e022b6ba38c1b948735439f39be0977dcd35/docs/variance_components.html" target="_blank">8729e02</a>
</td>
<td>
davismcc
</td>
<td>
2018-11-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/davismcc/fibroblast-clonality/0540cdbc554ab43e0ff60557b7ceb82fb13f3d75/docs/variance_components.html" target="_blank">0540cdb</a>
</td>
<td>
davismcc
</td>
<td>
2018-09-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/davismcc/fibroblast-clonality/f0ed980029a115234bc2edff312e5e52056d4eed/docs/variance_components.html" target="_blank">f0ed980</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/davismcc/fibroblast-clonality/ca3438f9274dbc7adf4aacc25b0fb017bc2d41fe/docs/variance_components.html" target="_blank">ca3438f</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/dc78a9541488651a4b9bcda20e016a0b849d0b94/analysis/variance_components.Rmd" target="_blank">dc78a95</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-29
</td>
<td>
Minor updates to analyses.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/davismcc/fibroblast-clonality/e573f2fcd4c0769acde01e24a2bb0c7fc3066ae9/docs/variance_components.html" target="_blank">e573f2f</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/davismcc/fibroblast-clonality/9ec2a595a73ee48db72d6c6d860dc45fb0192a36/docs/variance_components.html" target="_blank">9ec2a59</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/davismcc/fibroblast-clonality/36acf15b30f110282dd56b004c5d478d560b75e1/docs/variance_components.html" target="_blank">36acf15</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/d618fe52b90fd1d6e4f9613c8d7bbe3a3afde245/analysis/variance_components.Rmd" target="_blank">d618fe5</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-25
</td>
<td>
Updating analyses
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/davismcc/fibroblast-clonality/090c1b94750a83d2101ffaaa654dab8150e6284b/docs/variance_components.html" target="_blank">090c1b9</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/davismcc/fibroblast-clonality/d2e8b3145e4600ec75d0d0df9ad1e210ecb1bdd3/docs/variance_components.html" target="_blank">d2e8b31</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/davismcc/fibroblast-clonality/1489d32ff1b8d5073a4170baec953cdbe4fcc14a/docs/variance_components.html" target="_blank">1489d32</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-17
</td>
<td>
Add html files
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/c5c1416614dc3fbaf1d84af87b3b00004cd22e38/analysis/variance_components.Rmd" target="_blank">c5c1416</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-17
</td>
<td>
Adding variance components analysis.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries" class="section level2">
<h2>Load libraries</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">library</span>(viridis)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw">library</span>(tidyverse)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw">library</span>(ggpubr)</a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="kw">library</span>(ggrepel)</a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="kw">library</span>(cowplot)</a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="kw">library</span>(reshape2)</a>
<a class="sourceLine" id="cb1-7" data-line-number="7"><span class="kw">dir.create</span>(<span class="st">&quot;figures/variance_components&quot;</span>, <span class="dt">showWarnings =</span> <span class="ot">FALSE</span>, </a>
<a class="sourceLine" id="cb1-8" data-line-number="8">           <span class="dt">recursive =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
</div>
<div id="results-aggregated-across-all-lines" class="section level2">
<h2>Results aggregated across all lines</h2>
<p>Previously, we conducted variance components analysis combining scRNA-seq data across all lines. We used the <code>variancePartion</code> package, fitting assigned clone, line, plate (on which cells were processed) and cellular detection rate (cdr; proportion of genes per cell with non-zero expression) as random effects.</p>
<p>The output of the model gives us the proportion of variance explained, for each gene, by clone, line, plate, cdr and residuals.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">varPart.df &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(</a>
<a class="sourceLine" id="cb2-2" data-line-number="2">  <span class="kw">read.table</span>(<span class="st">&quot;data/variance_components/fit_per_gene_highVar.csv&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;,&quot;</span>))</a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="kw">colnames</span>(varPart.df) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;clone&quot;</span>, <span class="st">&quot;line&quot;</span>, <span class="st">&quot;plate&quot;</span>, <span class="st">&quot;cdr&quot;</span>, <span class="st">&quot;residuals&quot;</span>)</a>
<a class="sourceLine" id="cb2-4" data-line-number="4">varPart.df<span class="op">$</span>gene_id &lt;-<span class="st"> </span><span class="kw">rownames</span>(varPart.df)</a>
<a class="sourceLine" id="cb2-5" data-line-number="5">varPart.df &lt;-<span class="st"> </span><span class="kw">as_data_frame</span>(varPart.df)</a></code></pre></div>
<pre><code>Warning: `as_data_frame()` is deprecated, use `as_tibble()` (but mind the new semantics).
This warning is displayed once per session.</code></pre>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw">head</span>(varPart.df)</a></code></pre></div>
<pre><code># A tibble: 6 x 6
     clone    line   plate      cdr residuals gene_id               
     &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;    &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;                 
1 3.54e- 3 0.0179  0.0216  0.0124       0.945 ENSG00000001167_NFYA  
2 0.       0.167   0.0215  0.0354       0.776 ENSG00000001626_CFTR  
3 0.       0.00580 0.0143  0.000248     0.980 ENSG00000001629_ANKIB1
4 5.35e- 3 0.0194  0.0215  0.0152       0.939 ENSG00000001631_KRIT1 
5 9.73e-15 0.369   0.00582 0.0336       0.592 ENSG00000002016_RAD52 
6 0.       0.125   0.0827  0.0143       0.778 ENSG00000002330_BAD   </code></pre>
<p>We ran the analysis for 4998 highly variable genes (as identified using methods in the <code>scran</code> package).</p>
<p>We detect 194 genes with more than 5% of variance explained by clone and 1034 genes for which variance explained by clone is both greater than 1% and also greater than the variance explained by plate (a good proxy for technical effects on variability in gene expression between cells).</p>
<p>The median percentage of variance explained by clone is 0.487273, whereas the median percentage of variance explained by line is 5.4807953</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">varPart.mel &lt;-<span class="st"> </span>tidyr<span class="op">::</span><span class="kw">gather</span>(varPart.df, <span class="dt">key =</span> <span class="st">&quot;variable&quot;</span>, <span class="dt">value =</span> <span class="st">&quot;value&quot;</span>, </a>
<a class="sourceLine" id="cb6-2" data-line-number="2">                             <span class="op">-</span>gene_id)</a></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">dat_dir &lt;-<span class="st"> &quot;data/variance_components/donorVar/&quot;</span></a>
<a class="sourceLine" id="cb7-2" data-line-number="2"></a>
<a class="sourceLine" id="cb7-3" data-line-number="3">fig.violin &lt;-<span class="st"> </span>varPart.mel <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">variable =</span> <span class="kw">replace</span>(variable, variable <span class="op">==</span><span class="st"> &quot;cdr&quot;</span>, </a>
<a class="sourceLine" id="cb7-5" data-line-number="5">                                   <span class="st">&quot;cell. det. rate&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-6" data-line-number="6"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">variable =</span> <span class="kw">factor</span>(variable, </a>
<a class="sourceLine" id="cb7-7" data-line-number="7">                                  <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;residuals&quot;</span>, <span class="st">&quot;cell. det. rate&quot;</span>,</a>
<a class="sourceLine" id="cb7-8" data-line-number="8">                                             <span class="st">&quot;plate&quot;</span>, <span class="st">&quot;line&quot;</span>, <span class="st">&quot;clone&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-9" data-line-number="9"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> variable, <span class="dt">y =</span> <span class="dv">100</span> <span class="op">*</span><span class="st"> </span>value)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb7-10" data-line-number="10"><span class="st">  </span><span class="kw">geom_violin</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> variable), <span class="dt">scale =</span> <span class="st">&quot;width&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb7-11" data-line-number="11"><span class="st">  </span><span class="kw">geom_boxplot</span>(<span class="dt">width =</span> <span class="fl">0.07</span>, <span class="dt">fill =</span> <span class="st">&quot;grey&quot;</span>, <span class="dt">outlier.colour =</span> <span class="st">&quot;black&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb7-12" data-line-number="12"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Variance explained (%)&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb7-13" data-line-number="13"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb7-14" data-line-number="14"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;gray70&quot;</span>, <span class="st">&quot;#f7efe2&quot;</span>, <span class="st">&quot;#f9a603&quot;</span>, <span class="st">&quot;#f25c00&quot;</span>,</a>
<a class="sourceLine" id="cb7-15" data-line-number="15">                               <span class="st">&quot;#f70025&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb7-16" data-line-number="16"><span class="st">  </span><span class="kw">coord_flip</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb7-17" data-line-number="17"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb7-18" data-line-number="18"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">plot.title =</span> <span class="kw">element_text</span>(<span class="dt">hjust =</span> <span class="fl">0.5</span>))</a>
<a class="sourceLine" id="cb7-19" data-line-number="19"></a>
<a class="sourceLine" id="cb7-20" data-line-number="20"><span class="kw">ggsave</span>(<span class="st">&quot;figures/variance_components/varpart_violin_alllines.png&quot;</span>, </a>
<a class="sourceLine" id="cb7-21" data-line-number="21">       fig.violin, <span class="dt">height =</span> <span class="dv">4</span>, <span class="dt">width =</span> <span class="fl">6.5</span>, <span class="dt">dpi =</span> <span class="dv">300</span>)</a>
<a class="sourceLine" id="cb7-22" data-line-number="22"><span class="kw">ggsave</span>(<span class="st">&quot;figures/variance_components/varpart_violin_alllines.pdf&quot;</span>, </a>
<a class="sourceLine" id="cb7-23" data-line-number="23">       fig.violin, <span class="dt">height =</span> <span class="dv">4</span>, <span class="dt">width =</span> <span class="fl">6.5</span>, <span class="dt">dpi =</span> <span class="dv">300</span>)</a>
<a class="sourceLine" id="cb7-24" data-line-number="24"><span class="kw">ggsave</span>(<span class="st">&quot;figures/variance_components/varpart_violin_alllines_skinny.png&quot;</span>, </a>
<a class="sourceLine" id="cb7-25" data-line-number="25">       fig.violin, <span class="dt">height =</span> <span class="dv">4</span>, <span class="dt">width =</span> <span class="fl">4.5</span>, <span class="dt">dpi =</span> <span class="dv">300</span>)</a>
<a class="sourceLine" id="cb7-26" data-line-number="26"><span class="kw">ggsave</span>(<span class="st">&quot;figures/variance_components/varpart_violin_alllines_skinny.pdf&quot;</span>, </a>
<a class="sourceLine" id="cb7-27" data-line-number="27">       fig.violin, <span class="dt">height =</span> <span class="dv">4</span>, <span class="dt">width =</span> <span class="fl">4.5</span>, <span class="dt">dpi =</span> <span class="dv">300</span>)</a>
<a class="sourceLine" id="cb7-28" data-line-number="28"></a>
<a class="sourceLine" id="cb7-29" data-line-number="29">fig.violin</a></code></pre></div>
<p><img src="figure/variance_components.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/d2e8b3145e4600ec75d0d0df9ad1e210ecb1bdd3/docs/figure/variance_components.Rmd/unnamed-chunk-4-1.png" target="_blank">d2e8b31</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="co"># plotVarPart( varPart_mat )</span></a>
<a class="sourceLine" id="cb8-2" data-line-number="2"><span class="co"># idx_order &lt;- order(varPart_mat$clone, decreasing = TRUE)</span></a>
<a class="sourceLine" id="cb8-3" data-line-number="3"><span class="co"># plotPercentBars( varPart_mat[idx_order[1:10],] )</span></a></code></pre></div>
<div id="plot-line-component-against-clone-component" class="section level3">
<h3>Plot line component against clone component</h3>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">varPart.df <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> <span class="dv">100</span> <span class="op">*</span><span class="st"> </span>line, <span class="dt">y =</span> <span class="dv">100</span> <span class="op">*</span><span class="st"> </span>clone)) <span class="op">+</span></a>
<a class="sourceLine" id="cb9-3" data-line-number="3"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.5</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb9-4" data-line-number="4"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.7</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb9-5" data-line-number="5"><span class="st">  </span><span class="kw">coord_cartesian</span>(<span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">25</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb9-6" data-line-number="6"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Variance explained by line (%)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb9-7" data-line-number="7"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Variance explained by clone (%)&quot;</span>)</a></code></pre></div>
<p><img src="figure/variance_components.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/d2e8b3145e4600ec75d0d0df9ad1e210ecb1bdd3/docs/figure/variance_components.Rmd/unnamed-chunk-6-1.png" target="_blank">d2e8b31</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="results-for-individual-lines" class="section level2">
<h2>Results for individual lines</h2>
<div id="load-individual-lines" class="section level3">
<h3>Load individual lines</h3>
<p>We load a SingleCellExperiment object for each line containing expression data, cell metadata and gene metadata.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">params &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb10-2" data-line-number="2">params<span class="op">$</span>callset &lt;-<span class="st"> &quot;filt_lenient.cell_coverage_sites&quot;</span> </a>
<a class="sourceLine" id="cb10-3" data-line-number="3">lines &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;euts&quot;</span>, <span class="st">&quot;fawm&quot;</span>, <span class="st">&quot;feec&quot;</span>, <span class="st">&quot;fikt&quot;</span>, <span class="st">&quot;garx&quot;</span>, <span class="st">&quot;gesg&quot;</span>, <span class="st">&quot;heja&quot;</span>, <span class="st">&quot;hipn&quot;</span>, </a>
<a class="sourceLine" id="cb10-4" data-line-number="4">            <span class="st">&quot;ieki&quot;</span>, <span class="st">&quot;joxm&quot;</span>, <span class="st">&quot;kuco&quot;</span>, <span class="st">&quot;laey&quot;</span>, <span class="st">&quot;lexy&quot;</span>, <span class="st">&quot;naju&quot;</span>, <span class="st">&quot;nusw&quot;</span>, <span class="st">&quot;oaaz&quot;</span>, </a>
<a class="sourceLine" id="cb10-5" data-line-number="5">            <span class="st">&quot;oilg&quot;</span>, <span class="st">&quot;pipw&quot;</span>, <span class="st">&quot;puie&quot;</span>, <span class="st">&quot;qayj&quot;</span>, <span class="st">&quot;qolg&quot;</span>, <span class="st">&quot;qonc&quot;</span>, <span class="st">&quot;rozh&quot;</span>, <span class="st">&quot;sehl&quot;</span>, </a>
<a class="sourceLine" id="cb10-6" data-line-number="6">            <span class="st">&quot;ualf&quot;</span>, <span class="st">&quot;vass&quot;</span>, <span class="st">&quot;vuna&quot;</span>, <span class="st">&quot;wahn&quot;</span>, <span class="st">&quot;wetu&quot;</span>, <span class="st">&quot;xugn&quot;</span>, <span class="st">&quot;zoxy&quot;</span>)</a>
<a class="sourceLine" id="cb10-7" data-line-number="7"><span class="co">## Load SCE objects</span></a>
<a class="sourceLine" id="cb10-8" data-line-number="8">sce_unst_list &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb10-9" data-line-number="9"><span class="cf">for</span> (don <span class="cf">in</span> lines) {</a>
<a class="sourceLine" id="cb10-10" data-line-number="10">    sce_unst_list[[don]] &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">file.path</span>(<span class="st">&quot;data/sces/&quot;</span>, </a>
<a class="sourceLine" id="cb10-11" data-line-number="11">        <span class="kw">paste0</span>(<span class="st">&quot;sce_&quot;</span>, don, <span class="st">&quot;_with_clone_assignments.&quot;</span>, params<span class="op">$</span>callset, <span class="st">&quot;.rds&quot;</span>)))</a>
<a class="sourceLine" id="cb10-12" data-line-number="12">    <span class="kw">cat</span>(<span class="kw">paste</span>(<span class="st">&quot;reading&quot;</span>, don, <span class="st">&quot;:   &quot;</span>, <span class="kw">ncol</span>(sce_unst_list[[don]]), <span class="st">&quot;cells</span><span class="ch">\n</span><span class="st">&quot;</span>))</a>
<a class="sourceLine" id="cb10-13" data-line-number="13">}</a></code></pre></div>
<pre><code>reading euts :    79 cells
reading fawm :    53 cells
reading feec :    75 cells
reading fikt :    39 cells
reading garx :    70 cells
reading gesg :    105 cells
reading heja :    50 cells
reading hipn :    62 cells
reading ieki :    58 cells
reading joxm :    79 cells
reading kuco :    48 cells
reading laey :    55 cells
reading lexy :    63 cells
reading naju :    44 cells
reading nusw :    60 cells
reading oaaz :    38 cells
reading oilg :    90 cells
reading pipw :    107 cells
reading puie :    41 cells
reading qayj :    97 cells
reading qolg :    36 cells
reading qonc :    58 cells
reading rozh :    91 cells
reading sehl :    30 cells
reading ualf :    89 cells
reading vass :    37 cells
reading vuna :    71 cells
reading wahn :    82 cells
reading wetu :    77 cells
reading xugn :    35 cells
reading zoxy :    88 cells</code></pre>
<p>We load variance component analysis results for each line.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1">varPart_list &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb12-2" data-line-number="2"><span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">seq_along</span>(lines)) {</a>
<a class="sourceLine" id="cb12-3" data-line-number="3">  df &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste0</span>(<span class="st">&quot;data/variance_components/donorVar/&quot;</span>,</a>
<a class="sourceLine" id="cb12-4" data-line-number="4">                        lines[i], <span class="st">&quot;.var_part.var1.csv&quot;</span>))</a>
<a class="sourceLine" id="cb12-5" data-line-number="5">    <span class="kw">colnames</span>(df) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;clone&quot;</span>, <span class="st">&quot;plate&quot;</span>, <span class="st">&quot;cell. det. rate&quot;</span>, <span class="st">&quot;residuals&quot;</span>)</a>
<a class="sourceLine" id="cb12-6" data-line-number="6">    df<span class="op">$</span>gene_id &lt;-<span class="st"> </span><span class="kw">rownames</span>(df)</a>
<a class="sourceLine" id="cb12-7" data-line-number="7">    df<span class="op">$</span>line &lt;-<span class="st"> </span>lines[i]</a>
<a class="sourceLine" id="cb12-8" data-line-number="8">    varPart_list[[i]] &lt;-<span class="st"> </span><span class="kw">as_data_frame</span>(df)</a>
<a class="sourceLine" id="cb12-9" data-line-number="9">}</a>
<a class="sourceLine" id="cb12-10" data-line-number="10"><span class="kw">names</span>(varPart_list) &lt;-<span class="st"> </span>lines</a></code></pre></div>
<p>For each line we thus have a data.frame with the variance components results.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="kw">head</span>(varPart_list[[<span class="dv">1</span>]])</a></code></pre></div>
<pre><code># A tibble: 6 x 6
   clone   plate `cell. det. rate` residuals gene_id                  line 
   &lt;dbl&gt;   &lt;dbl&gt;             &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;                    &lt;chr&gt;
1 0      0.0727           0.0155       0.912 ENSG00000137804_NUSAP1   euts 
2 0.165  0                0.0439       0.791 ENSG00000115318_LOXL3    euts 
3 0      0.0311           0.000381     0.969 ENSG00000161888_SPC24    euts 
4 0.0274 0.00157          0.0518       0.919 ENSG00000171345_KRT19    euts 
5 0      0.0357           0.0222       0.942 ENSG00000089685_BIRC5    euts 
6 0      0                0.00906      0.991 ENSG00000197061_HIST1H4C euts </code></pre>
</div>
<div id="number-of-genes-with-substantial-clone-component" class="section level3">
<h3>Number of genes with substantial clone component</h3>
<p>We observe larger clone effects if we fit the linear mixed model for each line separately. A large number of genes for each line have &gt;5% of variance explained by clone (median: 825).</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1"><span class="kw">sort</span>(<span class="kw">sapply</span>(varPart_list, <span class="cf">function</span>(x) <span class="kw">sum</span>(x<span class="op">$</span>clone <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.05</span>)))</a></code></pre></div>
<pre><code>pipw vuna gesg euts oilg heja puie kuco wetu zoxy naju wahn qayj joxm qolg 
 331  347  382  478  532  548  559  561  566  584  609  635  651  741  760 
hipn xugn fawm ualf nusw garx oaaz lexy rozh vass feec fikt ieki laey sehl 
 825  844  856  881  901  920  948  962 1036 1096 1102 1314 1356 1392 1651 
qonc 
2162 </code></pre>
</div>
<div id="plot-results-for-individual-lines" class="section level3">
<h3>Plot results for individual lines</h3>
<p>For each line we will plot the variance explained by clone for the to 400 genes with most variance explained by clone (for that line).</p>
<p>First, we need to process the raw variance component results from each line.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1">n_top &lt;-<span class="st"> </span><span class="dv">400</span></a>
<a class="sourceLine" id="cb17-2" data-line-number="2">n_lines &lt;-<span class="st"> </span><span class="kw">length</span>(varPart_list)</a>
<a class="sourceLine" id="cb17-3" data-line-number="3"></a>
<a class="sourceLine" id="cb17-4" data-line-number="4">n_hv_genes &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="ot">NA</span>, n_lines)</a>
<a class="sourceLine" id="cb17-5" data-line-number="5">n_cells_use &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="ot">NA</span>, n_lines)</a>
<a class="sourceLine" id="cb17-6" data-line-number="6">n_genes_use &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="ot">NA</span>, n_lines)</a>
<a class="sourceLine" id="cb17-7" data-line-number="7">var_top_mean &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="ot">NA</span>, n_lines)</a>
<a class="sourceLine" id="cb17-8" data-line-number="8">var_top_list &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb17-9" data-line-number="9"><span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">seq_len</span>(<span class="kw">length</span>(varPart_list))) {</a>
<a class="sourceLine" id="cb17-10" data-line-number="10">    sort_idx &lt;-<span class="st"> </span><span class="kw">order</span>(varPart_list[[i]]<span class="op">$</span>clone, <span class="dt">decreasing =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb17-11" data-line-number="11">    var_top_list[[i]] &lt;-<span class="st"> </span>varPart_list[[i]][sort_idx[<span class="dv">1</span><span class="op">:</span>n_top], <span class="st">&quot;clone&quot;</span>, </a>
<a class="sourceLine" id="cb17-12" data-line-number="12">                                           drop =<span class="st"> </span><span class="ot">FALSE</span>]</a>
<a class="sourceLine" id="cb17-13" data-line-number="13">    var_top_mean[i] &lt;-<span class="st"> </span><span class="kw">mean</span>(var_top_list[[i]]<span class="op">$</span>clone, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb17-14" data-line-number="14">    n_hv_genes[i] &lt;-<span class="st"> </span><span class="kw">sum</span>(varPart_list[[i]]<span class="op">$</span>clone <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.25</span>, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb17-15" data-line-number="15">    min_var &lt;-<span class="st"> </span><span class="fl">1.0</span></a>
<a class="sourceLine" id="cb17-16" data-line-number="16">    cell_idx &lt;-<span class="st"> </span><span class="kw">which</span>(sce_unst_list[[lines[i]]]<span class="op">$</span>assigned <span class="op">!=</span><span class="st"> &quot;unassigned&quot;</span>)</a>
<a class="sourceLine" id="cb17-17" data-line-number="17">    n_cells_use[i] &lt;-<span class="st"> </span><span class="kw">length</span>(cell_idx)</a>
<a class="sourceLine" id="cb17-18" data-line-number="18">    n_genes_use[i] &lt;-<span class="st"> </span><span class="kw">nrow</span>(varPart_list[[i]])</a>
<a class="sourceLine" id="cb17-19" data-line-number="19">}</a>
<a class="sourceLine" id="cb17-20" data-line-number="20">don_sort_idx &lt;-<span class="st"> </span><span class="kw">order</span>(var_top_mean, <span class="dt">decreasing =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb17-21" data-line-number="21"></a>
<a class="sourceLine" id="cb17-22" data-line-number="22">df.line &lt;-<span class="st"> </span><span class="kw">data_frame</span>(<span class="dt">line_id =</span> lines, </a>
<a class="sourceLine" id="cb17-23" data-line-number="23">                       <span class="dt">clone_cells =</span> n_cells_use, </a>
<a class="sourceLine" id="cb17-24" data-line-number="24">                       <span class="dt">n_genes_use =</span> n_genes_use, </a>
<a class="sourceLine" id="cb17-25" data-line-number="25">                       <span class="dt">n_hv_genes =</span> n_hv_genes, </a>
<a class="sourceLine" id="cb17-26" data-line-number="26">                       <span class="dt">var_top_mean =</span> var_top_mean)[don_sort_idx, ]</a></code></pre></div>
<pre><code>Warning: `data_frame()` is deprecated, use `tibble()`.
This warning is displayed once per session.</code></pre>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1">df.line &lt;-<span class="st"> </span><span class="kw">cbind</span>(<span class="dt">rank =</span> <span class="kw">seq_len</span>(<span class="kw">nrow</span>(df.line)), df.line)</a>
<a class="sourceLine" id="cb19-2" data-line-number="2"></a>
<a class="sourceLine" id="cb19-3" data-line-number="3">var_full &lt;-<span class="st"> </span><span class="kw">c</span>()</a>
<a class="sourceLine" id="cb19-4" data-line-number="4">idx_full &lt;-<span class="st"> </span><span class="kw">c</span>()</a>
<a class="sourceLine" id="cb19-5" data-line-number="5">don_full &lt;-<span class="st"> </span><span class="kw">c</span>()</a>
<a class="sourceLine" id="cb19-6" data-line-number="6">rank_full &lt;-<span class="st"> </span><span class="kw">c</span>()</a>
<a class="sourceLine" id="cb19-7" data-line-number="7"><span class="cf">for</span> (ii <span class="cf">in</span> don_sort_idx) {        </a>
<a class="sourceLine" id="cb19-8" data-line-number="8">    var_full &lt;-<span class="st"> </span><span class="kw">c</span>(var_full, var_top_list[[ii]]<span class="op">$</span>clone)</a>
<a class="sourceLine" id="cb19-9" data-line-number="9">    idx_full &lt;-<span class="st"> </span><span class="kw">c</span>(idx_full, <span class="kw">seq_len</span>(n_top))</a>
<a class="sourceLine" id="cb19-10" data-line-number="10">    don_full &lt;-<span class="st"> </span><span class="kw">c</span>(don_full, <span class="kw">rep</span>(lines[ii], n_top))</a>
<a class="sourceLine" id="cb19-11" data-line-number="11">    rank_full &lt;-<span class="st"> </span><span class="kw">c</span>(rank_full, <span class="kw">rep</span>(ii, n_top))</a>
<a class="sourceLine" id="cb19-12" data-line-number="12">}</a>
<a class="sourceLine" id="cb19-13" data-line-number="13"></a>
<a class="sourceLine" id="cb19-14" data-line-number="14">line_info &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="st">&quot;other&quot;</span>, <span class="kw">length</span>(don_full))</a>
<a class="sourceLine" id="cb19-15" data-line-number="15">median_line &lt;-<span class="st"> </span>lines[don_sort_idx[<span class="kw">round</span>(<span class="kw">length</span>(don_sort_idx)<span class="op">/</span><span class="dv">2</span>)]]</a>
<a class="sourceLine" id="cb19-16" data-line-number="16"><span class="co"># line_info[don_full == median_line] &lt;- &quot;median&quot;</span></a>
<a class="sourceLine" id="cb19-17" data-line-number="17">line_info[don_full <span class="op">==</span><span class="st"> &quot;joxm&quot;</span>] &lt;-<span class="st"> &quot;joxm&quot;</span></a>
<a class="sourceLine" id="cb19-18" data-line-number="18"><span class="co"># print(median_line)</span></a>
<a class="sourceLine" id="cb19-19" data-line-number="19"><span class="co"># summary(n_hv_genes)</span></a>
<a class="sourceLine" id="cb19-20" data-line-number="20"></a>
<a class="sourceLine" id="cb19-21" data-line-number="21">df &lt;-<span class="st"> </span><span class="kw">data_frame</span>(<span class="dt">varPart =</span> var_full, <span class="dt">gene_rank =</span> idx_full, </a>
<a class="sourceLine" id="cb19-22" data-line-number="22">                 <span class="dt">lines =</span> don_full, <span class="dt">line_rank =</span> rank_full,</a>
<a class="sourceLine" id="cb19-23" data-line-number="23">                 <span class="dt">line_info =</span> line_info)</a>
<a class="sourceLine" id="cb19-24" data-line-number="24"><span class="kw">rm</span>(line_info)</a>
<a class="sourceLine" id="cb19-25" data-line-number="25"></a>
<a class="sourceLine" id="cb19-26" data-line-number="26">df &lt;-<span class="st"> </span>df <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb19-27" data-line-number="27"><span class="st">  </span><span class="kw">group_by</span>(gene_rank) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb19-28" data-line-number="28"><span class="st">  </span><span class="kw">summarise</span>(<span class="dt">varPart =</span> <span class="kw">median</span>(varPart)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb19-29" data-line-number="29"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">lines =</span> <span class="st">&quot;median&quot;</span>, <span class="dt">line_info =</span> <span class="st">&quot;median&quot;</span>, <span class="dt">line_rank =</span> <span class="ot">NA</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb19-30" data-line-number="30"><span class="st">  </span><span class="kw">bind_rows</span>(., df)</a></code></pre></div>
<p>The table below provides a summary of the per-line variance component results. We can read off the mean proportion of variance explained by clone for the top 400 genes for each line (<code>var_top_mean</code>), as well as other information for each line.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1">df.line</a></code></pre></div>
<pre><code>   rank line_id clone_cells n_genes_use n_hv_genes var_top_mean
1     1    feec          64        8243        342   0.50769558
2     2    xugn          34        7480        260   0.41529069
3     3    qonc          43        9882        420   0.35658891
4     4    fikt          36        9762        219   0.29630805
5     5    sehl          24        8792        198   0.27796718
6     6    ieki          26        8324        166   0.25419512
7     7    vass          37        7231        129   0.23747536
8     8    lexy          63       10154        123   0.23580397
9     9    oaaz          37        9946        105   0.22820218
10   10    nusw          20        9544        110   0.22397105
11   11    kuco          48       10021        121   0.22329573
12   12    euts          78       10210        128   0.21836347
13   13    wahn          77       10241        114   0.21728504
14   14    laey          55        7402         89   0.20944535
15   15    wetu          66        7177         79   0.20065196
16   16    hipn          49        7971         86   0.19522467
17   17    rozh          42        7819         57   0.18304904
18   18    heja          50        9315         57   0.16430667
19   19    qolg          36        8640         41   0.16297743
20   20    joxm          77       10095         45   0.15834547
21   21    garx          69        9571         25   0.15202524
22   22    pipw         107       10260         73   0.15192220
23   23    oilg          57        7875         44   0.14300906
24   24    qayj          59        6844         30   0.14024325
25   25    fawm          47        8537         17   0.13702877
26   26    ualf          88        7001         16   0.13288015
27   27    zoxy          82        6968         27   0.12433830
28   28    puie          41        8478         12   0.11222310
29   29    naju          44        9089          5   0.10986147
30   30    gesg         101       10036          5   0.08542351
31   31    vuna          71       10000          2   0.07668159</code></pre>
<p>Now we can plot a curve for each line showing the variance explained by clone for the top 400 genes.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1">df_labs &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(df, gene_rank <span class="op">==</span><span class="st"> </span><span class="dv">200</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb22-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">labs =</span> <span class="kw">ifelse</span>(varPart <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.33</span>, lines, <span class="st">&quot;&quot;</span>))</a>
<a class="sourceLine" id="cb22-3" data-line-number="3"></a>
<a class="sourceLine" id="cb22-4" data-line-number="4">fig.curve &lt;-<span class="st"> </span><span class="kw">ggplot</span>(df, <span class="kw">aes</span>(<span class="dt">colour =</span> line_info)) <span class="op">+</span></a>
<a class="sourceLine" id="cb22-5" data-line-number="5"><span class="st">  </span><span class="kw">geom_line</span>(<span class="kw">aes</span>(<span class="dt">x =</span> gene_rank, <span class="dt">y =</span> <span class="dv">100</span> <span class="op">*</span><span class="st"> </span>varPart, <span class="dt">color =</span> line_info, </a>
<a class="sourceLine" id="cb22-6" data-line-number="6">                <span class="dt">size =</span> line_info, <span class="dt">group =</span> lines)) <span class="op">+</span></a>
<a class="sourceLine" id="cb22-7" data-line-number="7"><span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;firebrick&quot;</span>, <span class="st">&quot;black&quot;</span>, <span class="st">&quot;gray60&quot;</span>),</a>
<a class="sourceLine" id="cb22-8" data-line-number="8">                     <span class="dt">name =</span> <span class="st">&quot;line&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb22-9" data-line-number="9"><span class="st">  </span><span class="kw">scale_size_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="fl">0.5</span>, <span class="dv">1</span>, <span class="fl">0.5</span>), <span class="dt">name =</span> <span class="st">&quot;line&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb22-10" data-line-number="10"><span class="st">  </span><span class="kw">geom_line</span>(<span class="dt">data =</span> df[df<span class="op">$</span>lines <span class="op">==</span><span class="st"> &quot;joxm&quot;</span>, ],</a>
<a class="sourceLine" id="cb22-11" data-line-number="11">            <span class="kw">aes</span>(<span class="dt">x =</span> gene_rank, <span class="dt">y =</span> <span class="dv">100</span> <span class="op">*</span><span class="st"> </span>varPart), <span class="dt">size =</span> <span class="fl">0.5</span>, <span class="dt">color =</span> <span class="st">&quot;firebrick&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb22-12" data-line-number="12"><span class="st">  </span><span class="kw">geom_line</span>(<span class="dt">data =</span> df[df<span class="op">$</span>lines <span class="op">==</span><span class="st"> &quot;median&quot;</span>, ],</a>
<a class="sourceLine" id="cb22-13" data-line-number="13">            <span class="kw">aes</span>(<span class="dt">x =</span> gene_rank, <span class="dt">y =</span> <span class="dv">100</span> <span class="op">*</span><span class="st"> </span>varPart), <span class="dt">size =</span> <span class="dv">1</span>, <span class="dt">color =</span> <span class="st">&quot;black&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb22-14" data-line-number="14"><span class="st">  </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">x =</span> gene_rank <span class="op">-</span><span class="st"> </span><span class="dv">15</span>, <span class="dt">y =</span> <span class="dv">100</span> <span class="op">*</span><span class="st"> </span>varPart, <span class="dt">label =</span> labs),</a>
<a class="sourceLine" id="cb22-15" data-line-number="15">                   <span class="dt">nudge_y =</span> <span class="dv">0</span>, <span class="dt">nudge_x =</span> <span class="dv">0</span>, </a>
<a class="sourceLine" id="cb22-16" data-line-number="16">                   <span class="dt">data =</span> df_labs, <span class="dt">colour =</span> <span class="st">&quot;gray40&quot;</span>, <span class="dt">size =</span> <span class="dv">2</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb22-17" data-line-number="17"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Top genes&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Variance explained by clone (%)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb22-18" data-line-number="18"><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlim</span>(<span class="dv">0</span>, <span class="dv">400</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb22-19" data-line-number="19"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="kw">c</span>(<span class="fl">0.8</span>, <span class="fl">0.8</span>))</a>
<a class="sourceLine" id="cb22-20" data-line-number="20"></a>
<a class="sourceLine" id="cb22-21" data-line-number="21"><span class="kw">ggsave</span>(<span class="st">&quot;figures/variance_components/varpart_curve_perline.png&quot;</span>, </a>
<a class="sourceLine" id="cb22-22" data-line-number="22">       fig.curve, <span class="dt">height =</span> <span class="dv">4</span>, <span class="dt">width =</span> <span class="fl">6.5</span>, <span class="dt">dpi =</span> <span class="dv">300</span>)</a>
<a class="sourceLine" id="cb22-23" data-line-number="23"><span class="kw">ggsave</span>(<span class="st">&quot;figures/variance_components/varpart_cruve_perline.pdf&quot;</span>, </a>
<a class="sourceLine" id="cb22-24" data-line-number="24">       fig.curve, <span class="dt">height =</span> <span class="dv">4</span>, <span class="dt">width =</span> <span class="fl">6.5</span>, <span class="dt">dpi =</span> <span class="dv">300</span>)</a>
<a class="sourceLine" id="cb22-25" data-line-number="25"><span class="kw">ggsave</span>(<span class="st">&quot;figures/variance_components/varpart_curve_perline_skinny.png&quot;</span>, </a>
<a class="sourceLine" id="cb22-26" data-line-number="26">       fig.curve <span class="op">+</span><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="kw">c</span>(<span class="fl">0.7</span>, <span class="fl">0.8</span>)),</a>
<a class="sourceLine" id="cb22-27" data-line-number="27">       <span class="dt">height =</span> <span class="dv">4</span>, <span class="dt">width =</span> <span class="fl">4.5</span>, <span class="dt">dpi =</span> <span class="dv">300</span>)</a>
<a class="sourceLine" id="cb22-28" data-line-number="28"><span class="kw">ggsave</span>(<span class="st">&quot;figures/variance_components/varpart_cruve_perline_skinny.pdf&quot;</span>, </a>
<a class="sourceLine" id="cb22-29" data-line-number="29">       fig.curve <span class="op">+</span><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="kw">c</span>(<span class="fl">0.7</span>, <span class="fl">0.8</span>)), </a>
<a class="sourceLine" id="cb22-30" data-line-number="30">       <span class="dt">height =</span> <span class="dv">4</span>, <span class="dt">width =</span> <span class="fl">4.5</span>, <span class="dt">dpi =</span> <span class="dv">300</span>)</a>
<a class="sourceLine" id="cb22-31" data-line-number="31"></a>
<a class="sourceLine" id="cb22-32" data-line-number="32">fig.curve</a></code></pre></div>
<p><img src="figure/variance_components.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/d2e8b3145e4600ec75d0d0df9ad1e210ecb1bdd3/docs/figure/variance_components.Rmd/unnamed-chunk-13-1.png" target="_blank">d2e8b31</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Each line has a substantial number of genes with a substantial proportion of variance explained by clone. For example, the number of genes with more than 25% of variance explained by line across lines:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1"><span class="kw">sapply</span>(varPart_list, <span class="cf">function</span>(x) <span class="kw">sum</span>(x<span class="op">$</span>clone <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.25</span>))</a></code></pre></div>
<pre><code>euts fawm feec fikt garx gesg heja hipn ieki joxm kuco laey lexy naju nusw 
 128   17  342  219   25    5   57   86  166   45  121   89  123    5  110 
oaaz oilg pipw puie qayj qolg qonc rozh sehl ualf vass vuna wahn wetu xugn 
 105   44   73   12   30   41  420   57  198   16  129    2  114   79  260 
zoxy 
  27 </code></pre>
<p>There are a large number of genes in each line with more than 5% of variance explained by clone:</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" data-line-number="1"><span class="kw">sapply</span>(varPart_list, <span class="cf">function</span>(x) <span class="kw">sum</span>(x<span class="op">$</span>clone <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.05</span>))</a></code></pre></div>
<pre><code>euts fawm feec fikt garx gesg heja hipn ieki joxm kuco laey lexy naju nusw 
 478  856 1102 1314  920  382  548  825 1356  741  561 1392  962  609  901 
oaaz oilg pipw puie qayj qolg qonc rozh sehl ualf vass vuna wahn wetu xugn 
 948  532  331  559  651  760 2162 1036 1651  881 1096  347  635  566  844 
zoxy 
 584 </code></pre>
</div>
</div>
<div id="plots-including-line-selection-information" class="section level2">
<h2>Plots including line selection information</h2>
<p>We have inferred selection dynamics from somatic variants detected from whole-exome sequencing data. We can overlay selection status on the plot above to look for any relationship between selection and variance explained by clone in top genes by line.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" data-line-number="1">line_selected &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;euts&quot;</span>, <span class="st">&quot;fawm&quot;</span>, <span class="st">&quot;fikt&quot;</span>, <span class="st">&quot;garx&quot;</span>, <span class="st">&quot;pipw&quot;</span>, <span class="st">&quot;puie&quot;</span>, <span class="st">&quot;qolg&quot;</span>, <span class="st">&quot;rozh&quot;</span>)</a>
<a class="sourceLine" id="cb27-2" data-line-number="2">line_neutral &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;hipn&quot;</span>, <span class="st">&quot;nusw&quot;</span>, <span class="st">&quot;qonc&quot;</span>, <span class="st">&quot;sehl&quot;</span>, <span class="st">&quot;ualf&quot;</span>, <span class="st">&quot;xugn&quot;</span>)</a>
<a class="sourceLine" id="cb27-3" data-line-number="3">line_undetermined &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;feec&quot;</span>, <span class="st">&quot;gesg&quot;</span>, <span class="st">&quot;heja&quot;</span>, <span class="st">&quot;ieki&quot;</span>, <span class="st">&quot;joxm&quot;</span>, <span class="st">&quot;kuco&quot;</span>, <span class="st">&quot;laey&quot;</span>, </a>
<a class="sourceLine" id="cb27-4" data-line-number="4">                        <span class="st">&quot;lexy&quot;</span>, <span class="st">&quot;naju&quot;</span>, <span class="st">&quot;oaaz&quot;</span>, </a>
<a class="sourceLine" id="cb27-5" data-line-number="5">                        <span class="st">&quot;oilg&quot;</span>, <span class="st">&quot;qayj&quot;</span>, <span class="st">&quot;vass&quot;</span>, <span class="st">&quot;vuna&quot;</span>, <span class="st">&quot;wahn&quot;</span>, <span class="st">&quot;wetu&quot;</span>, <span class="st">&quot;zoxy&quot;</span>)</a>
<a class="sourceLine" id="cb27-6" data-line-number="6">df<span class="op">$</span>selection &lt;-<span class="st"> &quot;undetermined&quot;</span></a>
<a class="sourceLine" id="cb27-7" data-line-number="7">df<span class="op">$</span>selection[df<span class="op">$</span>lines <span class="op">%in%</span><span class="st"> </span>line_neutral] &lt;-<span class="st"> &quot;neutral&quot;</span></a>
<a class="sourceLine" id="cb27-8" data-line-number="8">df<span class="op">$</span>selection[df<span class="op">$</span>lines <span class="op">%in%</span><span class="st"> </span>line_selected] &lt;-<span class="st"> &quot;selection&quot;</span></a>
<a class="sourceLine" id="cb27-9" data-line-number="9">df<span class="op">$</span>selection[df<span class="op">$</span>lines <span class="op">%in%</span><span class="st"> &quot;median&quot;</span>] &lt;-<span class="st"> &quot;median&quot;</span></a>
<a class="sourceLine" id="cb27-10" data-line-number="10">df<span class="op">$</span>selection &lt;-<span class="st"> </span><span class="kw">factor</span>(df<span class="op">$</span>selection, </a>
<a class="sourceLine" id="cb27-11" data-line-number="11">                       <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selection&quot;</span>, </a>
<a class="sourceLine" id="cb27-12" data-line-number="12">                                  <span class="st">&quot;median&quot;</span>))</a></code></pre></div>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" data-line-number="1">df_labs &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(df, gene_rank <span class="op">==</span><span class="st"> </span><span class="dv">200</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb28-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">labs =</span> <span class="kw">ifelse</span>(varPart <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.33</span>, lines, <span class="st">&quot;&quot;</span>))</a>
<a class="sourceLine" id="cb28-3" data-line-number="3"></a>
<a class="sourceLine" id="cb28-4" data-line-number="4">fig.curve &lt;-<span class="st"> </span><span class="kw">ggplot</span>(df, <span class="kw">aes</span>(<span class="dt">colour =</span> selection)) <span class="op">+</span></a>
<a class="sourceLine" id="cb28-5" data-line-number="5"><span class="st">  </span><span class="kw">geom_line</span>(<span class="kw">aes</span>(<span class="dt">x =</span> gene_rank, <span class="dt">y =</span> <span class="dv">100</span> <span class="op">*</span><span class="st"> </span>varPart, <span class="dt">color =</span> selection, </a>
<a class="sourceLine" id="cb28-6" data-line-number="6">                <span class="dt">size =</span> selection, <span class="dt">group =</span> lines)) <span class="op">+</span></a>
<a class="sourceLine" id="cb28-7" data-line-number="7"><span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span>  <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>, </a>
<a class="sourceLine" id="cb28-8" data-line-number="8">                                 <span class="st">&quot;black&quot;</span>), <span class="dt">name =</span> <span class="st">&quot;selection</span><span class="ch">\n</span><span class="st">dynamics&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb28-9" data-line-number="9"><span class="st">  </span><span class="kw">scale_size_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="fl">0.5</span>, <span class="fl">0.5</span>, <span class="fl">0.5</span>, <span class="dv">1</span>), <span class="dt">name =</span> <span class="st">&quot;selection</span><span class="ch">\n</span><span class="st">dynamics&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb28-10" data-line-number="10"><span class="st">  </span><span class="kw">geom_line</span>(<span class="dt">data =</span> df[df<span class="op">$</span>lines <span class="op">==</span><span class="st"> &quot;median&quot;</span>, ],</a>
<a class="sourceLine" id="cb28-11" data-line-number="11">            <span class="kw">aes</span>(<span class="dt">x =</span> gene_rank, <span class="dt">y =</span> <span class="dv">100</span> <span class="op">*</span><span class="st"> </span>varPart), <span class="dt">size =</span> <span class="dv">1</span>, <span class="dt">color =</span> <span class="st">&quot;black&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb28-12" data-line-number="12"><span class="st">  </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">x =</span> gene_rank <span class="op">-</span><span class="st"> </span><span class="dv">15</span>, <span class="dt">y =</span> <span class="dv">100</span> <span class="op">*</span><span class="st"> </span>varPart, <span class="dt">label =</span> labs),</a>
<a class="sourceLine" id="cb28-13" data-line-number="13">                   <span class="dt">nudge_y =</span> <span class="dv">0</span>, <span class="dt">nudge_x =</span> <span class="dv">0</span>, </a>
<a class="sourceLine" id="cb28-14" data-line-number="14">                   <span class="dt">data =</span> df_labs, <span class="dt">colour =</span> <span class="st">&quot;gray40&quot;</span>, <span class="dt">size =</span> <span class="dv">2</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb28-15" data-line-number="15"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Top genes&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Variance explained by clone (%)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb28-16" data-line-number="16"><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlim</span>(<span class="dv">0</span>, <span class="dv">400</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb28-17" data-line-number="17"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="kw">c</span>(<span class="fl">0.8</span>, <span class="fl">0.8</span>))</a>
<a class="sourceLine" id="cb28-18" data-line-number="18"></a>
<a class="sourceLine" id="cb28-19" data-line-number="19">fig.curve</a></code></pre></div>
<p><img src="figure/variance_components.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/davismcc/fibroblast-clonality/blob/d2e8b3145e4600ec75d0d0df9ad1e210ecb1bdd3/docs/figure/variance_components.Rmd/unnamed-chunk-17-1.png" target="_blank">d2e8b31</a>
</td>
<td>
davismcc
</td>
<td>
2018-08-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>There is not any obvious relationship between selection dynamics and variance explained by clone here.</p>
</div>
<div id="write-mean-variance-explained-to-file" class="section level2">
<h2>Write mean variance explained to file</h2>
<p>Write a table with the mean fraction of variance explained by clone for the top 400 genes per line out to file.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb29-1" data-line-number="1">idx &lt;-<span class="st"> </span><span class="kw">order</span>(var_top_mean, <span class="dt">decreasing =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb29-2" data-line-number="2">line_var_df &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">line =</span> lines[idx], </a>
<a class="sourceLine" id="cb29-3" data-line-number="3">                           <span class="dt">meanFraction400 =</span> var_top_mean[idx])</a>
<a class="sourceLine" id="cb29-4" data-line-number="4"><span class="kw">write.csv</span>(line_var_df, <span class="st">&quot;output/variance_components/line_var_top400.csv&quot;</span>)</a></code></pre></div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb30-1" data-line-number="1">devtools<span class="op">::</span><span class="kw">session_info</span>()</a></code></pre></div>
<pre><code>─ Session info ──────────────────────────────────────────────────────────
 setting  value                       
 version  R version 3.6.0 (2019-04-26)
 os       Ubuntu 18.04.3 LTS          
 system   x86_64, linux-gnu           
 ui       X11                         
 language (EN)                        
 collate  en_AU.UTF-8                 
 ctype    en_AU.UTF-8                 
 tz       Australia/Melbourne         
 date     2019-10-30                  

─ Packages ──────────────────────────────────────────────────────────────
 package              * version   date       lib source        
 assertthat             0.2.1     2019-03-21 [1] CRAN (R 3.6.0)
 backports              1.1.4     2019-04-10 [1] CRAN (R 3.6.0)
 Biobase              * 2.44.0    2019-05-02 [1] Bioconductor  
 BiocGenerics         * 0.30.0    2019-05-02 [1] Bioconductor  
 BiocParallel         * 1.18.1    2019-08-06 [1] Bioconductor  
 bitops                 1.0-6     2013-08-17 [1] CRAN (R 3.6.0)
 broom                  0.5.2     2019-04-07 [1] CRAN (R 3.6.0)
 callr                  3.3.2     2019-09-22 [1] CRAN (R 3.6.0)
 cellranger             1.1.0     2016-07-27 [1] CRAN (R 3.6.0)
 cli                    1.1.0     2019-03-19 [1] CRAN (R 3.6.0)
 colorspace             1.4-1     2019-03-18 [1] CRAN (R 3.6.0)
 cowplot              * 1.0.0     2019-07-11 [1] CRAN (R 3.6.0)
 crayon                 1.3.4     2017-09-16 [1] CRAN (R 3.6.0)
 DelayedArray         * 0.10.0    2019-05-02 [1] Bioconductor  
 desc                   1.2.0     2018-05-01 [1] CRAN (R 3.6.0)
 devtools               2.2.1     2019-09-24 [1] CRAN (R 3.6.0)
 digest                 0.6.21    2019-09-20 [1] CRAN (R 3.6.0)
 dplyr                * 0.8.3     2019-07-04 [1] CRAN (R 3.6.0)
 ellipsis               0.3.0     2019-09-20 [1] CRAN (R 3.6.0)
 evaluate               0.14      2019-05-28 [1] CRAN (R 3.6.0)
 fansi                  0.4.0     2018-10-05 [1] CRAN (R 3.6.0)
 forcats              * 0.4.0     2019-02-17 [1] CRAN (R 3.6.0)
 fs                     1.3.1     2019-05-06 [1] CRAN (R 3.6.0)
 generics               0.0.2     2018-11-29 [1] CRAN (R 3.6.0)
 GenomeInfoDb         * 1.20.0    2019-05-02 [1] Bioconductor  
 GenomeInfoDbData       1.2.1     2019-04-30 [1] Bioconductor  
 GenomicRanges        * 1.36.1    2019-09-06 [1] Bioconductor  
 ggplot2              * 3.2.1     2019-08-10 [1] CRAN (R 3.6.0)
 ggpubr               * 0.2.3     2019-09-03 [1] CRAN (R 3.6.0)
 ggrepel              * 0.8.1     2019-05-07 [1] CRAN (R 3.6.0)
 ggsignif               0.6.0     2019-08-08 [1] CRAN (R 3.6.0)
 git2r                  0.26.1    2019-06-29 [1] CRAN (R 3.6.0)
 glue                   1.3.1     2019-03-12 [1] CRAN (R 3.6.0)
 gridExtra              2.3       2017-09-09 [1] CRAN (R 3.6.0)
 gtable                 0.3.0     2019-03-25 [1] CRAN (R 3.6.0)
 haven                  2.1.1     2019-07-04 [1] CRAN (R 3.6.0)
 hms                    0.5.1     2019-08-23 [1] CRAN (R 3.6.0)
 htmltools              0.3.6     2017-04-28 [1] CRAN (R 3.6.0)
 httr                   1.4.1     2019-08-05 [1] CRAN (R 3.6.0)
 IRanges              * 2.18.3    2019-09-24 [1] Bioconductor  
 jsonlite               1.6       2018-12-07 [1] CRAN (R 3.6.0)
 knitr                  1.25      2019-09-18 [1] CRAN (R 3.6.0)
 labeling               0.3       2014-08-23 [1] CRAN (R 3.6.0)
 lattice                0.20-38   2018-11-04 [4] CRAN (R 3.5.1)
 lazyeval               0.2.2     2019-03-15 [1] CRAN (R 3.6.0)
 lifecycle              0.1.0     2019-08-01 [1] CRAN (R 3.6.0)
 lubridate              1.7.4     2018-04-11 [1] CRAN (R 3.6.0)
 magrittr             * 1.5       2014-11-22 [1] CRAN (R 3.6.0)
 Matrix                 1.2-17    2019-03-22 [1] CRAN (R 3.6.0)
 matrixStats          * 0.55.0    2019-09-07 [1] CRAN (R 3.6.0)
 memoise                1.1.0     2017-04-21 [1] CRAN (R 3.6.0)
 mgcv                   1.8-28    2019-03-21 [4] CRAN (R 3.5.3)
 modelr                 0.1.5     2019-08-08 [1] CRAN (R 3.6.0)
 munsell                0.5.0     2018-06-12 [1] CRAN (R 3.6.0)
 nlme                   3.1-139   2019-04-09 [4] CRAN (R 3.5.3)
 pillar                 1.4.2     2019-06-29 [1] CRAN (R 3.6.0)
 pkgbuild               1.0.5     2019-08-26 [1] CRAN (R 3.6.0)
 pkgconfig              2.0.3     2019-09-22 [1] CRAN (R 3.6.0)
 pkgload                1.0.2     2018-10-29 [1] CRAN (R 3.6.0)
 plyr                   1.8.4     2016-06-08 [1] CRAN (R 3.6.0)
 prettyunits            1.0.2     2015-07-13 [1] CRAN (R 3.6.0)
 processx               3.4.1     2019-07-18 [1] CRAN (R 3.6.0)
 ps                     1.3.0     2018-12-21 [1] CRAN (R 3.6.0)
 purrr                * 0.3.2     2019-03-15 [1] CRAN (R 3.6.0)
 R6                     2.4.0     2019-02-14 [1] CRAN (R 3.6.0)
 Rcpp                   1.0.2     2019-07-25 [1] CRAN (R 3.6.0)
 RCurl                  1.95-4.12 2019-03-04 [1] CRAN (R 3.6.0)
 readr                * 1.3.1     2018-12-21 [1] CRAN (R 3.6.0)
 readxl                 1.3.1     2019-03-13 [1] CRAN (R 3.6.0)
 remotes                2.1.0     2019-06-24 [1] CRAN (R 3.6.0)
 reshape2             * 1.4.3     2017-12-11 [1] CRAN (R 3.6.0)
 rlang                  0.4.0     2019-06-25 [1] CRAN (R 3.6.0)
 rmarkdown              1.15      2019-08-21 [1] CRAN (R 3.6.0)
 rprojroot              1.3-2     2018-01-03 [1] CRAN (R 3.6.0)
 rstudioapi             0.10      2019-03-19 [1] CRAN (R 3.6.0)
 rvest                  0.3.4     2019-05-15 [1] CRAN (R 3.6.0)
 S4Vectors            * 0.22.1    2019-09-09 [1] Bioconductor  
 scales                 1.0.0     2018-08-09 [1] CRAN (R 3.6.0)
 sessioninfo            1.1.1     2018-11-05 [1] CRAN (R 3.6.0)
 SingleCellExperiment * 1.6.0     2019-05-02 [1] Bioconductor  
 stringi                1.4.3     2019-03-12 [1] CRAN (R 3.6.0)
 stringr              * 1.4.0     2019-02-10 [1] CRAN (R 3.6.0)
 SummarizedExperiment * 1.14.1    2019-07-31 [1] Bioconductor  
 testthat               2.2.1     2019-07-25 [1] CRAN (R 3.6.0)
 tibble               * 2.1.3     2019-06-06 [1] CRAN (R 3.6.0)
 tidyr                * 1.0.0     2019-09-11 [1] CRAN (R 3.6.0)
 tidyselect             0.2.5     2018-10-11 [1] CRAN (R 3.6.0)
 tidyverse            * 1.2.1     2017-11-14 [1] CRAN (R 3.6.0)
 usethis                1.5.1     2019-07-04 [1] CRAN (R 3.6.0)
 utf8                   1.1.4     2018-05-24 [1] CRAN (R 3.6.0)
 vctrs                  0.2.0     2019-07-05 [1] CRAN (R 3.6.0)
 viridis              * 0.5.1     2018-03-29 [1] CRAN (R 3.6.0)
 viridisLite          * 0.3.0     2018-02-01 [1] CRAN (R 3.6.0)
 whisker                0.4       2019-08-28 [1] CRAN (R 3.6.0)
 withr                  2.1.2     2018-03-15 [1] CRAN (R 3.6.0)
 workflowr              1.4.0     2019-06-08 [1] CRAN (R 3.6.0)
 xfun                   0.9       2019-08-21 [1] CRAN (R 3.6.0)
 xml2                   1.2.2     2019-08-09 [1] CRAN (R 3.6.0)
 XVector                0.24.0    2019-05-02 [1] Bioconductor  
 yaml                   2.2.0     2018-07-25 [1] CRAN (R 3.6.0)
 zeallot                0.1.0     2018-01-28 [1] CRAN (R 3.6.0)
 zlibbioc               1.30.0    2019-05-02 [1] Bioconductor  

[1] /home/AD.SVI.EDU.AU/dmccarthy/R/x86_64-pc-linux-gnu-library/3.6
[2] /usr/local/lib/R/site-library
[3] /usr/lib/R/site-library
[4] /usr/lib/R/library</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
