---
title: "About"
output:
  workflowr::wflow_html:
    toc: false
---

## *Cardelino* : Integrating whole exomes and single-cell transcriptomes to reveal phenotypic impact of somatic variants

Decoding the clonal substructures of somatic tissues sheds light on cell growth,
development and differentiation in health, ageing and disease. Targeted DNA 
sequencing, whole exome sequencing, and most recently single-cell DNA sequencing
have been applied to reconstruct trees describing this clonal substructure. 
However, the functional and phenotypic differences between clonal populations 
identified have been difficult to characterise and remain unclear in most cases. 

To address this, we present cardelino (https://github.com/PMBio/cardelino), a 
computational method to assign single-cell transcriptome profiles to nodes in a 
clonal tree, based on variants identified from single-cell RNA-seq data. After 
validating the model on simulated data, we apply cardelino to matched 
single-cell RNA-seq and exome sequencing data from 32 human dermal fibroblast 
lines. 

The somatic mutation landscape reveals non-neutral clonal evolution in a subset 
of these healthy donors. In line with this evolutionary selection pressure, we 
identify gene expression differences between clones. Interestingly, cell cycle 
and proliferation pathways separate clones repeatedly in our fibroblast samples.
In summary, we develop and apply an approach that allows for reconstructing 
transcriptome profiles of clonal subpopulations within tissues. This has the 
potential to inform on mutational processes at single-cell resolution, both in 
cancer and healthy ageing. 

**Key findings:**

* A new approach for integrating DNA-seq and single-cell RNA-seq data to reconstruct clonal substructure and single-cell transcriptomes.
* A new computational method to map single-cell RNA-seq profiles to clones.
* Evidence for non-neutral evolution of clonal populations in human fibroblasts.
* Proliferation and cell cycle pathways are commonly distorted in mutated clonal
populations, with implications for cancer and ageing. 


## Authors

The full author list is as follows:

Davis J. McCarthy<sup>1,4\*</sup>, Raghd Rostom<sup>1,2,\*</sup>, Yuanhua Huang<sup>1\*</sup>, Daniel J. Kunz<sup>2,5,6</sup>, Petr Danecek<sup>2</sup>, Marc Jan Bonder<sup>1</sup>, Tzachi Hagai<sup>1,2</sup>, HipSci Consortium, Wenyi Wang, Daniel J. Gaffney<sup>2</sup>, Benjamin D. Simons<sup>5,6,7</sup>, Oliver Stegle<sup>1,3,#</sup>, Sarah A. Teichmann<sup>1,2,5,#</sup>

<sup>1</sup>European Molecular Biology Laboratory, European Bioinformatics Institute, Wellcome Genome Campus, CB10 1SD
Hinxton, Cambridge, UK; <sup>2</sup>Wellcome Sanger Institute, Wellcome Genome Campus, Hinxton, CB10 1SA, UK; <sup>3</sup>European Molecular Biology Laboratory, Genome Biology Unit, 69117 Heidelberg, Germany; <sup>4</sup>St Vincentâ€™s Institute of Medical Research, Fitzroy, Victoria 3065, Australia. <sup>5</sup>Cavendish Laboratory, Department of Physics, JJ Thomson Avenue, Cambridge, CB3 0HE, UK. <sup>6</sup>The Wellcome Trust/Cancer Research UK Gurdon Institute, University of Cambridge, Cambridge, CB2 1QN, UK. <sup>7</sup>The Wellcome Trust/Medical Research Council Stem Cell Institute, University of Cambridge, Cambridge, UK.

<sup>*</sup> These authors contributed equally to this work.

<sup>#</sup> Corresponding authors. 

